{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "51a596dae02a4ad1b52926bdd6fa3187": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_cb597df2a29b4dadb8ce67ef817cc413",
              "IPY_MODEL_6e8e216e073c4ce9977ec8503a26e328",
              "IPY_MODEL_8f0634cbd43e4bb4ab99cb4cd6275920"
            ],
            "layout": "IPY_MODEL_9a0672593157449f96bc33ecd12e7e93"
          }
        },
        "cb597df2a29b4dadb8ce67ef817cc413": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_33d910a3e342416fbaf61b59ccffb9b2",
            "placeholder": "​",
            "style": "IPY_MODEL_5c82ef28566f4015ae557d379748aa79",
            "value": "modules.json: 100%"
          }
        },
        "6e8e216e073c4ce9977ec8503a26e328": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_b4ba75df8d8240f1bbfa516519f90651",
            "max": 349,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_5f488a901f194ace800c359b2894a63d",
            "value": 349
          }
        },
        "8f0634cbd43e4bb4ab99cb4cd6275920": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8958a0703e7047e7850585f509332d56",
            "placeholder": "​",
            "style": "IPY_MODEL_59edad19fe834c2ca3ee185bb522b698",
            "value": " 349/349 [00:00&lt;00:00, 9.98kB/s]"
          }
        },
        "9a0672593157449f96bc33ecd12e7e93": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "33d910a3e342416fbaf61b59ccffb9b2": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "5c82ef28566f4015ae557d379748aa79": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "b4ba75df8d8240f1bbfa516519f90651": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "5f488a901f194ace800c359b2894a63d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "8958a0703e7047e7850585f509332d56": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "59edad19fe834c2ca3ee185bb522b698": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "837c4dfd878945229500728f7b70d677": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_6112addb7a644a6b9600cbbcc003f26c",
              "IPY_MODEL_9580ef0fb7bb42778936d0dbb7b903ef",
              "IPY_MODEL_04dfde4e3e4e4ea1a11ceae7ab851feb"
            ],
            "layout": "IPY_MODEL_2051f684d50546bab60d465a26311c7c"
          }
        },
        "6112addb7a644a6b9600cbbcc003f26c": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_6c2d050dd91a45a8b8a16aa5ca1dde92",
            "placeholder": "​",
            "style": "IPY_MODEL_f00aaeb5e1bc4c538c6de88e4dff155b",
            "value": "config_sentence_transformers.json: 100%"
          }
        },
        "9580ef0fb7bb42778936d0dbb7b903ef": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_624561237f114f67968740a564ddcb5b",
            "max": 116,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_080e6377d4494d1a8c89adb6a5259f19",
            "value": 116
          }
        },
        "04dfde4e3e4e4ea1a11ceae7ab851feb": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_6a889ec02e944748ad07b12d54c6373b",
            "placeholder": "​",
            "style": "IPY_MODEL_8badc599119f41ffb19b26353c389572",
            "value": " 116/116 [00:00&lt;00:00, 11.6kB/s]"
          }
        },
        "2051f684d50546bab60d465a26311c7c": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "6c2d050dd91a45a8b8a16aa5ca1dde92": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "f00aaeb5e1bc4c538c6de88e4dff155b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "624561237f114f67968740a564ddcb5b": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "080e6377d4494d1a8c89adb6a5259f19": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "6a889ec02e944748ad07b12d54c6373b": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8badc599119f41ffb19b26353c389572": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "8bd75bfdace44a4aa41b53c455027e91": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_739640d54b034d9595acd8b9dfb88468",
              "IPY_MODEL_bef636d678a64721af879bfbee0381f9",
              "IPY_MODEL_595c208686e143bca1000f4672495ab6"
            ],
            "layout": "IPY_MODEL_5c031b63a050466881f83ede11050473"
          }
        },
        "739640d54b034d9595acd8b9dfb88468": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_a00d5be1599c4799ae20a35671648944",
            "placeholder": "​",
            "style": "IPY_MODEL_af361eb4fd164262ab3b097263449d38",
            "value": "README.md: "
          }
        },
        "bef636d678a64721af879bfbee0381f9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_e6009157dd5949139111cbef36bd94e3",
            "max": 1,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_e837c7f9be1c4ffa92b1b287b471e176",
            "value": 1
          }
        },
        "595c208686e143bca1000f4672495ab6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_a02706cda4124d3cafeccc40231cb986",
            "placeholder": "​",
            "style": "IPY_MODEL_c05fa464b2674ba4934439d84a923d98",
            "value": " 10.5k/? [00:00&lt;00:00, 556kB/s]"
          }
        },
        "5c031b63a050466881f83ede11050473": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a00d5be1599c4799ae20a35671648944": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "af361eb4fd164262ab3b097263449d38": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "e6009157dd5949139111cbef36bd94e3": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "20px"
          }
        },
        "e837c7f9be1c4ffa92b1b287b471e176": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "a02706cda4124d3cafeccc40231cb986": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "c05fa464b2674ba4934439d84a923d98": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "e651c67896c2463d8cdeedb53c91c83a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_b6677361c0a94befbff3af410df0e035",
              "IPY_MODEL_7f775542d43e4e06a486c62b0d21abec",
              "IPY_MODEL_8e34a5d35e2e4e41b3251c94c90af894"
            ],
            "layout": "IPY_MODEL_083cba51fe1a4e6091a82253439cf3fb"
          }
        },
        "b6677361c0a94befbff3af410df0e035": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_365cf02d2a664a228851cd4197e23fdc",
            "placeholder": "​",
            "style": "IPY_MODEL_e754b45bc8054f038624b11fd2723e51",
            "value": "sentence_bert_config.json: 100%"
          }
        },
        "7f775542d43e4e06a486c62b0d21abec": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_d8db885614c54be6a8ec55dd4901b99b",
            "max": 53,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_63b8b0a416e644b0a6b93e49637b5335",
            "value": 53
          }
        },
        "8e34a5d35e2e4e41b3251c94c90af894": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_e8b147cc7e2d44a3bf7e5f2bc22e864d",
            "placeholder": "​",
            "style": "IPY_MODEL_a1b1c2153984495883bc7a8cca160620",
            "value": " 53.0/53.0 [00:00&lt;00:00, 5.14kB/s]"
          }
        },
        "083cba51fe1a4e6091a82253439cf3fb": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "365cf02d2a664a228851cd4197e23fdc": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "e754b45bc8054f038624b11fd2723e51": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "d8db885614c54be6a8ec55dd4901b99b": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "63b8b0a416e644b0a6b93e49637b5335": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "e8b147cc7e2d44a3bf7e5f2bc22e864d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a1b1c2153984495883bc7a8cca160620": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "90b60eaf86ab4fe08022b5f390266d7e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_c84b9d3422b7475498d726d1ef896df2",
              "IPY_MODEL_edb81105e1d04cb7b37ec7ce02172741",
              "IPY_MODEL_cc789ea1b02842fcb5aa0ac6b9ca35e6"
            ],
            "layout": "IPY_MODEL_bf0f1be4dace42d59e0c0faa01903a83"
          }
        },
        "c84b9d3422b7475498d726d1ef896df2": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_47803552b2c943968cf25695d34c5680",
            "placeholder": "​",
            "style": "IPY_MODEL_243b880d98b6469aa1c1d50a93d4e3f1",
            "value": "config.json: 100%"
          }
        },
        "edb81105e1d04cb7b37ec7ce02172741": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_bb9a89b7659747d38a2dd6b1816cd76c",
            "max": 612,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_a77c574237a54a51b352218fe8a6c456",
            "value": 612
          }
        },
        "cc789ea1b02842fcb5aa0ac6b9ca35e6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_e334e179c5854cbdac408a14d66dc28f",
            "placeholder": "​",
            "style": "IPY_MODEL_4fa8edb82b5b487bb4ceac33cfd4f3b0",
            "value": " 612/612 [00:00&lt;00:00, 48.1kB/s]"
          }
        },
        "bf0f1be4dace42d59e0c0faa01903a83": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "47803552b2c943968cf25695d34c5680": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "243b880d98b6469aa1c1d50a93d4e3f1": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "bb9a89b7659747d38a2dd6b1816cd76c": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a77c574237a54a51b352218fe8a6c456": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "e334e179c5854cbdac408a14d66dc28f": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "4fa8edb82b5b487bb4ceac33cfd4f3b0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "ac07b97003974be097777185e691fbb0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_952f307b8b7140a2bd706c4267c2c156",
              "IPY_MODEL_a79adac518c44df2a22978f32a5095e9",
              "IPY_MODEL_ed3d00416710410a9ff58a96292bcd28"
            ],
            "layout": "IPY_MODEL_bea74f071b6b4258b467aa12b091d12f"
          }
        },
        "952f307b8b7140a2bd706c4267c2c156": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_0f0da14b43fc4d7985a3ea946515f212",
            "placeholder": "​",
            "style": "IPY_MODEL_77974483806649d6bcd7d7f933a942a6",
            "value": "model.safetensors: 100%"
          }
        },
        "a79adac518c44df2a22978f32a5095e9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_7ad064c654ec4009a66c4f650cfb7f29",
            "max": 90868376,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_e73385d6dbfd4fa2866a5ad39a9bdacb",
            "value": 90868376
          }
        },
        "ed3d00416710410a9ff58a96292bcd28": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_c84aaea9a2264d08a0bc62decd7c4696",
            "placeholder": "​",
            "style": "IPY_MODEL_5544df020a004e15a4c70a707d2933d9",
            "value": " 90.9M/90.9M [00:50&lt;00:00, 1.91MB/s]"
          }
        },
        "bea74f071b6b4258b467aa12b091d12f": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "0f0da14b43fc4d7985a3ea946515f212": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "77974483806649d6bcd7d7f933a942a6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "7ad064c654ec4009a66c4f650cfb7f29": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "e73385d6dbfd4fa2866a5ad39a9bdacb": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "c84aaea9a2264d08a0bc62decd7c4696": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "5544df020a004e15a4c70a707d2933d9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "95358cb5bcc142fca0bc0dabb6cf4594": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_d38bc18df862400fa589552e988208db",
              "IPY_MODEL_76fa6fd5573a4f5db99504b5765f40e4",
              "IPY_MODEL_00dc68164a6145b9a8285314b4097900"
            ],
            "layout": "IPY_MODEL_91b6076aacd0468e97d40df17c907e30"
          }
        },
        "d38bc18df862400fa589552e988208db": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_3078f1c9403e424182f2dab06dd23c9e",
            "placeholder": "​",
            "style": "IPY_MODEL_d1758c960d544f76a49680289632f476",
            "value": "tokenizer_config.json: 100%"
          }
        },
        "76fa6fd5573a4f5db99504b5765f40e4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_bb8fcb60a2b84dc5afe52c6f0feec79e",
            "max": 350,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_396cecc6f8874f32a0b1d3b6e976cc9f",
            "value": 350
          }
        },
        "00dc68164a6145b9a8285314b4097900": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_d0f03df4e5074af5b8b9033461740f91",
            "placeholder": "​",
            "style": "IPY_MODEL_1ff54ca302ae46fa8de150c6d84f7c39",
            "value": " 350/350 [00:00&lt;00:00, 34.2kB/s]"
          }
        },
        "91b6076aacd0468e97d40df17c907e30": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "3078f1c9403e424182f2dab06dd23c9e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d1758c960d544f76a49680289632f476": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "bb8fcb60a2b84dc5afe52c6f0feec79e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "396cecc6f8874f32a0b1d3b6e976cc9f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "d0f03df4e5074af5b8b9033461740f91": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "1ff54ca302ae46fa8de150c6d84f7c39": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "facb372e5c584390aa98a65a98b7eb0a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_92dab5d55231413097ff6f1cc3746765",
              "IPY_MODEL_06a0b0be3c2d4c309cc52ed760827cfb",
              "IPY_MODEL_d45ae0cc9c43447caa9a84b5918432d9"
            ],
            "layout": "IPY_MODEL_131f143bacf54b4589ed6ebdb268e1fd"
          }
        },
        "92dab5d55231413097ff6f1cc3746765": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_9563938f0cfd4de480e55e89f4ccd23c",
            "placeholder": "​",
            "style": "IPY_MODEL_f574220d676a4454b85f50513d9a5f7a",
            "value": "vocab.txt: "
          }
        },
        "06a0b0be3c2d4c309cc52ed760827cfb": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_26b23d486c5e4dae825dab50432aba49",
            "max": 1,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_00ea3cdf7ed742d99234493f36b904ec",
            "value": 1
          }
        },
        "d45ae0cc9c43447caa9a84b5918432d9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_6350bf8025a1420cac4b2b75ea7e75e5",
            "placeholder": "​",
            "style": "IPY_MODEL_89716f5f44c34387b7cb0da6c8acfdf4",
            "value": " 232k/? [00:00&lt;00:00, 2.87MB/s]"
          }
        },
        "131f143bacf54b4589ed6ebdb268e1fd": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "9563938f0cfd4de480e55e89f4ccd23c": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "f574220d676a4454b85f50513d9a5f7a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "26b23d486c5e4dae825dab50432aba49": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "20px"
          }
        },
        "00ea3cdf7ed742d99234493f36b904ec": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "6350bf8025a1420cac4b2b75ea7e75e5": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "89716f5f44c34387b7cb0da6c8acfdf4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "9b2bb53b98b34b33862ed452d90e13a1": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_f7b8d4ebfbe84bba829c4c0990583963",
              "IPY_MODEL_c9f4c247e87042f99e4a12a8127ed405",
              "IPY_MODEL_ef386400977049818374dc03bea47c5a"
            ],
            "layout": "IPY_MODEL_d72ba1cf6d5d4995aa2ebda74ddf9b95"
          }
        },
        "f7b8d4ebfbe84bba829c4c0990583963": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_6e81859f445348ba90eb377926618459",
            "placeholder": "​",
            "style": "IPY_MODEL_135d54a37b724a3e8a3ab8b63be53b28",
            "value": "tokenizer.json: "
          }
        },
        "c9f4c247e87042f99e4a12a8127ed405": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_f280fd61c2944816b3e9667479ea1e61",
            "max": 1,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_03320f13b8254ecfad2ad6c1dda8258d",
            "value": 1
          }
        },
        "ef386400977049818374dc03bea47c5a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_869dabb5392f4aff98e14355fcf18b6d",
            "placeholder": "​",
            "style": "IPY_MODEL_3bea3c426a0b416fb18bbac91675bbc0",
            "value": " 466k/? [00:00&lt;00:00, 3.96MB/s]"
          }
        },
        "d72ba1cf6d5d4995aa2ebda74ddf9b95": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "6e81859f445348ba90eb377926618459": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "135d54a37b724a3e8a3ab8b63be53b28": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "f280fd61c2944816b3e9667479ea1e61": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "20px"
          }
        },
        "03320f13b8254ecfad2ad6c1dda8258d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "869dabb5392f4aff98e14355fcf18b6d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "3bea3c426a0b416fb18bbac91675bbc0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "c1cdb4864b8c41c5a4ffac11484dae16": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_3bd59b7b2bc64e2aab3d3ac1c760413d",
              "IPY_MODEL_ae87c180582247f68164a9686856a07b",
              "IPY_MODEL_90ee94ecd86740a7a16a0c301f33e859"
            ],
            "layout": "IPY_MODEL_d551e73fbb134ffda7f59488db2427bf"
          }
        },
        "3bd59b7b2bc64e2aab3d3ac1c760413d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_4fd2ae09013e40bf9c6ffea03bdc7111",
            "placeholder": "​",
            "style": "IPY_MODEL_ae7590009bc14463ac8938ae9b66fc2c",
            "value": "special_tokens_map.json: 100%"
          }
        },
        "ae87c180582247f68164a9686856a07b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_e56ae92014e546918bd7916255e17ce1",
            "max": 112,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_ee048e812a464635ae4e9f3320a412a6",
            "value": 112
          }
        },
        "90ee94ecd86740a7a16a0c301f33e859": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_5a36b49e872749a3bbd0119fdc4087b0",
            "placeholder": "​",
            "style": "IPY_MODEL_7f0a84666a394906a5b111e793edb175",
            "value": " 112/112 [00:00&lt;00:00, 9.81kB/s]"
          }
        },
        "d551e73fbb134ffda7f59488db2427bf": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "4fd2ae09013e40bf9c6ffea03bdc7111": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ae7590009bc14463ac8938ae9b66fc2c": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "e56ae92014e546918bd7916255e17ce1": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ee048e812a464635ae4e9f3320a412a6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "5a36b49e872749a3bbd0119fdc4087b0": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "7f0a84666a394906a5b111e793edb175": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "390a777355f14d3fb4ee317830bcd651": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_f231ec009b5e4875b1e214e5674a07d7",
              "IPY_MODEL_0e0687ce3b9b4983a7b903cab6461ca4",
              "IPY_MODEL_0ed0d6be33334da9aa4592c1687e5b43"
            ],
            "layout": "IPY_MODEL_8f2384b478234658b3ce0d551a0c91e0"
          }
        },
        "f231ec009b5e4875b1e214e5674a07d7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ab7b5fd477b8417db67b2440f7a1047d",
            "placeholder": "​",
            "style": "IPY_MODEL_4b63f43f5ab74376876c45f5ebf65fb9",
            "value": "config.json: 100%"
          }
        },
        "0e0687ce3b9b4983a7b903cab6461ca4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_1feda95f73014c93a1ee26dbdd020203",
            "max": 190,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_11231d1ed0d44e5587794dad7d5b81c0",
            "value": 190
          }
        },
        "0ed0d6be33334da9aa4592c1687e5b43": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_82bb37cd881542e4b1bf646b8319d4fc",
            "placeholder": "​",
            "style": "IPY_MODEL_ccb9dc54d8624199b660133ea6011e51",
            "value": " 190/190 [00:00&lt;00:00, 17.0kB/s]"
          }
        },
        "8f2384b478234658b3ce0d551a0c91e0": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ab7b5fd477b8417db67b2440f7a1047d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "4b63f43f5ab74376876c45f5ebf65fb9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "1feda95f73014c93a1ee26dbdd020203": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "11231d1ed0d44e5587794dad7d5b81c0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "82bb37cd881542e4b1bf646b8319d4fc": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ccb9dc54d8624199b660133ea6011e51": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "d03d5b9dd8b24318b69b543ad37d1fd4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_830cf1fd79e5453781dc6c0ca9fcbaf5",
              "IPY_MODEL_6f4fce1b24b443aaaed222ca0ca11b64",
              "IPY_MODEL_63c578dd205f493b8c2967684303d7f0"
            ],
            "layout": "IPY_MODEL_a70ea477ceb8437a9142a4d4a16a0d10"
          }
        },
        "830cf1fd79e5453781dc6c0ca9fcbaf5": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_35ec037126ab4d318500b75e23eb9c60",
            "placeholder": "​",
            "style": "IPY_MODEL_eb433b46efb745beb06bcf8a2222879b",
            "value": "Batches: 100%"
          }
        },
        "6f4fce1b24b443aaaed222ca0ca11b64": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_6b6c695cd465489aa1e08493ee5fb1d1",
            "max": 13,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_8be2370b12f648f5ac582e61ae806cd8",
            "value": 13
          }
        },
        "63c578dd205f493b8c2967684303d7f0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_7ff9edd14df345bf9489fdc8f2dd03fe",
            "placeholder": "​",
            "style": "IPY_MODEL_cc9a08733610408b81be5a882112fe5a",
            "value": " 13/13 [01:01&lt;00:00,  4.18s/it]"
          }
        },
        "a70ea477ceb8437a9142a4d4a16a0d10": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "35ec037126ab4d318500b75e23eb9c60": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "eb433b46efb745beb06bcf8a2222879b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "6b6c695cd465489aa1e08493ee5fb1d1": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8be2370b12f648f5ac582e61ae806cd8": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "7ff9edd14df345bf9489fdc8f2dd03fe": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "cc9a08733610408b81be5a882112fe5a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "d16cb3fb9499455f88cb89b3efc0c1c5": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_1670f5066d02400891f787719ec1b86f",
              "IPY_MODEL_444b866babf045869bc004b41bfb604a",
              "IPY_MODEL_51cc30baefac4a94b00c30432c1f012f"
            ],
            "layout": "IPY_MODEL_70fa290b0f2142c58bbd3e38bb1337eb"
          }
        },
        "1670f5066d02400891f787719ec1b86f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_eca118b4e9a44faeb243d0ef6369ea04",
            "placeholder": "​",
            "style": "IPY_MODEL_f743bddb5e6d4832b76d6ada022758d2",
            "value": "Batches: 100%"
          }
        },
        "444b866babf045869bc004b41bfb604a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_18e899bf051f457390875f54986ca345",
            "max": 1,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_534bb57c4abc440b95ef3f59d4b4ecca",
            "value": 1
          }
        },
        "51cc30baefac4a94b00c30432c1f012f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_f4a8e387a07d411f83ce23138a5516e2",
            "placeholder": "​",
            "style": "IPY_MODEL_9d7b50d40d0d4f1ba3523aff1c2ec1b6",
            "value": " 1/1 [00:00&lt;00:00, 20.29it/s]"
          }
        },
        "70fa290b0f2142c58bbd3e38bb1337eb": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "eca118b4e9a44faeb243d0ef6369ea04": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "f743bddb5e6d4832b76d6ada022758d2": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "18e899bf051f457390875f54986ca345": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "534bb57c4abc440b95ef3f59d4b4ecca": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "f4a8e387a07d411f83ce23138a5516e2": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "9d7b50d40d0d4f1ba3523aff1c2ec1b6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "6698907a3cde41d58cd638d315a8964e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_8ea9ba784d5b47a6b217f6e7defd0b1d",
              "IPY_MODEL_2c5423b6cc0e48b99209d800edaffb76",
              "IPY_MODEL_38487f3ff0cb44608847013d8ddd9b51"
            ],
            "layout": "IPY_MODEL_c6eed94d3e9745b7ac9f48f4ffb47c6d"
          }
        },
        "8ea9ba784d5b47a6b217f6e7defd0b1d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8ad56b7bed1c4cafad3e1daccf4e9c97",
            "placeholder": "​",
            "style": "IPY_MODEL_34d68f3c63fb4834b8b6dd0a2ab2913c",
            "value": "Batches: 100%"
          }
        },
        "2c5423b6cc0e48b99209d800edaffb76": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_d9d1dad68ce2458a9321fb6acabd406c",
            "max": 1,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_40b554f8d4244cceb030a04d8343eee4",
            "value": 1
          }
        },
        "38487f3ff0cb44608847013d8ddd9b51": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_5ac5560d9c074be08f8d9b51e8e4375c",
            "placeholder": "​",
            "style": "IPY_MODEL_4d459b47925b42a48938bdf82f2c5386",
            "value": " 1/1 [00:00&lt;00:00, 16.34it/s]"
          }
        },
        "c6eed94d3e9745b7ac9f48f4ffb47c6d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8ad56b7bed1c4cafad3e1daccf4e9c97": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "34d68f3c63fb4834b8b6dd0a2ab2913c": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "d9d1dad68ce2458a9321fb6acabd406c": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "40b554f8d4244cceb030a04d8343eee4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "5ac5560d9c074be08f8d9b51e8e4375c": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "4d459b47925b42a48938bdf82f2c5386": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        }
      }
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# **DATA INGESTION**"
      ],
      "metadata": {
        "id": "23_6tpyDF9bn"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -U langchain-community"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uI0gLZkEKlv1",
        "outputId": "945652c5-3975-4c1b-d51d-e9340ba59d57"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting langchain-community\n",
            "  Downloading langchain_community-0.3.30-py3-none-any.whl.metadata (3.0 kB)\n",
            "Requirement already satisfied: langchain-core<2.0.0,>=0.3.75 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (0.3.76)\n",
            "Requirement already satisfied: langchain<2.0.0,>=0.3.27 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (0.3.27)\n",
            "Requirement already satisfied: SQLAlchemy<3.0.0,>=1.4.0 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (2.0.43)\n",
            "Collecting requests<3.0.0,>=2.32.5 (from langchain-community)\n",
            "  Downloading requests-2.32.5-py3-none-any.whl.metadata (4.9 kB)\n",
            "Requirement already satisfied: PyYAML<7.0.0,>=5.3.0 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (6.0.2)\n",
            "Requirement already satisfied: aiohttp<4.0.0,>=3.8.3 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (3.12.15)\n",
            "Requirement already satisfied: tenacity!=8.4.0,<10.0.0,>=8.1.0 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (8.5.0)\n",
            "Collecting dataclasses-json<0.7.0,>=0.6.7 (from langchain-community)\n",
            "  Downloading dataclasses_json-0.6.7-py3-none-any.whl.metadata (25 kB)\n",
            "Requirement already satisfied: pydantic-settings<3.0.0,>=2.10.1 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (2.10.1)\n",
            "Requirement already satisfied: langsmith<1.0.0,>=0.1.125 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (0.4.28)\n",
            "Requirement already satisfied: httpx-sse<1.0.0,>=0.4.0 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (0.4.1)\n",
            "Requirement already satisfied: numpy>=1.26.2 in /usr/local/lib/python3.12/dist-packages (from langchain-community) (2.0.2)\n",
            "Requirement already satisfied: aiohappyeyeballs>=2.5.0 in /usr/local/lib/python3.12/dist-packages (from aiohttp<4.0.0,>=3.8.3->langchain-community) (2.6.1)\n",
            "Requirement already satisfied: aiosignal>=1.4.0 in /usr/local/lib/python3.12/dist-packages (from aiohttp<4.0.0,>=3.8.3->langchain-community) (1.4.0)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.12/dist-packages (from aiohttp<4.0.0,>=3.8.3->langchain-community) (25.3.0)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.12/dist-packages (from aiohttp<4.0.0,>=3.8.3->langchain-community) (1.7.0)\n",
            "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.12/dist-packages (from aiohttp<4.0.0,>=3.8.3->langchain-community) (6.6.4)\n",
            "Requirement already satisfied: propcache>=0.2.0 in /usr/local/lib/python3.12/dist-packages (from aiohttp<4.0.0,>=3.8.3->langchain-community) (0.3.2)\n",
            "Requirement already satisfied: yarl<2.0,>=1.17.0 in /usr/local/lib/python3.12/dist-packages (from aiohttp<4.0.0,>=3.8.3->langchain-community) (1.20.1)\n",
            "Collecting marshmallow<4.0.0,>=3.18.0 (from dataclasses-json<0.7.0,>=0.6.7->langchain-community)\n",
            "  Downloading marshmallow-3.26.1-py3-none-any.whl.metadata (7.3 kB)\n",
            "Collecting typing-inspect<1,>=0.4.0 (from dataclasses-json<0.7.0,>=0.6.7->langchain-community)\n",
            "  Downloading typing_inspect-0.9.0-py3-none-any.whl.metadata (1.5 kB)\n",
            "Requirement already satisfied: langchain-text-splitters<1.0.0,>=0.3.9 in /usr/local/lib/python3.12/dist-packages (from langchain<2.0.0,>=0.3.27->langchain-community) (0.3.11)\n",
            "Requirement already satisfied: pydantic<3.0.0,>=2.7.4 in /usr/local/lib/python3.12/dist-packages (from langchain<2.0.0,>=0.3.27->langchain-community) (2.11.9)\n",
            "Requirement already satisfied: jsonpatch<2.0,>=1.33 in /usr/local/lib/python3.12/dist-packages (from langchain-core<2.0.0,>=0.3.75->langchain-community) (1.33)\n",
            "Requirement already satisfied: typing-extensions>=4.7 in /usr/local/lib/python3.12/dist-packages (from langchain-core<2.0.0,>=0.3.75->langchain-community) (4.15.0)\n",
            "Requirement already satisfied: packaging>=23.2 in /usr/local/lib/python3.12/dist-packages (from langchain-core<2.0.0,>=0.3.75->langchain-community) (25.0)\n",
            "Requirement already satisfied: httpx<1,>=0.23.0 in /usr/local/lib/python3.12/dist-packages (from langsmith<1.0.0,>=0.1.125->langchain-community) (0.28.1)\n",
            "Requirement already satisfied: orjson>=3.9.14 in /usr/local/lib/python3.12/dist-packages (from langsmith<1.0.0,>=0.1.125->langchain-community) (3.11.3)\n",
            "Requirement already satisfied: requests-toolbelt>=1.0.0 in /usr/local/lib/python3.12/dist-packages (from langsmith<1.0.0,>=0.1.125->langchain-community) (1.0.0)\n",
            "Requirement already satisfied: zstandard>=0.23.0 in /usr/local/lib/python3.12/dist-packages (from langsmith<1.0.0,>=0.1.125->langchain-community) (0.25.0)\n",
            "Requirement already satisfied: python-dotenv>=0.21.0 in /usr/local/lib/python3.12/dist-packages (from pydantic-settings<3.0.0,>=2.10.1->langchain-community) (1.1.1)\n",
            "Requirement already satisfied: typing-inspection>=0.4.0 in /usr/local/lib/python3.12/dist-packages (from pydantic-settings<3.0.0,>=2.10.1->langchain-community) (0.4.1)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests<3.0.0,>=2.32.5->langchain-community) (3.4.3)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests<3.0.0,>=2.32.5->langchain-community) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests<3.0.0,>=2.32.5->langchain-community) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests<3.0.0,>=2.32.5->langchain-community) (2025.8.3)\n",
            "Requirement already satisfied: greenlet>=1 in /usr/local/lib/python3.12/dist-packages (from SQLAlchemy<3.0.0,>=1.4.0->langchain-community) (3.2.4)\n",
            "Requirement already satisfied: anyio in /usr/local/lib/python3.12/dist-packages (from httpx<1,>=0.23.0->langsmith<1.0.0,>=0.1.125->langchain-community) (4.10.0)\n",
            "Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.12/dist-packages (from httpx<1,>=0.23.0->langsmith<1.0.0,>=0.1.125->langchain-community) (1.0.9)\n",
            "Requirement already satisfied: h11>=0.16 in /usr/local/lib/python3.12/dist-packages (from httpcore==1.*->httpx<1,>=0.23.0->langsmith<1.0.0,>=0.1.125->langchain-community) (0.16.0)\n",
            "Requirement already satisfied: jsonpointer>=1.9 in /usr/local/lib/python3.12/dist-packages (from jsonpatch<2.0,>=1.33->langchain-core<2.0.0,>=0.3.75->langchain-community) (3.0.0)\n",
            "Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.12/dist-packages (from pydantic<3.0.0,>=2.7.4->langchain<2.0.0,>=0.3.27->langchain-community) (0.7.0)\n",
            "Requirement already satisfied: pydantic-core==2.33.2 in /usr/local/lib/python3.12/dist-packages (from pydantic<3.0.0,>=2.7.4->langchain<2.0.0,>=0.3.27->langchain-community) (2.33.2)\n",
            "Collecting mypy-extensions>=0.3.0 (from typing-inspect<1,>=0.4.0->dataclasses-json<0.7.0,>=0.6.7->langchain-community)\n",
            "  Downloading mypy_extensions-1.1.0-py3-none-any.whl.metadata (1.1 kB)\n",
            "Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.12/dist-packages (from anyio->httpx<1,>=0.23.0->langsmith<1.0.0,>=0.1.125->langchain-community) (1.3.1)\n",
            "Downloading langchain_community-0.3.30-py3-none-any.whl (2.5 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.5/2.5 MB\u001b[0m \u001b[31m25.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading dataclasses_json-0.6.7-py3-none-any.whl (28 kB)\n",
            "Downloading requests-2.32.5-py3-none-any.whl (64 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m64.7/64.7 kB\u001b[0m \u001b[31m3.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading marshmallow-3.26.1-py3-none-any.whl (50 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m50.9/50.9 kB\u001b[0m \u001b[31m2.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading typing_inspect-0.9.0-py3-none-any.whl (8.8 kB)\n",
            "Downloading mypy_extensions-1.1.0-py3-none-any.whl (5.0 kB)\n",
            "Installing collected packages: requests, mypy-extensions, marshmallow, typing-inspect, dataclasses-json, langchain-community\n",
            "  Attempting uninstall: requests\n",
            "    Found existing installation: requests 2.32.4\n",
            "    Uninstalling requests-2.32.4:\n",
            "      Successfully uninstalled requests-2.32.4\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "google-colab 1.0.0 requires requests==2.32.4, but you have requests 2.32.5 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed dataclasses-json-0.6.7 langchain-community-0.3.30 marshmallow-3.26.1 mypy-extensions-1.1.0 requests-2.32.5 typing-inspect-0.9.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install pypdf"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9sYCtmI4K5vJ",
        "outputId": "a20f1290-5252-4924-c007-cc068c8926e7"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting pypdf\n",
            "  Downloading pypdf-6.1.0-py3-none-any.whl.metadata (7.1 kB)\n",
            "Downloading pypdf-6.1.0-py3-none-any.whl (322 kB)\n",
            "\u001b[?25l   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/322.5 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K   \u001b[91m━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[91m╸\u001b[0m\u001b[90m━━━━━━━━━━━━\u001b[0m \u001b[32m225.3/322.5 kB\u001b[0m \u001b[31m6.6 MB/s\u001b[0m eta \u001b[36m0:00:01\u001b[0m\r\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m322.5/322.5 kB\u001b[0m \u001b[31m5.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: pypdf\n",
            "Successfully installed pypdf-6.1.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "from langchain_community.document_loaders import PyPDFLoader, PyMuPDFLoader\n",
        "from langchain.text_splitter import RecursiveCharacterTextSplitter\n",
        "from pathlib import Path"
      ],
      "metadata": {
        "id": "u_wduT7EY7aZ"
      },
      "execution_count": 15,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "### Read all the pdf's inside the directory\n",
        "def process_all_pdfs(pdf_directory):\n",
        "    \"\"\"Process all PDF files in a directory.\"\"\"\n",
        "    all_documents = []\n",
        "    pdf_dir = Path(pdf_directory)\n",
        "\n",
        "    # Find all PDF files recursively\n",
        "    pdf_files = list(pdf_dir.glob(\"**/*.pdf\"))\n",
        "\n",
        "    print(f\"Found {len(pdf_files)} PDF files to process\")\n",
        "\n",
        "    for pdf_file in pdf_files:\n",
        "        print(f\"\\nProcessing: {pdf_file.name}\")\n",
        "        try:\n",
        "            loader = PyPDFLoader(str(pdf_file))\n",
        "            documents = loader.load()\n",
        "\n",
        "            # Add source information to metadata\n",
        "            for doc in documents:\n",
        "                doc.metadata['source_file'] = pdf_file.name\n",
        "                doc.metadata['file_type'] = 'pdf'\n",
        "\n",
        "            all_documents.extend(documents)\n",
        "            print(f\"  ✓ Loaded {len(documents)} pages\")\n",
        "\n",
        "        except Exception as e:\n",
        "            print(f\"  ✗ Error: {e}\")\n",
        "\n",
        "    print(f\"\\nTotal documents loaded: {len(all_documents)}\")\n",
        "    return all_documents\n",
        "\n",
        "\n",
        "# Process all PDFs in the data directory\n",
        "all_pdf_documents = process_all_pdfs(\"/content/drive/MyDrive/Rag_Data\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "V6nR1XFTZKqF",
        "outputId": "769beb13-f325-4b8f-93b1-bf95ac0ed46a"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Found 6 PDF files to process\n",
            "\n",
            "Processing: s41598-022-12316-z.pdf\n",
            "  ✓ Loaded 8 pages\n",
            "\n",
            "Processing: s12911-022-01951-1.pdf\n",
            "  ✓ Loaded 16 pages\n",
            "\n",
            "Processing: s40620-023-01573-4.pdf\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "\r 17%|█▋        | 1/6 [1:02:48<5:14:03, 3768.60s/it]\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  ✓ Loaded 17 pages\n",
            "\n",
            "Processing: s40537-022-00657-5.pdf\n",
            "  ✓ Loaded 19 pages\n",
            "\n",
            "Processing: e002560.full.pdf\n",
            "  ✓ Loaded 7 pages\n",
            "\n",
            "Processing: 81878.pdf\n",
            "  ✓ Loaded 14 pages\n",
            "\n",
            "Total documents loaded: 81\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "all_pdf_documents[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eCqlLbfZLB8b",
        "outputId": "a3e53359-ba2e-4335-869c-0f96b56a71f1"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 0, 'page_label': '1', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='1\\nVol.:(0123456789)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports\\nMachine learning to predict end \\nstage kidney disease in chronic \\nkidney disease\\nQiong Bai1, Chunyan Su1, Wen Tang1* & Yike Li2*\\nThe purpose of this study was to assess the feasibility of machine learning (ML) in predicting the risk of \\nend-stage kidney disease (ESKD) from patients with chronic kidney disease (CKD). Data were obtained \\nfrom a longitudinal CKD cohort. Predictor variables included patients’ baseline characteristics and \\nroutine blood test results. The outcome of interest was the presence or absence of ESKD by the end \\nof 5 years. Missing data were imputed using multiple imputation. Five ML algorithms, including \\nlogistic regression, naïve Bayes, random forest, decision tree, and K-nearest neighbors were trained \\nand tested using fivefold cross-validation. The performance of each model was compared to that of \\nthe Kidney Failure Risk Equation (KFRE). The dataset contained 748 CKD patients recruited between \\nApril 2006 and March 2008, with the follow-up time of 6.3 ± 2.3 years. ESKD was observed in 70 \\npatients (9.4%). Three ML models, including the logistic regression, naïve Bayes and random forest, \\nshowed equivalent predictability and greater sensitivity compared to the KFRE. The KFRE had the \\nhighest accuracy, specificity, and precision. This study showed the feasibility of ML in evaluating the \\nprognosis of CKD based on easily accessible features. Three ML models with adequate performance \\nand sensitivity scores suggest a potential use for patient screenings. Future studies include external \\nvalidation and improving the models with additional predictor variables.\\nChronic kidney disease (CKD) is a significant healthcare burden that affects billions of individuals  worldwide1,2 \\nand makes a profound impact on global morbidity and  mortality3–5. In the United States, approximately 11% of \\nthe population or 37 million people suffer from CKD that results in an annual Medicare cost of $84  billion6. The \\nprevalence of this disease is estimated at 10.8% in China, affecting about 119.5 million  people7.\\nGradual loss of the kidney function can lead to end stage kidney disease (ESKD) in CKD patients, precipitat-\\ning the need for kidney replacement therapy (KRT). Timely intervention in those CKD patients who have a high \\nrisk of ESKD may not only improve these patients’ quality of life by delaying the disease progression, but also \\nreduce the morbidity, mortality and healthcare costs resulting from  KRT8,9. Because the disease progression is \\ntypically  silent10, a reliable prediction model for risk of ESKD at the early stage of CKD can be clinically essen -\\ntial. Such a model is expected to facilitate physicians in making personalized treatment decisions for high-risk \\npatients, thereby improving the overall prognosis and reducing the economic burden of this disease.\\nA few statistical models were developed to predict the likelihood of ESKD based on certain variables, \\nincluding age, gender, lab results, and most commonly, the estimated glomerular filtration rate (eGFR) and \\n albuminuria11,12. Although some of these models demonstrated adequate predictability in patients of a specific \\nrace, typically  Caucasians13–15, literature on their generalizability in other ethnic groups, such as Chinese, remains \\n scarce13,16. In addition, models based on non-urine variables, such as patients’ baseline characteristics and rou-\\ntine blood tests, have reportedly yield sufficient  performance17,18. Therefore, it may be feasible to predict ESKD \\nwithout urine tests, leading to a simplified model with equivalent reliability.\\nWith the advent of the big data era, new methods became available in developing a predictive model that \\nused to rely on traditional statistics. Machine learning (ML) is a subset of artificial intelligence (AI) that allows \\nthe computer to perform a specific task without explicit instructions. When used in predictive modeling, ML \\nalgorithm can be trained to capture the underlying patterns of the sample data and make predictions about the \\nnew data based on the acquired  information19. Compared to traditional statistics, ML represents more sophis -\\nticated math functions and usually results in better performance in predicting an outcome that is determined \\nOPEN\\n1Department of Nephrology, Peking University Third Hospital, 49 North Garden Rd, Haidian District, \\nBeijing 100191, People’s Republic of China. 2Department of Otolaryngology-Head and Neck Surgery, Bill Wilkerson \\nCenter, Vanderbilt University Medical Center, Nashville, TN, USA.  *email: tanggwen@126.com ; yike.li.1@\\nvumc.org')"
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# **EMBEDDING AND VECTORDATA-BASE**"
      ],
      "metadata": {
        "id": "6gtEYnIGL35B"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "###Libraries\n",
        "### sentence-transformers\n",
        "### faiss cpu\n",
        "### chromadb\n",
        "#installation\n",
        "!pip install sentence-transformers\n",
        "!pip install faiss-cpu\n",
        "!pip install chromadb"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "y6UXHOaJLa8B",
        "outputId": "c0df604a-77e1-4b18-c93d-3d0f39627a1b"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: sentence-transformers in /usr/local/lib/python3.12/dist-packages (5.1.0)\n",
            "Requirement already satisfied: transformers<5.0.0,>=4.41.0 in /usr/local/lib/python3.12/dist-packages (from sentence-transformers) (4.56.1)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.12/dist-packages (from sentence-transformers) (4.67.1)\n",
            "Requirement already satisfied: torch>=1.11.0 in /usr/local/lib/python3.12/dist-packages (from sentence-transformers) (2.8.0+cu126)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.12/dist-packages (from sentence-transformers) (1.6.1)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.12/dist-packages (from sentence-transformers) (1.16.2)\n",
            "Requirement already satisfied: huggingface-hub>=0.20.0 in /usr/local/lib/python3.12/dist-packages (from sentence-transformers) (0.35.0)\n",
            "Requirement already satisfied: Pillow in /usr/local/lib/python3.12/dist-packages (from sentence-transformers) (11.3.0)\n",
            "Requirement already satisfied: typing_extensions>=4.5.0 in /usr/local/lib/python3.12/dist-packages (from sentence-transformers) (4.15.0)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.12/dist-packages (from huggingface-hub>=0.20.0->sentence-transformers) (3.19.1)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.12/dist-packages (from huggingface-hub>=0.20.0->sentence-transformers) (2025.3.0)\n",
            "Requirement already satisfied: packaging>=20.9 in /usr/local/lib/python3.12/dist-packages (from huggingface-hub>=0.20.0->sentence-transformers) (25.0)\n",
            "Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.12/dist-packages (from huggingface-hub>=0.20.0->sentence-transformers) (6.0.2)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.12/dist-packages (from huggingface-hub>=0.20.0->sentence-transformers) (2.32.5)\n",
            "Requirement already satisfied: hf-xet<2.0.0,>=1.1.3 in /usr/local/lib/python3.12/dist-packages (from huggingface-hub>=0.20.0->sentence-transformers) (1.1.10)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (75.2.0)\n",
            "Requirement already satisfied: sympy>=1.13.3 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (1.13.3)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (3.5)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (3.1.6)\n",
            "Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.6.77 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (12.6.77)\n",
            "Requirement already satisfied: nvidia-cuda-runtime-cu12==12.6.77 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (12.6.77)\n",
            "Requirement already satisfied: nvidia-cuda-cupti-cu12==12.6.80 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (12.6.80)\n",
            "Requirement already satisfied: nvidia-cudnn-cu12==9.10.2.21 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (9.10.2.21)\n",
            "Requirement already satisfied: nvidia-cublas-cu12==12.6.4.1 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (12.6.4.1)\n",
            "Requirement already satisfied: nvidia-cufft-cu12==11.3.0.4 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (11.3.0.4)\n",
            "Requirement already satisfied: nvidia-curand-cu12==10.3.7.77 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (10.3.7.77)\n",
            "Requirement already satisfied: nvidia-cusolver-cu12==11.7.1.2 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (11.7.1.2)\n",
            "Requirement already satisfied: nvidia-cusparse-cu12==12.5.4.2 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (12.5.4.2)\n",
            "Requirement already satisfied: nvidia-cusparselt-cu12==0.7.1 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (0.7.1)\n",
            "Requirement already satisfied: nvidia-nccl-cu12==2.27.3 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (2.27.3)\n",
            "Requirement already satisfied: nvidia-nvtx-cu12==12.6.77 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (12.6.77)\n",
            "Requirement already satisfied: nvidia-nvjitlink-cu12==12.6.85 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (12.6.85)\n",
            "Requirement already satisfied: nvidia-cufile-cu12==1.11.1.6 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (1.11.1.6)\n",
            "Requirement already satisfied: triton==3.4.0 in /usr/local/lib/python3.12/dist-packages (from torch>=1.11.0->sentence-transformers) (3.4.0)\n",
            "Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.12/dist-packages (from transformers<5.0.0,>=4.41.0->sentence-transformers) (2.0.2)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.12/dist-packages (from transformers<5.0.0,>=4.41.0->sentence-transformers) (2024.11.6)\n",
            "Requirement already satisfied: tokenizers<=0.23.0,>=0.22.0 in /usr/local/lib/python3.12/dist-packages (from transformers<5.0.0,>=4.41.0->sentence-transformers) (0.22.0)\n",
            "Requirement already satisfied: safetensors>=0.4.3 in /usr/local/lib/python3.12/dist-packages (from transformers<5.0.0,>=4.41.0->sentence-transformers) (0.6.2)\n",
            "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.12/dist-packages (from scikit-learn->sentence-transformers) (1.5.2)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.12/dist-packages (from scikit-learn->sentence-transformers) (3.6.0)\n",
            "Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.12/dist-packages (from sympy>=1.13.3->torch>=1.11.0->sentence-transformers) (1.3.0)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.12/dist-packages (from jinja2->torch>=1.11.0->sentence-transformers) (3.0.2)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests->huggingface-hub>=0.20.0->sentence-transformers) (3.4.3)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests->huggingface-hub>=0.20.0->sentence-transformers) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests->huggingface-hub>=0.20.0->sentence-transformers) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests->huggingface-hub>=0.20.0->sentence-transformers) (2025.8.3)\n",
            "Collecting faiss-cpu\n",
            "  Downloading faiss_cpu-1.12.0-cp312-cp312-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl.metadata (5.1 kB)\n",
            "Requirement already satisfied: numpy<3.0,>=1.25.0 in /usr/local/lib/python3.12/dist-packages (from faiss-cpu) (2.0.2)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.12/dist-packages (from faiss-cpu) (25.0)\n",
            "Downloading faiss_cpu-1.12.0-cp312-cp312-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl (31.4 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m31.4/31.4 MB\u001b[0m \u001b[31m34.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: faiss-cpu\n",
            "Successfully installed faiss-cpu-1.12.0\n",
            "Collecting chromadb\n",
            "  Downloading chromadb-1.1.0-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.2 kB)\n",
            "Requirement already satisfied: build>=1.0.3 in /usr/local/lib/python3.12/dist-packages (from chromadb) (1.3.0)\n",
            "Requirement already satisfied: pydantic>=1.9 in /usr/local/lib/python3.12/dist-packages (from chromadb) (2.11.9)\n",
            "Collecting pybase64>=1.4.1 (from chromadb)\n",
            "  Downloading pybase64-1.4.2-cp312-cp312-manylinux1_x86_64.manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_5_x86_64.whl.metadata (8.7 kB)\n",
            "Requirement already satisfied: uvicorn>=0.18.3 in /usr/local/lib/python3.12/dist-packages (from uvicorn[standard]>=0.18.3->chromadb) (0.35.0)\n",
            "Requirement already satisfied: numpy>=1.22.5 in /usr/local/lib/python3.12/dist-packages (from chromadb) (2.0.2)\n",
            "Collecting posthog<6.0.0,>=2.4.0 (from chromadb)\n",
            "  Downloading posthog-5.4.0-py3-none-any.whl.metadata (5.7 kB)\n",
            "Requirement already satisfied: typing-extensions>=4.5.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (4.15.0)\n",
            "Collecting onnxruntime>=1.14.1 (from chromadb)\n",
            "  Downloading onnxruntime-1.23.0-cp312-cp312-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl.metadata (4.9 kB)\n",
            "Requirement already satisfied: opentelemetry-api>=1.2.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (1.37.0)\n",
            "Collecting opentelemetry-exporter-otlp-proto-grpc>=1.2.0 (from chromadb)\n",
            "  Downloading opentelemetry_exporter_otlp_proto_grpc-1.37.0-py3-none-any.whl.metadata (2.4 kB)\n",
            "Requirement already satisfied: opentelemetry-sdk>=1.2.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (1.37.0)\n",
            "Requirement already satisfied: tokenizers>=0.13.2 in /usr/local/lib/python3.12/dist-packages (from chromadb) (0.22.0)\n",
            "Collecting pypika>=0.48.9 (from chromadb)\n",
            "  Downloading PyPika-0.48.9.tar.gz (67 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m67.3/67.3 kB\u001b[0m \u001b[31m2.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: tqdm>=4.65.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (4.67.1)\n",
            "Requirement already satisfied: overrides>=7.3.1 in /usr/local/lib/python3.12/dist-packages (from chromadb) (7.7.0)\n",
            "Requirement already satisfied: importlib-resources in /usr/local/lib/python3.12/dist-packages (from chromadb) (6.5.2)\n",
            "Requirement already satisfied: grpcio>=1.58.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (1.75.0)\n",
            "Collecting bcrypt>=4.0.1 (from chromadb)\n",
            "  Downloading bcrypt-5.0.0-cp39-abi3-manylinux_2_34_x86_64.whl.metadata (10 kB)\n",
            "Requirement already satisfied: typer>=0.9.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (0.17.4)\n",
            "Collecting kubernetes>=28.1.0 (from chromadb)\n",
            "  Downloading kubernetes-33.1.0-py2.py3-none-any.whl.metadata (1.7 kB)\n",
            "Requirement already satisfied: tenacity>=8.2.3 in /usr/local/lib/python3.12/dist-packages (from chromadb) (8.5.0)\n",
            "Requirement already satisfied: pyyaml>=6.0.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (6.0.2)\n",
            "Collecting mmh3>=4.0.1 (from chromadb)\n",
            "  Downloading mmh3-5.2.0-cp312-cp312-manylinux1_x86_64.manylinux_2_28_x86_64.manylinux_2_5_x86_64.whl.metadata (14 kB)\n",
            "Requirement already satisfied: orjson>=3.9.12 in /usr/local/lib/python3.12/dist-packages (from chromadb) (3.11.3)\n",
            "Requirement already satisfied: httpx>=0.27.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (0.28.1)\n",
            "Requirement already satisfied: rich>=10.11.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (13.9.4)\n",
            "Requirement already satisfied: jsonschema>=4.19.0 in /usr/local/lib/python3.12/dist-packages (from chromadb) (4.25.1)\n",
            "Requirement already satisfied: packaging>=19.1 in /usr/local/lib/python3.12/dist-packages (from build>=1.0.3->chromadb) (25.0)\n",
            "Requirement already satisfied: pyproject_hooks in /usr/local/lib/python3.12/dist-packages (from build>=1.0.3->chromadb) (1.2.0)\n",
            "Requirement already satisfied: anyio in /usr/local/lib/python3.12/dist-packages (from httpx>=0.27.0->chromadb) (4.10.0)\n",
            "Requirement already satisfied: certifi in /usr/local/lib/python3.12/dist-packages (from httpx>=0.27.0->chromadb) (2025.8.3)\n",
            "Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.12/dist-packages (from httpx>=0.27.0->chromadb) (1.0.9)\n",
            "Requirement already satisfied: idna in /usr/local/lib/python3.12/dist-packages (from httpx>=0.27.0->chromadb) (3.10)\n",
            "Requirement already satisfied: h11>=0.16 in /usr/local/lib/python3.12/dist-packages (from httpcore==1.*->httpx>=0.27.0->chromadb) (0.16.0)\n",
            "Requirement already satisfied: attrs>=22.2.0 in /usr/local/lib/python3.12/dist-packages (from jsonschema>=4.19.0->chromadb) (25.3.0)\n",
            "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /usr/local/lib/python3.12/dist-packages (from jsonschema>=4.19.0->chromadb) (2025.9.1)\n",
            "Requirement already satisfied: referencing>=0.28.4 in /usr/local/lib/python3.12/dist-packages (from jsonschema>=4.19.0->chromadb) (0.36.2)\n",
            "Requirement already satisfied: rpds-py>=0.7.1 in /usr/local/lib/python3.12/dist-packages (from jsonschema>=4.19.0->chromadb) (0.27.1)\n",
            "Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.12/dist-packages (from kubernetes>=28.1.0->chromadb) (1.17.0)\n",
            "Requirement already satisfied: python-dateutil>=2.5.3 in /usr/local/lib/python3.12/dist-packages (from kubernetes>=28.1.0->chromadb) (2.9.0.post0)\n",
            "Requirement already satisfied: google-auth>=1.0.1 in /usr/local/lib/python3.12/dist-packages (from kubernetes>=28.1.0->chromadb) (2.38.0)\n",
            "Requirement already satisfied: websocket-client!=0.40.0,!=0.41.*,!=0.42.*,>=0.32.0 in /usr/local/lib/python3.12/dist-packages (from kubernetes>=28.1.0->chromadb) (1.8.0)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.12/dist-packages (from kubernetes>=28.1.0->chromadb) (2.32.5)\n",
            "Requirement already satisfied: requests-oauthlib in /usr/local/lib/python3.12/dist-packages (from kubernetes>=28.1.0->chromadb) (2.0.0)\n",
            "Requirement already satisfied: oauthlib>=3.2.2 in /usr/local/lib/python3.12/dist-packages (from kubernetes>=28.1.0->chromadb) (3.3.1)\n",
            "Requirement already satisfied: urllib3>=1.24.2 in /usr/local/lib/python3.12/dist-packages (from kubernetes>=28.1.0->chromadb) (2.5.0)\n",
            "Collecting durationpy>=0.7 (from kubernetes>=28.1.0->chromadb)\n",
            "  Downloading durationpy-0.10-py3-none-any.whl.metadata (340 bytes)\n",
            "Collecting coloredlogs (from onnxruntime>=1.14.1->chromadb)\n",
            "  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)\n",
            "Requirement already satisfied: flatbuffers in /usr/local/lib/python3.12/dist-packages (from onnxruntime>=1.14.1->chromadb) (25.2.10)\n",
            "Requirement already satisfied: protobuf in /usr/local/lib/python3.12/dist-packages (from onnxruntime>=1.14.1->chromadb) (5.29.5)\n",
            "Requirement already satisfied: sympy in /usr/local/lib/python3.12/dist-packages (from onnxruntime>=1.14.1->chromadb) (1.13.3)\n",
            "Requirement already satisfied: importlib-metadata<8.8.0,>=6.0 in /usr/local/lib/python3.12/dist-packages (from opentelemetry-api>=1.2.0->chromadb) (8.7.0)\n",
            "Requirement already satisfied: googleapis-common-protos~=1.57 in /usr/local/lib/python3.12/dist-packages (from opentelemetry-exporter-otlp-proto-grpc>=1.2.0->chromadb) (1.70.0)\n",
            "Collecting opentelemetry-exporter-otlp-proto-common==1.37.0 (from opentelemetry-exporter-otlp-proto-grpc>=1.2.0->chromadb)\n",
            "  Downloading opentelemetry_exporter_otlp_proto_common-1.37.0-py3-none-any.whl.metadata (1.8 kB)\n",
            "Collecting opentelemetry-proto==1.37.0 (from opentelemetry-exporter-otlp-proto-grpc>=1.2.0->chromadb)\n",
            "  Downloading opentelemetry_proto-1.37.0-py3-none-any.whl.metadata (2.3 kB)\n",
            "Requirement already satisfied: opentelemetry-semantic-conventions==0.58b0 in /usr/local/lib/python3.12/dist-packages (from opentelemetry-sdk>=1.2.0->chromadb) (0.58b0)\n",
            "Collecting backoff>=1.10.0 (from posthog<6.0.0,>=2.4.0->chromadb)\n",
            "  Downloading backoff-2.2.1-py3-none-any.whl.metadata (14 kB)\n",
            "Requirement already satisfied: distro>=1.5.0 in /usr/local/lib/python3.12/dist-packages (from posthog<6.0.0,>=2.4.0->chromadb) (1.9.0)\n",
            "Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.12/dist-packages (from pydantic>=1.9->chromadb) (0.7.0)\n",
            "Requirement already satisfied: pydantic-core==2.33.2 in /usr/local/lib/python3.12/dist-packages (from pydantic>=1.9->chromadb) (2.33.2)\n",
            "Requirement already satisfied: typing-inspection>=0.4.0 in /usr/local/lib/python3.12/dist-packages (from pydantic>=1.9->chromadb) (0.4.1)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.12/dist-packages (from rich>=10.11.0->chromadb) (4.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.12/dist-packages (from rich>=10.11.0->chromadb) (2.19.2)\n",
            "Requirement already satisfied: huggingface-hub<1.0,>=0.16.4 in /usr/local/lib/python3.12/dist-packages (from tokenizers>=0.13.2->chromadb) (0.35.0)\n",
            "Requirement already satisfied: click>=8.0.0 in /usr/local/lib/python3.12/dist-packages (from typer>=0.9.0->chromadb) (8.2.1)\n",
            "Requirement already satisfied: shellingham>=1.3.0 in /usr/local/lib/python3.12/dist-packages (from typer>=0.9.0->chromadb) (1.5.4)\n",
            "Collecting httptools>=0.6.3 (from uvicorn[standard]>=0.18.3->chromadb)\n",
            "  Downloading httptools-0.6.4-cp312-cp312-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.6 kB)\n",
            "Requirement already satisfied: python-dotenv>=0.13 in /usr/local/lib/python3.12/dist-packages (from uvicorn[standard]>=0.18.3->chromadb) (1.1.1)\n",
            "Collecting uvloop>=0.15.1 (from uvicorn[standard]>=0.18.3->chromadb)\n",
            "  Downloading uvloop-0.21.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)\n",
            "Collecting watchfiles>=0.13 (from uvicorn[standard]>=0.18.3->chromadb)\n",
            "  Downloading watchfiles-1.1.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.9 kB)\n",
            "Requirement already satisfied: websockets>=10.4 in /usr/local/lib/python3.12/dist-packages (from uvicorn[standard]>=0.18.3->chromadb) (15.0.1)\n",
            "Requirement already satisfied: cachetools<6.0,>=2.0.0 in /usr/local/lib/python3.12/dist-packages (from google-auth>=1.0.1->kubernetes>=28.1.0->chromadb) (5.5.2)\n",
            "Requirement already satisfied: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.12/dist-packages (from google-auth>=1.0.1->kubernetes>=28.1.0->chromadb) (0.4.2)\n",
            "Requirement already satisfied: rsa<5,>=3.1.4 in /usr/local/lib/python3.12/dist-packages (from google-auth>=1.0.1->kubernetes>=28.1.0->chromadb) (4.9.1)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.12/dist-packages (from huggingface-hub<1.0,>=0.16.4->tokenizers>=0.13.2->chromadb) (3.19.1)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.12/dist-packages (from huggingface-hub<1.0,>=0.16.4->tokenizers>=0.13.2->chromadb) (2025.3.0)\n",
            "Requirement already satisfied: hf-xet<2.0.0,>=1.1.3 in /usr/local/lib/python3.12/dist-packages (from huggingface-hub<1.0,>=0.16.4->tokenizers>=0.13.2->chromadb) (1.1.10)\n",
            "Requirement already satisfied: zipp>=3.20 in /usr/local/lib/python3.12/dist-packages (from importlib-metadata<8.8.0,>=6.0->opentelemetry-api>=1.2.0->chromadb) (3.23.0)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.12/dist-packages (from markdown-it-py>=2.2.0->rich>=10.11.0->chromadb) (0.1.2)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests->kubernetes>=28.1.0->chromadb) (3.4.3)\n",
            "Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.12/dist-packages (from anyio->httpx>=0.27.0->chromadb) (1.3.1)\n",
            "Collecting humanfriendly>=9.1 (from coloredlogs->onnxruntime>=1.14.1->chromadb)\n",
            "  Downloading humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)\n",
            "Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.12/dist-packages (from sympy->onnxruntime>=1.14.1->chromadb) (1.3.0)\n",
            "Requirement already satisfied: pyasn1<0.7.0,>=0.6.1 in /usr/local/lib/python3.12/dist-packages (from pyasn1-modules>=0.2.1->google-auth>=1.0.1->kubernetes>=28.1.0->chromadb) (0.6.1)\n",
            "Downloading chromadb-1.1.0-cp39-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (19.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m19.8/19.8 MB\u001b[0m \u001b[31m83.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading bcrypt-5.0.0-cp39-abi3-manylinux_2_34_x86_64.whl (278 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m278.2/278.2 kB\u001b[0m \u001b[31m14.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading kubernetes-33.1.0-py2.py3-none-any.whl (1.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.9/1.9 MB\u001b[0m \u001b[31m49.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading mmh3-5.2.0-cp312-cp312-manylinux1_x86_64.manylinux_2_28_x86_64.manylinux_2_5_x86_64.whl (103 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m103.3/103.3 kB\u001b[0m \u001b[31m6.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading onnxruntime-1.23.0-cp312-cp312-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl (17.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m17.3/17.3 MB\u001b[0m \u001b[31m82.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading opentelemetry_exporter_otlp_proto_grpc-1.37.0-py3-none-any.whl (19 kB)\n",
            "Downloading opentelemetry_exporter_otlp_proto_common-1.37.0-py3-none-any.whl (18 kB)\n",
            "Downloading opentelemetry_proto-1.37.0-py3-none-any.whl (72 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m72.5/72.5 kB\u001b[0m \u001b[31m4.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading posthog-5.4.0-py3-none-any.whl (105 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m105.4/105.4 kB\u001b[0m \u001b[31m6.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pybase64-1.4.2-cp312-cp312-manylinux1_x86_64.manylinux2014_x86_64.manylinux_2_17_x86_64.manylinux_2_5_x86_64.whl (71 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m71.6/71.6 kB\u001b[0m \u001b[31m4.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading backoff-2.2.1-py3-none-any.whl (15 kB)\n",
            "Downloading durationpy-0.10-py3-none-any.whl (3.9 kB)\n",
            "Downloading httptools-0.6.4-cp312-cp312-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (510 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m510.8/510.8 kB\u001b[0m \u001b[31m21.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading uvloop-0.21.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.7 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m4.7/4.7 MB\u001b[0m \u001b[31m60.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading watchfiles-1.1.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (452 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m452.2/452.2 kB\u001b[0m \u001b[31m18.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m46.0/46.0 kB\u001b[0m \u001b[31m2.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m86.8/86.8 kB\u001b[0m \u001b[31m5.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: pypika\n",
            "  Building wheel for pypika (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pypika: filename=pypika-0.48.9-py2.py3-none-any.whl size=53803 sha256=b6d573ad388846343fac47b373e09fccf26f3c4efbc7fa3229e5739466f208ff\n",
            "  Stored in directory: /root/.cache/pip/wheels/d5/3d/69/8d68d249cd3de2584f226e27fd431d6344f7d70fd856ebd01b\n",
            "Successfully built pypika\n",
            "Installing collected packages: pypika, durationpy, uvloop, pybase64, opentelemetry-proto, mmh3, humanfriendly, httptools, bcrypt, backoff, watchfiles, posthog, opentelemetry-exporter-otlp-proto-common, coloredlogs, onnxruntime, kubernetes, opentelemetry-exporter-otlp-proto-grpc, chromadb\n",
            "Successfully installed backoff-2.2.1 bcrypt-5.0.0 chromadb-1.1.0 coloredlogs-15.0.1 durationpy-0.10 httptools-0.6.4 humanfriendly-10.0 kubernetes-33.1.0 mmh3-5.2.0 onnxruntime-1.23.0 opentelemetry-exporter-otlp-proto-common-1.37.0 opentelemetry-exporter-otlp-proto-grpc-1.37.0 opentelemetry-proto-1.37.0 posthog-5.4.0 pybase64-1.4.2 pypika-0.48.9 uvloop-0.21.0 watchfiles-1.1.0\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "opentelemetry"
                ]
              },
              "id": "5d332040494e467f9110ee8ec78fc0cd"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "### Text splitting get into chunks\n",
        "\n",
        "def split_documents(documents,chunk_size=1000,chunk_overlap=200):\n",
        "    \"\"\"Split documents into smaller chunks for better RAG performance\"\"\"\n",
        "    text_splitter = RecursiveCharacterTextSplitter(\n",
        "        chunk_size=chunk_size,\n",
        "        chunk_overlap=chunk_overlap,\n",
        "        length_function=len,\n",
        "        separators=[\"\\n\\n\", \"\\n\", \" \", \"\"]\n",
        "    )\n",
        "    split_docs = text_splitter.split_documents(documents)\n",
        "    print(f\"Split {len(documents)} documents into {len(split_docs)} chunks\")\n",
        "\n",
        "    # Show example of a chunk\n",
        "    if split_docs:\n",
        "        print(f\"\\nExample chunk:\")\n",
        "        print(f\"Content: {split_docs[0].page_content[:200]}...\")\n",
        "        print(f\"Metadata: {split_docs[0].metadata}\")\n",
        "\n",
        "    return split_docs"
      ],
      "metadata": {
        "id": "6wqHvTR_YaMx"
      },
      "execution_count": 19,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import numpy as np\n",
        "from sentence_transformers import SentenceTransformer\n",
        "import chromadb\n",
        "from chromadb.config import Settings\n",
        "import uuid\n",
        "from typing import List, Dict, Any, Tuple\n",
        "from sklearn.metrics.pairwise import cosine_similarity"
      ],
      "metadata": {
        "id": "bRhrLlDDMlI7"
      },
      "execution_count": 9,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "class EmbeddingManager:\n",
        "  \"\"\" Handle Paper Embedding generation using Sentence Transformer\"\"\"\n",
        "  def __init__(self, model_name: str = \"all-MiniLM-L6-v2\"):\n",
        "    \"\"\"\n",
        "    Initialize embedding manager.\n",
        "\n",
        "    Args:\n",
        "        model_name (str): Hugging face model name for sentence embeddings\n",
        "    \"\"\"\n",
        "    self.model_name = model_name\n",
        "    self.model = None\n",
        "    self._load_model()\n",
        "\n",
        "  def _load_model(self):\n",
        "    \"\"\"\n",
        "    Load sentence transformer model.\n",
        "    \"\"\"\n",
        "    try:\n",
        "      print(f\"Loading Sentence Transformer model: {self.model_name}\")\n",
        "      self.model = SentenceTransformer(self.model_name)\n",
        "      print(f\"Model loaded successfully. Embedding dimension: {self.model.get_sentence_embedding_dimension()}\")\n",
        "    except Exception as e:\n",
        "      print(f\"Error loading Sentence Transformer model: {e}\")\n",
        "      raise\n",
        "\n",
        "  def generate_embeddings(self, texts: List[str]) -> List[np.ndarray]:\n",
        "    \"\"\"\n",
        "    Generate embeddings for a list of texts.\n",
        "\n",
        "    Args:\n",
        "        texts (List[str]): List of texts to generate embeddings for.\n",
        "\n",
        "    Returns:\n",
        "        List[np.ndarray]: List of embeddings.\n",
        "    \"\"\"\n",
        "    if not self.model:\n",
        "      raise ValueError(\"Model not loaded. Call load_model() first.\")\n",
        "\n",
        "    print(f\"Generating embeddings for {len(texts)} texts....\")\n",
        "    embeddings = self.model.encode(texts, show_progress_bar= True)\n",
        "    print(f\"Embeddings generated successfully. Shape: {embeddings.shape}\")\n",
        "    return embeddings\n",
        "\n",
        "#Initialize the embeddings\n",
        "embedding_manager = EmbeddingManager()\n",
        "embedding_manager"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 545,
          "referenced_widgets": [
            "51a596dae02a4ad1b52926bdd6fa3187",
            "cb597df2a29b4dadb8ce67ef817cc413",
            "6e8e216e073c4ce9977ec8503a26e328",
            "8f0634cbd43e4bb4ab99cb4cd6275920",
            "9a0672593157449f96bc33ecd12e7e93",
            "33d910a3e342416fbaf61b59ccffb9b2",
            "5c82ef28566f4015ae557d379748aa79",
            "b4ba75df8d8240f1bbfa516519f90651",
            "5f488a901f194ace800c359b2894a63d",
            "8958a0703e7047e7850585f509332d56",
            "59edad19fe834c2ca3ee185bb522b698",
            "837c4dfd878945229500728f7b70d677",
            "6112addb7a644a6b9600cbbcc003f26c",
            "9580ef0fb7bb42778936d0dbb7b903ef",
            "04dfde4e3e4e4ea1a11ceae7ab851feb",
            "2051f684d50546bab60d465a26311c7c",
            "6c2d050dd91a45a8b8a16aa5ca1dde92",
            "f00aaeb5e1bc4c538c6de88e4dff155b",
            "624561237f114f67968740a564ddcb5b",
            "080e6377d4494d1a8c89adb6a5259f19",
            "6a889ec02e944748ad07b12d54c6373b",
            "8badc599119f41ffb19b26353c389572",
            "8bd75bfdace44a4aa41b53c455027e91",
            "739640d54b034d9595acd8b9dfb88468",
            "bef636d678a64721af879bfbee0381f9",
            "595c208686e143bca1000f4672495ab6",
            "5c031b63a050466881f83ede11050473",
            "a00d5be1599c4799ae20a35671648944",
            "af361eb4fd164262ab3b097263449d38",
            "e6009157dd5949139111cbef36bd94e3",
            "e837c7f9be1c4ffa92b1b287b471e176",
            "a02706cda4124d3cafeccc40231cb986",
            "c05fa464b2674ba4934439d84a923d98",
            "e651c67896c2463d8cdeedb53c91c83a",
            "b6677361c0a94befbff3af410df0e035",
            "7f775542d43e4e06a486c62b0d21abec",
            "8e34a5d35e2e4e41b3251c94c90af894",
            "083cba51fe1a4e6091a82253439cf3fb",
            "365cf02d2a664a228851cd4197e23fdc",
            "e754b45bc8054f038624b11fd2723e51",
            "d8db885614c54be6a8ec55dd4901b99b",
            "63b8b0a416e644b0a6b93e49637b5335",
            "e8b147cc7e2d44a3bf7e5f2bc22e864d",
            "a1b1c2153984495883bc7a8cca160620",
            "90b60eaf86ab4fe08022b5f390266d7e",
            "c84b9d3422b7475498d726d1ef896df2",
            "edb81105e1d04cb7b37ec7ce02172741",
            "cc789ea1b02842fcb5aa0ac6b9ca35e6",
            "bf0f1be4dace42d59e0c0faa01903a83",
            "47803552b2c943968cf25695d34c5680",
            "243b880d98b6469aa1c1d50a93d4e3f1",
            "bb9a89b7659747d38a2dd6b1816cd76c",
            "a77c574237a54a51b352218fe8a6c456",
            "e334e179c5854cbdac408a14d66dc28f",
            "4fa8edb82b5b487bb4ceac33cfd4f3b0",
            "ac07b97003974be097777185e691fbb0",
            "952f307b8b7140a2bd706c4267c2c156",
            "a79adac518c44df2a22978f32a5095e9",
            "ed3d00416710410a9ff58a96292bcd28",
            "bea74f071b6b4258b467aa12b091d12f",
            "0f0da14b43fc4d7985a3ea946515f212",
            "77974483806649d6bcd7d7f933a942a6",
            "7ad064c654ec4009a66c4f650cfb7f29",
            "e73385d6dbfd4fa2866a5ad39a9bdacb",
            "c84aaea9a2264d08a0bc62decd7c4696",
            "5544df020a004e15a4c70a707d2933d9",
            "95358cb5bcc142fca0bc0dabb6cf4594",
            "d38bc18df862400fa589552e988208db",
            "76fa6fd5573a4f5db99504b5765f40e4",
            "00dc68164a6145b9a8285314b4097900",
            "91b6076aacd0468e97d40df17c907e30",
            "3078f1c9403e424182f2dab06dd23c9e",
            "d1758c960d544f76a49680289632f476",
            "bb8fcb60a2b84dc5afe52c6f0feec79e",
            "396cecc6f8874f32a0b1d3b6e976cc9f",
            "d0f03df4e5074af5b8b9033461740f91",
            "1ff54ca302ae46fa8de150c6d84f7c39",
            "facb372e5c584390aa98a65a98b7eb0a",
            "92dab5d55231413097ff6f1cc3746765",
            "06a0b0be3c2d4c309cc52ed760827cfb",
            "d45ae0cc9c43447caa9a84b5918432d9",
            "131f143bacf54b4589ed6ebdb268e1fd",
            "9563938f0cfd4de480e55e89f4ccd23c",
            "f574220d676a4454b85f50513d9a5f7a",
            "26b23d486c5e4dae825dab50432aba49",
            "00ea3cdf7ed742d99234493f36b904ec",
            "6350bf8025a1420cac4b2b75ea7e75e5",
            "89716f5f44c34387b7cb0da6c8acfdf4",
            "9b2bb53b98b34b33862ed452d90e13a1",
            "f7b8d4ebfbe84bba829c4c0990583963",
            "c9f4c247e87042f99e4a12a8127ed405",
            "ef386400977049818374dc03bea47c5a",
            "d72ba1cf6d5d4995aa2ebda74ddf9b95",
            "6e81859f445348ba90eb377926618459",
            "135d54a37b724a3e8a3ab8b63be53b28",
            "f280fd61c2944816b3e9667479ea1e61",
            "03320f13b8254ecfad2ad6c1dda8258d",
            "869dabb5392f4aff98e14355fcf18b6d",
            "3bea3c426a0b416fb18bbac91675bbc0",
            "c1cdb4864b8c41c5a4ffac11484dae16",
            "3bd59b7b2bc64e2aab3d3ac1c760413d",
            "ae87c180582247f68164a9686856a07b",
            "90ee94ecd86740a7a16a0c301f33e859",
            "d551e73fbb134ffda7f59488db2427bf",
            "4fd2ae09013e40bf9c6ffea03bdc7111",
            "ae7590009bc14463ac8938ae9b66fc2c",
            "e56ae92014e546918bd7916255e17ce1",
            "ee048e812a464635ae4e9f3320a412a6",
            "5a36b49e872749a3bbd0119fdc4087b0",
            "7f0a84666a394906a5b111e793edb175",
            "390a777355f14d3fb4ee317830bcd651",
            "f231ec009b5e4875b1e214e5674a07d7",
            "0e0687ce3b9b4983a7b903cab6461ca4",
            "0ed0d6be33334da9aa4592c1687e5b43",
            "8f2384b478234658b3ce0d551a0c91e0",
            "ab7b5fd477b8417db67b2440f7a1047d",
            "4b63f43f5ab74376876c45f5ebf65fb9",
            "1feda95f73014c93a1ee26dbdd020203",
            "11231d1ed0d44e5587794dad7d5b81c0",
            "82bb37cd881542e4b1bf646b8319d4fc",
            "ccb9dc54d8624199b660133ea6011e51"
          ]
        },
        "id": "GyNcdlU0NITE",
        "outputId": "2f2e32e7-15b4-46ff-cff3-be009258be6a"
      },
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Loading Sentence Transformer model: all-MiniLM-L6-v2\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.12/dist-packages/huggingface_hub/utils/_auth.py:94: UserWarning: \n",
            "The secret `HF_TOKEN` does not exist in your Colab secrets.\n",
            "To authenticate with the Hugging Face Hub, create a token in your settings tab (https://huggingface.co/settings/tokens), set it as secret in your Google Colab and restart your session.\n",
            "You will be able to reuse this secret in all of your notebooks.\n",
            "Please note that authentication is recommended but still optional to access public models or datasets.\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "modules.json:   0%|          | 0.00/349 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "51a596dae02a4ad1b52926bdd6fa3187"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "config_sentence_transformers.json:   0%|          | 0.00/116 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "837c4dfd878945229500728f7b70d677"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "README.md: 0.00B [00:00, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "8bd75bfdace44a4aa41b53c455027e91"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "sentence_bert_config.json:   0%|          | 0.00/53.0 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "e651c67896c2463d8cdeedb53c91c83a"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "config.json:   0%|          | 0.00/612 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "90b60eaf86ab4fe08022b5f390266d7e"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "model.safetensors:   0%|          | 0.00/90.9M [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "ac07b97003974be097777185e691fbb0"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "tokenizer_config.json:   0%|          | 0.00/350 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "95358cb5bcc142fca0bc0dabb6cf4594"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "vocab.txt: 0.00B [00:00, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "facb372e5c584390aa98a65a98b7eb0a"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "tokenizer.json: 0.00B [00:00, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "9b2bb53b98b34b33862ed452d90e13a1"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "special_tokens_map.json:   0%|          | 0.00/112 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "c1cdb4864b8c41c5a4ffac11484dae16"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "config.json:   0%|          | 0.00/190 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "390a777355f14d3fb4ee317830bcd651"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Model loaded successfully. Embedding dimension: 384\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<__main__.EmbeddingManager at 0x7c94ea4573e0>"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "class VectorStore:\n",
        "    \"\"\"Manages document embeddings in a ChromaDB vector store\"\"\"\n",
        "\n",
        "    def __init__(self, collection_name: str = \"pdf_documents\", persist_directory: str = \"/content/drive/MyDrive/Rag_Data/vector_store\"):\n",
        "        \"\"\"\n",
        "        Initialize the vector store\n",
        "\n",
        "        Args:\n",
        "            collection_name: Name of the ChromaDB collection\n",
        "            persist_directory: Directory to persist the vector store\n",
        "        \"\"\"\n",
        "        self.collection_name = collection_name\n",
        "        self.persist_directory = persist_directory\n",
        "        self.client = None\n",
        "        self.collection = None\n",
        "        self._initialize_store()\n",
        "\n",
        "    def _initialize_store(self):\n",
        "        \"\"\"Initialize ChromaDB client and collection\"\"\"\n",
        "        try:\n",
        "            # Create persistent ChromaDB client\n",
        "            os.makedirs(self.persist_directory, exist_ok=True)\n",
        "            self.client = chromadb.PersistentClient(path=self.persist_directory)\n",
        "\n",
        "            # Get or create collection\n",
        "            self.collection = self.client.get_or_create_collection(\n",
        "                name=self.collection_name,\n",
        "                metadata={\"description\": \"PDF document embeddings for RAG\"}\n",
        "            )\n",
        "            print(f\"Vector store initialized. Collection: {self.collection_name}\")\n",
        "            print(f\"Existing documents in collection: {self.collection.count()}\")\n",
        "\n",
        "        except Exception as e:\n",
        "            print(f\"Error initializing vector store: {e}\")\n",
        "            raise\n",
        "\n",
        "    def add_documents(self, documents: List[Any], embeddings: np.ndarray):\n",
        "        \"\"\"\n",
        "        Add documents and their embeddings to the vector store\n",
        "\n",
        "        Args:\n",
        "            documents: List of LangChain documents\n",
        "            embeddings: Corresponding embeddings for the documents\n",
        "        \"\"\"\n",
        "        if len(documents) != len(embeddings):\n",
        "            raise ValueError(\"Number of documents must match number of embeddings\")\n",
        "\n",
        "        print(f\"Adding {len(documents)} documents to vector store...\")\n",
        "        # Prepare data for ChromaDB\n",
        "        ids = []\n",
        "        metadatas = []\n",
        "        documents_text = []\n",
        "        embeddings_list = []\n",
        "\n",
        "        for i, (doc, embedding) in enumerate(zip(documents, embeddings)):\n",
        "            # Generate unique ID\n",
        "            doc_id = f\"doc_{uuid.uuid4().hex[:8]}_{i}\"\n",
        "            ids.append(doc_id)\n",
        "\n",
        "            # Prepare metadata\n",
        "            metadata = dict(doc.metadata)\n",
        "            metadata['doc_index'] = i\n",
        "            metadata['content_length'] = len(doc.page_content)\n",
        "            metadatas.append(metadata)\n",
        "\n",
        "            # Document content\n",
        "            documents_text.append(doc.page_content)\n",
        "\n",
        "            # Embedding\n",
        "            embeddings_list.append(embedding.tolist())\n",
        "\n",
        "        # Add to collection\n",
        "        try:\n",
        "            self.collection.add(\n",
        "                ids=ids,\n",
        "                embeddings=embeddings_list,\n",
        "                metadatas=metadatas,\n",
        "                 documents=documents_text\n",
        "            )\n",
        "            print(f\"Successfully added {len(documents)} documents to vector store\")\n",
        "            print(f\"Total documents in collection: {self.collection.count()}\")\n",
        "\n",
        "        except Exception as e:\n",
        "            print(f\"Error adding documents to vector store: {e}\")\n",
        "            raise\n",
        "\n",
        "vectorstore=VectorStore()\n",
        "vectorstore"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BKBnbpuiaelX",
        "outputId": "6a8d328f-9177-49a6-e6c9-7b40fff3c832"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Vector store initialized. Collection: pdf_documents\n",
            "Existing documents in collection: 0\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<__main__.VectorStore at 0x7c94e90dbf50>"
            ]
          },
          "metadata": {},
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "chunks=split_documents(all_pdf_documents)\n",
        "chunks"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "v-kblyheXJWm",
        "outputId": "29942630-97e5-4905-9320-956d01166bbc"
      },
      "execution_count": 23,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Split 81 documents into 413 chunks\n",
            "\n",
            "Example chunk:\n",
            "Content: 1\n",
            "Vol.:(0123456789)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\n",
            "www.nature.com/scientificreports\n",
            "Machine learning to predict end \n",
            "stage kidney disease in c...\n",
            "Metadata: {'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 0, 'page_label': '1', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 0, 'page_label': '1', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='1\\nVol.:(0123456789)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports\\nMachine learning to predict end \\nstage kidney disease in chronic \\nkidney disease\\nQiong Bai1, Chunyan Su1, Wen Tang1* & Yike Li2*\\nThe purpose of this study was to assess the feasibility of machine learning (ML) in predicting the risk of \\nend-stage kidney disease (ESKD) from patients with chronic kidney disease (CKD). Data were obtained \\nfrom a longitudinal CKD cohort. Predictor variables included patients’ baseline characteristics and \\nroutine blood test results. The outcome of interest was the presence or absence of ESKD by the end \\nof 5 years. Missing data were imputed using multiple imputation. Five ML algorithms, including \\nlogistic regression, naïve Bayes, random forest, decision tree, and K-nearest neighbors were trained \\nand tested using fivefold cross-validation. The performance of each model was compared to that of'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 0, 'page_label': '1', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='and tested using fivefold cross-validation. The performance of each model was compared to that of \\nthe Kidney Failure Risk Equation (KFRE). The dataset contained 748 CKD patients recruited between \\nApril 2006 and March 2008, with the follow-up time of 6.3 ± 2.3 years. ESKD was observed in 70 \\npatients (9.4%). Three ML models, including the logistic regression, naïve Bayes and random forest, \\nshowed equivalent predictability and greater sensitivity compared to the KFRE. The KFRE had the \\nhighest accuracy, specificity, and precision. This study showed the feasibility of ML in evaluating the \\nprognosis of CKD based on easily accessible features. Three ML models with adequate performance \\nand sensitivity scores suggest a potential use for patient screenings. Future studies include external \\nvalidation and improving the models with additional predictor variables.\\nChronic kidney disease (CKD) is a significant healthcare burden that affects billions of individuals  worldwide1,2'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 0, 'page_label': '1', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='validation and improving the models with additional predictor variables.\\nChronic kidney disease (CKD) is a significant healthcare burden that affects billions of individuals  worldwide1,2 \\nand makes a profound impact on global morbidity and  mortality3–5. In the United States, approximately 11% of \\nthe population or 37 million people suffer from CKD that results in an annual Medicare cost of $84  billion6. The \\nprevalence of this disease is estimated at 10.8% in China, affecting about 119.5 million  people7.\\nGradual loss of the kidney function can lead to end stage kidney disease (ESKD) in CKD patients, precipitat-\\ning the need for kidney replacement therapy (KRT). Timely intervention in those CKD patients who have a high \\nrisk of ESKD may not only improve these patients’ quality of life by delaying the disease progression, but also \\nreduce the morbidity, mortality and healthcare costs resulting from  KRT8,9. Because the disease progression is'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 0, 'page_label': '1', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='reduce the morbidity, mortality and healthcare costs resulting from  KRT8,9. Because the disease progression is \\ntypically  silent10, a reliable prediction model for risk of ESKD at the early stage of CKD can be clinically essen -\\ntial. Such a model is expected to facilitate physicians in making personalized treatment decisions for high-risk \\npatients, thereby improving the overall prognosis and reducing the economic burden of this disease.\\nA few statistical models were developed to predict the likelihood of ESKD based on certain variables, \\nincluding age, gender, lab results, and most commonly, the estimated glomerular filtration rate (eGFR) and \\n albuminuria11,12. Although some of these models demonstrated adequate predictability in patients of a specific \\nrace, typically  Caucasians13–15, literature on their generalizability in other ethnic groups, such as Chinese, remains'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 0, 'page_label': '1', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='race, typically  Caucasians13–15, literature on their generalizability in other ethnic groups, such as Chinese, remains \\n scarce13,16. In addition, models based on non-urine variables, such as patients’ baseline characteristics and rou-\\ntine blood tests, have reportedly yield sufficient  performance17,18. Therefore, it may be feasible to predict ESKD \\nwithout urine tests, leading to a simplified model with equivalent reliability.\\nWith the advent of the big data era, new methods became available in developing a predictive model that \\nused to rely on traditional statistics. Machine learning (ML) is a subset of artificial intelligence (AI) that allows \\nthe computer to perform a specific task without explicit instructions. When used in predictive modeling, ML \\nalgorithm can be trained to capture the underlying patterns of the sample data and make predictions about the \\nnew data based on the acquired  information19. Compared to traditional statistics, ML represents more sophis -'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 0, 'page_label': '1', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='new data based on the acquired  information19. Compared to traditional statistics, ML represents more sophis -\\nticated math functions and usually results in better performance in predicting an outcome that is determined \\nOPEN\\n1Department of Nephrology, Peking University Third Hospital, 49 North Garden Rd, Haidian District, \\nBeijing 100191, People’s Republic of China. 2Department of Otolaryngology-Head and Neck Surgery, Bill Wilkerson \\nCenter, Vanderbilt University Medical Center, Nashville, TN, USA.  *email: tanggwen@126.com ; yike.li.1@\\nvumc.org'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 1, 'page_label': '2', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='2\\nVol:.(1234567890)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports/\\nby a large set of variables with non-linear, complex  interactions20. ML has recently been applied in numerous \\nstudies and demonstrated high level of performance that surpassed traditional statistics and even  humans20–23.\\nThis article presents a proof-of-concept study with the major goal to establish ML models for predicting \\nthe risk of ESKD on a Chinese CKD dataset. The ML models were trained and tested based on easily obtain-\\nable variables, including the baseline characteristics and routine blood tests. Results obtained from this study \\nsuggest not only the feasibility of ML models in performing this clinically critical task, but also the potential in \\nfacilitating personalized medicine.\\nMaterials and methods\\nStudy population. The data used for this retrospective work were obtained from a longitudinal cohort'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 1, 'page_label': '2', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='facilitating personalized medicine.\\nMaterials and methods\\nStudy population. The data used for this retrospective work were obtained from a longitudinal cohort \\npreviously enrolled in an observational  study24,25. The major inclusion criteria for the cohort were adult CKD \\npatients (≥ 18\\xa0years old) with stable kidney functions for at least three months prior to recruitment. Patients \\nwere excluded if they had one or more of the following situations: (1) history of KRT in any form, including \\nhemodialysis, peritoneal dialysis or kidney transplantation; (2) any other existing condition deemed physically \\nunstable, including life expectancy < 6\\xa0months, acute heart failure, and advanced liver disease; (3) any pre-exist-\\ning malignancy. All patients were recruited from the CKD management clinic of Peking University Third Hos-\\npital between April 2006 and March 2008. Written informed consent was obtained from all patients. They were'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 1, 'page_label': '2', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='pital between April 2006 and March 2008. Written informed consent was obtained from all patients. They were \\ntreated according to routine clinical practice determined by the experienced nephrologists and observed until \\nDecember  31st, 2015. Detailed information regarding patient recruitment and management protocol has been \\ndescribed in a previous  publication24.\\nData acquisition. Patient characteristics included age, gender, education level, marriage status, and insur-\\nance status. Medical history comprised history of smoking, history of alcohol consumption, presence of each \\ncomorbid condition—diabetes, cardiovascular disease and hypertension. Clinical parameters contained body \\nmass index (BMI), systolic pressure and diastolic pressure. Blood tests consisted of serum creatinine, uric acid, \\nblood urea nitrogen, white blood cell count, hemoglobin, platelets count, alanine aminotransferase (ALT),'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 1, 'page_label': '2', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='blood urea nitrogen, white blood cell count, hemoglobin, platelets count, alanine aminotransferase (ALT), \\naspartate aminotransferase (AST), total protein, albumin, alkaline phosphatase (ALP), high-density lipoprotein, \\nlow-density lipoprotein, triglycerides, total cholesterol, calcium, phosphorus, potassium, sodium, chloride, and \\nbicarbonate. The estimated glomerular filtration rate and type of primary kidney disease were also used as pre-\\ndictors.\\nAll baseline variables were obtained at the time of subject enrollment. The primary study end point was \\nkidney failure which necessitated the use of any KRT. Subjects with the outcome of kidney failure were labeled \\nas ESKD+, and the rest ESKD−. Patients who died before reaching the study end point or lost to follow up were \\ndiscarded. Patients who developed ESKD after five years were labeled as ESKD−.\\nData preprocessing. All categorical variables, such as insurance status, education, and primary disease,'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 1, 'page_label': '2', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='discarded. Patients who developed ESKD after five years were labeled as ESKD−.\\nData preprocessing. All categorical variables, such as insurance status, education, and primary disease, \\nwere encoded using the one-hot approach. Any variable was removed from model development if the missing \\nvalues were greater than 50%. Missing data were handled using multiple imputation with five times of repetition, \\nleading to five slightly different imputed datasets where each of the missing values was randomly sampled from \\ntheir predictive distribution based on the observed data. On each imputed set, all models were trained and tested \\nusing a fivefold cross validation method. To minimize selection bias, subject assignment to train/test folds was \\nkept consistent across all imputed sets. Data were split in a stratified fashion to ensure the same distribution of \\nthe outcome classes (ESKD+ vs. ESKD−) in each subset as the entire set.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 1, 'page_label': '2', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='kept consistent across all imputed sets. Data were split in a stratified fashion to ensure the same distribution of \\nthe outcome classes (ESKD+ vs. ESKD−) in each subset as the entire set.\\nModel development. The model was trained to perform a binary classification task with the goal of gen-\\nerating the probability of ESKD+ based on the given features. Five ML algorithms were employed in this study, \\nincluding logistic regression, naïve Bayes, random forest, decision tree, and K-nearest neighbors. Grid search \\nwas performed to obtain the best hyperparameter combination for each algorithm.\\nAssessment of model performance. The performance of a classifiers was measured using accuracy, pre-\\ncision, recall, specificity, F1 score and area under the curve (AUC), as recommended by guidelines for results \\nreporting of clinical prediction  models26. All classifiers developed in this study were further compared with the'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 1, 'page_label': '2', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='reporting of clinical prediction  models26. All classifiers developed in this study were further compared with the \\nKidney Failure Risk Equation (KFRE), which estimates the 5-year risk of ESKD based on patient’s age, gender, \\nand  eGFR12. The KFRE is currently the most widely used model in predicting CKD progression to ESKD. The \\nreported outcome of a model represented the average performance of 5 test folds over all imputed sets.\\nStatistical analysis. Basic descriptive statistics were applied as deemed appropriate. Results are expressed \\nas frequencies and percentages for categorical variables; the mean ± standard deviation for continuous, normally \\ndistributed variables; and the median (interquartile range) for continuous variables that were not normally dis-\\ntributed. Patient characteristics were compared between the original dataset and the imputed sets using one-way \\nanalysis of variance (ANOV A). The AUC of each model was measured using the predicted probability. The opti-'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 1, 'page_label': '2', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='analysis of variance (ANOV A). The AUC of each model was measured using the predicted probability. The opti-\\nmal threshold of a classifier was determined based on the receiver operating characteristic (ROC) curve at the \\npoint with minimal distance to the upper left corner. For each ML model, this threshold was obtained during the \\ntraining process and applied unchangeably to the test set. For the KFRE, the threshold was set at a default value \\nof 0.5. Model development, performance evaluation and data analyses were all performed using  Python27. The \\nalpha level was set at 0.05.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='3\\nVol.:(0123456789)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports/\\nEthical approval. This research was conducted ethically in accordance with the World Medical Association \\nDeclaration of Helsinki. The study protocol has been approved by the Peking University Third Hospital Medical \\nScience Research Ethics Committee on human research (No. M2020132).\\nResults\\nCohort characteristics. The dataset contained a total of 748 subjects with the follow-up duration of \\n6.3 ± 2.3\\xa0years. The baseline characteristics are summarized in Table\\xa01. Most patients were in stage 2 (24.5%) or \\n3 (47.1%) CKD at baseline. ESKD was observed in 70 patients (9.4%), all of whom subsequently received KRT, \\nincluding hemodialysis in 49 patients, peritoneal dialysis in 17 and kidney transplantation in 4.\\nModel performance. Details of the five imputed sets are provided in the supplemental materials. There was'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='Model performance. Details of the five imputed sets are provided in the supplemental materials. There was \\nno significant difference between the imputed sets and the original dataset in each variable where missing data \\nwere replaced by imputed values. The hyperparameter settings for each classifier are displayed in Table\\xa0 2. The \\nbest overall performance, as measured by the AUC score, was achieved by the random forest algorithm (0.81, see \\nTable\\xa03). Nonetheless, this score and its 95% confidence interval had overlap with those of the other three mod-\\nels, including the logistic regression, naïve Bayes, and the KFRE (Fig.\\xa01). Interestingly, the KFRE model that was \\nbased on 3 simple variables, demonstrated not only a comparable AUC score but also the highest accuracy, speci-\\nficity, and precision. At the default threshold, however, the KFRE was one of the least sensitive models (47%).\\nDiscussion'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='ficity, and precision. At the default threshold, however, the KFRE was one of the least sensitive models (47%).\\nDiscussion\\nWith extensive utilization of electronic health record and recent progress in ML research, AI is expanding its \\nimpact on healthcare and has gradually changed the way clinicians pursue for problem-solving 28. Instead of \\nadopting a theory-driven strategy that requires a preformed hypothesis from prior knowledge, training an ML \\nmodel typically follows a data-driven approach that allows the model to learn from experience alone. Specifically, \\nthe model improves its performance iteratively on a training set by comparing the predictions to the ground \\ntruths and adjusting model parameters so as to minimize the distance between the predictions and the truths. In \\nnephrology, ML has demonstrated promising performances in predicting acute kidney injury or time to allograft'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='nephrology, ML has demonstrated promising performances in predicting acute kidney injury or time to allograft  \\nloss from clinical  features29,30, recognizing specific patterns in pathology  slides31,32, choosing an optimal dialysis \\n prescription33, or mining text in the electronic health record to find specific  cases34,35. Additionally, a few recent \\nstudies were performed to predict the progression of CKD using ML methods. These models were developed \\nto estimate the risk of short-term mortality following  dialysis36, calculate the future eGFR  values37, or assess \\nthe 24-h urinary protein  levels18. To our best knowledge, there hasn’t been any attempt to apply ML methods to \\npredict the occurrence of ESKD in CKD patients.\\nIn the present study, a prediction model for ESKD in CKD patients was explored using ML techniques. Most \\nclassifiers demonstrated adequate performance based on easily accessible patient information that is convenient'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='classifiers demonstrated adequate performance based on easily accessible patient information that is convenient \\nfor clinical translation. In general, three ML models, including the logistic regression, naïve Bayes and random \\nforest, showed non-inferior performance to the KFRE in this study. These findings imply ML as a feasible \\napproach for predicting disease progression in CKD, which could potentially guide physicians in establishing \\npersonalized treatment plans for this condition at an early stage. These ML models with higher sensitivity scores \\nmay also be practically favored in patient screening over the KFRE.\\nTo our best understanding, this study was also the first to validate the KFRE in CKD patients of Mainland \\nChina. The KFRE was initially developed and validated using North American patients with CKD stage 3–5 12. \\nThere were seven KFRE models that consisted of different combinations of predictor variables. The most com-'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='There were seven KFRE models that consisted of different combinations of predictor variables. The most com-\\nmonly used KFRE included a 4-variable model (age, gender, eGFR and urine ACR) or an 8-variable model \\n(age, gender, eGFR, urine ACR, serum calcium, phosphorous, bicarbonate, and albumin). Besides, there was a \\n3-variable model (age, gender, and eGFR) that required no urine ACR and still showed comparable performance \\nto the other models in the original article. Despite its favorable performance in prediction for ESKD in patients \\nof Western  countries14,15,38,39, the generalizability of KFRE in Asian population remained arguable following \\nthe suboptimal results revealed by some recent  papers13,40,41. In the current study, the KFRE was validated in a \\nChinese cohort with CKD stage 1–5 and showed an AUC of 0.80. This result indicated the KFRE was adequately \\napplicable to the Chinese CKD patients and even earlier disease stages. In particular, the high specificity score'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='applicable to the Chinese CKD patients and even earlier disease stages. In particular, the high specificity score \\n(0.95) may favor the use of this equation in ruling in patients who require close monitoring of disease progres -\\nsion. On the other hand, a low sensitivity (0.47) at the default threshold may suggest it may be less desirable than \\nthe other models for ruling out patients.\\nUrine test is a critical diagnostic approach for CKD. The level of albuminuria (i.e. ACR) has also been regarded \\nas a major predictor for disease progression and therefore used by most prognostic models. However, quantitative \\ntesting for albuminuria is not always available in China especially in rural areas, which precludes clinicians from \\nusing most urine-based models for screening patients. In this regard, several simplified models were developed \\nto predict CKD progression without the need of albuminuria. These models were based on patient characteristics'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='to predict CKD progression without the need of albuminuria. These models were based on patient characteristics \\n(e.g. age, gender, BMI, comorbidity) and/or blood work (e.g. creatinine/eGFR, BUN), and still able to achieve an \\nAUC of 0.87–0.8912,18 or a sensitivity of 0.8837. Such performance was largely consistent with the findings of this \\nstudy and comparable or even superior to some models incorporating urine  tests16,42. Altogether, it suggested a \\nreliable prediction for CKD progression may be obtained from routine clinical variables without urine measures. \\nThese models are expected to provide a more convenient screening tool for CKD patients in developing regions.\\nMissing data are such a common problem in ML research that they can potentially lead to a biased model \\nand undermine the validity of study outcomes. Traditional methods to handle missing data include complete'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 2, 'page_label': '3', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='and undermine the validity of study outcomes. Traditional methods to handle missing data include complete \\ncase analysis, missing indicator, single value imputation, sensitivity analyses, and model-based methods (e.g. \\nmixed models or generalized estimating equations) 43–45. In most scenarios, complete case analysis and single'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 3, 'page_label': '4', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='4\\nVol:.(1234567890)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports/\\nvalue imputation are favored by researchers primarily due to the ease of  implementation45–47. However, these \\nmethods may be associated with significant drawbacks. For example, by excluding samples with missing data \\nTable 1.  Baseline patient characteristics. SBP systolic blood pressure, DBP diastolic blood pressure, GN \\nglomerulonephritis, CIN chronic interstitial nephritis, BMI body mass index, eGFR estimated glomerular \\nfiltration rate, ALT alanine aminotransferase, AST aspartate transaminase, ALP alkaline phosphatase, CKD \\nchronic kidney disease, HDL-c high density lipoprotein cholesterol, LDL-c low density lipoprotein cholesterol, \\nCa × P calcium-phosphorus product.\\nVariables Original data\\nAge (years) 57.8 ± 17.6\\nGender (male/female) 419/329\\nSBP (mmHg) 129.5 ± 17.8\\nDBP (mmHg) 77.7 ± 11.1\\nBMI (kg/m2) 24.8 ± 3.7\\nPrimary disease'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 3, 'page_label': '4', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='Ca × P calcium-phosphorus product.\\nVariables Original data\\nAge (years) 57.8 ± 17.6\\nGender (male/female) 419/329\\nSBP (mmHg) 129.5 ± 17.8\\nDBP (mmHg) 77.7 ± 11.1\\nBMI (kg/m2) 24.8 ± 3.7\\nPrimary disease\\nPrimary GN 292 (39.0%)\\nDiabetes 224 (29.9%)\\nHypertension 97 (13.0%)\\nCIN 64 (8.6%)\\nOthers 18 (2.4%)\\nUnknown 53 (7.1%)\\nCreatinine (µmol/L) 130.0 (100.0, 163.0)\\nUrea (mmol/L) 7.9 (5.6, 10.4)\\nALT (U/L) 17.0 (12.0, 24.0)\\nAST (U/L) 18.0 (15.0, 22.0)\\nALP (U/L) 60.0 (50.0, 75.0)\\nTotal protein (g/L) 71.6 ± 8.4\\nAlbumin (g/L) 42.2 ± 5.6\\nUrine acid (µmol/L) 374.0 (301.0, 459.0)\\nCalcium (mmol/L) 2.2 ± 0.1\\nPhosphorous (mmol/L) 1.2 ± 0.2\\nCa × P  (mg2/dL2) 33.5 ± 5.6\\nBlood leukocyte  (109/L) 7.1 ± 2.4\\nHemoglobin (g/L) 131.0 ± 20.3\\nPlatelet  (109/L) 209.8 ± 57.1\\neGFR (ml/min/1.73m2) 46.1 (32.6, 67.7)\\nCKD stage\\nStage 1 58 (7.8%)\\nStage 2 183 (24.5%)\\nStage 3 352 (47.1%)\\nStage 4 119 (15.9%)\\nStage 5 36 (4.8%)\\nTotal cholesterol 5.1 (4.3, 5.9)\\nTriglyceride 1.8 (1.3, 2.6)\\nHDL-c 1.3 (1.1, 1.6)\\nLDL-c 3.0 (2.4, 3.7)'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 3, 'page_label': '4', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='Stage 1 58 (7.8%)\\nStage 2 183 (24.5%)\\nStage 3 352 (47.1%)\\nStage 4 119 (15.9%)\\nStage 5 36 (4.8%)\\nTotal cholesterol 5.1 (4.3, 5.9)\\nTriglyceride 1.8 (1.3, 2.6)\\nHDL-c 1.3 (1.1, 1.6)\\nLDL-c 3.0 (2.4, 3.7)\\nFasting glucose (mmol/L) 5.4 (4.9, 6.2)\\nPotassium (mmol/L) 4.3 ± 0.5\\nSodium (mmol/L) 140.2 ± 2.8\\nChlorine (mmol/L) 106.9 ± 3.7\\nBicarbonate (mmol/L) 25.9 ± 3.6\\nMedical history\\nHypertension 558 (74.6%)\\nDiabetes mellitus 415 (55.5%)\\nCardiovascular or cerebrovascular disease 177 (23.7%)\\nSmoking 91 (12.6%)'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 4, 'page_label': '5', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='5\\nVol.:(0123456789)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports/\\nfrom analyses, complete case analysis can result in reduction of model power, overestimation of benefit and \\nunderestimation of  harm43,46; Single value imputation replaces the missing data by a single value—typically \\nthe mean or mode of the complete cases, thereby increasing the homogeneity of data and overestimating the \\n precision43,48. In this regard, multiple imputation solves these problems by generating several different plausible \\nimputed datasets, which account for the uncertainty about the missing data and provide unbiased estimates of \\nthe true  effect49,50. It is deemed effective regardless of the pattern of  missingness43,51. Multiple imputation is now \\nwidely recognized as the standard method to deal with missing data in many areas of  research43,45. In the current'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 4, 'page_label': '5', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='widely recognized as the standard method to deal with missing data in many areas of  research43,45. In the current \\nstudy, a 5-set multiple imputation method was employed to obtain reasonable variability of the imputed data. \\nThe performance of each model was analyzed on each imputed set and pooled for the final result. These proce -\\ndures ensured that the model bias resulting from missing data was minimized. In the future, multiple imputa -\\ntion is expected to become a routine method for missing data handling in ML research, as the extra amount of \\nTable 2.  Hyperparameters of the algorithms.\\nAlgorithms Hyperparameters\\nLogistic regression penalty = ’l2’ , class_weight = ’balanced’ , max_iter = 100000, C = 10, solver = ’liblinear’\\nNaive Bayes type = ’multinomial’ , alpha = 150\\nDecision tree criterion = ’gini’ , splitter = ’best’ , max_depth = 16, max_features = 15, min_samples_leaf = 5, min_samples_\\nsplit = 0.0001'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 4, 'page_label': '5', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='Naive Bayes type = ’multinomial’ , alpha = 150\\nDecision tree criterion = ’gini’ , splitter = ’best’ , max_depth = 16, max_features = 15, min_samples_leaf = 5, min_samples_\\nsplit = 0.0001\\nRandom forest class_weight = ’balanced’ , criterion = ’gini’ , max_depth = 9, max_features = 17, min_samples_leaf = 6, min_samples_\\nsplit = 30, n_estimators = 32\\nK-nearest neighbors weights = ’ distance’ , metric = ’minkowski’ , n_neighbors = 16, leaf_size = 10\\nTable 3.  The performance of all algorithms. All\\xa0outcomes are expressed as mean and (95% confidence \\ninterval). KFRE kidney failure risk equation, AUC  area under the curve.\\nAccuracy Sensitivity Specificity Precision F1 Score AUC \\nLogistic regression 0.75 (0.72, 0.79) 0.79 (0.73, 0.85) 0.75 (0.71, 0.79) 0.26 (0.24, 0.29) 0.38 (0.36, 0.41) 0.79 (0.77, 0.82)\\nNaïve Bayes 0.86 (0.85, 0.87) 0.72 (0.68, 0.75) 0.87 (0.86, 0.89) 0.37 (0.35, 0.40) 0.49 (0.46, 0.51) 0.80 (0.77, 0.82)'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 4, 'page_label': '5', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='Naïve Bayes 0.86 (0.85, 0.87) 0.72 (0.68, 0.75) 0.87 (0.86, 0.89) 0.37 (0.35, 0.40) 0.49 (0.46, 0.51) 0.80 (0.77, 0.82)\\nRandom forest 0.82 (0.80, 0.85) 0.76 (0.71, 0.81) 0.83 (0.80, 0.86) 0.34 (0.30, 0.39) 0.46 (0.43, 0.49) 0.81 (0.78, 0.83)\\nK nearest neighbor 0.84 (0.81, 0.86) 0.60 (0.57, 0.64) 0.86 (0.83, 0.89) 0.35 (0.30, 0.40) 0.43 (0.40, 0.46) 0.73 (0.71, 0.75)\\nDecision tree 0.84 (0.82, 0.86) 0.44 (0.39, 0.49) 0.89 (0.86, 0.91) 0.33 (0.26, 0.40) 0.35 (0.32, 0.39) 0.66 (0.63, 0.68)\\nKFRE 0.90 (0.90, 0.91) 0.47 (0.42, 0.52) 0.95 (0.94, 0.96) 0.50 (0.45, 0.55) 0.48 (0.43, 0.52) 0.80 (0.78, 0.83)\\nFigure\\xa01.  ROC curves of the random forest algorithm and the KFRE model.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='6\\nVol:.(1234567890)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports/\\ncomputation associated with multiple imputation over those traditional methods can simply be fulfilled by the \\nhigh level of computational power required by ML.\\nAlthough ML has been shown to outperform traditional statistics in a variety of tasks by virtue of the model \\ncomplexity, some studies demonstrated no gain or even declination of performance compared to traditional \\nregression  methods52,53. In this study, the simple logistic regression model also yielded a comparable or even \\nsuperior predictability for ESKD to other ML algorithms. The most likely explanation is that the current dataset \\nonly had a small sample size and limited numbers of predictor variables, and the ESKD+ cases were relatively \\nrare. The lack of big data and imbalanced class distribution may have negative impact on the performance of'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='rare. The lack of big data and imbalanced class distribution may have negative impact on the performance of \\ncomplex ML algorithms, as they are typically data  hungry54. On the other hand, this finding could imply simple \\ninteractions among the predictor variables. In other words, the risk of ESKD may be largely influenced by only \\na limited number of factors in an uncomplicated fashion, which is consistent with some previous  findings12,18,55. \\nThe fact that the 3-variable KFRE, which is also a regression model, yielded equivalent outcomes to the best ML \\nmodels in this study may further support this implication. It is therefore indicated that traditional regression \\nmodels may continue to play a key role in disease risk prediction, especially when a small sample size, limited \\npredictor variables, or an imbalanced dataset is encountered. The fact that some of the complex ML models are'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='predictor variables, or an imbalanced dataset is encountered. The fact that some of the complex ML models are \\nsubject to the risk of overfitting and the lack of interpretability further favors the use of simple regression models, \\nwhich can be translated to explainable equations.\\nSeveral limitations should be noted. First, this cohort consisted of less than 1000 subjects and ESKD only \\noccurred in a small portion of them, both of which might have affected model performance as discussed earlier. \\nSecond, although this study aimed to assess the feasibility of a prediction model for ESKD without any urine \\nvariables, this was partially due to the lack of quantitative urine tests at our institute when this cohort was \\nestablished. As spot urine tests become increasingly popular, urine features such as ACR will be as accessible \\nand convenient as other lab tests. They are expected to play a critical role in more predictive models. Third, the'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='and convenient as other lab tests. They are expected to play a critical role in more predictive models. Third, the \\nKFRE was previously established on stages 3–5 CKD patients while the current cohort contained stages 1–5. This \\ndiscrepancy may have affected the KFRE performance. Forth, the generalizability of this model has not been \\ntested on any external data due to the lack of such resource in this early feasibility study. Therefore, additional \\nefforts are required to improve and validate this model before any clinical translation. Finally, although a simple \\nmodel without urine variables is feasible and convenient, model predictability may benefit from a greater variety \\nof clinical features, such as urine tests, imaging, or biopsy. Future works should include training ML models with \\nadditional features using a large dataset, and validating them on external patients.\\nIn conclusion, this study showed the feasibility of ML in evaluating the prognosis of CKD based on easily'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='additional features using a large dataset, and validating them on external patients.\\nIn conclusion, this study showed the feasibility of ML in evaluating the prognosis of CKD based on easily \\naccessible features. Logistic regression, naïve Bayes and random forest demonstrated comparable predictability \\nto the KFRE in this study. These ML models also had greater sensitivity scores that were potentially advanta-\\ngeous for patient screenings. Future studies include performing external validation and improving the model \\nwith additional predictor variables.\\nReceived: 19 July 2021; Accepted: 9 May 2022\\nReferences\\n 1. Zhang, L. et al. Trends in chronic kidney disease in China. N. Engl. J. Med. 375, 905–906. https:// doi. org/ 10. 1056/ NEJMc 16024 69 \\n(2016).\\n 2. Bello, A. K. et al. Effective CKD care in European countries: Challenges and opportunities for health policy. Am. J. Kidney Dis. 65, \\n15–25. https:// doi. org/ 10. 1053/j. ajkd. 2014. 07. 033 (2015).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='15–25. https:// doi. org/ 10. 1053/j. ajkd. 2014. 07. 033 (2015).\\n 3. Subbiah, A. K., Chhabra, Y . K. & Mahajan, S. Cardiovascular disease in patients with chronic kidney disease: A neglected subgroup. \\nHeart Asia 8, 56–61. https:// doi. org/ 10. 1136/ heart asia- 2016- 010809 (2016).\\n 4. Pecoits-Filho, R. et al. Interactions between kidney disease and diabetes: Dangerous liaisons. Diabetol. Metab. Syndr. 8, 50. https:// \\ndoi. org/ 10. 1186/ s13098- 016- 0159-z (2016).\\n 5. Weiner, D. E. et al. Chronic kidney disease as a risk factor for cardiovascular disease and all-cause mortality: A pooled analysis of \\ncommunity-based studies. J. Am. Soc. Nephrol. 15, 1307–1315. https:// doi. org/ 10. 1097/ 01. asn. 00001 23691. 46138. e2 (2004).\\n 6. Saran, R., Robinson, B., Abbott, K. C. et\\xa0al. US Renal Data System 2019 Annual Data Report: Epidemiology of kidney disease in \\nthe United States. Am. J. Kidney Dis. 75, A6–A7. https:// doi. org/ 10. 1053/j. ajkd. 2019. 09. 003 (2020).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='the United States. Am. J. Kidney Dis. 75, A6–A7. https:// doi. org/ 10. 1053/j. ajkd. 2019. 09. 003 (2020).\\n 7. Zhang, L. et al. Prevalence of chronic kidney disease in China: A cross-sectional survey. Lancet (London, England) 379, 815–822. \\nhttps:// doi. org/ 10. 1016/ S0140- 6736(12) 60033-6 (2012).\\n 8. Johns, T. S., Y ee, J., Smith-Jules, T., Campbell, R. C. & Bauer, C. Interdisciplinary care clinics in chronic kidney disease. BMC \\nNephrol. 16, 161. https:// doi. org/ 10. 1186/ s12882- 015- 0158-6 (2015).\\n 9. Lin, E., Chertow, G. M., Y an, B., Malcolm, E. & Goldhaber-Fiebert, J. D. Cost-effectiveness of multidisciplinary care in mild to \\nmoderate chronic kidney disease in the United States: A modeling study. PLoS Med. 15, e1002532. https:// doi. org/ 10. 1371/ journ \\nal. pmed. 10025 32 (2018).\\n 10. Zhong, J., Y ang, H. C. & Fogo, A. B. A perspective on chronic kidney disease progression. Am. J. Physiol. Ren. Physiol. 312, F375–'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='al. pmed. 10025 32 (2018).\\n 10. Zhong, J., Y ang, H. C. & Fogo, A. B. A perspective on chronic kidney disease progression. Am. J. Physiol. Ren. Physiol. 312, F375–\\nF384. https:// doi. org/ 10. 1152/ ajpre nal. 00266. 2016 (2017).\\n 11. Tangri, N. et al. A dynamic predictive model for progression of CKD. Am. J. Kidney Dis.  69, 514–520. https:// doi. org/ 10. 1053/j. \\najkd. 2016. 07. 030 (2017).\\n 12. Tangri, N. et al. A predictive model for progression of chronic kidney disease to kidney failure. JAMA 305, 1553–1559. https://  \\ndoi. org/ 10. 1001/ jama. 2011. 451 (2011).\\n 13. Tangri, N. et al. Multinational assessment of accuracy of equations for predicting risk of kidney failure: A meta-analysis. JAMA \\n315, 164–174. https:// doi. org/ 10. 1001/ jama. 2015. 18202 (2016).\\n 14. Major, R. W . et al. The Kidney Failure Risk Equation for prediction of end stage renal disease in UK primary care: An external'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 5, 'page_label': '6', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='14. Major, R. W . et al. The Kidney Failure Risk Equation for prediction of end stage renal disease in UK primary care: An external \\nvalidation and clinical impact projection cohort study. PLoS Med. 16, e1002955. https:// doi. org/ 10. 1371/ journ al. pmed. 10029 55 \\n(2019).\\n 15. Peeters, M. J. et al. Validation of the kidney failure risk equation in European CKD patients. Nephrol. Dial. Transplant. Off. Publ. \\nEur. Dial. Transpl. Assoc. Eur. Ren. Assoc. 28, 1773–1779. https:// doi. org/ 10. 1093/ ndt/ gft063 (2013).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='7\\nVol.:(0123456789)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports/\\n 16. Echouffo-Tcheugui, J. B. & Kengne, A. P . Risk models to predict chronic kidney disease and its progression: A systematic review. \\nPLoS Med. 9, e1001344. https:// doi. org/ 10. 1371/ journ al. pmed. 10013 44 (2012).\\n 17. Chang, H. L. et al. A predictive model for progression of CKD. Medicine (Baltimore) 98, e16186. https:// doi. org/ 10. 1097/ MD. 00000 \\n00000 016186 (2019).\\n 18. Xiao, J. et al. Comparison and development of machine learning tools in the prediction of chronic kidney disease progression. J. \\nTransl. Med. 17, 119. https:// doi. org/ 10. 1186/ s12967- 019- 1860-0 (2019).\\n 19. Song, H., Triguero, I. & Özcan, E. A review on the self and dual interactions between machine learning and optimisation. Prog. \\nArtif. Intell. 8, 143–165. https:// doi. org/ 10. 1007/ s13748- 019- 00185-z (2019).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='Artif. Intell. 8, 143–165. https:// doi. org/ 10. 1007/ s13748- 019- 00185-z (2019).\\n 20. Mortazavi, B. J. et al. Analysis of machine learning techniques for heart failure readmissions. Circ. Cardiovasc. Qual. Outcomes 9, \\n629–640. https:// doi. org/ 10. 1161/ CIRCO UTCOM ES. 116. 003039 (2016).\\n 21. Deo, R. C. Machine learning in medicine. Circulation 132, 1920–1930. https:// doi. org/ 10. 1161/ CIRCU LATIO NAHA. 115. 001593 \\n(2015).\\n 22. Weng, S. F ., Reps, J., Kai, J., Garibaldi, J. M. & Qureshi, N. Can machine-learning improve cardiovascular risk prediction using \\nroutine clinical data?. PLoS ONE 12, e0174944. https:// doi. org/ 10. 1371/ journ al. pone. 01749 44 (2017).\\n 23. Meiring, C. et al. Optimal intensive care outcome prediction over time using machine learning. PLoS ONE 13, e0206862. https:// \\ndoi. org/ 10. 1371/ journ al. pone. 02068 62 (2018).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='23. Meiring, C. et al. Optimal intensive care outcome prediction over time using machine learning. PLoS ONE 13, e0206862. https:// \\ndoi. org/ 10. 1371/ journ al. pone. 02068 62 (2018).\\n 24. Lai, X. et al. Outcomes of stage 1–5 chronic kidney disease in Mainland China. Ren. Fail. 36, 520–525. https:// doi. org/ 10. 3109/ \\n08860 22X. 2013. 875859 (2014).\\n 25. Bai, Q., Su, C. Y ., Zhang, A. H., Wang, T. & Tang, W . Loss of the normal gradient in arterial compliance and outcomes of chronic \\nkidney disease patients. Cardiorenal Med. 9, 297–307. https:// doi. org/ 10. 1159/ 00050 0479 (2019).\\n 26. Steyerberg, E. W . & Vergouwe, Y . Towards better clinical prediction models: Seven steps for development and an ABCD for valida-\\ntion. Eur. Heart J. 35, 1925–1931. https:// doi. org/ 10. 1093/ eurhe artj/ ehu207 (2014).\\n 27. Python: A dynamic, open source programming language. https:// www. python. org/ (2019).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='tion. Eur. Heart J. 35, 1925–1931. https:// doi. org/ 10. 1093/ eurhe artj/ ehu207 (2014).\\n 27. Python: A dynamic, open source programming language. https:// www. python. org/ (2019).\\n 28. Yu, K., Beam, A. L. & Kohane, I. S. Artificial intelligence in healthcare. Nat. Biomed. Eng. 2, 719–731. https:// doi. org/ 10. 1038/ \\ns41551- 018- 0305-z (2018).\\n 29. Mohamadlou, H. et al. Prediction of acute kidney injury with a machine learning algorithm using electronic health record data. \\nCan. J. Kidney Health Dis. 5, 2054358118776326. https:// doi. org/ 10. 1177/ 20543 58118 776326 (2018).\\n 30. Lee, H. C. et al. Derivation and validation of machine learning approaches to predict acute kidney injury after cardiac surgery. J. \\nClin. Med. https:// doi. org/ 10. 3390/ jcm71 00322 (2018).\\n 31. Hermsen, M. et al. Deep learning-based histopathologic assessment of kidney tissue. J. Am. Soc. Nephrol. 30, 1968–1979. https:// \\ndoi. org/ 10. 1681/ ASN. 20190 20144 (2019).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='31. Hermsen, M. et al. Deep learning-based histopathologic assessment of kidney tissue. J. Am. Soc. Nephrol. 30, 1968–1979. https:// \\ndoi. org/ 10. 1681/ ASN. 20190 20144 (2019).\\n 32. Boor, P . Artificial intelligence in nephropathology. Nat. Rev. Nephrol. 16, 4–6. https:// doi. org/ 10. 1038/ s41581- 019- 0220-x (2020).\\n 33. Barbieri, C. et al. Development of an artificial intelligence model to guide the management of blood pressure, fluid volume, and \\ndialysis dose in end-stage kidney disease patients: Proof of concept and first clinical assessment. Kidney Dis. (Basel) 5, 28–33. \\nhttps:// doi. org/ 10. 1159/ 00049 3479 (2019).\\n 34. Garcelon, N., Burgun, A., Salomon, R. & Neuraz, A. Electronic health records for the diagnosis of rare diseases. Kidney Int.  97, \\n676–686. https:// doi. org/ 10. 1016/j. kint. 2019. 11. 037 (2020).\\n 35. Rajkomar, A. et al. Scalable and accurate deep learning with electronic health records. NPJ Digit. Med.  1, 18. https:// doi. org/ 10.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='35. Rajkomar, A. et al. Scalable and accurate deep learning with electronic health records. NPJ Digit. Med.  1, 18. https:// doi. org/ 10. \\n1038/ s41746- 018- 0029-1 (2018).\\n 36. Akbilgic, O. et al. Machine learning to identify dialysis patients at high death risk. Kidney Int. Rep. 4, 1219–1229. https:// doi. org/ \\n10. 1016/j. ekir. 2019. 06. 009 (2019).\\n 37. Zhao, J., Gu, S. & McDermaid, A. Predicting outcomes of chronic kidney disease from EMR data based on Random Forest Regres-\\nsion. Math. Biosci. 310, 24–30. https:// doi. org/ 10. 1016/j. mbs. 2019. 02. 001 (2019).\\n 38. Tangri, N., Ferguson, T. & Komenda, P . Pro: Risk scores for chronic kidney disease progression are robust, powerful and ready for \\nimplementation. Nephrol. Dial. Transplant. Off. Publ. Eur. Dial. Transpl. Assoc. Eur. Ren. Assoc. 32, 748–751. https:// doi. org/ 10. \\n1093/ ndt/ gfx067 (2017).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='implementation. Nephrol. Dial. Transplant. Off. Publ. Eur. Dial. Transpl. Assoc. Eur. Ren. Assoc. 32, 748–751. https:// doi. org/ 10. \\n1093/ ndt/ gfx067 (2017).\\n 39. Marks, A. et al. Looking to the future: Predicting renal replacement outcomes in a large community cohort with chronic kidney \\ndisease. Nephrol. Dial. Transplant. Off. Publ. Eur. Dial. Transpl. Assoc. Eur. Ren. Assoc. 30, 1507–1517. https:// doi. org/ 10. 1093/ ndt/ \\ngfv089 (2015).\\n 40. Wang, Y . et al. Validation of the kidney failure risk equation for end-stage kidney disease in Southeast Asia. BMC Nephrol. 20, 451. \\nhttps:// doi. org/ 10. 1186/ s12882- 019- 1643-0 (2019).\\n 41. Y amanouchi, M. et al. Value of adding the renal pathological score to the kidney failure risk equation in advanced diabetic \\nnephropathy. PLoS ONE 13, e0190930. https:// doi. org/ 10. 1371/ journ al. pone. 01909 30 (2018).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='nephropathy. PLoS ONE 13, e0190930. https:// doi. org/ 10. 1371/ journ al. pone. 01909 30 (2018).\\n 42. Lin, C. C. et al. Development and validation of a risk prediction model for end-stage renal disease in patients with type 2 diabetes. \\nSci. Rep. 7, 10177. https:// doi. org/ 10. 1038/ s41598- 017- 09243-9 (2017).\\n 43. Pedersen, A. B. et al. Missing data and multiple imputation in clinical epidemiological research. Clin. Epidemiol. 9, 157–166. https:// \\ndoi. org/ 10. 2147/ CLEP . S1297 85 (2017).\\n 44. Sterne, J. A. et al. Multiple imputation for missing data in epidemiological and clinical research: Potential and pitfalls. BMJ (Clin. \\nRes. Ed.) 338, b2393. https:// doi. org/ 10. 1136/ bmj. b2393 (2009).\\n 45. Jakobsen, J. C., Gluud, C., Wetterslev, J. & Winkel, P . When and how should multiple imputation be used for handling missing \\ndata in randomised clinical trials: A practical guide with flowcharts. BMC Med. Res. Methodol.  17, 162. https:// doi. org/ 10. 1186/'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='data in randomised clinical trials: A practical guide with flowcharts. BMC Med. Res. Methodol.  17, 162. https:// doi. org/ 10. 1186/ \\ns12874- 017- 0442-1 (2017).\\n 46. Karadaghy, O. A., Shew, M., New, J. & Bur, A. M. Development and assessment of a machine learning model to help predict survival \\namong patients with oral squamous cell carcinoma. JAMA Otolaryngol. Head Neck Surg. 145, 1115–1120. https:// doi. org/ 10. 1001/ \\njamao to. 2019. 0981 (2019).\\n 47. Bur, A. M. et al. Machine learning to predict occult nodal metastasis in early oral squamous cell carcinoma. Oral Oncol. 92, 20–25. \\nhttps:// doi. org/ 10. 1016/j. oralo ncolo gy. 2019. 03. 011 (2019).\\n 48. Zhang, Z. Missing data imputation: Focusing on single imputation. Ann. Transl. Med. 4, 9. https:// doi. org/ 10. 3978/j. issn. 2305- 5839. \\n2015. 12. 38 (2016).\\n 49. Graham, J. W . Missing data analysis: Making it work in the real world. Annu. Rev. Psychol. 60, 549–576. https:// doi. org/ 10. 1146/'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='2015. 12. 38 (2016).\\n 49. Graham, J. W . Missing data analysis: Making it work in the real world. Annu. Rev. Psychol. 60, 549–576. https:// doi. org/ 10. 1146/ \\nannur ev. psych. 58. 110405. 085530 (2009).\\n 50. Klebanoff, M. A. & Cole, S. R. Use of multiple imputation in the epidemiologic literature. Am. J. Epidemiol. 168, 355–357. https:// \\ndoi. org/ 10. 1093/ aje/ kwn071% JAmer icanJ ourna lofEp idemi ology (2008).\\n 51. Azur, M. J., Stuart, E. A., Frangakis, C. & Leaf, P . J. Multiple imputation by chained equations: What is it and how does it work?. \\nInt. J. Methods Psychiatr. Res. 20, 40–49. https:// doi. org/ 10. 1002/ mpr. 329 (2011).\\n 52. Desai, R. J., Wang, S. V ., Vaduganathan, M., Evers, T. & Schneeweiss, S. Comparison of machine learning methods with traditional \\nmodels for use of administrative claims with electronic medical records to predict heart failure outcomes. JAMA Netw. Open  3, \\ne1918962. https:// doi. org/ 10. 1001/ jaman etwor kopen. 2019. 18962 (2020).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 6, 'page_label': '7', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='e1918962. https:// doi. org/ 10. 1001/ jaman etwor kopen. 2019. 18962 (2020).\\n 53. Christodoulou, E. et al. A systematic review shows no performance benefit of machine learning over logistic regression for clinical \\nprediction models. J. Clin. Epidemiol. 110, 12–22. https:// doi. org/ 10. 1016/j. jclin epi. 2019. 02. 004 (2019).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 7, 'page_label': '8', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='8\\nVol:.(1234567890)Scientific Reports |         (2022) 12:8377  | https://doi.org/10.1038/s41598-022-12316-z\\nwww.nature.com/scientificreports/\\n 54. Nusinovici, S. et al. Logistic regression was as good as machine learning for predicting major chronic diseases. J. Clin. Epidemiol. \\n122, 56–69. https:// doi. org/ 10. 1016/j. jclin epi. 2020. 03. 002 (2020).\\n 55. Nusinovici, S. et al. Logistic regression was as good as machine learning for predicting major chronic diseases. J. Clin. Epidemiol. \\nhttps:// doi. org/ 10. 1016/j. jclin epi. 2020. 03. 002 (2020).\\nAuthor contributions\\nQ.B. was involved in the data collection, data analysis, and drafting the manuscript. C.S. was involved in data \\ncollection. W .T. conceptualized the idea, interpreted the results and wrote part of the draft. Y .L. conceptualized \\nthe idea, analyzed the data, performed all coding, evaluated all machine learning models, drafted and edited \\nthe manuscript.\\nFunding'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 7, 'page_label': '8', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='the idea, analyzed the data, performed all coding, evaluated all machine learning models, drafted and edited \\nthe manuscript.\\nFunding\\nThis work was supported by PKU-Baidu Fund (2020BD030 to Wen Tang), and by fund from China International \\nMedical Foundation (Z-2017-24-2037 to Wen Tang). The funders had no role in study design, data collection \\nand analysis, decision to publish, or preparation of the manuscript.\\nCompeting interests \\nThe authors declare no competing interests.\\nAdditional information\\nSupplementary Information The online version contains supplementary material available at https:// doi. org/ \\n10. 1038/ s41598- 022- 12316-z.\\nCorrespondence and requests for materials should be addressed to W .T.\\xa0or\\xa0Y .L.\\nReprints and permissions information is available at www.nature.com/reprints.\\nPublisher’s note Springer Nature remains neutral with regard to jurisdictional claims in published maps and \\ninstitutional affiliations.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 7, 'page_label': '8', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='Publisher’s note Springer Nature remains neutral with regard to jurisdictional claims in published maps and \\ninstitutional affiliations.\\nOpen Access  This article is licensed under a Creative Commons Attribution 4.0 International \\nLicense, which permits use, sharing, adaptation, distribution and reproduction in any medium or \\nformat, as long as you give appropriate credit to the original author(s) and the source, provide a link to the \\nCreative Commons licence, and indicate if changes were made. The images or other third party material in this \\narticle are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line to the \\nmaterial. If material is not included in the article’s Creative Commons licence and your intended use is not \\npermitted by statutory regulation or exceeds the permitted use, you will need to obtain permission directly from'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-05-19T09:40:50+05:30', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-05-19T08:13:56+02:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z', 'author': 'Qiong Bai', 'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1038/s41598-022-12316-z', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf', 'total_pages': 8, 'page': 7, 'page_label': '8', 'source_file': 's41598-022-12316-z.pdf', 'file_type': 'pdf'}, page_content='permitted by statutory regulation or exceeds the permitted use, you will need to obtain permission directly from \\nthe copyright holder. To view a copy of this licence, visit http:// creat iveco mmons. org/ licen ses/ by/4. 0/.\\n© The Author(s) 2022'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 0, 'page_label': '1', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Lei\\xa0et\\xa0al. \\nBMC Medical Informatics and Decision Making          (2022) 22:205  \\nhttps://doi.org/10.1186/s12911-022-01951-1\\nRESEARCH\\nMachine learning algorithms’ accuracy \\nin\\xa0predicting kidney disease progression: \\na\\xa0systematic review and\\xa0meta-analysis\\nNuo Lei1†, Xianlong Zhang2†, Mengting Wei1, Beini Lao1, Xueyi Xu1, Min Zhang1, Huifen Chen1, Yanmin Xu1, \\nBingqing Xia1, Dingjun Zhang1, Chendi Dong1, Lizhe Fu3, Fang Tang3 and Yifan Wu2* \\nAbstract \\nBackground: Kidney disease progression rates vary among patients. Rapid and accurate prediction of kidney disease \\noutcomes is crucial for disease management. In recent years, various prediction models using Machine Learning \\n(ML) algorithms have been established in nephrology. However, their accuracy have been inconsistent. Therefore, we \\nconducted a systematic review and meta-analysis to investigate the diagnostic accuracy of ML algorithms for kidney \\ndisease progression.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 0, 'page_label': '1', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='conducted a systematic review and meta-analysis to investigate the diagnostic accuracy of ML algorithms for kidney \\ndisease progression.\\nMethods: We searched PubMed, EMBASE, Cochrane Central Register of Controlled Trials, the Chinese Biomedi-\\ncine Literature Database, Chinese National Knowledge Infrastructure, Wanfang Database, and the VIP Database for \\ndiagnostic studies on ML algorithms’ accuracy in predicting kidney disease prognosis, from the establishment of \\nthese databases until October 2020. Two investigators independently evaluate study quality by QUADAS-2 tool and \\nextracted data from single ML algorithm for data synthesis using the bivariate model and the hierarchical summary \\nreceiver operating characteristic (HSROC) model.\\nResults: Fifteen studies were left after screening, only 6 studies were eligible for data synthesis. The sample size of \\nthese 6 studies was 12,534, and the kidney disease types could be divided into chronic kidney disease (CKD) and'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 0, 'page_label': '1', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='these 6 studies was 12,534, and the kidney disease types could be divided into chronic kidney disease (CKD) and \\nImmunoglobulin A Nephropathy, with 5 articles using end-stage renal diseases occurrence as the primary outcome. \\nThe main results indicated that the area under curve (AUC) of the HSROC was 0.87 (0.84–0.90) and ML algorithm \\nexhibited a strong specificity, 95% confidence interval and heterogeneity  (I2) of (0.87, 0.84–0.90,  [I2 99.0%]) and a weak \\nsensitivity of (0.68, 0.58–0.77,  [I2 99.7%]) in predicting kidney disease deterioration. And the the results of subgroup \\nanalysis indicated that ML algorithm’s AUC for predicting CKD prognosis was 0.82 (0.79–0.85), with the pool sensitivity \\nof (0.64, 0.49–0.77,  [I2 99.20%]) and pool specificity of (0.84, 0.74–0.91,  [I2 99.84%]). The ML algorithm’s AUC for predict-\\ning IgA nephropathy prognosis was 0.78 (0.74–0.81), with the pool sensitivity of (0.74, 0.71–0.77,  [I2 7.10%]) and pool'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 0, 'page_label': '1', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='ing IgA nephropathy prognosis was 0.78 (0.74–0.81), with the pool sensitivity of (0.74, 0.71–0.77,  [I2 7.10%]) and pool \\nspecificity of (0.93, 0.91–0.95,  [I2 83.92%]).\\n© The Author(s) 2022. Open Access This article is licensed under a Creative Commons Attribution 4.0 International License, which \\npermits use, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the \\noriginal author(s) and the source, provide a link to the Creative Commons licence, and indicate if changes were made. The images or \\nother third party material in this article are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line \\nto the material. If material is not included in the article’s Creative Commons licence and your intended use is not permitted by statutory \\nregulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 0, 'page_label': '1', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='regulation or exceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this \\nlicence, visit http:// creat iveco mmons. org/ licen ses/ by/4. 0/. The Creative Commons Public Domain Dedication waiver (http:// creat iveco \\nmmons. org/ publi cdoma in/ zero/1. 0/) applies to the data made available in this article, unless otherwise stated in a credit line to the data.\\nOpen Access\\n†Nuo Lei and Xianlong Zhang contributed equally to this work and should be \\nconsidered co-first authors\\n*Correspondence:  wuyifan007@gzucm.edu.cn\\n2 Department of Nephrology, Guangdong Provincial Hospital of Chinese \\nMedicine, The Second Affiliated Hospital of Guangzhou University of Chinese \\nMedicine, Guangzhou, China\\nFull list of author information is available at the end of the article'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 1, 'page_label': '2', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 2 of 16Lei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\nBackground\\nChronic kidney disease (CKD) affects 8–16% of the \\nworld’s population, and has become a global public \\nhealth problem as its prevalence has increased [1, 2]. As \\nthe 10th leading cause of death in the world [3], 1.2 mil -\\nlion people died from CKD in 2017 globally [4]. Kidney \\ninjury is an irreversible process, any form of kidney dis -\\nease can progress to end-stage renal diseases (ESRD), \\nand may require Renal Replacement Therapy (RRT) for \\nresidual renal function damage [5, 6]. Patients who finally \\nprogress to ESRD or undertake RRT suffer from heavy \\neconomic pressure [7]. However, observational studies \\nhave shown that the speed and severity of kidney disease \\nprogression varies [8, 9]. Therefore, early identification of \\ngroups at high-risk of kidney disease progression accu -\\nrately and delaying kidney function deterioration, have'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 1, 'page_label': '2', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='progression varies [8, 9]. Therefore, early identification of \\ngroups at high-risk of kidney disease progression accu -\\nrately and delaying kidney function deterioration, have \\nbecome an important focus in kidney disease manage -\\nment [10–12].\\nConsidering the increasing prevalence and variegated \\nseverity of disease progression, kidney disease patients \\nmust be managed through stratification. An accurate \\ndisease prognosis prediction model may assist medical \\nstaff in early intervention for high-risk patients with poor \\nprognosis. Management strategies should be adopted \\nbased on the predictable outcome. In order to promote \\nearly identification of patients at high risk of kidney func-\\ntion deterioration, researchers have conducted numerous \\nstudies exploring the risk factors, and have established \\nseveral risk prediction models [13, 14].\\nThese models have performed well in internal valida -\\ntion, but their capacity for generalization is uncertain'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 1, 'page_label': '2', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='several risk prediction models [13, 14].\\nThese models have performed well in internal valida -\\ntion, but their capacity for generalization is uncertain \\nbecause only a portion of the studies have been externally \\nvalidated. As a new tool for big data analysis, machine \\nlearning (ML) has emerged in the field of medicine in \\nrecent years [15]. ML allows the construction of an algo -\\nrithm that can learn, predict, and improve with experi -\\nence [16] based on big data. It has immense potential in \\nexploring risk factors for disease progression and pre -\\ndicting patients’ prognosis. Several ML algorithm-based \\nprediction models have been successful in predicting kid-\\nney function during a specific period of time, and shown \\ngreater capacity for generalization than previous statisti -\\ncal models.\\nThough ML algorithms can extract meaningful pat -\\nterns from big data, several problems remain in clinical \\npractice. Firstly, selecting suitable models in clinical prac-'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 1, 'page_label': '2', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='cal models.\\nThough ML algorithms can extract meaningful pat -\\nterns from big data, several problems remain in clinical \\npractice. Firstly, selecting suitable models in clinical prac-\\ntice is challenging, due to the lack of evidence. Because \\nprevious systematic reviews pertaining to prognostic \\nprediction models have neither focused on ML algo -\\nrithms, nor have they extracted data for further analysis. \\nSecondly, researchers have used an array of ML algo -\\nrithms, predictors, and outcome indicators to construct \\nprediction models. Finally, there is a dearth of high qual -\\nity research demonstrating the accuracy and reliability \\nof ML algorithm-based prediction models. Thus, experi -\\nenced clinicians rely more on their own knowledge and \\nexperience when judging patients’ prognosis.\\nConsidering the potentiality and problems pertain -\\ning to ML algorithms in the field of nephrology, there \\nis a need for a summary of current research on ML'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 1, 'page_label': '2', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Considering the potentiality and problems pertain -\\ning to ML algorithms in the field of nephrology, there \\nis a need for a summary of current research on ML \\nalgorithm-based prognostic prediction models for the \\ndeterioration of various kidney diseases. Therefore, we \\nconducted this systematic review by reviewing relevant \\nstudies and extracting data for a meta-analysis. In doing \\nso, we investigated ML algorithms’ accuracy in predicting \\nkidney disease progression.\\nMethods\\nThe methods and results of this review are presented \\naccording to the Preferred Reporting Items for System -\\natic reviews and Meta-analyses statement (PRISMA). \\nThe review protocol was previously registered on PROS -\\nPERO (International Prospective Register of Systematic \\nReviews) with the CRD (Centre of Reviews and Dissemi -\\nnation), report number CRD42020156213.\\nEligibility criteria\\nThe inclusion criteria were:\\n1. Clinical studies of diagnostic tests of accuracy.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 1, 'page_label': '2', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Reviews) with the CRD (Centre of Reviews and Dissemi -\\nnation), report number CRD42020156213.\\nEligibility criteria\\nThe inclusion criteria were:\\n1. Clinical studies of diagnostic tests of accuracy.\\n2. Participants with kidney disease, aged 18\\xa0 years or \\nolder.\\n3. Studies that used ML algorithms.\\n4. Study outcome reflected kidney disease deteriora -\\ntion, including the doubling of serum creatinine, \\nsudden estimated Glomerular Filtration Rate (eGFR) \\ndeterioration, urinary protein level aggravation, \\nESRD occurrence, RRT initiation, cardiovascular \\nevents and all-cause mortality.\\nConclusion: Taking advantage of big data, ML algorithm-based prediction models have high accuracy in predicting \\nkidney disease progression, we recommend ML algorithms as an auxiliary tool for clinicians to determine proper treat-\\nment and disease management strategies.\\nKeywords: Artificial intelligence, Machine learning algorithm, Prediction models, Chronic kidney disease, CKD'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 1, 'page_label': '2', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='ment and disease management strategies.\\nKeywords: Artificial intelligence, Machine learning algorithm, Prediction models, Chronic kidney disease, CKD \\nprogression, Immunoglobulin A nephropathy'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 2, 'page_label': '3', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 3 of 16\\nLei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n \\n5. Studies from which indicators could be extracted that \\npertained to diagnostic test accuracy, such as accu -\\nracy, specificity, sensitivity, true positive (TP), false \\npositive (FP), true negative (TN), false negative (FN), \\nArea Under Curve(AUC) and C-statistic.\\n6. Studies published in either English or Chinese.\\nSearch strategy\\nWe searched PubMed, EMBASE, Cochrane Central \\nRegister of Controlled Trials, the Chinese Biomedi -\\ncine Literature Database, Chinese National Knowledge \\nInfrastructure, Wanfang Database, and VIP Database by \\nusing both free-text terms and Medical Subject Headings \\n(MeSH) terms for studies limited to humans, without any \\nlanguage restrictions, from the establishment of these \\ndatabases until October 2020. The detailed search strate -\\ngies are listed in the Additional file\\xa0 1. The last search was \\nperformed on October 31, 2020.\\nStudy selection'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 2, 'page_label': '3', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='databases until October 2020. The detailed search strate -\\ngies are listed in the Additional file\\xa0 1. The last search was \\nperformed on October 31, 2020.\\nStudy selection\\nThe records retrieved from the search were imported to \\nNoteExpress 3.2.0. Two authors (M. T. Wei and N. L.) \\nindependently screened the records by title and abstract \\nafter deduplication. Then, the full texts of the selected \\nrecords were read independently by two researchers \\n(M. T. Wei and N. L.). At each stage of selection, disa -\\ngreements were arbitrated by a third reviewer (X. L. \\nZhang) and resolved by consensus. After that, each \\nresearcher created an Excel spreadsheet of the articles to \\nbe excluded, and their exclusion reasons, before we com -\\npiled a final list of included articles.\\nData extraction\\nTwo independent reviewers (M. T. Wei and N. L.) \\nextracted data using a customized extraction form. \\nConsidering an individual article may use several ML \\nalgorithms to build prediction models separately or in'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 2, 'page_label': '3', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='extracted data using a customized extraction form. \\nConsidering an individual article may use several ML \\nalgorithms to build prediction models separately or in \\ncombination, in order to explicate the performance of \\na single ML algorithm, we extracted data in the unit of \\na single ML algorithm rather than a single article. The \\nextracted data included the TP , FP , FN and TN numbers \\nof patients with kidney disease progression predicted by \\nan ML algorithm, and the ML algorithm’s accuracy, sen -\\nsitivity, specificity, positive predictive value (PPV) and \\nnegative predictive value (NPV). During the data extrac -\\ntion process, we used RevMan 5.2 for data conversion.\\nAssessment methodology quality\\nWe used the Quality Assessment of Diagnostic Accuracy \\nStudies 2 (QUADAS-2) tool to assess the quality of the \\nincluded studies. Two reviewers adjusted both the sign -\\naling questions and the assessment questions to build \\na specific version of the tool, according to our study’s'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 2, 'page_label': '3', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='included studies. Two reviewers adjusted both the sign -\\naling questions and the assessment questions to build \\na specific version of the tool, according to our study’s \\nobjective. We then tested the tool, and when we achieved \\ngood agreement, we determined that it would be the final \\nversion of the review tool. Both authors then used it to \\nindependently assess the risk of bias and the applicability \\nof all included studies. Disagreements were resolved by \\nconsensus.\\nData synthesis\\nConsidering the bivariate nature of the data, we used \\nboth the bivariate model and the hierarchical summary \\nreceiver operating characteristic (HSROC) method for \\ndata synthesis. The bivariate model, which preserved the \\nbivariate nature of the data, was used to summarize the \\nindex tests’ hierarchical sensitivity and specificity. The \\nHSROC model, which could convert the bivariate data \\ninto univariate data, was used to determine the index \\ntests’ overall accuracy. In the absence of covariates, the'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 2, 'page_label': '3', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='HSROC model, which could convert the bivariate data \\ninto univariate data, was used to determine the index \\ntests’ overall accuracy. In the absence of covariates, the \\nbivariate model was equivalent to the HSROC model \\n[17].\\nTo judge whether there was a threshold effect between \\nthe studies, we used the correlation coefficient between \\nthe logit transformed sensitivity and specificity generated \\nby the bivariate model, and the asymmetry parameter β \\ngenerated by the HSROC model. Where a negative cor -\\nrelation coefficient or β = 0 showed an expected trade-off  \\nbetween sensitivity and specificity across thresholds, test \\naccuracy could be represented by the expected accuracy \\n(logDOR) [18]. Then we used the bivariate model to esti -\\nmate the pool sensitivity and specificity, and generated a \\nforest plot. After that, we generated the HSROC curves \\nand their 95% prediction intervals via the HSROC model. \\nWe also calculated the HSROC’s AUC and diagnostic'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 2, 'page_label': '3', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='forest plot. After that, we generated the HSROC curves \\nand their 95% prediction intervals via the HSROC model. \\nWe also calculated the HSROC’s AUC and diagnostic \\nodds ratios (DORs) to evaluate the overall accuracy.\\nAn  I2 statistic was used to explore the heterogeneity. \\nThere is potential heterogeneity when the  I2 is greater \\nthan 50%. Heterogeneity was also examined visually \\nthrough HSROC plots. In order to identify any sources \\nof heterogeneity, we conducted a meta regression when \\nthere were more than 10 single ML algorithms included. \\nThen, we followed with a subgroup analysis after identi -\\nfying any sources of heterogeneity. Finally, we conducted \\nsensitivity analysis, and combined the data after eliminat-\\ning outliers and data with small sample sizes to assess the \\nindex test’s stability. We used Deeks’ funnel plot asym -\\nmetry test to evaluate publication bias. And the data syn -\\nthesis was conducted with RevMan 5.2 and Stata (version'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 2, 'page_label': '3', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='index test’s stability. We used Deeks’ funnel plot asym -\\nmetry test to evaluate publication bias. And the data syn -\\nthesis was conducted with RevMan 5.2 and Stata (version \\n15.0), using the \"Metandi\" and \"Midas\" packages.\\nResults\\nWe retrieved a total of 184,052 articles from the litera -\\nture databases. After removing duplicates, we screened \\n180,958 records by title and abstract, and excluded'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 3, 'page_label': '4', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 4 of 16Lei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n180,612. Then, we evaluated the full texts of the remain -\\ning 188 articles based on the study eligibility criteria. \\nUltimately, 15 studies were included in our systematic \\nreview. However, we were unable to extract the specific \\nTP , FP , FN and TN data from 9 of the articles. Therefore, \\nonly 6 articles were eligible for data synthesis. A detailed \\nflow diagram with the study selection process and rea -\\nsons for exclusion is shown in Fig.\\xa01.\\nClinical application\\nThe total sample size of the 15 articles was 115,155, and \\nthe mean age was 59.28\\xa0 years old. 23 ML algorithms \\nFig. 1 PRISMA flow diagram of study selection process'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 4, 'page_label': '5', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 5 of 16\\nLei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n \\nwere used to construct progression models with 6 types \\nof predictors—demographics, comorbidities, laboratory \\ndata from blood and urine samples, renal biopsy pathol -\\nogy, and therapeutic regimen. These algorithms’ accuracy \\nvaried, as did the evaluation indexes used to evaluate the \\naccuracy (see Table\\xa01).\\nKidney disease types\\nThe various kidney diseases in the included articles could \\nbe classified into 3 categories: CKD, Immunoglobulin A \\nNephropathy (IgAN) and diabetic nephropathy. Studies \\non CKD (43.75%) and IgAN (37.5%) accounted for the \\nlargest proportions.\\nThe CKD sample size was 17,862, with a mean age of \\n61.93\\xa0years old and stage 3–4 in 7 articles. Utilization of \\nthe Logistics Regression (LR), Support Vector Machine \\n(SVM) and Random Forest (RF) algorithm accounted for \\nproportions of 15.3%, 10.7% and 9.2%, respectively. The'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 4, 'page_label': '5', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='the Logistics Regression (LR), Support Vector Machine \\n(SVM) and Random Forest (RF) algorithm accounted for \\nproportions of 15.3%, 10.7% and 9.2%, respectively. The \\nRF algorithm was the most accurate algorithm for pre -\\ndicting CKD prognosis and its AUC was 0.878.\\nThe IgAN sample size was 6127 with a mean age of \\n34.7\\xa0 years old in 6 articles. Utilization of the Artificial \\nNeural Network (ANN), LR and Decision Tree (DT) \\nalgorithms accounted for proportions of 29.0%, 19.3% \\nand 16.1%, respectively. The ANN algorithm was the \\nmost accurate algorithm for predicting IgAN prognosis, \\nwith an AUC of 0.933.\\nPredictors in\\xa0the\\xa0models\\nMost of the included ML algorithms used demography, \\ncomorbidities, and laboratory data from blood and urine \\nsamples as predictors, among which, age, sex, hyperten -\\nsion, serum creatinine and 24-h urinary protein were \\ncommon predictors. However, Dovgan [28] only used \\ncomorbidities to construct prediction models. Chen [26]'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 4, 'page_label': '5', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='sion, serum creatinine and 24-h urinary protein were \\ncommon predictors. However, Dovgan [28] only used \\ncomorbidities to construct prediction models. Chen [26] \\nand Schena [30] applied renal biopsy pathology and types \\nof drug therapy as predictors of ML algorithms to estab -\\nlish accurate prediction models. See Additional file\\xa0 1 for \\ndetails.\\nOutcome indicators\\nIn 12 studies, ESRD was the primary outcome and was \\ndefined as such, (1) eGFR < 15\\xa0 ml/min/1.73\\xa0  m2; (2) the \\ninitiation of RRT; (3) renal transplantation. In addition, \\nXiao et\\xa0al. [25] used the severity of proteinuria, Feng et\\xa0al. \\n[23] and Masaki et\\xa0al. [27] used the progression of CKD \\nstages as outcome indicators for renal disease progres -\\nsion, respectively.\\nMakeup of\\xa0the\\xa06 eligible articles\\nThe 6 eligible articles for data synthesis mainly focus on \\nCKD and IgAN with a total sample size of 12, 534 and \\na mean age of 42.77\\xa0 years old. 18 ML algorithms were \\nused to construct progression models with 6 types of'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 4, 'page_label': '5', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='CKD and IgAN with a total sample size of 12, 534 and \\na mean age of 42.77\\xa0 years old. 18 ML algorithms were \\nused to construct progression models with 6 types of \\npredictors mentioned above. Except for Xiao et\\xa0 al. [25], \\nthe other 5 researchers took ESRD/RRT as the primary \\noutcomes. Utilization of the ANN, LR, and RF algo -\\nrithm accounted for proportions of 22.5%, 6.4% and 6.4%, \\nrespectively. The optimal prognosis model for predict -\\ning IgAN progression was constructed by ANN algo -\\nrithm. And the optimal prognosis model for predicting \\nCKD progression was constructed by LR algorithm (see \\nTable\\xa01).\\nQuality assessment\\nWe assessed the studies’ quality with the QUADAS-2 \\ntool. Figure\\xa0 2 depicts the risk of bias graph, while Fig.\\xa0 3 \\npresents the risk of bias summary. Bias of the included \\nstudies comes primarily from the domains of Index test \\nand Flow and timing. Bias of application concerns per -\\ntained primarily to the Index test. Of all the included'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 4, 'page_label': '5', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='studies comes primarily from the domains of Index test \\nand Flow and timing. Bias of application concerns per -\\ntained primarily to the Index test. Of all the included \\nprimary studies, none of the articles were judged as “low \\nrisk” on all bias-related domains; 5 (33.3%) were judged \\nas “low concern” on all applicability domains.\\nResults of\\xa0data synthesis\\nExtracted data was synthesized using the bivariate model \\nand the HSROC model without accounting for possi -\\nble covariates explaining heterogeneity. The correlation \\ncoefficient was—0.53 and asymmetrical parameter β \\nwas 0.015 (P > 0.05) which indicated a trade-off between \\nsensitivity and specificity. The ML algorithms exhibited \\na pool sensitivity of 0.68 (0.58–0.77) and a pool specific -\\nity of 0.87 (0.84–0.90) (Fig.\\xa0 4A). The AUC of the HSROC \\ncurve was 0.87 (0.84–0.90) (Fig.\\xa0 5A), and the DOR was \\n16.34.\\nThe  I2 for pool sensitivity and specificity were 99.0% \\nand 99.7%, respectively, which indicated the poten -'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 4, 'page_label': '5', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='curve was 0.87 (0.84–0.90) (Fig.\\xa0 5A), and the DOR was \\n16.34.\\nThe  I2 for pool sensitivity and specificity were 99.0% \\nand 99.7%, respectively, which indicated the poten -\\ntial heterogeneity. Considering that there were multiple \\nsources of heterogeneity, we conducted a META regres -\\nsion and determined that kidney disease types, algorithm \\nand parameters, dataset, predictors and race were the \\ninfluential factors (See Fig.\\xa0 6). After that, we conducted a \\nsubgroup analysis.\\nAccording to the results of subgroup analysis based \\non kidney disease types, the ML algorithm’s AUC and \\nDOR for predicting CKD prognosis was 0.82 (0.79–0.85) \\nand 9.31, respectively. The pool sensitivity was 0.64 \\n(0.49–0.77) with an  I2 of 99.20%, and the pool specific -\\nity was 0.84 (0.74–0.91) with an  I2 of 99.84% (Figs.\\xa0 4B, \\n5B). The ML algorithm’s AUC and DOR for predicting \\nIgA nephropathy prognosis was 0.78 (0.74–0.81) and \\n39.27,respectively. The pool sensitivity was 0.74 (0.71–'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 4, 'page_label': '5', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='5B). The ML algorithm’s AUC and DOR for predicting \\nIgA nephropathy prognosis was 0.78 (0.74–0.81) and \\n39.27,respectively. The pool sensitivity was 0.74 (0.71–\\n0.77) with an  I2 of 7.10%, and the pool specificity was \\n0.93 (0.91–0.95) with an  I2 of 83.92% (Figs.\\xa0 4C, 5C). See'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 5, 'page_label': '6', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 6 of 16Lei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\nTable 1 Clinical characteristics of the included studies\\nStudies Country N Ages \\n(years)\\nMen (%) KD/\\nOutcome\\nFollow \\nup \\ntimes(Y)\\nReporting \\ndataset\\nML \\nalgorithm\\nOptimal \\nmodel\\nAUC Accuracy C statistic Analysis TP FP FN TN\\nGoto [19], \\n2009\\nJapan 790 28.14 41.8 IgAN/\\ndoubling of \\nSCR\\n10.0 Training DTs DTs 0.830 N/A N/A N/A N/A N/A N/A N/A\\nDiciolla \\n[20], 2015\\nItaly\\nNorway\\nJapan\\n1040 34.56 69.13 IgAN/ESRD N/A Training ANN\\nNFS\\nSVM\\nDTs\\nANN N/A 90.1 N/A (1) ANN\\n(2) NFS\\n(3) SVM\\n(4) DTs\\n177\\n175\\n180\\n178\\n35\\n63\\n97\\n67\\n64\\n66\\n61\\n63\\n764\\n736\\n702\\n732\\nPesce [21], \\n2016\\nItaly 546 32.5 68.13 IgAN/ESRD 5.0* Testing ANN ANN 89.9 91.8 N/A (1) ANN 14 2 7 87\\nNorway 441 38.2 72.56 IgAN/ESRD 6.0* Testing ANN ANN 93.3 92.1 N/A (2) ANN 23 3 4 59\\nJapan 53 32.7 50.94 IgAN/ESRD 3.0* Testing ANN ANN 100 90.9 N/A (3) ANN 1 1 0 9\\nCheng [22], \\n2017\\nTaiwan,\\nChina\\n463 65 N/A CKD/ESRD 1.5 Training C4.5 CART \\nSVM Ada-\\nboost\\nCART +'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 5, 'page_label': '6', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Japan 53 32.7 50.94 IgAN/ESRD 3.0* Testing ANN ANN 100 90.9 N/A (3) ANN 1 1 0 9\\nCheng [22], \\n2017\\nTaiwan,\\nChina\\n463 65 N/A CKD/ESRD 1.5 Training C4.5 CART \\nSVM Ada-\\nboost\\nCART + \\nAdaboost\\n0.715 0.662 N/A (1) C4.5\\n(2) CART \\n(3) SVM\\n73\\n76\\n80\\n57\\n48\\n45\\n59\\n56\\n52\\n75\\n84\\n87\\nFeng [23], \\n2018\\nChina 119 N/A N/A CKD/CKD \\nSTAGE\\nN/A Training DTs DTs N/A 0.9827 N/A N/A N/A N/A N/A N/A\\nLiu [24], \\n2018\\nChina 262 32.71 47.3 IgAN/ESRD 3.0 Training RF RF 0.9729 N/A N/A N/A N/A N/A N/A N/A\\nHelena [8], \\n2019\\nU.S 4640 60.19 N/A CKD/ESRD 3.7 Testing Lasso Lasso N/A N/A 0.863 N/A N/A N/A N/A N/A\\nXiao [25], \\n2019\\nChina 551 58.15 N/A CKD/PRO 3.0 Testing LR\\nElasticN\\nLasso Ridge \\nSVM\\nRF\\nKNN ANN \\nXGBoost\\nLR 0.873 0.82 N/A (1) LR\\n(2) ElasticN\\n(3) Lasso\\n(4) Ridge\\n(5) SVM\\n(6) RF\\n(7) KNN\\n(8) ANN’\\n(9)XGBoost\\n54\\n52\\n52\\n51\\n53\\n52\\n49\\n52\\n55\\n8\\n7\\n7\\n8\\n8\\n8\\n12\\n8\\n8\\n12\\n14\\n14\\n14\\n13\\n14\\n17\\n14\\n11\\n37\\n38\\n38\\n37\\n37\\n37\\n33\\n37\\n37\\nChen [26], \\n2020\\nChina 2047 34.80 49.25 IgAN/\\nESRD\\n5.01 Train-\\ning + Test-\\ning\\nLR SVM DTs \\nANN RF \\nXGBoost'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 5, 'page_label': '6', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='51\\n53\\n52\\n49\\n52\\n55\\n8\\n7\\n7\\n8\\n8\\n8\\n12\\n8\\n8\\n12\\n14\\n14\\n14\\n13\\n14\\n17\\n14\\n11\\n37\\n38\\n38\\n37\\n37\\n37\\n33\\n37\\n37\\nChen [26], \\n2020\\nChina 2047 34.80 49.25 IgAN/\\nESRD\\n5.01 Train-\\ning + Test-\\ning\\nLR SVM DTs \\nANN RF \\nXGBoost\\nXGBoost N/A N/A 0.89 N/A N/A N/A N/A N/A\\nMasaki [27], \\n2020\\nJapan 30,810 N/A N/A DKD/stage \\nof DKD\\n0..5 Training LR CNN LR 0.722 N/A N/A N/A N/A N/A N/A N/A'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 6, 'page_label': '7', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 7 of 16\\nLei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n \\nCKD chronic kidney disease, DKD diabetes kidney disease, IgAN Immunoglobulin A Nephropathy, ESRD end stage renal disease, RRT  renal replace therapy, PRO proteinuria, TP true positive, FP false positive, FN false \\nnegative, TN true negative, AUC  area under the curve, DTs Decision Trees, ANN artificial neural network, NFS neural fuzzy systems, SVM support vector machine, RF random forest, LR logistic regression, ElasticN Elastic Net, \\nKNN K Nearest Neighbors, NN Nearest Neighbors, SGD Stochastic Gradient Descent, BDTs Bagging Decision Trees, CNN convolutional neural network, CART  classification and regression, Lasso Lasso regression, Ridge Ridge \\nregression, NNET neural network, FNN Feed-forward neural network\\n*Median times\\nTable 1 (continued)\\nStudies Country N Ages \\n(years)\\nMen (%) KD/\\nOutcome\\nFollow \\nup \\ntimes(Y)\\nReporting \\ndataset\\nML \\nalgorithm\\nOptimal \\nmodel'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 6, 'page_label': '7', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='*Median times\\nTable 1 (continued)\\nStudies Country N Ages \\n(years)\\nMen (%) KD/\\nOutcome\\nFollow \\nup \\ntimes(Y)\\nReporting \\ndataset\\nML \\nalgorithm\\nOptimal \\nmodel\\nAUC Accuracy C statistic Analysis TP FP FN TN\\nDovgan \\n[28], 2020\\nTaiwan, \\nChina\\n8492 N/A N/A CKD/\\nRRT \\n1.0 Training LR XGBoost\\nSGD\\nSVM\\nBDTs\\nANN\\nRF\\nBayes\\nDTs\\nNN\\nLR 0.778 N/A N/A (1) LR\\n(2)XGBoost\\n(3) SGD\\n651\\n60\\n152\\n1628\\n52\\n164\\n394\\n985\\n893\\n5819\\n7395\\n7283\\nNagaraj \\n[29], 2020\\nNether-\\nlands\\n11,789 62.75 N/A DKD/ESRD 2.7 Testing LR SVM RF \\nFNN\\nFNN 0.84 N/A N/A N/A N/A N/A N/A N/A\\nSchena \\n[30], 2020\\nItaly 948 40.6 72.2 IgAN/ESRD 7.4 Training ANN ANN 0.89 0.83 N/A (1) ANN 36 8 6 117\\nYuan [31], \\n2020\\nChina 1090 50.01 56.3 CKD/ESRD 4.0 Training LR\\nRF\\nSVM\\nNNET\\nRF 0.878 N/A N/A N/A N/A N/A N/A N/A\\nZhou (9), \\n2020\\nChina 2507 70.6 N/A CKD/ESRD 3.0 Training LR LR N/A N/A 0.69 N/A N/A N/A N/A N/A'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 7, 'page_label': '8', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 8 of 16Lei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\nTable\\xa02, Additional file\\xa0 2 and Additional file\\xa0 3 for more \\ndetails of subgroup analysis.\\nWe found outliers by observing the forest plot and \\nHSROC curve. We performed sensitivity analysis and \\nreapplied the bivariate and HSROC model after exclud -\\ning the outliers. Which showed that ML algorithm’s AUC \\nand DOR was 0.83 (0.80–0.90) and 16.80, respectively. \\nThe pool sensitivity was 0.74 (0.70–0.77) with an  I2 of \\n74.93%, and the pool specificity was 0.86 (0.80–0.90) with \\nan  I2 of 99.84% (see Figs.\\xa0 4D, 5D). Additionally, we found \\nthat Deek’s funnel plot (Fig.\\xa0 7) was symmetrical, and \\nthere was no evidence of publication bias in asymmetric \\ntests (P = 0.07).\\nDiscussion\\nOur study indicates that ML algorithms did not pool a \\nbalance between sensitivity and specificity, which had \\nexceptional accuracy, with an AUC of 0.87, and strong \\nspecificity (0.88), but weak sensitivity (0.68) in predict -'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 7, 'page_label': '8', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='balance between sensitivity and specificity, which had \\nexceptional accuracy, with an AUC of 0.87, and strong \\nspecificity (0.88), but weak sensitivity (0.68) in predict -\\ning adverse outcomes, progress to ESRD or initiation of \\nRRT, among both CKD and IgAN patients. This result \\nindicates that recent ML algorithms have low misdiag -\\nnosis rates, but significant probability of missed diagno -\\nsis, which means that its ability to detect patients with \\nkidney function progression is not strong enough. ML \\nalgorithms need optimization because we aim to iden -\\ntify patients at risk. The main reason for the decrease in \\nsensitivity may be the low end-point incidence rate with \\ninsufficient follow up time. As shown in the results, only \\n16.3% of patients reached the end point. In previous stud-\\nies, a mean follow-up time of at least 5\\xa0 years has been \\nrequired to project whether patients with CKD would \\nprogress to ESRD [32]. However, in the studies included'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 7, 'page_label': '8', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='ies, a mean follow-up time of at least 5\\xa0 years has been \\nrequired to project whether patients with CKD would \\nprogress to ESRD [32]. However, in the studies included \\nfor data synthesis, the mean follow up times were just 1.5 \\nand 3.0\\xa0 years, in Cheng’ s and Xiao’ s research, respec -\\ntively. Sufficient follow-up time is needed to establish \\nprediction models.\\nThe technical superiority of ML algorithm-based pre -\\ndiction models over traditional models is well estab -\\nlished. Most previous models have performed well in \\ninternal validation, but their capacity for generalization is \\nuncertain because only a portion of the studies have been \\nexternally validated. As shown in the subgroup analy -\\nsis, we found that in the test set group, the pool sensi -\\ntivity was 0.79 and the pool specificity was 0.86, which \\nindicated an exceptional accuracy in external verifica -\\ntion. This is because the goal of machine learning is to fit \\nthe models with new samples [33]. Furthermore, in the'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 7, 'page_label': '8', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='indicated an exceptional accuracy in external verifica -\\ntion. This is because the goal of machine learning is to fit \\nthe models with new samples [33]. Furthermore, in the \\nprocess of using ML algorithms, it is necessary to first \\ndivide the data into a training set and a test set. After \\nlearning some potential rules from the training set, the \\nrudimentary model can be verified on the test set. In the \\nincluded studies, the validation set was used for verifica -\\ntion in modeling. However, not all articles reported the \\nresults of ML algorithms performed on validation sets, as \\nthere is no unified standard for reporting [34]. In order \\nto standardize the research reporting process, and to gain \\na more comprehensive understanding of ML algorithms, \\nwe suggest that future research report the results of both \\ntraining sets and validation sets. They also show a greater \\ncapacity for generalization than traditional statistical \\nmethods. However, the small sample size in Pesce’s study’'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 7, 'page_label': '8', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='training sets and validation sets. They also show a greater \\ncapacity for generalization than traditional statistical \\nmethods. However, the small sample size in Pesce’s study’ \\ns test set may have led to high accuracy.\\nHowever, when modeling, ML algorithm type should \\nbe chosen deliberately. Our study focused on CKD and \\nIgAN, ANN and XGBoost algorithm have been utilized \\nsuccessfully in the field of IgAN [14]. However, regard -\\nless of the studies on early stage CKD diagnosis [35–38], \\nthere is a lack of studies predicting its outcome. Our \\nstudy has shown that it is feasible to use ML algorithms \\nto build progression models for CKD. Recent studies have \\nfocused on the RF, LR and SVM algorithms which can \\nproduce accurate predictions and deserve further study. \\nAs for the strengths, the RF and SVM are classification \\nFig. 2 Risk of bias and application concerns graph for the included studies.Red, yellow and green frames correspond to high, unclear and low risk \\nof bias, respectively'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 8, 'page_label': '9', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 9 of 16\\nLei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n \\nalgorithms can produce a qualitative index which can \\nintuitively reflect the occurrence of an outcome event \\nby summarizing and classifying the data characteris -\\ntics. However, the LR algorithm can utilize as regression \\nalgorithm and can predict the probability of an outcome \\nevent. But they also have weaknesses. The probability \\ncannot be known when using the classification algorithm, \\nwhile the regression algorithm-based prediction models \\ncannot produce direct conclusions. Thus, it is necessary \\nto determine the cut-off point.\\nAdditionally, which predictors to use to construct \\nprediction models is undergoing debate. As shown in \\nour subgroup analysis, renal biopsy pathology plays an \\nimportant role in prognosis predicting. Studies that used \\npathology had a pool sensitivity of 0.71 (0.66–0.76) and a \\npool specificity of 0.89 (0.80–0.94). While those without'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 8, 'page_label': '9', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='important role in prognosis predicting. Studies that used \\npathology had a pool sensitivity of 0.71 (0.66–0.76) and a \\npool specificity of 0.89 (0.80–0.94). While those without \\nFig. 3 Risk of bias and application concerns summary for the included studies. (+) indicates low risk of bias, (?) indicates unclear risk of bias, (−) \\nindicates high risk of bias'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 9, 'page_label': '10', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 10 of 16Lei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\npathology had a pool sensitivity of 0.65 (0.46–0.81) and \\na pool specificity of 0.87 (0.78–0.93). Which indicate \\nthat high quality pathology data optimized the accuracy \\nof prediction models. However, only a small minority of \\npatients can provide pathology data. Because renal biopsy \\nis an invasive manipulation, for which not all patients \\nhave indication. Furthermore, there are great differences \\nin renal biopsy specimen preparation and diagnosis for \\nthere is no unified or standardized pathological diagnosis \\nmode.\\nHowever, there are evidences indicated that CKD \\nprognostic ML prediction models using laboratory data \\nfrom blood and urine samples are also accurate [22]. \\nMoreover, CKD patients have the most comorbidities \\nFig. 4 Coupled forest plots for sensitivity and specificity. A All single-unit ML algorithms. B CKD subgroup. C IgAN subgroup. D Sensitivity analysis'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 9, 'page_label': '10', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Fig. 4 Coupled forest plots for sensitivity and specificity. A All single-unit ML algorithms. B CKD subgroup. C IgAN subgroup. D Sensitivity analysis \\nafter eliminating outliers and data with small sample sizes. The gray square with a black point in the center showed study specific estimates of \\nsensitivity and specificity. The width of solid black line showed their 95% confidence intervals. The diamond at the bottom of the figure was a \\ncombination of single-unit ML algorithm.The center of diamond represented the point estimates, and the width of diamond represented 95% \\nconfidence intervals\\n(See figure on next page.)\\nFig. 5 HSROC curve with 95% confidence region and prediction region. A All single-unit ML algorithms with AUC of 0.87. B CKD subgroup with \\nAUC of 0.82. C IgAN subgroup with AUC of 0.78. D Sensitivity analysis after eliminating outliers and data with small sample sizes with AUC of 0.83.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 9, 'page_label': '10', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='AUC of 0.82. C IgAN subgroup with AUC of 0.78. D Sensitivity analysis after eliminating outliers and data with small sample sizes with AUC of 0.83. \\nEach circle represents a single-unit ML algorithm. The curve represents the summary receiver operating characteristic curve for all single-unit \\nML algorithm. The red square represents the summary estimate of test performance. The zone outlines represent the 95% confidence and 95% \\nprediction regions of the summary estimate, respectively'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 10, 'page_label': '11', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 11 of 16\\nLei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n \\nFig. 5 (See legend on previous page.)'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 11, 'page_label': '12', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 12 of 16Lei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n[39] correlated with disease prognosis, and there is evi -\\ndence of comorbidities’ effectiveness for modeling [9, 40]. \\nWe believe that with the development of the electronic \\nmedical record (EMR) system, [15] the quantity of \\ncomorbidity data will grow with its quality improves. \\nTherefore, exploring the use of laboratory data from \\nFig. 6 Univariate meta-regression plot of all single-unit ML algorithms. The red point represents the result of the individual combination of the \\nsubgroup into which each independent variable is divided. The width of solid black line showed their 95% confidence intervals. “*” means that the \\neffects of independent variables on the pool sensitivity and specificity were statistically significant'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 12, 'page_label': '13', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 13 of 16\\nLei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n \\nblood and urine samples and comorbidities as the predic-\\ntors for modeling.\\nAs for the outcomes, ESRD occurrence and the time \\nto start RRT were the end points which caught the most \\nattention from researchers. Researchers seem to have \\nTable 2 Summary of meta-analysis and subgroup analysis\\nDTA diagnostic test accuracy, KD kidney disease, CKD chronic kidney disease, IgAN Immunoglobulin A Nephropathy, ML machine learning, Y Yes, N No\\na P < 0.01\\nSubgroup Number of ML \\nalgorithms\\nSensitivity (95% CI) Specificity (95% CI) AUC (95% CI) Correlation \\ncoefficient\\nβ DOR\\nTotal DTA 23 0.68 (0.58–0.77) 0.88 (0.83–0.92) 0.87 (0.84–0.90) − 0.53 0.015 16.34\\nType of KD\\nCKD 15 0.64 (0.49–0.77) 0.84 (0.74–0.91) 0.82 (0.79–0.85) − 0.77 − 0.036 9.31\\nIgAN 8 0.74 (0.71–0.77) 0.93 (0.91–0.95) 0.78 (0.74–0.81) − 1.0 3.781 39.27\\nML algorithm type'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 12, 'page_label': '13', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Type of KD\\nCKD 15 0.64 (0.49–0.77) 0.84 (0.74–0.91) 0.82 (0.79–0.85) − 0.77 − 0.036 9.31\\nIgAN 8 0.74 (0.71–0.77) 0.93 (0.91–0.95) 0.78 (0.74–0.81) − 1.0 3.781 39.27\\nML algorithm type\\nClassification 16 0.64 (0.50–0.76) 0.87 (0.79–0.92) 0.84 (0.81–0.87) − 0.66 0.021 11.75\\nRegression 7 0.80 (0.74–0.84) 0.91 (0.86–0.95) N/A 1.0 6.044 41.09\\nDataset type\\nTraining set 11 0.56 (0.37–0.73) 0.90 (0.80–0.95) 0.83 (0.80–0.86) − 0.57 0.074 11.40\\nTesting set 12 0.79 (0.76–0.82) 0.86 (0.81–0.90) 0.81 (0.77–0.84) − 1.0 3.693 23.33\\nPathology\\nY 11 0.71 (0.66–0.76) 0.89 (0.80–0.94) N/A 1 1.086a 19.46\\nN 12 0.65 (0.46–0.81) 0.87 (0.78–0.93) 0.86 (0.83–0.89) − 0.53 − 0.172 12.92\\nRace\\nAsian 16 0.64 (0.49–0.77) 0.84 (0.75–0.91) 0.82 (0.79–0.86) − 0.76 − 0.042 9.53\\nNot Asian 7 0.74 (0.71–0.77) 0.93 (0.91–0.95) 0.78 (0.74–0.81) − 1 3.806 10.95\\nFig. 7 Deek’s funnel plot of all single-unit ML algorithms. Each circle represents a single-unit ML algorithm'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 13, 'page_label': '14', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 14 of 16Lei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\nbeen less concerned about Major Adverse Cardiovascu -\\nlar Events (MACE) and all-cause mortality. Considering \\nthat MACE is the leading cause of death in kidney dis -\\nease patients [7], we believe that using ML algorithms to \\npredict risk MACE occurrence is also meaningful.\\nAfter data synthesis, we found significant heterogeneity \\nbetween the studies. This may have been due to hetero -\\ngeneity of kidney diseases, algorithms and parameters, \\ndatasets, predictors or race. Thus, we should be cautious \\nwhen interpreting the results.\\nThe 15 studies in our systematic review have a mod -\\nerate to severe risk of bias in methodology. This was \\nbecause we did not have enough information to deter -\\nmine whether the researchers had interpreted the results \\nknowing patients’ outcome. And in some articles, some \\ndata was left out of the analysis because the dataset'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 13, 'page_label': '14', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='mine whether the researchers had interpreted the results \\nknowing patients’ outcome. And in some articles, some \\ndata was left out of the analysis because the dataset \\nneeded to be divided into a training set and a test set (see \\nAdditional file\\xa01 for more details).\\nHowever, we also found studies with low risk of bias. \\nFrom these, we found several prediction models with \\nhigh accuracy which had used common clinical data as \\npredictors. These included Diciolla’ s ANN model whose \\naccuracy was 0.901, Liu’ s RF model whose AUC was \\n0.9729, Chen’s XGBoost model whose C-Statistic was \\n0.89 and Yuan’ s RF model whose AUC was 0.878. Based \\non the results above, we offer several recommendations \\nfor clinicians. When predicting whether IgAN patients \\nwill progress to ESRD, we recommend either Diciolla’s \\nANN model or Chen’s XGBoost model, assuming we \\ncan obtain patients’ renal biopsy pathology data. How -\\never, when this data is unavailable, we recommend Liu’s'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 13, 'page_label': '14', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='ANN model or Chen’s XGBoost model, assuming we \\ncan obtain patients’ renal biopsy pathology data. How -\\never, when this data is unavailable, we recommend Liu’s \\nRF model. When predicting whether CKD patients will \\nprogress to ESRD, we suggest Yuan’s RF model. However, \\nnote that Yuan’s model is only suitable for CKD stage 3 \\npatients.\\nLimitations\\n(1) We found that the sources of heterogeneity were \\nmultifaceted, and the high heterogeneity persisted \\nafter subgroup analysis. Furthermore, the covari -\\nate had a significant influence on the pool specific -\\nity. We also believe that ML algorithm type is an \\nimportant source of heterogeneity. We found that \\nmultiple types of ML algorithms were utilized, but \\nfew studies focused on one. This makes it impos -\\nsible to collect enough data to evaluate the perfor -\\nmance of a specific type of ML algorithm. There -\\nfore, we cannot eliminate the heterogeneity, and \\nfurther studies are needed.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 13, 'page_label': '14', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='sible to collect enough data to evaluate the perfor -\\nmance of a specific type of ML algorithm. There -\\nfore, we cannot eliminate the heterogeneity, and \\nfurther studies are needed.\\n(2) We utilized data transformation during data extrac-\\ntion. This may have resulted in bias because most \\nof the included studies reported a mean accuracy \\nindex without specific TP , FP , FN or TN data.\\n(3) Considering the difference in generalizability, our \\nresults might not reflect the actual power of ML \\nalgorithms. This is because we synthesized data \\nextracted from the training set and test set, which \\nneed improvement in future studies.\\n(4) The combination of multiple ML algorithms is \\nsuperior to utilizing a single ML algorithm. How -\\never, we only synthesized data extracted from a \\nsingle ML algorithm. This may have caused us to \\nunderestimate the accuracy. This is because we can-\\nnot get enough data, since few studies have com -\\nbined more than two ML algorithms during mod -'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 13, 'page_label': '14', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='single ML algorithm. This may have caused us to \\nunderestimate the accuracy. This is because we can-\\nnot get enough data, since few studies have com -\\nbined more than two ML algorithms during mod -\\neling. Furthermore, the type of ML algorithms they \\nutilized for combination varied.\\n(5) The last search was performed on October 31, 2020. \\nAfter that, we spent almost 10\\xa0 months screening \\nthe retrieved studies, which could affect the timeli -\\nness of this study.\\nConclusion\\nML algorithms are a tool for unearthing the rules of big \\ndata, and prediction models which incorporate them \\nhave exceptional accuracy in predicting kidney disease \\npatients’ poor prognosis during clinical practice. The \\nuse of ML algorithms can help clinicians detect patients \\nat high risk of kidney function progression in the early \\nstages. In this way, they can receive treatment and man -\\nagement in time. In sum, we suggest the gradual incor -\\nporation of ML algorithm-based prediction models into'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 13, 'page_label': '14', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='stages. In this way, they can receive treatment and man -\\nagement in time. In sum, we suggest the gradual incor -\\nporation of ML algorithm-based prediction models into \\nclinical practice.\\nAbbreviations\\nML: Machine learning; HSROC model: The hierarchical summary receiver \\noperating characteristic model; CKD: Chronic kidney disease; IgAN: Immu-\\nnoglobulin A nephropathy; ESRD: End stage renal diseases; AUC : Area under \\ncurve; RRT : Renal replacement therapy; PRISMA: Preferred reporting items for \\nsystematic reviews and meta-analyses statement; eGFR: Estimated glomerular \\nfiltration rate; MeSH: Medical subject headings; TP: True positive; FP: False \\npositive; TN: True negative; FN: False negative; PPV: Positive predictive value; \\nNPV: Negative predictive value; QUADAS-2: Quality assessment of diagnostic \\naccuracy studies 2; DOR: Diagnostic odds ratios; LR: Logistics regression; SVM: \\nSupport vector machine; RF: Random forest; ANN: Artificial neural network; DT:'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 13, 'page_label': '14', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='accuracy studies 2; DOR: Diagnostic odds ratios; LR: Logistics regression; SVM: \\nSupport vector machine; RF: Random forest; ANN: Artificial neural network; DT: \\nDecision tree; EMR: Electronic medical record; DKD: Diabetes kidney disease; \\nPRO: Proteinuria; NFS: Neural fuzzy systems; ElasticN: Elastic net; KNN: K nearest \\nneighbors; NN: Nearest neighbors; SGD: Stochastic gradient descent; BDTs: \\nBagging decision trees; CNN: Convolutional neural network; CART : Classifica-\\ntion and regression; Lasso: Lasso regression; Ridge: Ridge regression; NNET: \\nNeural network; FNN: Feed-forward neural network; DTA: Diagnostic test \\naccuracy; KD: Kidney disease.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 15 of 16\\nLei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n \\nSupplementary Information\\nThe online version contains supplementary material available at https:// doi. \\norg/ 10. 1186/ s12911- 022- 01951-1.\\nAdditional file\\xa01: Method S1–S4. Search Strategies. Method S5. Exclu-\\nsion criteria for articles. Method S6. QUADAS-2 coding manual for primary \\nstudies included. Table\\xa0S1. QUADAS-2 gradings for each primary study. \\nTable\\xa0S2. Predictors used in each primary study.\\nAdditional file\\xa02: Figure S1. HSROC curve for classification algorithm \\ngroup. Figure S2. HSROC curve for regression algorithm group. Figure \\nS3. HSROC curve for training set group. Figure S4. HSROC curve for test \\nset group.\\nAdditional file\\xa03: Figure S5. HSROC curve for subgroup used renal \\nbiopsy pathology as a predictor. Figure S6. HSROC curve for subgroup \\ndid not use renal biopsy pathology as a predictor. Figure S7. HSROC curve \\nfor Asian group. Figure S8. HSROC curve for non-Asian group.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='did not use renal biopsy pathology as a predictor. Figure S7. HSROC curve \\nfor Asian group. Figure S8. HSROC curve for non-Asian group.\\nAcknowledgements\\nWe acknowledge professor Xusheng Liu, head of Department of Nephrology, \\nGuangdong Provincial Hospital of Chinese Medicine, for his indispensable \\nhelp in managing this project.\\nAuthor contributions\\nNL, XZ and YW designed the study and drafted the manuscript; NL, MW, BL \\nand XX conducted literature research, study selection, quality assessment and \\ndata extraction; NL, XZ and MZ conducted the statistical analysis; HC, YX, BX, \\nDZ, CD, LF and FT participated in data interpretation and critically revised the \\nmanuscript. All authors approved the final version of the manuscript.\\nFunding\\nThis study was supported by the National Key Research and Develop-\\nment Program of China: Establishment and Evaluation an Exposed Omics \\nbased Prediction Model on CKD Risk and Benefit Factors (Project No.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='ment Program of China: Establishment and Evaluation an Exposed Omics \\nbased Prediction Model on CKD Risk and Benefit Factors (Project No. \\n2019YFE0196300); Department of science and technology of Guangdong \\nProvincial project: Construction of popularization of science and effectiveness \\nevaluation tool for chronic kidney disease based on “Internet plus” (Project No. \\n2020A1414050048) and First-class universities and disciplines of the world and \\ncollaborative innovation of disciplines in high-level universities workgroup \\n(Project No. 2021xk66).\\nAvailability of data and materials\\nAll data generated or analysed during this study are included in this \\nmanuscript [and its supplementary information files], and available from the \\ncorresponding author upon reasonable request. Links of database analysed \\nin this manuscript. 1. PubMed, PubMed (http:// nih. gov/). 2. EMBASE, Embase. \\n3. Cochrane Central Register of Controlled Trials, Cochrane|Trusted evidence.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='in this manuscript. 1. PubMed, PubMed (http:// nih. gov/). 2. EMBASE, Embase. \\n3. Cochrane Central Register of Controlled Trials, Cochrane|Trusted evidence. \\nInformed decisions. Better health. 4. Chinese Biomedicine Literature Database, \\nhttps:// sinom ed. ac. cn/. 5. Chinese National Knowledge Infrastructure, https:// \\ncnki. net/. 6. Wanfang Database, https:// wanfa ngdata. com. cn/ index. html. 7. VIP \\nDatabase, https:// www. cqvip. com/.\\nDeclarations\\nEthics approval and consent to participate\\nAll methods were performed in accordance with the relevant guidelines and \\nregulations. But none sought as this was a systematic review of published \\nstudies.\\nConsent for publication\\nNot applicable.\\nCompeting interests\\nThe authors declare that they have no competing interests.\\nAuthor details\\n1 The Second Clinical Medical College of Guangzhou University of Chinese \\nMedicine, Guangzhou, China. 2 Department of Nephrology, Guangdong'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Author details\\n1 The Second Clinical Medical College of Guangzhou University of Chinese \\nMedicine, Guangzhou, China. 2 Department of Nephrology, Guangdong \\nProvincial Hospital of Chinese Medicine, The Second Affiliated Hospital \\nof Guangzhou University of Chinese Medicine, Guangzhou, China. 3 Chronic \\nDisease Management Department, Guangdong Provincial Hospital of Chinese \\nMedicine, The Second Affiliated Hospital of Guangzhou University of Chinese \\nMedicine, Guangzhou, China. \\nReceived: 15 December 2021   Accepted: 18 July 2022\\nReferences\\n 1. Coresh J, Turin TC, Matsushita K, et al. Decline in estimated gomerular \\nfiltration rate and subsequent risk of end stage renal diseases and mortal-\\nity. JAMA. 2014;311(24):2518–31. https:// doi. org/ 10. 1001/ jama. 2014. 6634.\\n 2. Jhac V, Garcia G, Iseki K, et al. Chronic kidney disease: global dimension \\nand perspective. Lancet. 2013;382:260–72. https:// doi. org/ 10. 1016/ \\nS0140- 6736(13) 60687- X1.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='2. Jhac V, Garcia G, Iseki K, et al. Chronic kidney disease: global dimension \\nand perspective. Lancet. 2013;382:260–72. https:// doi. org/ 10. 1016/ \\nS0140- 6736(13) 60687- X1.\\n 3. World Health Organization. World Health Statistics 2019 Monitoring \\nHealth for The SDGs, Sustainable Development Goals. Geneva: World \\nHealth Organization; 2019.Licence: CCBY-NC-SA3.0IGO.https:// apps. who. \\nint/ iris/ bitst ream/ handle/ 10665/ 324835/ 97892 41565 707- eng. pdf? seque \\nnce= 9& isAll owed=y.\\n 4. GBD Chronic Kidney Disease Collaboration. Global, regional, and national \\nburden of chronic kidney disease, 1990–2017: a systematic analysis for \\nthe Global Burden of Disease Study. Lancet. 2017;2020:1–25. https:// doi. \\norg/ 10. 1016/ S0140- 6736(19) 32977-0.\\n 5. Scott J, Danile E, Andrew S, et al. National kidney foundation’s primer on \\nkidney disease. 7th ed. New York City: Elsevier; 2018. p. 2–18.\\n 6. Lo LJ, Go AS, Chertow GM, et al. Dialysis-requiring acute renal failure'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='kidney disease. 7th ed. New York City: Elsevier; 2018. p. 2–18.\\n 6. Lo LJ, Go AS, Chertow GM, et al. Dialysis-requiring acute renal failure \\nincreases the risk of progressive chronic kidney disease. Kidney Int. \\n2009;76:9–893. https:// doi. org/ 10. 1038/ ki. 2009. 289.\\n 7. United Stated Renal Data System. 2018 USRDS annual data report: execu-\\ntive summary. Am J kidney Dis. 2019;73:A9–22. https:// doi. org/ 10. 1053/j. \\najkd. 2019. 01. 002.\\n 8. Helena U, Zacharias MA, et al. A novel metabolic signature to predict the \\nrequirement of dialysis or renal transplantation in patients with chronic \\nkidney disease. J Proteome Res. 2018;2:1–42. https:// doi. org/ 10. 1021/ acs. \\njprot eome. 8b009 83.\\n 9. Fang Z, Avrum G, Djordje G, Jelena G, Zoran O. Use of disease embedding \\ntechnique to predict the risk of progression to end-stage renal disease. \\nJ Biomed Inform. 2020;105:103409. https:// doi. org/ 10. 1016/j. jbi. 2020. \\n103409.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='technique to predict the risk of progression to end-stage renal disease. \\nJ Biomed Inform. 2020;105:103409. https:// doi. org/ 10. 1016/j. jbi. 2020. \\n103409.\\n 10. KDIGO workgroup. KDIGO2012 clinical practice guideline for the evalu-\\nation and management of chronic kidney disease. Kidney Int Suppl. \\n2013;3:1–150.\\n 11. UK National Institute of Health and Clinical Excellence. chronic kidney \\ndisease early identification and management of chronic kidney disease in \\nadults in primary and secondary care. NICE Clin Guidel. 2014;182:1–59.\\n 12. KDOQI Workgroup. KDOQI clinical practice guideline for nutrition in CKD: \\n2020 update. Am J Kidney Dis. 2020;76:S1–107.\\n 13. Navdeep T, Georgios DK, Lesley AI, et al. Risk prediction models for \\npatients with chronic kidney disease: a systematic review. Ann Intern \\nMed. 2013;158:596–603. https:// doi. org/ 10. 7326/ 0003- 4819- 158-8- 20130 \\n4160- 00004.\\n 14. Chava LR, Ype J, Friedo WD, Merel D. Towards the best kidney failure'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Med. 2013;158:596–603. https:// doi. org/ 10. 7326/ 0003- 4819- 158-8- 20130 \\n4160- 00004.\\n 14. Chava LR, Ype J, Friedo WD, Merel D. Towards the best kidney failure \\nprediction tool: a systematic review and selection aid. Nephrol Dial Trans-\\nplant. 2020;35:1527–38. https:// doi. org/ 10. 1093/ ndt/ gfz018.\\n 15. Peter BJ, Lars JJ, Søren B. Mining electronic health records: towards better \\nresearch applications and clinical care. Nat Rev Genet. 2012;13:395–405. \\nhttps:// doi. org/ 10. 1038/ nrg32 08.\\n 16. Jordan MI, Mitchell TM. Machine learning: trends, perspectives, and pros-\\npects. Science. 2015;349:255–60. https:// doi. org/ 10. 1126/ scien ce. aaa84 \\n15.\\n 17. van Enst WA, Ochodo E, Scholten RJ, Hooft L, Leeflang MM. Investigation \\nof publication bias in meta-analyses of diagnostic test accuracy: a meta-\\nepidemiological study. BMC Med Res Methodol. 2014;14:70. https:// doi. \\norg/ 10. 1186/ 1471- 2288- 14- 70.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 14, 'page_label': '15', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='of publication bias in meta-analyses of diagnostic test accuracy: a meta-\\nepidemiological study. BMC Med Res Methodol. 2014;14:70. https:// doi. \\norg/ 10. 1186/ 1471- 2288- 14- 70.\\n 18. Macaskill P , Gatsonis C, Deeks JJ, Harbord RM, Takwoingi Y. Chapter 10: \\nAnalysing and Presenting Results. In: Deeks JJ, Bossuyt PM, Gatsonis C \\n(editors), Cochrane Handbook for Systematic Reviews of Diagnostic Test'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 15, 'page_label': '16', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='Page 16 of 16Lei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n•\\n \\nfast, convenient online submission\\n •\\n  \\nthorough peer review by experienced researchers in your ﬁeld\\n• \\n \\nrapid publication on acceptance\\n• \\n \\nsupport for research data, including large and complex data types\\n•\\n  \\ngold Open Access which fosters wider collaboration and increased citations \\n \\nmaximum visibility for your research: over 100M website views per year •\\n  At BMC, research is always in progress.\\nLearn more biomedcentral.com/submissions\\nReady to submit y our researc hReady to submit y our researc h  ?  Choose BMC and benefit fr om: ?  Choose BMC and benefit fr om: \\nAccuracy Version 1.0. The Cochrane Collaboration, 2010. Available from: \\nhttp:// srdta. cochr ane. org/.\\n 19. Goto M, Kawamura T, Wakai K, Ando M, Endoh M, Tomino Y. Risk stratifica-\\ntion for progression of IgA nephropathy using a decision tree induction'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 15, 'page_label': '16', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='http:// srdta. cochr ane. org/.\\n 19. Goto M, Kawamura T, Wakai K, Ando M, Endoh M, Tomino Y. Risk stratifica-\\ntion for progression of IgA nephropathy using a decision tree induction \\nalgorithm. Nephrol Dial Transpl. 2009;24:1242–7. https:// doi. org/ 10. 1093/ \\nndt/ gfn610.\\n 20. Diciolla M, Binetti G, Di Noia T, et al. Patient classification and outcome \\nprediction in IgA nephropathy. Comput Biol Med. 2015;66:278–86. \\nhttps:// doi. org/ 10. 1016/j. compb iomed. 2015. 09. 003.\\n 21. Francesco P , Mattea D, Giulio B, et al. Clinical decision support system for \\nend stage kidney disease risk estimation in IgA nephropathy patients. \\nNephrol Dial Transplant. 2015. https:// doi. org/ 10. 1093/ ndt/ gfv232.\\n 22. Li-Chen C, Ya-Han H, Shr-Han C. Applying the temporal abstraction tech-\\nnique to the prediction of chronic kidney disease progression. J Med Syst. \\n2017;41:85–97. https:// doi. org/ 10. 1007/ s10916- 017- 0732-5.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 15, 'page_label': '16', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='nique to the prediction of chronic kidney disease progression. J Med Syst. \\n2017;41:85–97. https:// doi. org/ 10. 1007/ s10916- 017- 0732-5.\\n 23. Miao F, Xiaorong Q, Zhi L. Progression prediction model of chronic kidney \\ndisease based on decision tree ant path optimization and logistic regres-\\nsion. Jisuanji yu Xiandaihua. 2018;272:117–21.\\n 24. Yexin L, Yan Z, Di L, et al. Prediction of ESRD in IgA nephropathy patients \\nfrom an Asian Cohort: a random forest model. Kidney Blood Press Res. \\n2018;43:1852–64. https:// doi. org/ 10. 1159/ 00049 5818.\\n 25. Jing X, Ruifeng D, Xiulin X, et al. Comparison and development of \\nmachine learning tools in the prediction of chronic kidney disease \\nprogression. J Transl Med. 2019;17:119. https:// doi. org/ 10. 1186/ \\ns12967- 019- 1860-0.\\n 26. Tingyu C, Xiang L, Yingxue L, et al. Prediction and risk stratification of \\nkidney outcomes in IgA nephropathy. Am J kidney Dis. 2019;74:1–10. \\nhttps:// doi. org/ 10. 1053/j. ajkd. 2019. 02. 016.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 15, 'page_label': '16', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='kidney outcomes in IgA nephropathy. Am J kidney Dis. 2019;74:1–10. \\nhttps:// doi. org/ 10. 1053/j. ajkd. 2019. 02. 016.\\n 27. Masaki O, Takayuki K, Masaki M, Kyoichi H, Atsushi S, Reitaro T. Feature \\nset for a prediction model of diabetic kidney disease progression. Stud \\nHealth Technol Inform. 2020;270:1289–90. https:// doi. org/ 10. 3233/ SHTI2 \\n00406.\\n 28. Erik D, Anton G, Mitja L, et al. Using machine learning models to predict \\nthe initiation of renal replacement therapy among chronic kidney disease \\npatients. PLoS ONE. 2020;15(6):e0233976. https:// doi. org/ 10. 1371/ journ al. \\npone. 02339 76.\\n 29. Sunil BN, Michelle JP , Wenjun J, Hiddo LH. Machine learning based early \\nprediction of end-stage renal disease in patients with diabetic kidney \\ndisease using clinical trials data. Diabetes Obes Metab. 2020;22:2479–86. \\nhttps:// doi. org/ 10. 1111/ dom. 14178.\\n 30. Francesco PS, Vito WA, Joseph T, et al. Development and testing of an arti-'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 15, 'page_label': '16', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='disease using clinical trials data. Diabetes Obes Metab. 2020;22:2479–86. \\nhttps:// doi. org/ 10. 1111/ dom. 14178.\\n 30. Francesco PS, Vito WA, Joseph T, et al. Development and testing of an arti-\\nficial intelligence tool for predicting end stage kidney disease in patients \\nwith immunoglobulin A nephropathy. Kidney Int. 2020;46:1–26. https:// \\ndoi. org/ 10. 1016/j. kint. 2020. 07. 046.\\n 31. Qiongjing Y, Haixia Z, Yanyun X, et al. Development of prognostic model \\nfor patients at CKD Stage 3a and 3b in South Central China using compu-\\ntational intelligence. Clin Exp Nephrol. 2020;24:865–75. https:// doi. org/ \\n10. 1007/ s10157- 020- 01909-5.\\n 32. Ming-Hsien T, Chen-Yang H, Ming-Yen L, et al. Incidence, prevalence, and \\nduration of chronic kidney disease in taiwan: results from a community-\\nbased screening program of 106,094 individuals. Nephron. 2018;140:175–\\n84. https:// doi. org/ 10. 1159/ 00049 1708.\\n 33. Zhou ZH. Machine learning. Beijing: Tsinghua University Press; 2016. p. \\n1–409.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 15, 'page_label': '16', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='84. https:// doi. org/ 10. 1159/ 00049 1708.\\n 33. Zhou ZH. Machine learning. Beijing: Tsinghua University Press; 2016. p. \\n1–409.\\n 34. Jie MA, Collins GS, Verbakel EW, et al. A systematic review shows no per-\\nformance benefit of machine learning over logistic regression for clinical \\nperdition models. J Clin Epidemiol. 2019;110:12–22.\\n 35. Adeola AO, Wangqing G. XGBoost model for chronic kidney disease \\ndiagnosis. IEEE/ACM Trans Comput Biol Bioinf. 2020;17:2131–40. https:// \\ndoi. org/ 10. 1109/ TCBB. 2019. 29110 71.\\n 36. Njoud AA, Hajra FS, Nuha RK, et al. Neural network and support vector \\nmachine for the prediction of chronic kidney disease: a comparative \\nstudy. Comput Biol Med. 2019;109:101–11. https:// doi. org/ 10. 1016/j. \\ncompb iomed. 2019. 04. 017.\\n 37. Akben SB. Early stage chronic kidney disease diagnosis by applying data \\nmining methods to urinalysis. Blood Anal Dis Hist IRBM. 2018;39:353–8. \\nhttps:// doi. org/ 10. 1016/j. irbm. 2018. 09. 004.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Adobe InDesign 15.0 (Windows)', 'creationdate': '2022-08-01T07:19:39+00:00', 'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-08-01T10:44:42+02:00', 'trapped': '/False', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1', 'author': 'Nuo Lei', 'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s12911-022-01951-1', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf', 'total_pages': 16, 'page': 15, 'page_label': '16', 'source_file': 's12911-022-01951-1.pdf', 'file_type': 'pdf'}, page_content='mining methods to urinalysis. Blood Anal Dis Hist IRBM. 2018;39:353–8. \\nhttps:// doi. org/ 10. 1016/j. irbm. 2018. 09. 004.\\n 38. Zewei C, Zhuoyong Z, Ruohua Z, Yuhong X, Peter BH. Diagnosis of \\npatients with chronic kidney disease by using two fuzzy classifiers. \\nChemom Intell Lab Syst. 2016;153:140–5. https:// doi. org/ 10. 1016/j. chemo \\nlab. 2016. 03. 004.\\n 39. Marcello T, Natasha W, Braden JM, et al. Comparison of the complexity \\nof patients seen by different medical subspecialists in a universal health \\ncare system. JAMA Netw Open. 2018;1:e184 852-e184 852. https:// doi. \\norg/ 10. 1001/ jaman etwor kopen. 2018. 4852.\\n 40. Anrew LB, Isaac SK. Big data and machine learning in health care. JAMA. \\n2018;319:1317. https:// doi. org/ 10. 1001/ jama. 2017. 18391.\\nPublisher’s Note\\nSpringer Nature remains neutral with regard to jurisdictional claims in pub-\\nlished maps and institutional affiliations.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 0, 'page_label': '1101', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Vol.:(0123456789)1 3\\nJournal of Nephrology (2023) 36:1101–1117 \\nhttps://doi.org/10.1007/s40620-023-01573-4\\nSYSTEMATIC REVIEWS\\nPredict, diagnose, and\\xa0treat chronic kidney disease with\\xa0machine \\nlearning: a\\xa0systematic literature review\\nFrancesco\\xa0Sanmarchi1\\xa0· Claudio\\xa0Fanconi2,3\\xa0· Davide\\xa0Golinelli1\\xa0· Davide\\xa0Gori1\\xa0· Tina\\xa0Hernandez‑Boussard2\\xa0· \\nAngelo\\xa0Capodici1,2 \\nReceived: 6 August 2022 / Accepted: 1 January 2023 / Published online: 14 February 2023 \\n© The Author(s) 2023, corrected publication 2023\\nAbstract\\nObjectives In this systematic review we aimed at assessing how artificial intelligence (AI), including machine learning (ML) \\ntechniques have been deployed to predict, diagnose, and treat chronic kidney disease (CKD). We systematically reviewed \\nthe available evidence on these\\xa0innovative techniques to improve CKD diagnosis and patient management.\\nMethods We included English language studies retrieved from PubMed.\\xa0The review is\\xa0therefore to be classified as a “rapid'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 0, 'page_label': '1101', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Methods We included English language studies retrieved from PubMed.\\xa0The review is\\xa0therefore to be classified as a “rapid \\nreview”, since it includes one database only, and has language restrictions; the novelty and importance\\xa0of the issue make \\nmissing relevant papers unlikely. We extracted 16 variables, including: main aim, studied population, data source, sample \\nsize, problem type (regression, classification), predictors used, and performance metrics.\\xa0We followed\\xa0the Preferred Report-\\ning Items for Systematic Reviews (PRISMA) approach; all main steps were done in duplicate.\\xa0\\nResults From a total of 648 studies initially retrieved, 68 articles met the inclusion\\xa0criteria.\\nModels, as reported by authors, performed well, but the reported metrics were not homogeneous across articles and therefore \\ndirect comparison was not feasible. The most common aim was prediction of prognosis, followed by diagnosis of CKD.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 0, 'page_label': '1101', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='direct comparison was not feasible. The most common aim was prediction of prognosis, followed by diagnosis of CKD. \\nAlgorithm generalizability, and testing on diverse populations was rarely taken into account. Furthermore, the clinical \\nevaluation and validation of the models/algorithms was perused; only a fraction of the included studies, 6 out of 68, were \\nperformed in a clinical context.\\nConclusions Machine learning is a\\xa0promising tool\\xa0for the prediction of risk, diagnosis, and therapy management for \\nCKD\\xa0patients. Nonetheless, future work is needed to address the interpretability, generalizability, and fairness of the models \\nto ensure the safe application of such technologies in routine clinical practice.\\n * Angelo Capodici \\n angelo.capodici@studio.unibo.it\\n1 Department of\\xa0Biomedical and\\xa0Neuromotor Science, Alma \\nMater Studiorum, University of\\xa0Bologna, Via San Giacomo \\n12, 40126\\xa0Bologna, Italy\\n2 Department of\\xa0Medicine (Biomedical Informatics), Stanford'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 0, 'page_label': '1101', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1 Department of\\xa0Biomedical and\\xa0Neuromotor Science, Alma \\nMater Studiorum, University of\\xa0Bologna, Via San Giacomo \\n12, 40126\\xa0Bologna, Italy\\n2 Department of\\xa0Medicine (Biomedical Informatics), Stanford \\nUniversity, School of\\xa0Medicine, Stanford, CA, USA\\n3 Department of\\xa0Electrical Engineering and\\xa0Information \\nTechnology, ETH Zurich, Zurich, Switzerland'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 1, 'page_label': '1102', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1102 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\nGraphical abstract\\nKeywords Chronic kidney disease\\xa0· Machine learning\\xa0· Artificial intelligence\\xa0· Systematic review'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 2, 'page_label': '1103', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1103Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\nIntroduction\\nChronic Kidney Disease (CKD) is a state of progressive loss \\nof kidney function ultimately resulting in the need for renal \\nreplacement therapy (dialysis or transplantation) [1 ]. It is \\ndefined as the presence of kidney damage or an estimated \\nglomerular filtration rate less than 60\\xa0ml/min per 1.73  m2, \\npersisting for 3\\xa0months or more [2]. CKD prevalence is grow-\\ning worldwide, along with demographic and epidemiologi-\\ncal transitions [3]. The implications of this disease are enor-\\nmous for our society in terms of quality of life and the overall \\nsustainability of national health systems. Worldwide, CKD \\naccounted for 2,968,600 (1%) disability-adjusted life-years \\nand 2,546,700 (1% to 3%) life-years lost in 2012 [4]. There-\\nfore, it is of the utmost importance to assess how to promptly \\nand adequately diagnose and treat patients with CKD.\\nThe causes of CKD vary globally. The most common pri-'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 2, 'page_label': '1103', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='fore, it is of the utmost importance to assess how to promptly \\nand adequately diagnose and treat patients with CKD.\\nThe causes of CKD vary globally. The most common pri-\\nmary diseases causing CKD and ultimately kidney failure are \\ndiabetes mellitus, hypertension, and primary glomerulone-\\nphritis, representing 70–90% of the total primary causes [1, \\n2, 4]. Although these three causes are at the top of the CKD \\netiology charts, other features are involved in CKD pathophys-\\niology (e.g., pollution, infections and autoimmune diseases) \\n[5–9]. Similarly, there are numerous factors that play a role in \\nCKD progression, namely non-modifiable risk factors (e.g., \\nage, gender, ethnicity) and modifiable ones (e.g., systolic and \\ndiastolic blood pressure, proteinuria) [1, 2, 4–9].\\nGiven how dauntingly vast the number of factors that can \\nplay a significant role in the etiology and progression of CKD \\nis, it can be difficult to correctly assess the individual risk of'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 2, 'page_label': '1103', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Given how dauntingly vast the number of factors that can \\nplay a significant role in the etiology and progression of CKD \\nis, it can be difficult to correctly assess the individual risk of \\nCKD and its progression. Naturally, as with any complex prob-\\nlem, humans seek simplification, and therefore the question \\nshifts to what to take into account when assessing CKD risk. \\nThanks to new methodological techniques, we now have the \\nability to improve our diagnostic and predictive capabilities.\\nArtificial Intelligence (AI) is the capacity of human-built \\nmachines to manifest complex decision-making or data analy-\\nsis in a similar or augmented fashion in comparison to human \\nintelligence [10]. Machine Learning (ML) is the collection of \\nalgorithms that empower models to learn from data, and there-\\nfore to undertake complex tasks through complex calculations \\n[11–15]. In recent years AI and ML have offered enticing solu-\\ntions to clinical problems, such as how to perform a diagnosis'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 2, 'page_label': '1103', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='fore to undertake complex tasks through complex calculations \\n[11–15]. In recent years AI and ML have offered enticing solu-\\ntions to clinical problems, such as how to perform a diagnosis \\nfrom sparse and seemingly contrasting data, or how to predict \\na prognosis [16]. Given the enormous potential of ML, and \\nits capacity to learn from data, researchers have tried to apply \\nits capacities to resolve complex problems, such as predicting \\nCKD diagnosis and prognosis, and managing its treatment.\\nIn this complex scenario, we aimed to systematically \\nreview the published studies that applied machine learning \\nin the diagnosis and prediction, prognosis, and treatment of \\nCKD patients. In doing so, the primary objective is to describe \\nhow ML models and variables have been used to predict, \\ndiagnose and treat CKD, as well as what results have been \\nachieved in this field.\\nMethods\\nSearch strategy and\\xa0selection criteria\\nWe conducted a systematic literature review, following the'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 2, 'page_label': '1103', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='diagnose and treat CKD, as well as what results have been \\nachieved in this field.\\nMethods\\nSearch strategy and\\xa0selection criteria\\nWe conducted a systematic literature review, following the \\nPreferred Reporting Items for Systematic Reviews (PRISMA) \\napproach [17], including studies that applied ML algorithms \\nto CKD forecasting, diagnosis, prognosis, and treatment. This \\nsystematic review’s outcomes of interest are machine learn-\\ning models, features used, performances and uses regarding \\ndiagnosis, prognosis and treatment of CKD. The review itself \\nand its protocol were not registered.\\nThe initial search was implemented on October 20, 2021. \\nThe search query consisted of terms considered pertinent by \\nthe authors.\\nWe searched for publications on PubMed using the follow-\\ning search string: “((artificial intelligence[Title/Abstract]) OR \\n(machine learning[Title/Abstract]) OR (computational*[Title/\\nAbstract]) OR (deep learning[Title/Abstract])) AND ((ckd)'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 2, 'page_label': '1103', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='ing search string: “((artificial intelligence[Title/Abstract]) OR \\n(machine learning[Title/Abstract]) OR (computational*[Title/\\nAbstract]) OR (deep learning[Title/Abstract])) AND ((ckd) \\nOR (chronic kidney disease) OR (chronic kidney injury) OR \\n(chronic kidney) OR (chronic renal) OR (end stage renal) OR \\n(end stage kidney) OR (ESKD) OR (ESRD) OR (CKJ) OR \\n(CKI) OR (((renal) OR (kidney)) AND (failure)))”.\\nWe included articles for review if they were in\\xa0vivo studies \\n(human-based), which applied AI & ML techniques in order \\nto assess the diagnosis, prognosis, or therapy of CKD patients \\nand reported original data. We did not limit our inclusion cri-\\nteria to any specific study design, nor to any outcome of inter-\\nest, as our main goal was to be as inclusive as possible, and \\nwe wanted to capture all available evidence from any study \\ndesign and any outcome of interest.\\nWe excluded studies that were not in English, those focus-'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 2, 'page_label': '1103', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='we wanted to capture all available evidence from any study \\ndesign and any outcome of interest.\\nWe excluded studies that were not in English, those focus-\\ning on animals, reviews, systematic reviews, opinions, editori-\\nals, and case reports. We decided to exclude in\\xa0vitro studies \\n(conducted on cellular substrates) and studies focusing on \\nanimals, in order to summarize the current evidence on the \\napplication of ML models on humans.\\nData extraction\\nData were extracted by two independent reviewers (AC and \\nFS). Disagreement on extracted data was discussed with an \\nindependent arbiter (DGol).\\nThe following data were extracted from each included article \\n(main text and/or supplementary material): author(s) name, date \\nof publication, first author affiliation (country and region), main \\nstudy objective, objective category (risk, diagnosis, progno-\\nsis, and treatment), prognosis category, study population, data \\nsource, sample size, problem type (regression, classification),'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 3, 'page_label': '1104', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1104 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\nmachine learning algorithms examined in the study, predictor \\ncategories, number of predictors used, predictor list, perfor-\\nmance metrics, final conclusions, use in clinical context and the \\n5 most important model features. When more than one model \\nwas considered in the study, the one the authors deemed best \\nwas extracted. Performance metrics always refer to the models’ \\nperformance on test sets.\\nQuality and\\xa0risk assessment\\nEvaluation of the included studies was performed using \\nboth PROBAST [18] and the Guidelines for developing and \\nreporting machine learning predictive models in biomedical \\nresearch developed by Luo and colleagues [19].\\nResults\\nIncluded studies\\nOf the 648 articles retrieved from PubMed, 421 were ruled \\nout after title screening, and 140 were excluded after abstract \\nscreening; a total of 87 articles were selected for full-text \\nscreening (Fig.\\xa01). Of these 87 studies, 68 were included in'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 3, 'page_label': '1104', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='out after title screening, and 140 were excluded after abstract \\nscreening; a total of 87 articles were selected for full-text \\nscreening (Fig.\\xa01). Of these 87 studies, 68 were included in \\nthe final set of articles (Table\\xa01) [20–87].\\nFig. 1  PRISMA flow-chart'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 4, 'page_label': '1105', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1105Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\nTable 1  Extracts of the main findings\\nAuthors Country Year Sample size Main Aim Model task Selected model Performance \\nmetric(s)\\nClinical \\nDeploy-\\nment\\nAkl et\\xa0al Africa 2001 30 Prognosis Regression ANN – No\\nKusiak et\\xa0al North America 2005 188 Prognosis Classification Decision tree Accuracy: 85 No\\nChen et\\xa0al Asia 2007 153 Prognosis Regression ANN – No\\nLuo et\\xa0al North America 2013 66,633 Prognosis Classification Hidden Markov \\nModel\\n– No\\nEscandell-\\nMontero et\\xa0al\\nEurope 2014 128 Therapy Regression Markov decision \\nprocesses\\n– No\\nMartínez-Mar-\\ntínez et\\xa0al\\nEurope 2014 13,011 Prognosis Regression Ensemble model MSE: 0.90\\nMAE: 0.67\\nNo\\nBarbieri et\\xa0al Europe 2015 4135 Prognosis Regression ANN MSE: 0.75\\nMAE: 0.55\\nNo\\nSingh et\\xa0al North America 2015 6435 Prognosis Classification ANN AUC: 0.72\\nSensitivity: 54\\nNo\\nBarbieri et\\xa0al Europe 2016 752 Therapy Regression ANN – Yes\\nChen et\\xa0al Asia 2016 400 Risk/forecast Classification Support vector \\nmachine'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 4, 'page_label': '1105', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Sensitivity: 54\\nNo\\nBarbieri et\\xa0al Europe 2016 752 Therapy Regression ANN – Yes\\nChen et\\xa0al Asia 2016 400 Risk/forecast Classification Support vector \\nmachine\\nAccuracy: 99\\nSensitivity: 100\\nSpecificity: 99\\nNo\\nNorouzi et\\xa0al Asia 2016 465 Diagnosis + Prog-\\nnosis\\nRegression ANN MSE: 54.88\\nMAE: 5.50\\nNo\\nRodriguez et\\xa0al Europe 2016 1758 Prognosis Regression Random forest – No\\nGoldstein et\\xa0al North America 2017 18,846 Prognosis Classification LASSO regres-\\nsion\\nAUC: 0.84 No\\nPolat et\\xa0al Europe 2017 400 Diagnosis Classification Support vector \\nmachine\\nAUC: 0.99\\nSensitivity: 98\\nNo\\nKleiman et\\xa0al North America 2018 401 Prognosis Classification Random forest AUC: 0.86\\nAccuracy: 54\\nNo\\nKolachalama et\\xa0al North America 2018 171 Diagnosis Classification CNN AUC: 0.91 No\\nTang et\\xa0al Asia 2018 173 Risk/forecast Classification Random forest – No\\nAkbilgic et\\xa0al North America 2019 27,615 Prognosis Classification Random forest AUC: 0.69 Yes'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 4, 'page_label': '1105', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Tang et\\xa0al Asia 2018 173 Risk/forecast Classification Random forest – No\\nAkbilgic et\\xa0al North America 2019 27,615 Prognosis Classification Random forest AUC: 0.69 Yes\\nAlmansour et\\xa0al Africa 2019 400 Diagnosis Classification ANN Accuracy: 99\\nSensitivity: 99\\nSpecificity: 100\\nNo\\nElhoseny et\\xa0al Africa 2019 400 Diagnosis Classification ANN Accuracy: 95\\nSensitivity: 96\\nSpecificity: 93\\nNo\\nForné et\\xa0al Europe 2019 1366 Prognosis Classification Random forest AUC: 0.74 No\\nGalloway et\\xa0al North America 2019 449,380 Prognosis Classification CNN AUC: 0.87\\nGuo et\\xa0al South America 2019 703 Diagnosis + Prog-\\nnosis\\nClassification LASSO regres-\\nsion\\nAccuracy: 99 Yes\\nHan et\\xa0al Asia 2019 1370 Risk/forecast Classification Random forest Accuracy: 93\\nSensitivity: 80\\nSpecificity: 95\\nNo\\nHuang et\\xa0al Asia 2019 400 Diagnosis Classification ANN Accuracy: 99\\nSensitivity: 99\\nSpecificity: 99\\nNo\\nKanda et\\xa0al Asia 2019 7465 Prognosis Classification Support vector \\nmachine\\nAccuracy: 89 No'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 4, 'page_label': '1105', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='No\\nHuang et\\xa0al Asia 2019 400 Diagnosis Classification ANN Accuracy: 99\\nSensitivity: 99\\nSpecificity: 99\\nNo\\nKanda et\\xa0al Asia 2019 7465 Prognosis Classification Support vector \\nmachine\\nAccuracy: 89 No\\nKannan et\\xa0al North America 2019 171 Diagnosis Classification CNN Accuracy: 95\\nSensitivity: 56\\nSpecificity: 99\\nNo\\nKuo et\\xa0al Asia 2019 1299 Diagnosis Classification CNN Accuracy: 86 No\\nLin et\\xa0al Asia 2019 48,153 Prognosis Classification Random forest MSE: 0.75\\nMAE: 0.51\\nNo'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 5, 'page_label': '1106', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1106 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\nTable 1  (continued)\\nAuthors Country Year Sample size Main Aim Model task Selected model Performance \\nmetric(s)\\nClinical \\nDeploy-\\nment\\nNavaneeth et\\xa0al Asia 2019 104 Diagnosis Classification CNN Accuracy: 98\\nSensitivity: 98\\nSpecificity: 98\\nNo\\nYu et\\xa0al North America 2019 703 Diagnosis Classification ANN Accuracy: 99 No\\nAldhyani et\\xa0al Africa 2020 768 Diagnosis Classification Support vector \\nmachine\\nAccuracy: 100\\nSensitivity: 100\\nSpecificity: 100\\nNo\\nBelur Nagaraj et\\xa0al Europe 2020 11,789 Prognosis Classification ANN AUC: 0.82 No\\nChen et\\xa0al Asia 2020 101 Diagnosis Classification Support vector \\nmachine\\nAccuracy: 90\\nSensitivity: 100\\nSpecificity: 79\\nNo\\nDovgan et\\xa0al Europe 2020 8492 Prognosis Classification XGBoost AUC: 0.78\\nSensitivity: 62\\nSpecificity: 78\\nNo\\nGarcia-Mon-\\ntemayor et\\xa0al\\nEurope 2020 1571 Prognosis Classification Random forest AUC: 0.7\\nAccuracy: 73\\nNo\\nGlazyrin et\\xa0al Asia 2020 48 Diagnosis Classification K nearest neigh-\\nbor'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 5, 'page_label': '1106', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Specificity: 78\\nNo\\nGarcia-Mon-\\ntemayor et\\xa0al\\nEurope 2020 1571 Prognosis Classification Random forest AUC: 0.7\\nAccuracy: 73\\nNo\\nGlazyrin et\\xa0al Asia 2020 48 Diagnosis Classification K nearest neigh-\\nbor\\nAccuracy: 87 No\\nHuang et\\xa0al Europe 2020 3080 Risk/forecast Classification Random forest AUC: 0.86 No\\nInaguma et\\xa0al Asia 2020 19,894 Prognosis Classification Random forest AUC: 0.73 No\\nJeong et\\xa0al Asia 2020 134,895 Diagnosis Classification ANN Accuracy: 99 No\\nKanda et\\xa0al Asia 2020 79,860 Prognosis Classification Ensemble model Accuracy: 95\\nSensitivity: 91\\nSpecificity: 99\\nYes\\nKomaru et\\xa0al Asia 2020 101 Prognosis Classification Hierarchical \\nclustering\\nAUC: 0.8 Yes\\nKumar et\\xa0al Asia 2020 400 Diagnosis Classification Genetic algo-\\nrithms\\nAccuracy: 99\\nSensitivity: 99\\nSpecificity: 100\\nNo\\nNoh et\\xa0al Asia 2020 1730 Prognosis Classification ANN AUC: 0.86 No\\nNusinovici et\\xa0al Asia 2020 6762 Risk/forecast Classification Logistic regres-\\nsion\\nAUC: 0.90\\nSensitivity: 80\\nSpecificity: 60\\nNo'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 5, 'page_label': '1106', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='No\\nNoh et\\xa0al Asia 2020 1730 Prognosis Classification ANN AUC: 0.86 No\\nNusinovici et\\xa0al Asia 2020 6762 Risk/forecast Classification Logistic regres-\\nsion\\nAUC: 0.90\\nSensitivity: 80\\nSpecificity: 60\\nNo\\nOgunleye et\\xa0al Africa 2020 400 Diagnosis Classification XGBoost Accuracy: 100\\nSensitivity: 100\\nSpecificity: 100\\nNo\\nPellicer-Valero \\net\\xa0al\\nEurope 2020 110,758 Prognosis Regression RNN MSE: 0.72\\nMAE: 0.65\\nNo\\nRoth et\\xa0al Europe 2020 12,761 Risk/forecast Classification RNN AUC: 0.96 No\\nSabanayagam et\\xa0al Asia 2020 5188 Diagnosis Classification ANN AUC: 0.91 No\\nSegal et\\xa0al Asia 2020 550,000 Prognosis Classification XGBoost AUC: 0.93\\nSensitivity: 72\\nSpecificity: 96\\nNo\\nShih et\\xa0al Asia 2020 19,270 Risk/forecast Classification Decision tree AUC: 0.79\\nAccuracy: 82\\nSensitivity: 67\\nSpecificity: 79\\nNo\\nSong et\\xa0al North America 2020 14,039 Risk/forecast Classification Gradient boosting \\nmachine\\nAUC: 0.83\\nSensitivity: 83\\nSpecificity: 78\\nNo'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 5, 'page_label': '1106', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Accuracy: 82\\nSensitivity: 67\\nSpecificity: 79\\nNo\\nSong et\\xa0al North America 2020 14,039 Risk/forecast Classification Gradient boosting \\nmachine\\nAUC: 0.83\\nSensitivity: 83\\nSpecificity: 78\\nNo\\nVitsios et\\xa0al Europe 2020 12,713 Risk/forecast Classification Random forest AUC: 0.84 No\\nWeber et\\xa0al Europe 2020 785 Diagnosis Classification ANN ACU: 0.91\\nSensitivity: 100\\nSpecificity: 82\\nNo'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 6, 'page_label': '1107', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1107Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\nMost of the included articles (n  = 51) were published \\nfrom 2019 to 2021. Among the 68 articles selected for data \\nextraction, the majority were published by authors from \\norganizations based in Asia (n  = 33; 48.5%). The remain-\\ning articles were published by authors from Europe (n = 17; \\n25%), North America (n = 12; 17.6%), Africa (n = 5; 7.35%) \\nand South America (n  = 1; 1.47%). The analyzed studies \\nwere classified as observational.\\nStudy aim\\nA total of 28 studies focused on the use of ML algorithms \\nin disease prognosis analysis, 21 investigated the use of \\nML techniques on diagnosis (4 evaluated both), 12 evalu-\\nated the risk of developing the disease, and 3 investigated \\nthe use of ML in CKD treatment. Among the articles \\nfocusing on prognosis, the majority studied the applica-\\ntion of ML in evaluating CKD progression (n  = 13) and \\nmortality (n  = 8).\\nStudy populations and\\xa0sample size'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 6, 'page_label': '1107', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='focusing on prognosis, the majority studied the applica-\\ntion of ML in evaluating CKD progression (n  = 13) and \\nmortality (n  = 8).\\nStudy populations and\\xa0sample size\\nThe most commonly investigated study population con-\\nsisted of patients with CKD and healthy subjects (n  = 26; \\n38.2%), followed by patients with CKD only (n = 16; 23.5%) \\nand patients with CKD treated with hemodialysis (n  = 12; \\n17.6%). The sample size investigated in the selected articles \\nvaried from a minimum of 30 individuals to a maximum of \\n550,000 (median = 776; IQR 400–12,020).\\nData sources\\nThe majority of the included articles analyzed data obtained \\nfrom single-hospital registries (n = 33; 48.5%), datasets pro-\\nvided by universities (n = 15; 22.1%), and datasets collected \\nin multi-center studies (n = 12, 17.6%). Five studies analyzed \\nhealth insurance data (7.35%) and 3 studies used data pro-\\nvided by national health services (4.41%).\\nThe most commonly used data were various combinations'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 6, 'page_label': '1107', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='health insurance data (7.35%) and 3 studies used data pro-\\nvided by national health services (4.41%).\\nThe most commonly used data were various combinations \\nof demographic data along with individual clinical char -\\nacteristics and laboratory data (n  = 60; 82.24%), followed \\nTable 1  (continued)\\nAuthors Country Year Sample size Main Aim Model task Selected model Performance \\nmetric(s)\\nClinical \\nDeploy-\\nment\\nWu et\\xa0al Asia 2020 508 Risk/forecast Classification XGBoost AUC: 0.76 No\\nXin et\\xa0al Asia 2020 163 Diagnosis + Prog-\\nnosis\\nClassification XGBoost AUC: 0.96\\nSensitivity: 92\\nNo\\nYuan et\\xa0al Asia 2020 1090 Prognosis Classification Random forest AUC: 0.88\\nAccuracy: 85\\nNo\\nDaniel et\\xa0al Europe 2021 60 Prognosis Classification CNN Accuracy: 99\\nSensitivity: 93\\nSpecificity: 99\\nNo\\nJeong et\\xa0al Asia 2021 586 Prognosis Classification Random forest Sensitivity: 68 No\\nKrishnamurthy \\net\\xa0al\\nAsia 2021 90,000 Risk/forecast Classification CNN AUC: 0.95\\nAccuracy: 89\\nSensitivity: 94\\nSpecificity: 88\\nNo'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 6, 'page_label': '1107', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Krishnamurthy \\net\\xa0al\\nAsia 2021 90,000 Risk/forecast Classification CNN AUC: 0.95\\nAccuracy: 89\\nSensitivity: 94\\nSpecificity: 88\\nNo\\nOhara et\\xa0al Asia 2021 440 Therapy Classification RNN Accuracy: 95 No\\nParab et\\xa0al Asia 2021 57 Prognosis Regression ANN MSE: 2.06 No\\nPeng et\\xa0al Asia 2021 198 Diagnosis Regression DNN MSE: 11.62 No\\nRashed-Al-Mah-\\nfuz et\\xa0al\\nAsia 2021 400 Diagnosis Classification Random forest AUC: 0.97\\nAccuracy: 97\\nSensitivity: 96\\nSpecificity: 99\\nNo\\nSchena et\\xa0al Europe 2021 758 Diagnosis + Prog-\\nnosis\\nClassification ANN Accuracy: 80 No\\nSenan et\\xa0al Asia 2021 400 Diagnosis Classification Random forest Accuracy: 100\\nSensitivity: 100\\nNo\\nShang et\\xa0al North America 2021 2350 Diagnosis Classification Ensemble model Sensitivity: 87\\nSpecificity: 97\\nNo\\nZhang et\\xa0al Asia 2021 115,344 Risk/forecast Classification ANN AUC: 0.89 Yes'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 7, 'page_label': '1108', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1108 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\nby data obtained by medical imaging technologies (n  = 5; \\n7.35%) and genomic data (n = 3; 4.41%).\\nModels\\nThe number of models tested and reported in each article \\nvaried from a minimum of 1 model to a maximum of 10 \\n(mean = 3). The most frequently tested model class was tree \\nalgorithms (n = 58, 33.53%), such as random forest (n = 27, \\n15.61%), decision trees (n = 10, 5.78%) and extreme gradient \\nboosting (n = 9, 5.20). Subsequently, neural networks (NNs) \\nwere often inspected (n = 44, 16.18%), especially the multi-\\nlayer perceptron (MLP) (n  = 28, 16.18%). Another popular \\nchoice of machine learning model class was Support Vector \\nMachines (n = 25, 14.45%) and logistic regression ( n = 18, \\n10.45%) with various regularizations. Another\\xa0popular \\nmethod that we did not classify into a\\xa0larger model class was \\nthe non-parametric k-Nearest Neighbors\\xa0algorithm ( n = 8, \\n2.31%). The complete list of models can be found in Table\\xa02.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 7, 'page_label': '1108', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='method that we did not classify into a\\xa0larger model class was \\nthe non-parametric k-Nearest Neighbors\\xa0algorithm ( n = 8, \\n2.31%). The complete list of models can be found in Table\\xa02.\\nAll the articles implemented supervised learning algo-\\nrithms, 57 (83.8%) of them addressed classification tasks \\nand 11 (16.2%) regression tasks.\\nTable 2  List of machine \\nlearning models used in the \\nselected papers\\nThe models were also classified in larger model families to present a general overview. Some models that \\nwe were not able to classify in larger model families were classified as “Others”\\nModel class Specific model n %\\nNeural networks Feedforward NN/multilayer perceptron (MLP) 44 28 25.43 16.18\\nConvolutional NN (CNN) 9 5.20\\nRecurrent NN and long short-term memory NN (RNN) 5 2.89\\nAuto-encoder 1 0.58\\nExtreme learning machine 1 0.58\\nTree Algorithms Random forest 58 27 33.53 15.61\\nDecision trees 10 5.78\\nExtreme gradient boosting (XGBoost) 9 5.20\\nGradient boosting machine 5 2.89'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 7, 'page_label': '1108', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Auto-encoder 1 0.58\\nExtreme learning machine 1 0.58\\nTree Algorithms Random forest 58 27 33.53 15.61\\nDecision trees 10 5.78\\nExtreme gradient boosting (XGBoost) 9 5.20\\nGradient boosting machine 5 2.89\\nBagged decision trees 3 1.73\\nExtremely randomized trees 2 1.16\\nLight gradient boosting machine 1 0.58\\nAdaptive boosting machine 1 0.58\\nCategorical boost 1 0.58\\nSupport Vector Machines Support vector machines 25 22 14.45 12.72\\nGenetic algorithm based on SVM 1 0.58\\nParticle swarm optimization SVM 1 0.58\\nSimulated annealing particle swarm optimization SVM 1 0.58\\nLogistic Regression Logistic regression 18 13 10.45 7.51\\nLASSO logistic regression 3 1.73\\nRidge logistic regression 1 0.58\\nElastic net logistic regression 1 0.58\\nOthers k-Nearest neighbors (kNN) 28 8 16.18 4.62\\nGaussian Naïve Bayes 4 2.31\\nEnsemble model 3 1.73\\nLinear regression 2 1.16\\n(Adaptive) Neuro-fuzzy Inference System 2 1.16\\nPartial Least Square Regression 1 0.58\\nHidden Markov Model (HMM) 1 0.58\\nk-Means 1 0.58'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 7, 'page_label': '1108', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Ensemble model 3 1.73\\nLinear regression 2 1.16\\n(Adaptive) Neuro-fuzzy Inference System 2 1.16\\nPartial Least Square Regression 1 0.58\\nHidden Markov Model (HMM) 1 0.58\\nk-Means 1 0.58\\nCox regression 1 0.58\\nHierarchical clustering 1 0.58\\nGenetic programming 1 0.58\\nLinear discriminant analysis (LDA) 1 0.58\\nMarkov decision process (MDP) 1 0.58\\nHierarchical clustering 1 0.58'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 8, 'page_label': '1109', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1109Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\nThe majority of the included articles (n  = 52) speci-\\nfied the total number of features used to train the models. \\nThese models used a highly variable number of features, \\nranging from 4 to 6624 (median = 24; IQR = 17—46). Of \\nthe 68 included studies, 55 specified the variables used in \\nthe models (n  = 130). The most frequently used features \\nare reported in Fig.\\xa0 2.\\nFig. 2  Occurrence of variables in the selected articles, divided per aim\\nTable 3  List of metrics and \\ntheir occurrence in number \\nand percentages of the selected \\npapers\\nFurthermore, the last column specifies for which task the metric is used\\nName n % Task\\nAccuracy 30 17.05 Classification\\nROC AUC/C statistic 30 17.05 Classification\\nSensitivity/recall 29 16.48 Classification\\nSpecificity 24 13.64 Classification\\nPrecision/positive predictive value (PPV) 16 9.09 Classification\\nF1 score 14 7.95 Classification\\nMatthews correlation coefficient 7 3.98 Classification'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 8, 'page_label': '1109', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Specificity 24 13.64 Classification\\nPrecision/positive predictive value (PPV) 16 9.09 Classification\\nF1 score 14 7.95 Classification\\nMatthews correlation coefficient 7 3.98 Classification\\nMean absolute error (MAE) 6 3.41 Regression\\nRoot mean squared error (RMSE) 5 2.84 Regression\\nNegative predictive value (NPV) 3 1.70 Classification\\nR2/coefficient of determination 3 1.70 Regression\\nMean squared error (MSE) 2 1.14 Regression\\nPrecision-recall AUC (AUPRC) 2 1.14 Regression\\nBayesian information criterion (BIC) 1 0.57 Regression/classification\\nCohen’s kappa statistic 1 0.57 Classification\\nJaccard index/intersection over union 1 0.57 Classification\\nNormalized mean squared error (NMSE) 1 0.57 Regression\\nq2 Statistic 1 0.57 Regression'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 9, 'page_label': '1110', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1110 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\nPerformance metrics\\nThe most common performance metrics were accuracy \\n(n = 30, 17.05%) and the area under the receiver operating \\ncharacteristic curve (often also referred to as ROC-AUC, \\nAUROC, AUC, or C-statistic) (n  = 30, 17.05%). Subse-\\nquently, other classification metrics, such as sensitivity \\n(n = 29, 16.48%), specificity (n  = 24, 13.64%), precision \\n(n = 16, 9.09%), and F1-score (n  = 14, 7.95%) were often \\nused to compare the machine learning models. Note that \\nall the aforementioned metrics, except ROC AUC, were \\nused for classification and required establishing a risk \\nthreshold as a decision boundary. ROC AUC conversely \\ndid not require setting a decision threshold as it was calcu-\\nlated by iterating over all the decision thresholds. In terms \\nof regression, the most used metrics for comparison were \\nmean absolute error (n  = 6, 3.41%) and root mean squared \\nerror (n  = 5, 2.84%). The full list of the metrics and how'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 9, 'page_label': '1110', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='of regression, the most used metrics for comparison were \\nmean absolute error (n  = 6, 3.41%) and root mean squared \\nerror (n  = 5, 2.84%). The full list of the metrics and how \\noften they occurred can be found in Table\\xa0 3.\\nBest performing models, and\\xa0their performances\\nIn the included articles, neural networks were the models \\nthat commonly performed best (n  = 28, 41.18%) compared \\nto the median performance of other models, such as MLP \\n(n = 18, 26.47%) and convolutional neural networks (n  = 7, \\n24.53%). Tree-based algorithms performed best (n  = 24, \\n35.29%); these algorithms included Random Forest (n = 16, \\n23.53%) and Extreme Gradient Boosting (n = 5, 7.35%). The \\nresults for Support Vector Machines (n = 5, 7.35%) were also \\nnoteworthy. A complete list of the best performing models \\nin the selected papers can be found in Table\\xa04.\\nIn terms of performance, we compared the metrics of \\nprediction models, diagnostic models and risk prediction'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 9, 'page_label': '1110', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='in the selected papers can be found in Table\\xa04.\\nIn terms of performance, we compared the metrics of \\nprediction models, diagnostic models and risk prediction \\nmodels separately. Of the 25 (36.76%) machine learning \\nmodels for diagnosis, 19 papers reported accuracy. Three \\nmodels reported the highest accuracy of 1.00 while the low-\\nest reported accuracy is 0.80 (mean = 0.95, median = 0.98). \\nSensitivity was reported 15 times, with a maximum of \\n1.00, a minimum of 0.56, a mean of 0.95 and a median \\nof 0.99. In addition, specificity was reported in 13 cases \\n(max = 1.00, min = 0.79, mean = 0.96, median = 0.99). The \\nROC-AUC was reported in 6 papers (max = 0.99, min = 0.91, \\nmean = 0.941, median = 0.94).\\nFor the prediction models (n  = 32, 47.06%), 15 papers \\nreported the ROC-AUC with a maximum of 0.96 and a \\nminimum of 0.69 (mean = 0.82, median = 0.82). Ten papers \\nreported accuracy, ranging from 0.54 to 0.99, with a mean of \\n0.85 and a median of 0.87. Sensitivity was reported 8 times,'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 9, 'page_label': '1110', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='minimum of 0.69 (mean = 0.82, median = 0.82). Ten papers \\nreported accuracy, ranging from 0.54 to 0.99, with a mean of \\n0.85 and a median of 0.87. Sensitivity was reported 8 times, \\nranging from 0.54 to 0.93 (mean = 0.765, median = 0.76), \\nand specificity was reported 5 times (max = 0.99, min = 0.78, \\nmean = 0.917, median = 0.96).\\nNext, the risk prediction models (n = 12, 17.65%) showed \\nROC-AUC 9 times (max = 0.96, min = 0.76, mean = 0.864, \\nmedian = 0.86) and accuracy 4 times (max = 0.99, \\nmin = 0.82, mean = 0.901, median = 0.91).\\nFinally, 3 (4.41%) papers focused on therapy, one of \\nwhich reported an accuracy of 0.95, while the other two \\nfocused on outcome differences (p-values).\\nMost common variables and\\xa0most important ones\\nThe total number of variables used in the included studies \\nwas 813. The five most common ones were: Blood Pressure \\nTable 4  List of the best \\nperforming models throughout \\nthe selected papers, classified by \\nmodel family\\nModel class Model n %'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 9, 'page_label': '1110', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='was 813. The five most common ones were: Blood Pressure \\nTable 4  List of the best \\nperforming models throughout \\nthe selected papers, classified by \\nmodel family\\nModel class Model n %\\nNeural networks Feedforward NN/multilayer perceptron (MLP) 28 18 41.18 26.47\\nConvolutional NN (CNN) 7 10.29\\nRecurrent NN and long short-term memory NN (RNN) 3 4.41\\nTree algorithms Random forest 24 16 35.29 23.53\\nExtreme gradient boosting (XGBoost) 5 7.35\\nDecision tree 2 2.94\\nGradient boosting machine 1 1.47\\nSupport vector machines Support vector machines 5 5 7.35 7.35\\nLogistic regression LASSO logistic regression 3 2 4.41 2.94\\nLogistic regression 1 1.47\\nOthers Ensemble model 8 3 11.76 4.41\\nk-Nearest Neighbors (kNN) 1 1.47\\nGenetic algorithms 1 1.47\\nHierarchical clustering 1 1 1.47\\nHidden Markov Model 1 1.47\\nMarkov decision processes 1 1.47'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 10, 'page_label': '1111', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1111Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\n(n = 62, 7.63%), Age (n  = 45, 5.54%), Hemoglobin (n  = 37, \\n4.55%), Creatinine (serum) (n = 31, 3.81%) and Sex (n = 31, \\n3.81%).\\nNonetheless, to better capture how variables were used \\nin the selected papers, we classified the variables into\\xa04 \\nsubsets (CKD Prognosis, CKD Diagnosis, Risk of Devel-\\noping CKD, CKD Treatment) based on the primary aim the \\nauthors stated their model would have\\xa0attempted to achieve.\\nRegarding CKD Prognosis, 342 variables were used out \\nof 813 total (42%). The most common ones were: Blood \\nPressure (n  = 24, 7%), Age (n  = 19, 5,56%), Cholesterol \\n(serum) (n = 18, 5.26%), Sex (n = 14, 4%) and Hemoglobin \\n(blood) (n  = 13, 3.8%), with the most important variables \\nbeing: Age, Hemoglobin and Proteinuria.\\nConcerning CKD Diagnosis, 311 variables were used \\nout of 813 total (38.25%). The most common ones were: \\nBlood Pressure ( n = 22, 7%), Hemoglobin (blood) ( n = 19,'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 10, 'page_label': '1111', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Concerning CKD Diagnosis, 311 variables were used \\nout of 813 total (38.25%). The most common ones were: \\nBlood Pressure ( n = 22, 7%), Hemoglobin (blood) ( n = 19, \\n6.1%), Pus Cell General—used to indicate the number of \\ndead white cells in urine—(n  = 18, 5.79%), Age (n  = 14, \\n4.50%) and Glucose (serum) (n  = 14, 4.50%). The most \\nimportant variables in this case were Albumin, Creatinine, \\nand Hemoglobin.\\nWith regard to Risk of Developing CKD, 137 variables \\nwere used out of 813 total (16.85%). The most common ones \\nwere: Blood Pressure (n  = 12, 8.75%), Age (n  = 9, 6.57%), \\nSex (n = 7, 5.11%), History of Cardiovascular Disease (n = 6, \\n4.38%) and estimated Glomerular Filtration Rate (eGFR) \\n(n = 6, 4.38%). The most important variables were Age, GFR \\nand Blood Pressure.\\nFinally, regarding CKD Treatment, 23 variables were \\nused out of 813 total (2.83%). The most common ones were: \\nBlood Iron (n = 5, 21.74%), Hemoglobin (n = 3, 13%), Drugs'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 10, 'page_label': '1111', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='and Blood Pressure.\\nFinally, regarding CKD Treatment, 23 variables were \\nused out of 813 total (2.83%). The most common ones were: \\nBlood Iron (n = 5, 21.74%), Hemoglobin (n = 3, 13%), Drugs \\nUsed (n = 2, 8.70%), MCV (n = 2, 8.70%) and White Blood \\nCells (blood) (n = 2, 8.70%). Regarding this aim, no weights \\nwere listed in the examined articles.\\nThe complete spreadsheet with all variables and percent-\\nages can be found in Supplemental Material, together with \\nthe most important variables, divided per aim.\\nFairness\\nOther than using PROBAST to assess risk of bias, we also \\nassessed fairness based on how the authors explicitly used \\nvariables. In some studies, variables were not fully listed, \\nand in such cases, if the variable (sex, or race/ethnicity) was \\nnot indexed, we considered the feature as not included in \\nthe general model.\\nOut of 68 studies, 43 included gender in the model and \\n12 included race/ethnicity. When Non-Hispanic Whites were'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 10, 'page_label': '1111', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='not indexed, we considered the feature as not included in \\nthe general model.\\nOut of 68 studies, 43 included gender in the model and \\n12 included race/ethnicity. When Non-Hispanic Whites were \\npart of the assessed cohort, they were the majority group, \\nranging from 87 to 31%. Ten out of 68 studies addressed \\nboth gender and race/ethnicity, and included these variables \\nin the model.\\nRace/ethnicity was included in 4 out of 12 studies predict-\\ning risk, in 5 out of 28 studies predicting prognosis, and in 3 \\nout of 21 studies classifying diagnosis. It was never included \\nin models investigating prognosis and diagnosis combined, \\nand therapeutics.\\nClinical Deployment\\nRegarding Diagnosis, just one\\xa0model was actually deployed \\nin a clinical environment [60]. The authors applied a lasso \\nregression with metabolites as features, achieving an accu-\\nracy of 99%; the authors used data from a real clinical con-\\ntext, and therefore they deployed and evaluated their model'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 10, 'page_label': '1111', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='regression with metabolites as features, achieving an accu-\\nracy of 99%; the authors used data from a real clinical con-\\ntext, and therefore they deployed and evaluated their model \\nperformance on a clinical context, nevertheless, they did \\nnot validate their model. Regarding Prognosis, just 3 stud-\\nies were conducted in a clinical setting [49, 50, 62]. Komaru \\net\\xa0al. [49] predicted 1-year mortality following the start of \\nhemodialysis through hierarchical clustering and achieved \\nan AUC of 0.8; the authors used data from a clinical pro-\\nspective study to deploy and evaluate their model. Further -\\nmore, they validated the used clusters. Kanda et\\xa0al. [50] \\napplied a support vector machine model onto a real popula-\\ntion in an observational study to deploy and evaluate their \\nmodel. The authors achieved an accuracy of 89% through 13 \\nvariables; unfortunately, they did not disclose the weights of \\nthe variables nor did they validate the model, and therefore'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 10, 'page_label': '1111', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='model. The authors achieved an accuracy of 89% through 13 \\nvariables; unfortunately, they did not disclose the weights of \\nthe variables nor did they validate the model, and therefore \\nwe do not know which variables were the most important. \\nAkbilgic et\\xa0al. [62] used a model based on a Random Forest \\nalgorithm, and achieved an AUC of 0.69; the most impor -\\ntant features were eGFR, Spontaneous Bacterial Peritonitis, \\nAge, Diastolic Blood Pressure and BUN. The authors used \\ndata from a real clinical context to deploy and evaluate their \\nmodel; furthermore, they validated their results and model \\ninternally. Regarding Risk of developing CKD, one study’s \\nmodel was used in a clinical context [42]. The authors used \\na NN, achieving an AUC of 0.89, using retinal images as \\nfeatures from a clinical context to deploy, evaluate and vali-\\ndate their model. Finally, regarding CKD Treatment, one \\nstudy’s model was used in a clinical environment [26]; they'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 10, 'page_label': '1111', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='features from a clinical context to deploy, evaluate and vali-\\ndate their model. Finally, regarding CKD Treatment, one \\nstudy’s model was used in a clinical environment [26]; they \\npresented their results through differences in achieved values \\nby their algorithms, and the best performance was achieved \\nby a NN. They evaluated the model with clinical data, but \\ndid not validate it.\\nQuality assessment\\nAccording to the PROBAST assessment tool [18], most \\nof the included articles showed an overall low risk of \\nbias (n = 48; 67.6%), and 65 (91.5%) of the included arti-\\ncles showed low applicability. Moreover, only 8.5% of \\nthe included studies scored less than 70% in the reporting \\nguidelines for machine learning predictive models in bio-\\nmedical research developed by Luo and colleagues [19]. The'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 11, 'page_label': '1112', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1112 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\ncomplete quality assessment can be found in Supplemental \\nMaterial.\\nDiscussion\\nThis systematic review describes how machine learning has \\nbeen used for CKD. Six overarching themes were found, \\neach of which underlines the need for further consideration \\nby the scientific community.\\nFirst, despite the ever-growing number of studies focus-\\ning on the topic, a staggeringly low amount are being con-\\nsidered for actual clinical implementation. In this review, \\njust 5 out of 68 articles tried to deploy their model in a real \\nclinical setting. This might indicate either that the technol-\\nogy is not ready yet, or, considering 4 of these 5 articles \\nwere published in the last 3\\xa0years, that the technology is \\njust starting to creep into real clinical settings. Recent evi-\\ndence suggests that it is paramount to test newly developed \\nalgorithms in clinical settings before trying to deploy them'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 11, 'page_label': '1112', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='just starting to creep into real clinical settings. Recent evi-\\ndence suggests that it is paramount to test newly developed \\nalgorithms in clinical settings before trying to deploy them \\n[88]. Despite promising laboratory results, clinical transla-\\ntion is not always guaranteed. As an example, when study -\\ning the feasibility of providing an automated electronic \\nalarm for acute kidney injury in different clinical settings, \\nsubstantial heterogeneity in the findings among hospitals \\nwas described, with the worrying result of a significantly \\nincreased risk of death for some hospitals [89].\\nSecond, as expected, the most important features were \\nprofoundly related to the main aim the authors were pur -\\nsuing. In this regard, there were no surprises in the stud-\\nied topics as the most important features were related to \\nconditions known to lead to CKD diagnosis, worsening of \\nprognosis and risk of developing CKD (e.g., age, comor -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 11, 'page_label': '1112', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='ied topics as the most important features were related to \\nconditions known to lead to CKD diagnosis, worsening of \\nprognosis and risk of developing CKD (e.g., age, comor -\\nbidities, systolic and diastolic blood pressure and eGFR \\nvalues).\\nThird, a lack of consistency in reporting results was \\nfound. Most of the studies chose to report accuracy, but \\nthis was not the norm. Furthermore, while accuracy pro-\\nvides information on model performance, it fails to consider \\nclass imbalance and data representation. This is extremely \\nimportant as accuracy in highly unbalanced datasets can be \\nvery high by always predicting the same binary outcome \\nbecause of a flawed model. For instance, considering a low \\nprevalence disease, if the algorithm is flawed for it always \\npredicts a negative event, the accuracy will be high, but the \\nveracity of the model will not [90]. As a result, AUCs and \\nROCs better measure the model precision without requir -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 11, 'page_label': '1112', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='predicts a negative event, the accuracy will be high, but the \\nveracity of the model will not [90]. As a result, AUCs and \\nROCs better measure the model precision without requir -\\ning the definition of a risk threshold. Twenty-nine authors \\nchose to express their results including AUCs and ROCs: \\nthe minimum value was 0.69 and the maximum was 0.99 \\n(mean: 0.83, median: 0.84). These results best express how \\nprecise the algorithms were and confirm the overall high \\nperformance of the assessed models.\\nFourth, a common conundrum regarding feature selec-\\ntion and output was found in studies assessing CKD diag-\\nnosis. The definition of CKD requires certain variables \\nto be present in order to make a diagnosis, thus including \\nthose variables in the model might be considered mandatory. \\nNonetheless, including those variables forces the model to \\nstreamline its decision process to a simple match in altered \\nvalues, effectively transforming a complex machine learning'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 11, 'page_label': '1112', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Nonetheless, including those variables forces the model to \\nstreamline its decision process to a simple match in altered \\nvalues, effectively transforming a complex machine learning \\nmodel into a linear decision flow-chart, the performance of \\nwhich will always be stellar.\\nThis phenomenon is especially clear in four of the stud-\\nies this systematic review assessed [36, 39, 46, 47]. In these \\nstudies, the same database [91] is used, and accuracy, sensi-\\ntivity, specificity, and ROC-AUC are never below 98%. We \\nbelieve researchers should carefully assess the variables used \\nin their machine learning models to make sure that no data \\nleakage is present between features and results.\\nFifth, model bias and fairness were almost never consid-\\nered. This is critical, as both biased and unfair models will \\nnot achieve the same results in different demographics, and \\ntheir societal impact could exasperate disparities in certain \\npopulations. These issues need to be further explored before'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 11, 'page_label': '1112', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='not achieve the same results in different demographics, and \\ntheir societal impact could exasperate disparities in certain \\npopulations. These issues need to be further explored before \\nany model can be implemented at point of care.\\nFinally, among the included studies, only 6 evaluated \\ntheir models in a clinical setting [26, 42, 49, 50, 60, 62], \\nand only 3 were validated [42, 49, 62]. These studies showed \\npromising results and did not report any unintended conse-\\nquences after evaluation and/or validation. Notwithstand-\\ning the robust results described by the authors, as discussed \\nbefore, recent evidence suggests that it is paramount to test \\nnewly developed algorithms in clinical settings to avoid \\nadverse or unintended consequences [ 88, 89]. Taking into \\naccount the pinnacle of importance of validating ones’ \\nresults in real clinical contexts and not just “in lab”, in read-\\ning their results, their generalizability has to be questioned,'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 11, 'page_label': '1112', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='account the pinnacle of importance of validating ones’ \\nresults in real clinical contexts and not just “in lab”, in read-\\ning their results, their generalizability has to be questioned, \\nespecially since no multi-center validations were described \\namong the validated models.\\nThis systematic review presents a few limitations: first, \\nonly one database (PubMed) was used to collect studies of \\ninterest. It should be noted that systematic reviews are usu-\\nally exhorted to use at least two databases as stated by the \\nPRISMA statement. Nonetheless, as PubMed has grown to \\nbe one of the most used search engines for medical sciences \\nthis limitation should be self-amending. Secondly, this sys-\\ntematic review assessed only papers written in English since \\nEnglish is the most widely adopted and commonly used lan-\\nguage for the publication of medical papers.\\nIn addition to these limitations, due to this review’s \\ndesign, all in\\xa0vitro studies (on cellular substrates) were'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 11, 'page_label': '1112', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='guage for the publication of medical papers.\\nIn addition to these limitations, due to this review’s \\ndesign, all in\\xa0vitro studies (on cellular substrates) were \\nexcluded. Consequently, the evidence presented in this \\nreview is not to be interpreted as definitive for all things con-\\ncerning CKD, since in\\xa0vitro studies (on cellular substrates),'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 12, 'page_label': '1113', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1113Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\nthe insight of which is critical in understanding pathogenetic \\nas well as therapeutic mechanisms, were not assessed.\\nLastly, the majority of included studies did not evalu-\\nate the integration of ML models in daily clinical practice, \\ntherefore the results and discussion have to be considered \\nlargely from an academic standpoint. Despite these limita-\\ntions, we feel this review advances the knowledge on the \\ncurrent state of data-driven algorithms to advance CKD \\ndiagnosis, prognosis and treatment.\\nDespite the potential benefits, the application of machine \\nlearning for CKD diagnosis, prognosis, and treatment pre -\\nsents several issues, namely fairness, model and result inter-\\npretability [ 90], and the lack of validated models. Result \\ninterpretability concerns reflect the inability to explain \\nwhich aspects of the dataset used in the training phase led \\nto a predicted result in a particular case [92, 93]. Therefore,'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 12, 'page_label': '1113', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content=\"interpretability concerns reflect the inability to explain \\nwhich aspects of the dataset used in the training phase led \\nto a predicted result in a particular case [92, 93]. Therefore, \\nas the trend in machine learning techniques moves from tra-\\nditional algorithms (e.g., lasso regressions, support vector \\nmachine, and decision trees), to more complex ones (e.g., \\nensemble algorithms and deep learning), the interpretability \\nconcerns become more pronounced [90]. Notably, research-\\ners highlighted the need for explainability and for models \\nthat could have a significant impact on patients' health [94, \\n95]. These models should be reported using best practice \\nreporting guidelines such as the Transparent Reporting of \\na Multivariate Prediction Model for Individual Prognosis \\nor Diagnosis (TRIPOD) [94] or MINimum Information for \\nMedical AI Reporting (MINIMAR) [97]. Transparent and \\naccurate reports are also fundamental in advancing multi-\"),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 12, 'page_label': '1113', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content=\"or Diagnosis (TRIPOD) [94] or MINimum Information for \\nMedical AI Reporting (MINIMAR) [97]. Transparent and \\naccurate reports are also fundamental in advancing multi-\\ncenter validations of the applied models, which in turn is an \\nessential step to ensure that only safe and sound models are \\napplied on a large scale.\\nMost of the studies failed to report on the ethical issues \\nrevolving around their model development; the impact on the \\npatient's well-being can also be affected by algorithmic bias \\n[98, 99] and this can be worse in certain underrepresented \\npopulations. This concern is closely related to the general-\\nizability of the developed model [100– 102]. Specifically, \\nretrospective data that are usually used during the training \\nphase often have significant biases towards subgroups of \\nindividuals that have been defined by factors such as age, \\ngender, educational level, socioeconomic status, and location \\n[98]. The issues of fairness and bias in algorithms should be\"),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 12, 'page_label': '1113', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='individuals that have been defined by factors such as age, \\ngender, educational level, socioeconomic status, and location \\n[98]. The issues of fairness and bias in algorithms should be \\nevaluated by investigating the models’ performance within \\npopulation subgroups.\\nThis systematic review underlines the potential benefits \\nand pitfalls of ML in the diagnosis, prognosis, and manage-\\nment of CKD. We found that most of the studies included \\nin this systematic review reported that ML offers invalu-\\nable help to clinicians allowing them to make informed \\ndecisions and provide better care to their patients; nonethe-\\nless most of those articles were not actually piloted in real \\nlife settings, and therefore, notwithstanding the excellent \\nmodel performance results reported by authors, the tech-\\nnology might not be ready for mass real-time adoption or \\nimplementation.\\nAlthough future work is needed to address the viability, \\ninterpretability, generalizability, and fairness issues, to allow'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 12, 'page_label': '1113', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='nology might not be ready for mass real-time adoption or \\nimplementation.\\nAlthough future work is needed to address the viability, \\ninterpretability, generalizability, and fairness issues, to allow \\na safer translation of these models for use in daily clinical \\npractice, the implementation of these techniques could fur -\\nther enhance the effective management of hospital resources \\nin a timely and efficient manner by potentially identifying \\npatients at high risk for adverse events and the need for addi-\\ntional resources.\\nWe hope the summarized evidence from this article will \\nfacilitate\\xa0implementation of ML approaches in the\\xa0clinical \\npractice.\\nSupplementary Information The online version contains supplemen-\\ntary material available at https:// doi. org/ 10. 1007/ s40620- 023- 01573-4.\\nAcknowledgements None.\\nAuthors contribution FS and AC had the idea, extracted, and analyzed \\nthe data and wrote the manuscript. CF analyzed the data and wrote the'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 12, 'page_label': '1113', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Acknowledgements None.\\nAuthors contribution FS and AC had the idea, extracted, and analyzed \\nthe data and wrote the manuscript. CF analyzed the data and wrote the \\nmanuscript. DGol, DGor, helped in results interpretation. THB revised \\nthe manuscript and helped in results interpretation. AC supervised the \\nentire process.\\nFunding Open access funding provided by Alma Mater Studiorum - \\nUniversità di Bologna within the CRUI-CARE Agreement.\\nData availability Statement Data that support the findings of this study \\nare available upon reasonable request from the corresponding author, \\nAC.\\nDeclarations \\nDisclosure statement The authors did not receive support from any \\norganization for the submitted work. The authors do not have any con-\\nflicts of interest to report.\\nEthics approval Not Applicable.\\nOpen Access This article is licensed under a Creative Commons Attri-\\nbution 4.0 International License, which permits use, sharing, adapta-'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 12, 'page_label': '1113', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content=\"Ethics approval Not Applicable.\\nOpen Access This article is licensed under a Creative Commons Attri-\\nbution 4.0 International License, which permits use, sharing, adapta-\\ntion, distribution and reproduction in any medium or format, as long \\nas you give appropriate credit to the original author(s) and the source, \\nprovide a link to the Creative Commons licence, and indicate if changes \\nwere made. The images or other third party material in this article are \\nincluded in the article's Creative Commons licence, unless indicated \\notherwise in a credit line to the material. If material is not included in \\nthe article's Creative Commons licence and your intended use is not \\npermitted by statutory regulation or exceeds the permitted use, you will \\nneed to obtain permission directly from the copyright holder. To view a \\ncopy of this licence, visit http:// creat iveco mmons. org/ licen ses/ by/4. 0/.\\nReferences\\n 1. Webster AC, Nagler EV, Morton RL et\\xa0al (2017) Chronic kidney\"),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 12, 'page_label': '1113', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='copy of this licence, visit http:// creat iveco mmons. org/ licen ses/ by/4. 0/.\\nReferences\\n 1. Webster AC, Nagler EV, Morton RL et\\xa0al (2017) Chronic kidney \\ndisease. Lancet Lond Engl 389(10075):1238–1252. https:// doi.  \\norg/ 10. 1016/ S0140- 6736(16) 32064-5'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1114 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\n 2. Chen TK, Knicely DH, Grams ME (2019) Chronic kidney dis-\\nease diagnosis and management. JAMA 322(13):1294–1304. \\nhttps:// doi. org/ 10. 1001/ jama. 2019. 14745\\n 3. Global, regional, and national burden of chronic kidney disease, \\n1990–2017: a systematic analysis for the Global Burden of Dis-\\nease Study 2017. Lancet Lond Engl. 2020;395(10225):709–733. \\ndoi:https:// doi. org/ 10. 1016/ S0140- 6736(20) 30045-3\\n 4. Vaidya SR, Aeddula NR. Chronic Renal Failure. In: StatPearls. \\nStatPearls Publishing; 2022. Accessed July 28, 2022. http:// www. \\nncbi. nlm. nih. gov/ books/ NBK53 5404/\\n 5. Romagnani P, Remuzzi G, Glassock R et\\xa0al (2017) Chronic \\nkidney disease. Nat Rev Dis Primer 3:17088. https:// doi. org/ 10. \\n1038/ nrdp. 2017. 88\\n 6. Thomas R, Kanso A, Sedor JR (2008) Chronic kidney disease \\nand its complications. Prim Care 35(2):329–vii. https:// doi. org/ \\n10. 1016/j. pop. 2008. 01. 008'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1038/ nrdp. 2017. 88\\n 6. Thomas R, Kanso A, Sedor JR (2008) Chronic kidney disease \\nand its complications. Prim Care 35(2):329–vii. https:// doi. org/ \\n10. 1016/j. pop. 2008. 01. 008\\n 7. Fraser SD, Blakeman T (2016) Chronic kidney disease: identi-\\nfication and management in primary care. Pragmatic Obs Res \\n7:21–32. https:// doi. org/ 10. 2147/ POR. S97310\\n 8. Chronic Kidney Disease: Overview. Institute for Quality and Effi-\\nciency in Health Care (IQWiG); 2018. Accessed July 28, 2022. \\nhttps:// www. ncbi. nlm. nih. gov/ books/ NBK49 2977/\\n 9. Kazancioğlu R (2013) Risk factors for chronic kidney disease: an \\nupdate. Kidney Int Suppl 3(4):368–371. https:// doi. org/ 10. 1038/ \\nkisup. 2013. 79\\n 10. Topol EJ (2019) High-performance medicine: the convergence of \\nhuman and artificial intelligence. Nat Med 25(1):44–56. https:// \\ndoi. org/ 10. 1038/ s41591- 018- 0300-7\\n 11. Nichols JA, Herbert Chan HW, Baker MAB (2018) Machine \\nlearning: applications of artificial intelligence to imaging and'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='doi. org/ 10. 1038/ s41591- 018- 0300-7\\n 11. Nichols JA, Herbert Chan HW, Baker MAB (2018) Machine \\nlearning: applications of artificial intelligence to imaging and \\ndiagnosis. Biophys Rev 11(1):111–118. https:// doi. org/ 10. 1007/ \\ns12551- 018- 0449-9\\n 12. Sidey-Gibbons JAM, Sidey-Gibbons CJ (2019) Machine learning \\nin medicine: a practical introduction. BMC Med Res Methodol \\n19:64. https:// doi. org/ 10. 1186/ s12874- 019- 0681-4\\n 13. Peterson DJ, Ostberg NP, Blayney DW et\\xa0al (2021) Machine \\nlearning applied to electronic health records: identification of \\nchemotherapy patients at high risk for preventable emergency \\ndepartment visits and hospital admissions. JCO Clin Cancer \\nInform 5:1106–1126. https:// doi. org/ 10. 1200/ CCI. 21. 00116\\n 14. Lenain R, Seneviratne MG, Bozkurt S et\\xa0al (2019) Machine \\nlearning approaches for extracting stage from pathology reports \\nin prostate cancer. Stud Health Technol Inform 264:1522–1523. \\nhttps:// doi. org/ 10. 3233/ SHTI1 90515'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='learning approaches for extracting stage from pathology reports \\nin prostate cancer. Stud Health Technol Inform 264:1522–1523. \\nhttps:// doi. org/ 10. 3233/ SHTI1 90515\\n 15. Cahan EM, Hernandez-Boussard T, Thadaney-Israni S et\\xa0al \\n(2019) Putting the data before the algorithm in big data address-\\ning personalized healthcare. NPJ Digit Med 2:78. https:// doi. org/ \\n10. 1038/ s41746- 019- 0157-2\\n 16. Rajpurkar P, Chen E, Banerjee O et\\xa0al (2022) AI in health \\nand medicine. Nat Med 28(1):31–38. https:// doi. org/ 10. 1038/ \\ns41591- 021- 01614-0\\n 17. Page MJ, McKenzie JE, Bossuyt PM et\\xa0al (2021) The PRISMA \\n2020 statement: an updated guideline for reporting systematic \\nreviews. BMJ 372:71. https:// doi. org/ 10. 1136/ bmj. n71\\n 18. Wolff RF, Moons KGM, Riley RD et\\xa0al (2019) PROBAST: a tool \\nto assess the risk of bias and applicability of prediction model \\nstudies. Ann Intern Med 170(1):51–58. https:// doi. org/ 10. 7326/ \\nM18- 1376'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='to assess the risk of bias and applicability of prediction model \\nstudies. Ann Intern Med 170(1):51–58. https:// doi. org/ 10. 7326/ \\nM18- 1376\\n 19. Luo W, Phung D, Tran T et\\xa0al (2016) Guidelines for develop -\\ning and reporting machine learning predictive models in bio-\\nmedical research: a multidisciplinary view. J Med Internet Res \\n18(12):e323. https:// doi. org/ 10. 2196/ jmir. 5870\\n 20. Goldstein BA, Pomann GM, Winkelmayer WC et\\xa0al (2017) \\nA comparison of risk prediction methods using repeated \\nobservations: an application to electronic health records for \\nhemodialysis. Stat Med 36(17):2750–2763. https:// doi. org/ 10. \\n1002/ sim. 7308\\n 21. Sabanayagam C, Xu D, Ting DSW et\\xa0al (2020) A deep learning \\nalgorithm to detect chronic kidney disease from retinal photo-\\ngraphs in community-based populations. Lancet Digit Health \\n2(6):e295–e302. https:// doi.  org/ 10. 1016/  S2589- 7500(20) \\n30063-7\\n 22. Rodriguez M, Salmeron MD, Martin-Malo A et\\xa0al (2016) A new'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='graphs in community-based populations. Lancet Digit Health \\n2(6):e295–e302. https:// doi.  org/ 10. 1016/  S2589- 7500(20) \\n30063-7\\n 22. Rodriguez M, Salmeron MD, Martin-Malo A et\\xa0al (2016) A new \\ndata analysis system to quantify associations between biochemi-\\ncal parameters of chronic kidney disease-mineral bone disease. \\nPLoS ONE 11(1):e0146801. https:// doi. org/ 10. 1371/ journ al. \\npone. 01468 01\\n 23. Barbieri C, Mari F, Stopper A et\\xa0al (2015) A new machine learn-\\ning approach for predicting the response to anemia treatment in \\na large cohort of End Stage Renal Disease patients undergoing \\ndialysis. Comput Biol Med 61:56–61. https:// doi. org/ 10. 1016/j. \\ncompb iomed. 2015. 03. 019\\n 24. Kumar A, Sinha N, Bhardwaj A (2020) A novel fitness function \\nin genetic programming for medical data classification. J Biomed \\nInform 112:103623. https:// doi. org/ 10. 1016/j. jbi. 2020. 103623\\n 25. Peng H, Zhu H, Ieong CWA et\\xa0al (2021) A two-stage neural'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='in genetic programming for medical data classification. J Biomed \\nInform 112:103623. https:// doi. org/ 10. 1016/j. jbi. 2020. 103623\\n 25. Peng H, Zhu H, Ieong CWA et\\xa0al (2021) A two-stage neural \\nnetwork prediction of chronic kidney disease. IET Syst Biol \\n15(5):163–171. https:// doi. org/ 10. 1049/ syb2. 12031\\n 26. Barbieri C, Molina M, Ponce P et\\xa0al (2016) An international \\nobservational study suggests that artificial intelligence for clini-\\ncal decision support optimizes anemia management in hemo-\\ndialysis patients. Kidney Int 90(2):422–429. https:// doi. org/ 10. \\n1016/j. kint. 2016. 03. 036\\n 27. Kanda E, Epureanu BI, Adachi T et\\xa0al (2020) Application of \\nexplainable ensemble artificial intelligence model to categoriza-\\ntion of hemodialysis-patient and treatment using nationwide-real-\\nworld data in Japan. PLoS ONE 15(5):e0233491. https:// doi. org/ \\n10. 1371/ journ al. pone. 02334 91\\n 28. Yu H, Samuels DC, Zhao YY, Guo Y (2019) Architectures and'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='world data in Japan. PLoS ONE 15(5):e0233491. https:// doi. org/ \\n10. 1371/ journ al. pone. 02334 91\\n 28. Yu H, Samuels DC, Zhao YY, Guo Y (2019) Architectures and \\naccuracy of artificial neural network for disease classification \\nfrom omics data. BMC Genom 20(1):167. https:// doi. org/ 10.  \\n1186/ s12864- 019- 5546-z\\n 29. Lin SY, Hsieh MH, Lin CL et\\xa0al (2019) Artificial intelligence \\nprediction model for the cost and mortality of renal replacement \\ntherapy in aged and super-aged populations in Taiwan. J Clin \\nMed. https:// doi. org/ 10. 3390/ jcm80 70995\\n 30. Ohara T, Ikeda H, Sugitani Y et\\xa0al (2021) Artificial intelli-\\ngence supported anemia control system (AISACS) to prevent \\nanemia in maintenance hemodialysis patients. Int J Med Sci \\n18(8):1831–1839. https:// doi. org/ 10. 7150/ ijms. 53298\\n 31. Akl AI, Sobh MA, Enab YM et\\xa0al (2001) Artificial intelligence: \\na new approach for prescription and monitoring of hemodi-\\nalysis therapy. Am J Kidney Dis Off J Natl Kidney Found'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='31. Akl AI, Sobh MA, Enab YM et\\xa0al (2001) Artificial intelligence: \\na new approach for prescription and monitoring of hemodi-\\nalysis therapy. Am J Kidney Dis Off J Natl Kidney Found \\n38(6):1277–1283. https:// doi. org/ 10. 1053/ ajkd. 2001. 29225\\n 32. Kolachalama VB, Singh P, Lin CQ et\\xa0al (2018) Association \\nof pathological fibrosis with renal survival using deep neural \\nnetworks. Kidney Int Rep 3(2):464–475. https://  doi. org/ 10. \\n1016/j. ekir. 2017. 11. 002\\n 33. Daniel AJ, Buchanan CE, Allcock T et\\xa0al (2021) Automated \\nrenal segmentation in healthy and chronic kidney disease sub-\\njects using a convolutional neural network. Magn Reson Med \\n86(2):1125–1136. https:// doi. org/ 10. 1002/ mrm. 28768\\n 34. Kuo CC, Chang CM, Liu KT et\\xa0al (2019) Automation of the \\nkidney function prediction and classification through ultra-\\nsound-based kidney imaging using deep learning. NPJ Digit \\nMed 2:29. https:// doi. org/ 10. 1038/ s41746- 019- 0104-2'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 13, 'page_label': '1114', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='kidney function prediction and classification through ultra-\\nsound-based kidney imaging using deep learning. NPJ Digit \\nMed 2:29. https:// doi. org/ 10. 1038/ s41746- 019- 0104-2\\n 35. Parab J, Sequeira M, Lanjewar M et\\xa0al (2021) Backpropagation \\nneural network-based machine learning model for prediction \\nof blood urea and glucose in CKD patients. IEEE J Transl Eng \\nHealth Med 9:4900608. https:// doi. org/ 10. 1109/ JTEHM. 2021. \\n30797 14'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1115Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\n 36. Chen Z, Zhang X, Zhang Z (2016) Clinical risk assessment \\nof patients with chronic kidney disease by using clinical data \\nand multivariate models. Int Urol Nephrol 48(12):2069–2075. \\nhttps:// doi. org/ 10. 1007/ s11255- 016- 1346-4\\n 37. Rashed-Al-Mahfuz M, Haque A, Azad A et\\xa0al (2021) Clinically \\napplicable machine learning approaches to identify attributes \\nof chronic kidney disease (CKD) for use in low-cost diagnostic \\nscreening. IEEE J Transl Eng Health Med 9:4900511. https://  \\ndoi. org/ 10. 1109/ JTEHM. 2021. 30736 29\\n 38. Roth JA, Radevski G, Marzolini C et\\xa0al (2021) Cohort-derived \\nmachine learning models for individual prediction of chronic \\nkidney disease in people living with human immunodeficiency \\nvirus: a prospective multicenter cohort study. J Infect Dis \\n224(7):1198–1208. https:// doi. org/ 10. 1093/ infdis/ jiaa2 36\\n 39. Huang ML, Chou YC (2019) Combining a gravitational search'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='virus: a prospective multicenter cohort study. J Infect Dis \\n224(7):1198–1208. https:// doi. org/ 10. 1093/ infdis/ jiaa2 36\\n 39. Huang ML, Chou YC (2019) Combining a gravitational search \\nalgorithm, particle swarm optimization, and fuzzy rules to \\nimprove the classification performance of a feed-forward neu -\\nral network. Comput Methods Programs Biomed. 180:105016. \\nhttps:// doi. org/ 10. 1016/j. cmpb. 2019. 105016\\n 40. Jeong B, Cho H, Kim J et\\xa0al (2020) Comparison between sta-\\ntistical models and machine learning methods on classifica-\\ntion for highly imbalanced multiclass kidney data. Diagn Basel \\nSwitz. https:// doi. org/ 10. 3390/ diagn ostic s1006 0415\\n 41. Xin G, Zhou G, Zhang W et\\xa0al (2020) Construction and valida-\\ntion of predictive model to identify critical genes associated \\nwith advanced kidney disease. Int J Genomics 2020:7524057. \\nhttps:// doi. org/ 10. 1155/ 2020/ 75240 57\\n 42. Zhang K, Liu X, Xu J et\\xa0al (2021) Deep-learning models'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='with advanced kidney disease. Int J Genomics 2020:7524057. \\nhttps:// doi. org/ 10. 1155/ 2020/ 75240 57\\n 42. Zhang K, Liu X, Xu J et\\xa0al (2021) Deep-learning models \\nfor the detection and incidence prediction of chronic kid-\\nney disease and type 2 diabetes from retinal fundus images. \\nNat Biomed Eng 5(6):533–545. https:// doi. org/ 10. 1038/  \\ns41551- 021- 00745-6\\n 43. Schena FP, Anelli VW, Trotta J et\\xa0al (2021) Development and \\ntesting of an artificial intelligence tool for predicting end-stage \\nkidney disease in patients with immunoglobulin A nephropathy. \\nKidney Int 99(5):1179–1188. https:// doi. org/ 10. 1016/j. kint. 2020. \\n07. 046\\n 44. Galloway CD, Valys AV, Shreibati JB et\\xa0al (2019) Development \\nand validation of a deep-learning model to screen for hyper -\\nkalemia from the electrocardiogram. JAMA Cardiol 4(5):428–\\n436. https:// doi. org/ 10. 1001/ jamac ardio. 2019. 0640\\n 45. Yuan Q, Zhang H, Xie Y et\\xa0al (2020) Development of prognos-'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='kalemia from the electrocardiogram. JAMA Cardiol 4(5):428–\\n436. https:// doi. org/ 10. 1001/ jamac ardio. 2019. 0640\\n 45. Yuan Q, Zhang H, Xie Y et\\xa0al (2020) Development of prognos-\\ntic model for patients at CKD stage 3a and 3b in South Central \\nChina using computational intelligence. Clin Exp Nephrol \\n24(10):865–875. https:// doi. org/ 10. 1007/ s10157- 020- 01909-5\\n 46. Polat H, Danaei Mehr H, Cetin A (2017) Diagnosis of chronic \\nkidney disease based on support vector machine by feature \\nselection methods. J Med Syst 41(4):55. https:// doi. org/ 10.  \\n1007/ s10916- 017- 0703-x\\n 47. Senan EM, Al-Adhaileh MH, Alsaade FW et\\xa0al (2021) Diagno-\\nsis of chronic kidney disease using effective classification algo-\\nrithms and recursive feature elimination techniques. J Healthc \\nEng 2021:1004767. https:// doi. org/ 10. 1155/ 2021/ 10047 67\\n 48. Pellicer-Valero OJ, Cattinelli I, Neri L et\\xa0al (2020) Enhanced \\nprediction of hemoglobin concentration in a very large cohort'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='Eng 2021:1004767. https:// doi. org/ 10. 1155/ 2021/ 10047 67\\n 48. Pellicer-Valero OJ, Cattinelli I, Neri L et\\xa0al (2020) Enhanced \\nprediction of hemoglobin concentration in a very large cohort \\nof hemodialysis patients by means of deep recurrent neural \\nnetworks. Artif Intell Med. 107:101898. https:// doi. org/ 10.  \\n1016/j. artmed. 2020. 101898\\n 49. Komaru Y, Yoshida T, Hamasaki Y et\\xa0al (2020) Hierarchical \\nclustering analysis for predicting 1-year mortality after starting \\nhemodialysis. Kidney Int Rep 5(8):1188–1195. https:// doi. org/ \\n10. 1016/j. ekir. 2020. 05. 007\\n 50. Kanda E, Kanno Y, Katsukawa F (2019) Identifying progres-\\nsive CKD from healthy population using Bayesian network and \\nartificial intelligence: a worksite-based cohort study. Sci Rep \\n9(1):5082. https:// doi. org/ 10. 1038/ s41598- 019- 41663-7\\n 51. Singh A, Nadkarni G, Gottesman O et\\xa0al (2015) Incorporating \\ntemporal EHR data in predictive models for risk stratification'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='9(1):5082. https:// doi. org/ 10. 1038/ s41598- 019- 41663-7\\n 51. Singh A, Nadkarni G, Gottesman O et\\xa0al (2015) Incorporating \\ntemporal EHR data in predictive models for risk stratification \\nof renal function deterioration. J Biomed Inform 53:220–228. \\nhttps:// doi. org/ 10. 1016/j. jbi. 2014. 11. 005\\n 52. Inaguma D, Kitagawa A, Yanagiya R et\\xa0al (2020) Increasing \\ntendency of urine protein is a risk factor for rapid eGFR decline \\nin patients with CKD: a machine learning-based prediction \\nmodel by using a big database. PLoS ONE 15(9):e0239262. \\nhttps:// doi. org/ 10. 1371/ journ al. pone. 02392 62\\n 53. Elhoseny M, Shankar K, Uthayakumar J (2019) Intelligent \\ndiagnostic prediction and classification system for chronic \\nkidney disease. Sci Rep 9(1):9583. https:// doi. org/ 10. 1038/  \\ns41598- 019- 46074-2\\n 54. Nusinovici S, Tham YC, Chak Yan MY et\\xa0al (2020) Logis-\\ntic regression was as good as machine learning for predicting \\nmajor chronic diseases. J Clin Epidemiol 122:56–69. https://'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='54. Nusinovici S, Tham YC, Chak Yan MY et\\xa0al (2020) Logis-\\ntic regression was as good as machine learning for predicting \\nmajor chronic diseases. J Clin Epidemiol 122:56–69. https://  \\ndoi. org/ 10. 1016/j. jclin epi. 2020. 03. 002\\n 55. Song X, Waitman LR, Yu AS et\\xa0al (2020) Longitudinal risk \\nprediction of chronic kidney disease in diabetic patients using \\na temporal-enhanced gradient boosting machine: retrospective \\ncohort study. JMIR Med Inform 8(1):e15510. https://  doi. org/ \\n10. 2196/ 15510\\n 56. Tang Y, Zhang W, Zhu M et\\xa0al (2018) Lupus nephritis pathol-\\nogy prediction with clinical indices. Sci Rep 8(1):10231. \\nhttps:// doi. org/ 10. 1038/ s41598- 018- 28611-7\\n 57. Segal Z, Kalifa D, Radinsky K et\\xa0al (2020) Machine learn-\\ning algorithm for early detection of end-stage renal dis-\\nease. BMC Nephrol 21(1):518. https:// doi. org/ 10. 1186/  \\ns12882- 020- 02093-0\\n 58. Forné C, Cambray S, Bermudez-Lopez M et\\xa0al (2020) Machine'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='ease. BMC Nephrol 21(1):518. https:// doi. org/ 10. 1186/  \\ns12882- 020- 02093-0\\n 58. Forné C, Cambray S, Bermudez-Lopez M et\\xa0al (2020) Machine \\nlearning analysis of serum biomarkers for cardiovascular risk \\nassessment in chronic kidney disease. Clin Kidney J 13(4):631–\\n639. https:// doi. org/ 10. 1093/ ckj/ sfz094\\n 59. Huang J, Huth C, Covic M et\\xa0al (2020) Machine learning \\napproaches reveal metabolic signatures of incident chronic kid-\\nney disease in individuals with prediabetes and type 2 diabetes. \\nDiabetes 69(12):2756–2765. https:// doi. org/ 10. 2337/ db20- 0586\\n 60. Guo Y, Yu H, Chen D et\\xa0al (2019) Machine learning distilled \\nmetabolite biomarkers for early stage renal injury. Metabo-\\nlomics Off J Metabolomic Soc 16(1):4. https:// doi. org/ 10. 1007/ \\ns11306- 019- 1624-0\\n 61. Krishnamurthy S, Ks K, Dovgan E et\\xa0al (2021) Machine learn-\\ning prediction models for chronic kidney disease using national \\nhealth insurance claim data in Taiwan. Healthc Basel Switz.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='61. Krishnamurthy S, Ks K, Dovgan E et\\xa0al (2021) Machine learn-\\ning prediction models for chronic kidney disease using national \\nhealth insurance claim data in Taiwan. Healthc Basel Switz. \\nhttps:// doi. org/ 10. 3390/ healt hcare 90505 46\\n 62. Akbilgic O, Obi Y, Potukuchi PK et\\xa0al (2019) Machine learning \\nto identify dialysis patients at high death risk. Kidney Int Rep \\n4(9):1219–1229. https:// doi. org/ 10. 1016/j. ekir. 2019. 06. 009\\n 63. Belur Nagaraj S, Pena MJ, Ju W et\\xa0al (2020) Machine-learning-\\nbased early prediction of end-stage renal disease in patients with \\ndiabetic kidney disease using clinical trials data. Diabetes Obes \\nMetab 22(12):2479–2486. https:// doi. org/ 10. 1111/ dom. 14178\\n 64. Vitsios D, Petrovski S (2020) Mantis-ml: disease-agnostic gene \\nprioritization from high-throughput genomic screens by stochas-\\ntic semi-supervised learning. Am J Hum Genet 106(5):659–678. \\nhttps:// doi. org/ 10. 1016/j. ajhg. 2020. 03. 012'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 14, 'page_label': '1115', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='prioritization from high-throughput genomic screens by stochas-\\ntic semi-supervised learning. Am J Hum Genet 106(5):659–678. \\nhttps:// doi. org/ 10. 1016/j. ajhg. 2020. 03. 012\\n 65. Shang N, Khan A, Polubriaginof F et\\xa0al (2021) Medical records-\\nbased chronic kidney disease phenotype for clinical care and “big \\ndata” observational and genetic studies. NPJ Digit Med 4(1):70. \\nhttps:// doi. org/ 10. 1038/ s41746- 021- 00428-1\\n 66. Luo L, Small D, Stewart WF et\\xa0al (2013) Methods for estimat-\\ning kidney disease stage transition probabilities using electronic \\nmedical records. EGEMS Wash DC 1(3):1040. https:// doi. org/ \\n10. 13063/ 2327- 9214. 1040\\n 67. Almansour NA, Syed HF, Khayat NR et\\xa0al (2019) Neural network \\nand support vector machine for the prediction of chronic kidney'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='1116 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\ndisease: a comparative study. Comput Biol Med 109:101–111. \\nhttps:// doi. org/ 10. 1016/j. compb iomed. 2019. 04. 017\\n 68. Chen CA, Li YC, Lin YF et\\xa0al (2007) Neuro-fuzzy technology \\nas a predictor of parathyroid hormone level in hemodialysis \\npatients. Tohoku J Exp Med 211(1):81–87. https:// doi. org/ 10.  \\n1620/ tjem. 211. 81\\n 69. Escandell-Montero P, Chermisi M, Martínez-Martínez JM \\net\\xa0al (2014) Optimization of anemia treatment in hemodialysis \\npatients via reinforcement learning. Artif Intell Med 62(1):47–\\n60. https:// doi. org/ 10. 1016/j. artmed. 2014. 07. 004\\n 70. Weber C, Röschke L, Modersohn L et\\xa0al (2020) Optimized \\nidentification of advanced chronic kidney disease and absence \\nof kidney disease by combining different electronic health data \\nresources and by applying machine learning strategies. J Clin \\nMed. https:// doi. org/ 10. 3390/ jcm90 92955\\n 71. Garcia-Montemayor V, Martin-Malo A, Barbieri C et\\xa0al (2021)'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='resources and by applying machine learning strategies. J Clin \\nMed. https:// doi. org/ 10. 3390/ jcm90 92955\\n 71. Garcia-Montemayor V, Martin-Malo A, Barbieri C et\\xa0al (2021) \\nPredicting mortality in hemodialysis patients using machine \\nlearning analysis. Clin Kidney J 14(5):1388–1395. https:// doi.  \\norg/ 10. 1093/ ckj/ sfaa1 26\\n 72. Norouzi J, Yadollahpour A, Mirbagheri SA et\\xa0al (2016) Predict-\\ning renal failure progression in chronic kidney disease using inte-\\ngrated intelligent fuzzy expert system. Comput Math Methods \\nMed 2016:6080814. https:// doi. org/ 10. 1155/ 2016/ 60808 14\\n 73. Kusiak A, Dixon B, Shah S (2005) Predicting survival time for \\nkidney dialysis patients: a data mining approach. Comput Biol \\nMed 35(4):311–327. https:// doi. org/ 10. 1016/j. compb iomed.  \\n2004. 02. 004\\n 74. Jeong YS, Kim J, Kim D et\\xa0al (2021) Prediction of postoperative \\ncomplications for patients of end stage renal disease. Sensors. \\nhttps:// doi. org/ 10. 3390/ s2102 0544'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='2004. 02. 004\\n 74. Jeong YS, Kim J, Kim D et\\xa0al (2021) Prediction of postoperative \\ncomplications for patients of end stage renal disease. Sensors. \\nhttps:// doi. org/ 10. 3390/ s2102 0544\\n 75. Martínez-Martínez JM, Escandell-Montero P, Barbieri C et\\xa0al \\n(2014) Prediction of the hemoglobin level in hemodialysis \\npatients using machine learning techniques. Comput Methods \\nPrograms Biomed 117(2):208–217. https://  doi. org/ 10. 1016/j. \\ncmpb. 2014. 07. 001\\n 76. Noh J, Yoo KD, Bae W et\\xa0al (2020) Prediction of the mortal-\\nity risk in peritoneal dialysis patients using machine learning \\nmodels: a nation-wide prospective cohort in Korea. Sci Rep \\n10(1):7470. https:// doi. org/ 10. 1038/ s41598- 020- 64184-0\\n 77. Glazyrin YE, Veprintsev DV, Ler IA et\\xa0al (2020) Proteomics-\\nbased machine learning approach as an alternative to conven-\\ntional biomarkers for differential diagnosis of chronic kidney \\ndiseases. Int J Mol Sci. https:// doi. org/ 10. 3390/ ijms2 11348 02'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='based machine learning approach as an alternative to conven-\\ntional biomarkers for differential diagnosis of chronic kidney \\ndiseases. Int J Mol Sci. https:// doi. org/ 10. 3390/ ijms2 11348 02\\n 78. Navaneeth B, Suchetha M (2019) PSO optimized 1-D CNN-SVM \\narchitecture for real-time detection and classification applica -\\ntions. Comput Biol Med 108:85–92. https://  doi. org/ 10. 1016/j. \\ncompb iomed. 2019. 03. 017\\n 79. Chen C, Yang L, Li H et\\xa0al (2020) Raman spectroscopy combined \\nwith multiple algorithms for analysis and rapid screening of \\nchronic renal failure. Photodiagnosis Photodyn Ther. 30:101792. \\nhttps:// doi. org/ 10. 1016/j. pdpdt. 2020. 101792\\n 80. Han X, Zheng X, Wang Y et\\xa0al (2019) Random forest can \\naccurately predict the development of end-stage renal disease \\nin immunoglobulin a nephropathy patients. Ann Transl Med \\n7(11):234. https:// doi. org/ 10. 21037/ atm. 2018. 12. 11\\n 81. Shih CC, Lu CJ, Chen GD et\\xa0al (2020) Risk prediction for early'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='in immunoglobulin a nephropathy patients. Ann Transl Med \\n7(11):234. https:// doi. org/ 10. 21037/ atm. 2018. 12. 11\\n 81. Shih CC, Lu CJ, Chen GD et\\xa0al (2020) Risk prediction for early \\nchronic kidney disease: results from an adult health examination \\nprogram of 19,270 individuals. Int J Environ Res Public Health. \\nhttps:// doi. org/ 10. 3390/ ijerp h1714 4973\\n 82. Kannan S, Morgan LA, Liang B et\\xa0al (2019) Segmentation of \\nglomeruli within trichrome images using deep learning. Kidney \\nInt Rep 4(7):955–962. https:// doi. org/ 10. 1016/j. ekir. 2019. 04. 008\\n 83. Aldhyani THH, Alshebami AS, Alzahrani MY (2020) Soft \\nclustering for enhancing the diagnosis of chronic diseases over \\nmachine learning algorithms. J Healthc Eng 2020:4984967. \\nhttps:// doi. org/ 10. 1155/ 2020/ 49849 67\\n 84. Kleiman RS, LaRose ER, Badger JC et\\xa0al (2018) Using machine \\nlearning algorithms to predict risk for development of calciphy-\\nlaxis in patients with chronic kidney disease. AMIA Jt Summits'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='84. Kleiman RS, LaRose ER, Badger JC et\\xa0al (2018) Using machine \\nlearning algorithms to predict risk for development of calciphy-\\nlaxis in patients with chronic kidney disease. AMIA Jt Summits \\nTransl Sci Proc AMIA Jt Summits Transl Sci 2017:139–146\\n 85. Dovgan E, Gradišek A, Luštrek M et\\xa0al (2020) Using machine \\nlearning models to predict the initiation of renal replacement \\ntherapy among chronic kidney disease patients. PLoS ONE \\n15(6):e0233976. https:// doi. org/ 10. 1371/ journ al. pone. 02339 76\\n 86. Wu X, Yuan X, Wang W et\\xa0al (2020) Value of a machine learning \\napproach for predicting clinical outcomes in young patients with \\nhypertension. Hypertens Dallas Tex. 75(5):1271–1278. https://  \\ndoi. org/ 10. 1161/ HYPER TENSI ONAHA. 119. 13404\\n 87. Ogunleye A, Wang QG (2020) XGBoost model for chronic kid-\\nney disease diagnosis. IEEE/ACM Trans Comput Biol Bioinform \\n17(6):2131–2140. https:// doi. org/ 10. 1109/ TCBB. 2019. 29110 71'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='87. Ogunleye A, Wang QG (2020) XGBoost model for chronic kid-\\nney disease diagnosis. IEEE/ACM Trans Comput Biol Bioinform \\n17(6):2131–2140. https:// doi. org/ 10. 1109/ TCBB. 2019. 29110 71\\n 88. Connell A, Black G, Montgomery H et\\xa0al (2019) Implementation \\nof a digitally enabled care pathway (part 2): qualitative analysis \\nof experiences of health care professionals. J Med Internet Res \\n21(7):e13143. https:// doi. org/ 10. 2196/ 13143\\n 89. Wilson FP, Martin M, Yamamoto Y et\\xa0al (2021) Electronic health \\nrecord alerts for acute kidney injury: multicenter, randomized \\nclinical trial. BMJ 372:4786. https:// doi. org/ 10. 1136/ bmj. m4786\\n 90. Röösli E, Bozkurt S, Hernandez-Boussard T (2022) Peeking into \\na black box, the fairness and generalizability of a MIMIC-III \\nbenchmarking model. Sci Data 9(1):24. https:// doi. org/ 10. 1038/ \\ns41597- 021- 01110-7\\n 91. UCI Machine Learning Repository. Accessed July 28, 2022. \\nhttps:// archi ve. ics. uci. edu/ ml/ index. php'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='s41597- 021- 01110-7\\n 91. UCI Machine Learning Repository. Accessed July 28, 2022. \\nhttps:// archi ve. ics. uci. edu/ ml/ index. php\\n 92. Linardatos P, Papastefanopoulos V, Kotsiantis S (2020) Explain-\\nable AI: a review of machine learning interpretability methods. \\nEntropy 23(1):18. https:// doi. org/ 10. 3390/ e2301 0018\\n 93. Murdoch WJ, Singh C, Kumbier K et\\xa0al (2019) Definitions, meth-\\nods, and applications in interpretable machine learning. Proc Natl \\nAcad Sci U S A 116(44):22071–22080. https:// doi. org/ 10. 1073/ \\npnas. 19006 54116\\n 94. Amann J, Blasimme A, Vayena E et\\xa0al (2020) Explainability for \\nartificial intelligence in healthcare: a multidisciplinary perspec-\\ntive. BMC Med Inform Decis Mak 20:310. https:// doi.  org/ 10. \\n1186/ s12911- 020- 01332-6\\n 95. Payrovnaziri SN, Chen Z, Rengifo-Moreno P et\\xa0 al (2020) \\nExplainable artificial intelligence models using real-world elec-\\ntronic health record data: a systematic scoping review. J Am Med'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='95. Payrovnaziri SN, Chen Z, Rengifo-Moreno P et\\xa0 al (2020) \\nExplainable artificial intelligence models using real-world elec-\\ntronic health record data: a systematic scoping review. J Am Med \\nInform Assoc JAMIA 27(7):1173–1185. https:// doi. org/ 10. 1093/ \\njamia/ ocaa0 53\\n 96. Collins GS, Reitsma JB, Altman DG et\\xa0al (2015) Transparent \\nreporting of a multivariable prediction model for individual \\nprognosis or diagnosis (TRIPOD): the TRIPOD Statement. BMC \\nMed 13(1):1. https:// doi. org/ 10. 1186/ s12916- 014- 0241-z\\n 97. Hernandez-Boussard T, Bozkurt S, Ioannidis JPA et\\xa0al (2020) \\nMINIMAR (MINimum Information for Medical AI Report-\\ning): developing reporting standards for artificial intelligence in \\nhealth care. J Am Med Inform Assoc JAMIA 27(12):2011–2015. \\nhttps:// doi. org/ 10. 1093/ jamia/ ocaa0 88\\n 98. Gianfrancesco MA, Tamang S, Yazdany J et\\xa0al (2018) Potential \\nbiases in machine learning algorithms using electronic health'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 15, 'page_label': '1116', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content='https:// doi. org/ 10. 1093/ jamia/ ocaa0 88\\n 98. Gianfrancesco MA, Tamang S, Yazdany J et\\xa0al (2018) Potential \\nbiases in machine learning algorithms using electronic health \\nrecord data. JAMA Intern Med 178(11):1544–1547. https:// doi. \\norg/ 10. 1001/ jamai ntern med. 2018. 3763\\n 99. Panch T, Mattie H, Atun R (2002) Artificial intelligence and \\nalgorithmic bias: implications for health systems. J Glob Health \\n9(2):20318. https:// doi. org/ 10. 7189/ jogh. 09. 020318\\n 100. Ramspek CL, Jager KJ, Dekker FW et\\xa0al (2020) External vali-\\ndation of prognostic models: what, why, how, when and where? \\nClin Kidney J 14(1):49–58. https:// doi. org/ 10. 1093/ ckj/ sfaa1 88\\n 101. Steyerberg EW, Bleeker SE, Moll HA et\\xa0al (2003) Internal and \\nexternal validation of predictive models: a simulation study of'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows)', 'creator': 'Springer', 'creationdate': '2023-02-22T14:59:25+05:30', 'author': 'Francesco Sanmarchi', 'crossmarkdomains[1]': 'springer.com', 'crossmarkdomains[2]': 'springerlink.com', 'crossmarkdomainexclusive': 'true', 'crossmarkmajorversiondate': '2010-04-23', 'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review', 'moddate': '2023-05-23T18:34:02+05:30', 'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4', 'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review', 'doi': '10.1007/s40620-023-01573-4', 'robots': 'noindex', 'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf', 'total_pages': 17, 'page': 16, 'page_label': '1117', 'source_file': 's40620-023-01573-4.pdf', 'file_type': 'pdf'}, page_content=\"1117Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\nbias and precision in small samples. J Clin Epidemiol 56(5):441–\\n447. https:// doi. org/ 10. 1016/ s0895- 4356(03) 00047-7\\n 102. Riley RD, Ensor J, Snell KIE et\\xa0al (2016) External validation \\nof clinical prediction models using big datasets from e-health \\nrecords or IPD meta-analysis: opportunities and challenges. BMJ \\n353:3140. https:// doi. org/ 10. 1136/ bmj. i3140\\nPublisher's Note Springer Nature remains neutral with regard to \\njurisdictional claims in published maps and institutional affiliations.\"),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 0, 'page_label': '1', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Chronic kidney disease prediction using \\nmachine learning techniques\\nDibaba Adeba Debal1* and Tilahun Melak Sitote2 \\nIntroduction\\nChronic kidney disease (CKD) is non-communicable disease that has significantly \\ncontributed to morbidity, mortality and admission rate of patients worldwide [2]. It is \\nquickly expanding and becoming one of the major causes of death all over the world. \\nA report from 1990 to 2013 indicated that the global yearly life loss caused by CKD \\nincreased by 90% and it is the 13th leading cause of death in the world [1]. 850 million \\npeople throughout the world\\xa0are likely to have kidney diseases from different factors [3]. \\nAccording to the report of world kidney day of 2019, at least 2.4 million people die every \\nyear due to kidney related disease. Currently, it is the 6th fastest-growing cause of death \\nworldwideCKD is becoming a challenging public health problem with increasing prev -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 0, 'page_label': '1', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='year due to kidney related disease. Currently, it is the 6th fastest-growing cause of death \\nworldwideCKD is becoming a challenging public health problem with increasing prev -\\nalence worldwide. Its burden is even higher in low-income countries where detection, \\nAbstract \\nGoal three of the UN’s Sustainable Development Goal is good health and well-being \\nwhere it clearly emphasized that non-communicable diseases is emerging challenge. \\nOne of the objectives is to reduce premature mortality from non-communicable dis-\\nease by third in 2030. Chronic kidney disease (CKD) is among the significant contribu-\\ntor to morbidity and mortality from non-communicable diseases that can affected \\n10–15% of the global population. Early and accurate detection of the stages of CKD \\nis believed to be vital to minimize impacts of patient’s health complications such \\nas hypertension, anemia (low blood count), mineral bone disorder, poor nutritional'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 0, 'page_label': '1', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='is believed to be vital to minimize impacts of patient’s health complications such \\nas hypertension, anemia (low blood count), mineral bone disorder, poor nutritional \\nhealth, acid base abnormalities, and neurological complications with timely interven-\\ntion through appropriate medications. Various researches have been carried out using \\nmachine learning techniques on the detection of CKD at the premature stage. Their \\nfocus was not mainly on the specific stages prediction. In this study, both binary and \\nmulti classification for stage prediction have been carried out. The prediction models \\nused include Random Forest (RF), Support Vector Machine (SVM) and Decision Tree \\n(DT). Analysis of variance and recursive feature elimination using cross validation have \\nbeen applied for feature selection. Evaluation of the models was done using tenfold \\ncross-validation. The results from the experiments indicated that RF based on recursive'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 0, 'page_label': '1', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='been applied for feature selection. Evaluation of the models was done using tenfold \\ncross-validation. The results from the experiments indicated that RF based on recursive \\nfeature elimination with cross validation has better performance than SVM and DT.\\nKeywords: Chronic Kidney Disease (CKD), Machine Learning, Random Forest (RF), \\nSupport Vector Machine (SVM)\\nOpen Access\\n© The Author(s) 2022. Open Access This article is licensed under a Creative Commons Attribution 4.0 International License, which permits \\nuse, sharing, adaptation, distribution and reproduction in any medium or format, as long as you give appropriate credit to the original \\nauthor(s) and the source, provide a link to the Creative Commons licence, and indicate if changes were made. The images or other third \\nparty material in this article are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line to the mate-'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 0, 'page_label': '1', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='party material in this article are included in the article’s Creative Commons licence, unless indicated otherwise in a credit line to the mate-\\nrial. If material is not included in the article’s Creative Commons licence and your intended use is not permitted by statutory regulation or \\nexceeds the permitted use, you will need to obtain permission directly from the copyright holder. To view a copy of this licence, visit http:// \\ncreat iveco mmons. org/ licen ses/ by/4. 0/.\\nRESEARCH\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109  \\nhttps://doi.org/10.1186/s40537-022-00657-5\\nJournal of Big Data\\n*Correspondence:   \\ndibabaadeba44@gmail.com\\n1 Department of Information \\nScience, College of Computing, \\nMadda Walabu University, Robe, \\nEthiopia\\n2 Department of Computer \\nScience and Engineering, \\nSchool of Electrical Engineering \\nand Computing, Adama Science \\nand Technology University, \\nAdama, Ethiopia'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 1, 'page_label': '2', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 2 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nprevention and treatment remain low [2]. Kidney disease is serious public health prob -\\nlem in Ethiopia affecting hundreds of thousands of people irrespective of age, sex [4]. \\nThe lack of safe water, appropriate diet, and physical activities is believed have contrib -\\nuted. Additionally, communities living in rural area have limited knowledge about the \\nCKD. According to WHO report of 2017, the number of deaths in Ethiopia due to kid -\\nney disease was 4,875. It is 0.77% of total deaths that has ranked the country 138th in the \\nworld. The age-adjusted death rate is 8.46 per 100,000 of the population and the death \\nrate increased to 12.70 per 100,000 that has ranked the country 109 in 2018 [3].\\nNational kidney foundation classifies stages of CKD into five based on the abnormal \\nkidney function and reduced Glomerular Filtration Rate (GFR), which measures a level'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 1, 'page_label': '2', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='National kidney foundation classifies stages of CKD into five based on the abnormal \\nkidney function and reduced Glomerular Filtration Rate (GFR), which measures a level \\nof kidney function,. The mildest stage (stage 1 and stage 2) is known with only a few \\nsymptoms and stage 5 is considered as end-stage or kidney failure. The Renal Replace -\\nment Therapy (RRT) cost for total kidney failure is very expensive. The treatment is not \\nalso available in most developing countries like Ethiopia. As a result, the management of \\nkidney failure and its complications is very difficult in developing countries due to short-\\nage of facilities, physicians, and the high cost to get the treatment [4, 5]. Hence, early \\ndetection of CKD is very essential to minimize the economic burden and maximize the \\neffectiveness of treatments [6]. Predictive analysis using machine learning techniques \\ncan be helpful through an early detection of CKD for efficient and timely interventions'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 1, 'page_label': '2', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='effectiveness of treatments [6]. Predictive analysis using machine learning techniques \\ncan be helpful through an early detection of CKD for efficient and timely interventions \\n[7]. In this study, Random Forest (RF), Support Vector Machine (SVM) and Decision \\nTree (DT) have been used to detect CKD. Most of previous researches focused on two \\nclasses, which make treatment recommendations difficult because the type of treatment \\nto be given is based on the severity of CKD.\\nRelated works\\nDifferent machine-learning techniques have been used for effective classification of \\nchronic kidney disease from patients’ data.\\nCharleonnan et\\xa0 al. [8] did comparison of the predictive models such as K-nearest \\nneighbors (KNN), support vector machine (SVM), logistic regression (LR), and decision \\ntree (DT) on Indians Chronic Kidney Disease (CKD) dataset in order to select best clas -\\nsifier for predicting chronic kidney disease. They have identified that SVM has the high -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 1, 'page_label': '2', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='tree (DT) on Indians Chronic Kidney Disease (CKD) dataset in order to select best clas -\\nsifier for predicting chronic kidney disease. They have identified that SVM has the high -\\nest classification accuracy of 98.3% and highest sensitivity of 0.99.\\nSalekin and Stankovic [9] did evaluation of classifiers such as K-NN, RF and ANN on \\na dataset of 400. Wrapper feature selection were implemented and five features were \\nselected for model construction in the study. The highest classification accuracy is 98% \\nby RF and a RMSE of 0.11. S. Tekale et\\xa0al. [10] worked on “Prediction of Chronic Kidney \\nDisease Using Machine Learning Algorithm” with a dataset consists of 400 instances and \\n14 features. They have used decision tree and support vector machine. The dataset has \\nbeen preprocessed and the number of features has been reduced from 25 to 14. SVM is \\nstated as a better model with an accuracy of 96.75%.\\nXiao et\\xa0 al. [11] proposed prediction of\\xa0 chronic kidney disease progression using'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 1, 'page_label': '2', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='stated as a better model with an accuracy of 96.75%.\\nXiao et\\xa0 al. [11] proposed prediction of\\xa0 chronic kidney disease progression using \\nlogistic regression, Elastic Net, lasso regression, ridge regression, support vector \\nmachine, random forest, XGBoost, neural network and k-nearest neighbor and com -\\npared the models based on their performance. They have used 551 patients’ history \\ndata with proteinuria with 18 features and classified the outcome as mild, moderate,'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 2, 'page_label': '3', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 3 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\nsevere. They have concluded that Logistic regression performed better with AUC of \\n0.873, sensitivity and specificity of 0.83 and 0.82, respectively.\\nMohammed and Beshah [13] conducted their research on developing a self-learn -\\ning knowledge-based system for diagnosis and treatment of the first three stages of \\nchronic kidney have been conducted using machine learning. A small number of data \\nhave been used in this research and they have developed prototype which enables the \\npatient to query KBS to see the delivery of advice. They used decision tree in order to \\ngenerate the rules. The overall performance of the prototype has been stated as 91% \\naccurate.\\nPriyanka et\\xa0al. [12] carried out chronic kidney disease prediction through naive bayes. \\nThey have tested using other algorithms such as KNN (K-Nearest Neighbor Algorithm), \\nSVM (Support Vector Machines), Decision tree, and ANN (Artificial Neural Network)'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 2, 'page_label': '3', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='They have tested using other algorithms such as KNN (K-Nearest Neighbor Algorithm), \\nSVM (Support Vector Machines), Decision tree, and ANN (Artificial Neural Network) \\nand they have got Naïve Bayes with better accuracy of 94.6% when compared to other \\nalgorithms.\\nAlmasoud and Ward [13] aimed in their work to test the ability of machine learning \\nalgorithms for the prediction of chronic kidney disease using subset of features. They \\nused Pearson correlation, ANOVA, and Cramer’s V test to select predictive features. \\nThey have done modeling using LR, SVM, RF, and GB machine learning algorithms. \\nFinally, they concluded that Gradient Boosting has the highest accuracy with an F-meas-\\nure of 99.1.\\nYashfi [14] proposed to predict the risk of CKD using machine learning algorithms by \\nanalyzing the data of CKD patients. Random Forest and Artificial Neural Network have \\nbeen used. They have extracted 20 out of 25 features and applied RF and ANN. RF has'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 2, 'page_label': '3', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='analyzing the data of CKD patients. Random Forest and Artificial Neural Network have \\nbeen used. They have extracted 20 out of 25 features and applied RF and ANN. RF has \\nbeen identified with the highest accuracy of 97.12%.\\nRady and Anwar [15] carried out the comparison of Probabilistic Neural Networks \\n(PNN), Multilayer Perceptron (MLP), Support Vector Machine (SVM), and Radial Basis \\nFunction (RBF) algorithms to predict kidney disease stages. The researchers conducted \\ntheir research on a small size dataset and few numbers of features. The result of this \\npaper shows that the Probabilistic Neural Networks algorithm gives the highest overall \\nclassification accuracy percentage of 96.7%.\\nAlsuhibany et\\xa0al. [16] presented ensemble of deep learning based clinical decision sup-\\nport systems (EDL-CDSS) for CKD diagnosis in the IoT environment. The presented \\ntechnique involves Adaptive Synthetic (ADASYN) technique for outlier detection pro -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 2, 'page_label': '3', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='port systems (EDL-CDSS) for CKD diagnosis in the IoT environment. The presented \\ntechnique involves Adaptive Synthetic (ADASYN) technique for outlier detection pro -\\ncess and employed ensemble of three models, namely, deep belief network (DBN), kernel \\nextreme learning machine (KELM), and convolutional neural network with gated recur -\\nrent unit (CNN-GRU).\\nQuasi-oppositional butterfly optimization algorithm (QOBOA) technique is also \\nemployed in the study for hyperparameter tuning of DBN and CNN-GRU. The research-\\ners have concluded that EDL-CDSS method has the capability of proficiently detecting \\nthe presence of CKD in the IoT environment.\\nPoonia et\\xa0al. [17] employed Various machine learning algorithms, including k-nearest \\nneighbors algorithm (KNN), artificial neural networks (ANN), support vector machines \\n(SVM), naive bayes (NB), and Logistic Regression as well as Re-cursive Feature Elimina -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 2, 'page_label': '3', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='neighbors algorithm (KNN), artificial neural networks (ANN), support vector machines \\n(SVM), naive bayes (NB), and Logistic Regression as well as Re-cursive Feature Elimina -\\ntion (RFE) and Chi-Square test feature-selection techniques. Publicly available dataset of \\nhealthy and kidney disease patients were used to build and analyze prediction models.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 3, 'page_label': '4', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 4 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nThe study found that a logistic regression-based prediction model with optimal features \\nchosen using the Chi-Square technique had the highest accuracy of 98.75%.\\nVinod [18] carried out the assessment of seven supervised machine learning algo -\\nrithms namely K-Nearest Neighbor, Decision Tree, Support vector Machine, Random \\nForest, Neural Network, Naïve Bayes and Logistic Regression to find the most suitable \\nmodel for BCD prediction based on different performance evaluation. Finally, the result \\nshowed that k-NN is the best performer on the BCD dataset with 97% accuracy.\\nThe above reviews indicates that several studies have been conducted on chronic kid -\\nney disease prediction using machine-learning techniques. There are various parameters \\nwhich play important role in improving model performance like dataset size, quality of'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 3, 'page_label': '4', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='ney disease prediction using machine-learning techniques. There are various parameters \\nwhich play important role in improving model performance like dataset size, quality of \\ndataset and the time dataset collected. This study focuses on chronic kidney disease pre -\\ndiction using machine learning models based on the dataset with big size and recent \\nthan online available dataset collected from St. Paulo’s Hospital in Ethiopia with five \\nclasses: notckd, mild, moderate, severe, and ESRD and binary classes: ckd and notckd \\nby applying machine-learning models. Most previously conducted researches focused \\non two classes, which make treatment recommendations difficult because the type of \\ntreatment to be given is based on the stages. Table\\xa0 1 below shows the summary of some \\nrelated works.\\nTable 1 Summary of some related works\\nNo. Author Technique applied Claimed outcome Draw back\\n1 Salekin and Stankovic \\n[9]\\nK-NN, RF, and NN, \\nWrapper approach and \\nEmbedded approach'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 3, 'page_label': '4', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='related works.\\nTable 1 Summary of some related works\\nNo. Author Technique applied Claimed outcome Draw back\\n1 Salekin and Stankovic \\n[9]\\nK-NN, RF, and NN, \\nWrapper approach and \\nEmbedded approach\\nDetection F1-score of RF 99.8 Small dataset size with \\nmissing values was used; \\nSeverity level prediction \\nwas not included\\n2 Tekale et al. [10] DT and SVM DT and SVM with an accuracy \\nof 91.75 and 96.75 respec-\\ntively\\nDataset size need to \\nincreased, Severity level \\nprediction was not \\nincluded. Only to classi-\\nfiers result compared\\n3 Priyanka et al. [12] NB, KNN, SVM, DT, and \\nANN. NB\\nNB, KNN, SVM, DT, and ANN. \\nNB accuracy is 94.6%\\nSmall size dataset. \\nNo stages prediction. \\nFeature extraction was \\nnot carried out and clas-\\nsification accuracy needs \\nimprovement\\n4 Yashfi [14] RF and ANN RF and ANN with an accuracy \\nof 97.12% and 94.5%\\nSmall size dataset and no \\nstages prediction\\n5 Rady and Anwar [15] PNN, MLP , SVM, RBF PNN, MLP , SVM, RBF accuracy \\nof 96.7, 60.7, 87, and 51.5'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 3, 'page_label': '4', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='of 97.12% and 94.5%\\nSmall size dataset and no \\nstages prediction\\n5 Rady and Anwar [15] PNN, MLP , SVM, RBF PNN, MLP , SVM, RBF accuracy \\nof 96.7, 60.7, 87, and 51.5 \\nrespectively\\nSmall data size. the \\nalgorithms, which is not \\nappropriate for small \\ndataset size were used\\n6 Alsuhibany et al. [16] EDL-CDSS technique, \\nADASYN based outlier \\ndetection and QOBOA \\nbased hyperparameter \\ntuning\\nThe proposed EDL-CDSS \\nmethod has depicted the \\nother approaches with the \\nsuperior sensy and specy \\nof 0.9680 and 0. 9702 \\ncompared to ACO, FNC, KELM, \\nCNN-GRU, DBN, DT, MLP and \\nD-ACO\\nThe size of benchmark \\ndataset is small. IoT \\nand benchmark data \\nintegration is not clearly \\nexplained. The study \\nfocus only on binary \\nclassification (ckd or \\nnot-ckd)\\n7 Poonia et al. [17] KNN, ANN, SVM, NB \\nand LR with Chi-Square \\nand RFE\\nKNN, ANN, SVM, \\nNB, LR accuracy of \\n66.25%,65%,97.5%,95%,97.5% \\nrespectively\\nSmall size dataset and no \\nstages prediction. Accu-\\nracy needs improvement'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 4, 'page_label': '5', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 5 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\nMaterials and\\xa0method\\nData source and\\xa0description\\nThe data source for this study is St. Paulo’s Hospital. It is the second-largest public \\nhospital in Ethiopia which admits large number of patients with chronic diseases. \\nThere are dialysis treatment and kidney transplant center in the hospital. As it has \\nbeen shown in Table\\xa02 , the dataset for this study is patients’ records of chronic kidney \\ndisease from patients admitted to the renal ward during 2018 to 2019. Some of them \\nwere obtained from the same patient history data at different times of different stages. \\nTo prepare the dataset and understand features, interviews of domain experts have \\nbeen conducted. The dataset contains 1718 instances with 19 features where 12 are \\nnumeric and 7 are nominal. As the detail have been shown in Table\\xa0 3, the features in \\nthe dataset include Age, Gender, blood pressure, specific gravity, chloride, sodium,'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 4, 'page_label': '5', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='numeric and 7 are nominal. As the detail have been shown in Table\\xa0 3, the features in \\nthe dataset include Age, Gender, blood pressure, specific gravity, chloride, sodium, \\npotassium, blood urine nitrogen, serum creatinine, hemoglobin, red blood cell count, \\nwhite blood cell count, mean cell volume, platelet count, hypertension, diabetic mel -\\nlitus, anemia and heart disease. When we see multi class distribution, 441 (25.67%) \\ninstances are end-stage renal disease stage or stage five, 399 (23.22%) are at a severe \\nstage or stage four, 354 (20.61%) are at a moderate stage or stage three, 248 (14.44%) \\nare at a mild stage or stage two, and 276 (16.07%) have no chronic kidney disease or \\nnormal. The class distribution for binary class is 1442 (83.93%) ckd (stage 1 to 5) and \\n276 (16.07%) notcckd. The binary-class distribution is imbalanced. Oversampling data \\nresampling technique have been used to balance the value of the minority class with'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 4, 'page_label': '5', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='276 (16.07%) notcckd. The binary-class distribution is imbalanced. Oversampling data \\nresampling technique have been used to balance the value of the minority class with \\nthe value of the majority class. After employing the resampling technique, the total \\nsize of binary class dataset, become 2888.\\nPreprocessing\\nReal world data is often inconsistent which can affect the performances of mod -\\nels. Preprocessing the data before it is fed into classifiers is vital part of developing \\nTable 2 Dataset description\\nDescription Values\\nData source St. Paul’s \\nHospital, \\nAddis Ababa, \\nEthiopia\\nPeriod 2018 to 2019\\nNo. of instances 1718\\nNo. of features 19\\nClass Both binary \\nand multi \\nclass (5 \\nstages)\\nEnd-stage renal disease stage (stage five) 25.67%\\nSevere stage (stage four) 23.22%\\nModerate stage (stage three) 20.61%\\nMild stage (stage two) 14.44\\nPercentage of negative samples 16.07%'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 5, 'page_label': '6', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 6 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nmachine-learning model. Similarly, the dataset for this study contains missing values \\nthat needs to be handled appropriately. It has to also be in a suitable format for mod -\\neling. Hence, pre-processing has been conducted as it has been shown in Fig.\\xa01 .\\nCleaning Noisy Data: removing outlier and smoothening noisy data is an important \\npart of preprocessing. Outliers are values that lies away from the range of the rest of \\nthe values. In clinical data, outliers may arise from the natural variance of data. The \\npotential outliers are the data points that fall above Q3 + 1.5(IQR) and below Q1 − 1.5 \\n(IQR), where Q1 is the first quartile, Q3 is the third quartile, and IQR = Q3 − Q1 [19].\\nHandling Missing Values: data is not always available (or missed) due to equipment \\nmalfunction, inconsistent with other recorded data and thus deleted, not entered into \\nTable 3 Features description'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 5, 'page_label': '6', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Handling Missing Values: data is not always available (or missed) due to equipment \\nmalfunction, inconsistent with other recorded data and thus deleted, not entered into \\nTable 3 Features description\\nSymbols Features full name Type Class Missing values in %\\nAge Age Numeric Predictor 0\\nGender Gender Nominal Predictor 0\\nBp Blood pressure Numeric Predictor 0.058207\\nSg Specific gravity Nominal Predictor 0.058207\\nChl Chloride Numeric Predictor 0.116414\\nSod Sodium Numeric Predictor 0.232829\\nPot Potassium Numeric Predictor 0.116414\\nBun Blood Urea Nitrogen Numeric Predictor 0.232829\\nScr Serum Creatinine Numeric Predictor 0.058207\\nHgb Hemoglobin Numeric Predictor 0.232829\\nRbcc Red blood cell count Numeric Predictor 0.232829\\nWbcc White blood cell count Numeric Predictor 0.232829\\nMcv Mean cell volume Numeric Predictor 6.111758\\nPltc Platelet count Numeric Predictor 7.275902\\nHtn Hypertension Nominal Predictor 0\\nDm Diabetes Mellitus Nominal Predictor 0\\nAne Anemia Nominal Predictor 0'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 5, 'page_label': '6', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Pltc Platelet count Numeric Predictor 7.275902\\nHtn Hypertension Nominal Predictor 0\\nDm Diabetes Mellitus Nominal Predictor 0\\nAne Anemia Nominal Predictor 0\\nHd Heart disease Nominal Predictor 0\\nckd_status ckd_status Nominal Target 0\\nFig. 1 Chronic kidney disease dataset preprocessing steps'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 6, 'page_label': '7', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 7 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\nthe database due to misunderstanding, some data may not be considered important at \\nthe time of entry.\\nPatient data often has missing diagnostic test results that would help to predict the \\nlikelihood of diagnoses or predict treatment’s effectiveness [20]. The missing values have \\nan impact on the performance of the prediction model. There are several ways of han -\\ndling missing values including dropping missing values and filling missing values. Some -\\ntimes missing values are ignored if they are small percentage i.e., if missing data under \\n10%. But it is not considered healthy for the model because the missing value can be \\nan important feature contributing to the model development. Sometimes the missing \\nvalues can be also replaced by zero, which will bring no change to the model. To handle \\nthese missing values the mean, an average of the observed features used in this study,'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 6, 'page_label': '7', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='values can be also replaced by zero, which will bring no change to the model. To handle \\nthese missing values the mean, an average of the observed features used in this study, \\nbecause the missing features are numeric and mean imputation is better for numerical \\nmissing values.\\nHandling Categorical Data: In this step, data has been transformed into the required \\nformat. The nominal data converted into numerical data of the form 0 and 1. For \\ninstance, ‘Gender’ has the nominal value that can be labeled as 0-for female and 1-for \\nmale. After preprocessing the data then the resultant CSV file comprises all the integer \\nand float values for different CKD related features.\\nNormalization: It is important to scale numerical features before fitting to any mod -\\nels, as scaling is mandatory for some techniques such as nearest neighbors, SVMs, and \\ndeep learning [21]. There are different techniques of scaling and in this study Z-score'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 6, 'page_label': '7', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='els, as scaling is mandatory for some techniques such as nearest neighbors, SVMs, and \\ndeep learning [21]. There are different techniques of scaling and in this study Z-score \\nnormalization (or zero-mean normalization) have been used. The values for a feature are \\nnormalized based on the mean and standard deviation. It is as follows:\\nwhere z is Z-score, x is feature value, µ is mean value and σ is standard deviation.\\nFeature selection\\nIdentify subset of relevant predictive features is important for quality result [22]. Feature \\nselection is the process of selecting most important predictive features to use them as \\ninput for models. It is important preprocessing step to deal with the problem of high \\ndimensionality. Hence, the main aim of feature selection is to select the subset of fea -\\ntures that are relevant and independent of each other for training the model [23]. Simi -\\nlarly, feature selection is crucial to develop chronic kidney disease predictive model. This'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 6, 'page_label': '7', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='tures that are relevant and independent of each other for training the model [23]. Simi -\\nlarly, feature selection is crucial to develop chronic kidney disease predictive model. This \\nreduces the dimensionality and complexity of the data and makes the model be faster, \\nmore effective and accurate. Hence, feature selection algorithm have been used to select \\nrelevant features after the construction of the dataset.\\nFilter, wrappers, or embedded techniques are widely used for feature selection in dif -\\nferent clinical datasets including chronic kidney disease. A filter method is a technique \\nthat is independent of a classification algorithm and uses general characteristics of data \\nfor evaluating and selecting relevant features. The filter method works independently of \\nthe learning algorithm to remove irrelevant features and analyses properties of a data -\\nset to chooses relevant features [24]. It is widely used approach due to its less complex'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 6, 'page_label': '7', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='the learning algorithm to remove irrelevant features and analyses properties of a data -\\nset to chooses relevant features [24]. It is widely used approach due to its less complex \\nin nature. With wrapper method, relevant features are selected using the classification \\n(1)z = x − µ\\nσ'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 7, 'page_label': '8', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 8 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nalgorithm. It is better than filter feature selection technique in terms of accuracy. How -\\never, it requires higher processing time. In this study, univariate feature selection method \\nfrom filter methods have been selected because it is fast, efficient and scalable. Recursive \\nfeature elimination with cross-validation (RFECV) has been used from wrapper feature \\nselection method.\\nUnivariate Feature Selection (UFS): This method is popular, simplistic and fastest fea -\\nture selection method used in healthcare dataset. It considers each feature separately to \\ndetermine the strength of the relationship of the feature with the dependent variable. It \\nis fast, scalable and independent of classifier. Different options are there for univariate \\nalgorithms such as Pearson correlation, information gain, chi-square, ANOVA (Analysis \\nof Variance). In this study, feature selection was done using\\xa0 the ANOVA as shown in'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 7, 'page_label': '8', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='algorithms such as Pearson correlation, information gain, chi-square, ANOVA (Analysis \\nof Variance). In this study, feature selection was done using\\xa0 the ANOVA as shown in \\nEq.\\xa02.\\nwhere F is ANOVA\\xa0coefficient, MST is mean\\xa0sum\\xa0of\\xa0squares\\xa0of\\xa0treatment and MSE is \\nmean\\xa0sum\\xa0of\\xa0squares\\xa0error.\\nRecursive Feature Elimination with Cross-Validation (RFECV): An optimization algo -\\nrithm to develop a trained machine-learning model with relevant and selected features \\nby repeatedly eliminating irrelevant features. It repetitively creates the model, keeps \\naside the worst performing feature at each iteration, and builds the next model with the \\nremaining features until the features are completed to select best subset of features [25]. \\nIt eliminates the redundant and weak feature whose deletion least affects the training \\nand keeps the independent and strong feature to improve the generalization perfor -\\nmance of the model [26]. This method uses the iterative procedure for feature ranking'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 7, 'page_label': '8', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='and keeps the independent and strong feature to improve the generalization perfor -\\nmance of the model [26]. This method uses the iterative procedure for feature ranking \\nand to find out the features that have been evaluated as most important.\\xa0Because this \\ntechnique work interacting with a machine learning model, it first builds the model on \\nthe entire set of features and ranked the feature according to its importance.\\nMachine learning models\\nThe aim of the study was to predict chronic kidney disease using machine-learning tech-\\nniques. Three machine learning algorithms; Random Forest, Support Vector Machine \\nand Decision Tree have been used in this study. The algorithms were selected based on \\ntheir popularity in chronic kidney disease prediction and their performance of classifica-\\ntion on previous research works [12, 27–33].\\nRandom Forest: Random Forest is ensemble learning that consists of several collec -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 7, 'page_label': '8', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='tion on previous research works [12, 27–33].\\nRandom Forest: Random Forest is ensemble learning that consists of several collec -\\ntions of decision trees. It is used for both classification and regression. This model com -\\nprises of a number of decision trees and outputs the class target that is the highest voting \\nresults of the target output by each tree [28]. Random Forest uses both bagging and ran -\\ndom feature selection to build the tree and creates an uncorrelated forest of trees. Its \\nprediction by the group is more accurate than that of any individual tree. After it builds \\nthe forest, test instances are permeated down through each tree and trees make their \\nrespective prediction of class [33]. The Random Forest pseudocode is shown in Fig.\\xa02.\\nSupport vector machine (SVM): Support Vector Machine is one of the promi -\\nnent and convenient supervised machine-learning algorithm that can be used for \\n(2)F = MST\\nMSE'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 8, 'page_label': '9', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 9 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\nclassification, learning and prediction. A set of hyperplanes are built to classify all \\ninput in high dimensional data. A discrete hyperplane is created in the signifier space \\nof the training data and compounds are classified based on the side of the hyperplane \\n[30]. Hyperplanes are decision boundaries that separate the data points. Support vec -\\ntors are data points that are closer to the hyperplane and determine the position and \\norientation of the hyperplane. SVMs have been mainly proposed to deal with binary \\nclassification, but nowadays many researchers have tried to apply it to multiclass clas -\\nsification because there are a huge amount of data to be classified into more than \\ntwo classes in today’s world. SVM solves multiclass problems through the two most \\npopular approaches; one-versus-rest and one-vs-one. In this study, one-versus-'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 8, 'page_label': '9', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='two classes in today’s world. SVM solves multiclass problems through the two most \\npopular approaches; one-versus-rest and one-vs-one. In this study, one-versus-\\nrest has been used. For multi classification, we used OVR with the SVM algorithm. \\nThis method separates each class from the rest of the classes in the dataset. Besides, \\nbecause Linear SVC is used in this study, one vs rest is an appropriate method with \\nLinear SVC. The pseudocode of SVM have been shown in Fig.\\xa03 .\\nDecision Tree (DT): It is one of the most popular supervised machine-learning \\nalgorithms that can be used for classification. Decision Tree solves the problem of \\nmachine learning by transforming the data into a tree representation through sorted \\nfeature values. Each node in a decision tree denotes features in an instance to be \\nclassified, and each leaf node represents a class label the instances belong to. This \\nmodel uses a tree structure to split the dataset based on the condition as a predictive'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 8, 'page_label': '9', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='classified, and each leaf node represents a class label the instances belong to. This \\nmodel uses a tree structure to split the dataset based on the condition as a predictive \\nmodel that maps observations about an item to make a decision on the target value of \\ninstances [34]. Decision Tree pseudocode is shown in Fig.\\xa0 4 and decision making in \\nbinary class of chronic kidney disease have been shown in Fig.\\xa05 .\\nFig. 2 Random forest pseudocode [18]\\nFig. 3 SVM pseudocode [18]'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 9, 'page_label': '10', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 10 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nFig. 4 Decision tree pseudocode [18]\\nFig. 5 Decision making in binary class of chronic kidney disease\\nFig. 6 Model building flow diagram'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 10, 'page_label': '11', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 11 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\nFigure\\xa06 shows the flow of model building using three machine-learning algorithms with \\ntenfold cross validation. The machine learning models were developed for both multiclass \\nand binary classification. The best performing model was selected as best machine learning \\nmodel from the three algorithms for each classification.\\nPrediction model evaluation\\nPerformance evaluation is the critical step of developing an accurate machine-learning \\nmodel. Prediction model shall to be evaluated to ensure that the model fits the dataset and \\nwork well on unseen data. The aim of the performance evaluation is to estimate the gener-\\nalization accuracy of a model on unseen/out-of-sample data. Cross-Validation (CV) is one \\nof the performance evaluation methods for evaluating and comparing models by dividing \\ndata into partitions. The original dataset was partitioned into k equal size subsamples called'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 10, 'page_label': '11', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='of the performance evaluation methods for evaluating and comparing models by dividing \\ndata into partitions. The original dataset was partitioned into k equal size subsamples called \\nfolds: nine used to train a model and one used to test or validate the model. This process \\nrepeated k times and the average performance will be taken. Tenfold cross-validation have \\nbeen used in this study. Different performance evaluation metrics including accuracy, pre-\\ncision, recall, f1-score, sensitivity, specificity have been computed.\\n• True positive (TP): are the condition when both actual value and predicted value are \\npositive.\\n• True negative (TN): are the condition when both the actual value of the data point and \\nthe predicted are negative.\\n• False positive (FP): These are the cases when the actual value of the data point was nega-\\ntive and the predicted is positive.\\n• False negative (FN): are the cases when the actual value of the data point is positive and \\nthe predicted is negative.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 10, 'page_label': '11', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='tive and the predicted is positive.\\n• False negative (FN): are the cases when the actual value of the data point is positive and \\nthe predicted is negative.\\nAccuracy\\nAccuracy implies the ability of the classification algorithm to predict the classes of the data-\\nset correctly. It is the measure of how close or near the predicted value is to the actual or \\ntheoretical value [35]. Generally, accuracy is the measure of the ratio of correct predictions \\nover the total number of instances. The equation of accuracy is shown in Eq.\\xa03.\\nPrecision\\nPrecision measure the true values correctly predicted from the total predicted values in the \\nactual class. Precision quantifies the ability of the classifiers to not label a\\xa0negative\\xa0example \\nas\\xa0positive. The equation of precision is shown in Eq.\\xa04.\\nMacro average is used for multiclass classification because it gives equal weight for \\neach class. The equation of macro average precision is shown in Eq.\\xa05.\\n(3)Accuracy= TP + TN\\nTP + FP + TN + FN'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 10, 'page_label': '11', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Macro average is used for multiclass classification because it gives equal weight for \\neach class. The equation of macro average precision is shown in Eq.\\xa05.\\n(3)Accuracy= TP + TN\\nTP + FP + TN + FN\\n(4)Precision= TP\\nTP + FP'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 11, 'page_label': '12', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 12 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nRecall\\nRecall measure the rate of positive values that are correctly classified. Recall answers the \\nquestion of what proportion of\\xa0 actual positives\\xa0 are correctly classified. The equation of \\nrecall is shown in Eq.\\xa06.\\nSince the macro average is used in order to compute the recall value of the models, macro \\naverage recall is calculated as follows (Eq.\\xa07).\\nF‑measure\\nF-measure is also called F1-score is the harmonic mean between recall and precision. The \\nequation of F1-score is shown in Eq.\\xa08.\\nThe macro average of F1_score is calculated as follows (Eq.\\xa09).\\nSensitivity\\nSensitivity is also called True Positive Rate. Sensitivity is the mean proportion of actual true \\npositives that are correctly identified [36]. The equation of sensitivity is shown in Eq.\\xa010.\\nSpecificity\\nSpecificity is also called True Negative Rate. It is used to measure the fraction of negative'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 11, 'page_label': '12', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Specificity\\nSpecificity is also called True Negative Rate. It is used to measure the fraction of negative \\nvalues that are correctly classified. The equation of sensitivity is shown in Eq.\\xa011.\\n(5)Precision_macro = 1\\n|C|\\n∑|C|\\ni=1\\nTPi\\nTPi + FPi\\n(6)Recall= TP\\nTP + FN\\n(7)Recall_macro = 1\\n|C|\\n∑|C |\\ni=1\\nTPi\\nTPi + FNi\\n(8)F 1score = 2 ∗ Precision∗Recall\\nPrecision+ Recall\\n(9)F 1score_macro = 2 ∗ Precision_macro∗Recall_macro\\nPrecision_macro+ Recall_macro\\n(10)Sensitivity= TP\\nTP + FN\\n(11)Speciﬁcity= TN\\nTN + FP'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 12, 'page_label': '13', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 13 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\nResults and\\xa0discussions\\nFeature selection\\nThe features selection process using the two methods; UFS and RFECV resulted two dif -\\nferent sets of features. The resulted subset of features have been used in the training of \\nRF, SVM, and DT. The selected features for both five-class and binary-class are different \\nbecause recursive feature elimination with cross-validation can automatically eliminate \\nless predictive features iteratively depending on the model. Table\\xa04 shows the number of \\nselected features of binary-class and five-class that set the size of the dataset respectively.\\nEvaluation results\\nThe Experiment has been carried out based on two feature selection methods and three \\nclassifiers for both binary and five-class classification for 18 models. Training and testing \\non these models have been executed using tenfold cross-validation. In tenfold cross-val -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 12, 'page_label': '13', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='classifiers for both binary and five-class classification for 18 models. Training and testing \\non these models have been executed using tenfold cross-validation. In tenfold cross-val -\\nidation, the dataset is partitioned randomly into ten equal size sets. Training the models \\nwas then done by using 10–1 folds and test using one remaining fold. The process is \\niterative for each fold. The obtained results are presented in binary and five-class classifi-\\ncation models. Modeling was first carried out for both binary and five class classification \\nusing preprocessed dataset without applying feature selection methods. Then, modeling \\nhas been experimented by applying the two feature selection methods as it has been dis -\\ncussed in the following sections.\\nBinary classification models evaluation results\\nThese classification models were built using the two-class dataset that was converted \\nfrom the five-class dataset. The target classes are notckd and ckd. The models are trained'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 12, 'page_label': '13', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='These classification models were built using the two-class dataset that was converted \\nfrom the five-class dataset. The target classes are notckd and ckd. The models are trained \\nand tested using tenfold CV and other performance evaluation metrics of CV. As it has \\nbeen discussed previously, modeling was first conducted on the preprocessed dataset \\nwithout applying feature selection methods. Then, feature selection techniques have \\nbeen implemented to select the most predictive features. Feature selection was imple -\\nmented using UFS and RFECV. The performance measures of each test for RF, SVM, and \\nDT models before feature selection and after feature selection method are presented in \\nTable\\xa05.\\nAn accuracy of 99.8% resulted in the RF with RFECV model with selected 8 features is \\nthe highest. The result of models before applying feature selection is graphically shown \\nin Fig.\\xa07.\\nTable 4 Feature selection results'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 12, 'page_label': '13', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='the highest. The result of models before applying feature selection is graphically shown \\nin Fig.\\xa07.\\nTable 4 Feature selection results\\nFeature selection methods Machine learning algorithm Selected features (Binary) Selected \\nfeatures (five-\\nClass)\\nRFECV RF 8 9\\nSVM 9 10\\nDT 16 7\\nUFS (ANOVA) RF 14 14\\nSVM 14 14\\nDT 14 14'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 13, 'page_label': '14', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 14 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nFurtherly, we have carried out hyperparameter optimization using grid search using \\ncross validation on SVM for binary dataset. This has been done without feature selec -\\ntion. The performance has been significantly improved to 99.83% which is almost the \\nsame with the highest RF with RFECV. Another experiment on the binary classification \\nis with Extreme Gradient Boosting (XGBoost). It is a powerful machine learning algo -\\nrithm that can be used to solve classification and regression problems. The performance \\nis 98.96% which is not better than RF with RFECV’s performance.\\nMulticlass classification models evaluation results\\nThe multiclass models were similarly built using the preprocessed five-class dataset. The \\nmodels are trained and tested using tenfold CV and evaluated with other performance \\nevaluation metrics. The performance metrics result of each trained model; RF, SVM,'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 13, 'page_label': '14', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='models are trained and tested using tenfold CV and evaluated with other performance \\nevaluation metrics. The performance metrics result of each trained model; RF, SVM, \\nand DT have been presented for without and with feature selection method as shown in \\nTable\\xa06. Models were first trained and tested with all features without applying feature \\nselection methods and then we apply the feature selection methods.\\nTable\\xa06 shows the CV performance metrics results for three classifiers of multiclass \\ndataset before and after applying feature selection. The accuracy is 79% from RF with \\nRFECV with selected 9 features. The result of models after applying feature selection \\nTable 5 Binary classification without and with feature selection\\nModels No of features Acc Prec Rec F1 Sen Spec\\nRF 18 99.7 99.9 99.5 99.7 99.5 99.9\\nRF with RFECV 8 99.8 99.9 99.7 99.8 99.7 99.9\\nRF with UFS 14 99.7 99.9 99.5 99.7 99.5 99.9\\nSVM 18 96.9 99.1 94.7 96.8 94.7 99.1\\nSVM with RFECV 9 95.5 98.7 92.2 95.3 92.2 98.8'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 13, 'page_label': '14', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='RF with RFECV 8 99.8 99.9 99.7 99.8 99.7 99.9\\nRF with UFS 14 99.7 99.9 99.5 99.7 99.5 99.9\\nSVM 18 96.9 99.1 94.7 96.8 94.7 99.1\\nSVM with RFECV 9 95.5 98.7 92.2 95.3 92.2 98.8\\nSVM with UFS 14 96.7 99.4 97.7 98.6 97.7 99.4\\nDT 18 98.5 99.4 97.6 98.5 97.6 99.4\\nDT with RFECV 16 98.6 99.4 97.7 98.6 97.7 99.4\\nDT with UFS 14 98.5 99.4 97.6 98.5 97.6 99.4\\nFig. 7 Binary class classification without feature selection'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 14, 'page_label': '15', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 15 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\nis graphically shown in Fig.\\xa0 8. Similarly, we have carried out hyperparameter optimi -\\nzation using grid search using cross validation on SVM for multiclass dataset. This has \\nbeen done without feature selection. The performance has been significantly improved \\nto 78.78% which is not better than the highest performing model RF with RFECV. \\nAnother experiment on the multiclass classification is with Extreme Gradient Boosting \\n(XGBoost). It is a powerful machine learning algorithm that can be used to solve classifi-\\ncation and regression problems. The performance is 82.56% which is better than RF with \\nRFECV’s performance.\\nDiscussions\\nChronic kidney disease is a global health threat and becoming silent killer in Ethiopia \\n[6]. Many people die or suffer severely by the disease mainly due to lack of awareness \\nabout the disease and inability to detect early. Thus, early prediction of chronic kidney'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 14, 'page_label': '15', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='[6]. Many people die or suffer severely by the disease mainly due to lack of awareness \\nabout the disease and inability to detect early. Thus, early prediction of chronic kidney \\ndisease is believed to be helpful to slow the progress of the disease. Machine learning \\nplays a vital role in early disease identification and prediction. It supports the decision of \\nmedical experts by enabling them to diagnose the disease fast and accurately.\\nTable 6 Multiclass classification results without and with feature selection\\nModels No. of \\nfeatures\\nAcc Prec Rec F1 Sen Spec\\nRF 18 78.3 79.5 77.4 77.3 77.4 94.5\\nRF with RFECV 9 79.0 79.8 78.1 77.9 78.1 94.7\\nRF with UFS 14 77.6 78.9 76.6 76.7 76.7 94.3\\nSVM 18 63.0 60.9 61.5 59.9 61.6 90.6\\nSVM with RFECV 10 62.2 58.7 60.7 56.5 60.8 90.5\\nSVM with UFS 14 61.5 59.2 60.5 58.4 60.5 90.2\\nDT 18 77.5 79.6 76.4 76.2 76.5 94.3\\nDT with RFECV 7 78.0 80 76.9 76.6 76.9 94.4\\nDT with UFS 14 77.9 79.8 76.7 76.4 76.7 94.4\\n0\\n10\\n20\\n30\\n40\\n50\\n60\\n70\\n80\\n90\\n100'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 14, 'page_label': '15', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='DT 18 77.5 79.6 76.4 76.2 76.5 94.3\\nDT with RFECV 7 78.0 80 76.9 76.6 76.9 94.4\\nDT with UFS 14 77.9 79.8 76.7 76.4 76.7 94.4\\n0\\n10\\n20\\n30\\n40\\n50\\n60\\n70\\n80\\n90\\n100\\nRandom Forest SVMD ecision Tree\\nAccuracy Precision Recall F1-Score Sensi/g415vity Speciﬁcity\\nFig. 8 Multiclass classification results after applying RFECV'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 15, 'page_label': '16', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 16 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nIn this study, chronic kidney disease prediction has been carried out using machine \\nlearning techniques. The datasets consists of 19 features with numerical and nominal \\nvalues along with the class to which each instance belongs. The dataset had missing val -\\nues and these missing values were handled at the preprocessing step. After preprocess -\\ning, we have prepared two datasets for two tasks; binary classification and multiclass \\nclassification that has five-classes. The dataset comprises 1718 instances. The binary \\nclass dataset was created by converting the five-class dataset to two class labels; notckd \\nand ckd. All classes except notckd converted to ckd in the binary class dataset. After pre-\\nparing the dataset to binary classification, it has been observed that the class labels were \\nimbalanced and we applied the data resampling technique to balance the dataset.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 15, 'page_label': '16', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='paring the dataset to binary classification, it has been observed that the class labels were \\nimbalanced and we applied the data resampling technique to balance the dataset.\\nThree machine-learning models and two feature selection methods were used. The \\nmodel evaluation was done using tenfold cross-validation and other performance \\nevaluation metrics such as precision, recall, F1-score, sensitivity, and specificity. The \\nconfusion matrix was used to show the correctly and incorrectly classified classes to \\nevaluate the performance of the model. Initially, we applied machine learning classifiers \\nwithout feature selection for both binary class and five classes. The machine learning \\nmodels used in this study are Random Forest, Support Vector Machine, and Decision \\ntree. Then, feature selection techniques were applied along with the models in order to \\nselect predictive. RFECV and UFS were the two feature selection methods applied to'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 15, 'page_label': '16', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='tree. Then, feature selection techniques were applied along with the models in order to \\nselect predictive. RFECV and UFS were the two feature selection methods applied to \\nselect the relevant features. UFS is the filtering method, it works independently of the \\nmachine learning model and RFECV is the wrapper method that works depending on \\nthe machine-learning algorithm. RFECV is also automatic feature selection method that \\ncan select features without specifying the number of features to be selected. The number \\nof selected features is different from model to model as it has been shown in the results. \\nThere were features that has been frequently selected in each feature sets, which indi -\\ncates that they are the most predictive features and have strong predictive relation to the \\nclass. The most frequently selected features using both feature selection methods in all \\nmodels are serum creatinine (Scr), blood urine Nitrogen (Bun), Hemoglobin (Hgb), and'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 15, 'page_label': '16', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='class. The most frequently selected features using both feature selection methods in all \\nmodels are serum creatinine (Scr), blood urine Nitrogen (Bun), Hemoglobin (Hgb), and \\nSpecific Gravity (Sg). Pltc, Rbcc, Wbcc, Mcv, Dm, and Htn are the next most frequently \\nselected features.\\nSeveral related studies have been conducted to predict chronic kidney disease using \\nmachine learning and other techniques globally. However, there were very limited works \\nin the context of Ethiopia. The focuses of global studies were mainly on binary classifica-\\ntion. There were very limited attempt on multiclass classification. Additionally, the size \\nof the datasets used were small which makes them susceptible for overfitting and diffi -\\ncult for comparison.\\nSalekin and Stankovic [9] proposed the method of detecting chronic kidney disease \\nusing K-NN, RF and NN, analysed the characteristics of 24 features, and sorted their'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 15, 'page_label': '16', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='cult for comparison.\\nSalekin and Stankovic [9] proposed the method of detecting chronic kidney disease \\nusing K-NN, RF and NN, analysed the characteristics of 24 features, and sorted their \\npredictability. They have used the dataset of 400 instances comprising 250 ckd and 150 \\nnotckd. They have used five features for final model construction and they have evalu -\\nated the performance of their model using tenfold cross-validation. Compared to the \\nproposed solution in this study, this work used small size dataset and only focus on \\nbinary classification (ckd and notckd). Even though dataset used in the previous work \\nand dataset used in proposed study are not the same, the proposed model in this study \\nhas higher performance compared to the previous work.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 16, 'page_label': '17', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 17 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\nIn the study conducted by Almasoud and Ward [13] Logistic regression, support vec -\\ntor machines, random forest, and gradient boosting algorithms and feature selection \\nmethods such as the ANOVA test, the Pearson’s correlation, and the Cramer’s V test \\nwere implemented to detect chronic kidney disease. A dataset with 400 instances has \\nbeen used. Compared to the proposed model performance of binary class classifica -\\ntion, the performance of the model in this work is inferior. The kidney disease stages \\nprediction model was developed by Rady and Anwar using PNN, MLP , SVM and RBF \\n[15]. They evaluated and compared the performance of the models with accuracy. They \\nreported accuracy accuracy result were 96.7% for PNN, 87% for RBF, 60.7% for SVM \\nand 51.5% for MLP . They used dataset of 361 instances with 25 features including the'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 16, 'page_label': '17', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='reported accuracy accuracy result were 96.7% for PNN, 87% for RBF, 60.7% for SVM \\nand 51.5% for MLP . They used dataset of 361 instances with 25 features including the \\ntarget class. The total data used in this work has been the data of patients with ckd and \\nthen they calculated the eGFR to identify stages of the disease. PNN with an accuracy of \\n96.7% and RBF with an accuracy of 87% developed by Rady and Anwar [15] shows better \\nperformance than the model with best performance in this study which is RF based on \\nRFECV with accuracy of 79% even though models in [15] and two models build in this \\nstudy are different. However, the SVM model built in this study perform better than the \\nSVM model built by Rady and Anwar [15]\\nIn this study, proposed models were evaluated using tenfold cross-validation and other \\nperformance evaluation metrics such as precision, recall, f1_scor, sensitivity, and speci -\\nficity. Four machine-learning models; RF, SVM, DT and Extreme Gradient Boosting'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 16, 'page_label': '17', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='performance evaluation metrics such as precision, recall, f1_scor, sensitivity, and speci -\\nficity. Four machine-learning models; RF, SVM, DT and Extreme Gradient Boosting \\n(XGBoost) have been implemented before applying feature selection. SVM resulted the \\nhighest accuracy of 99.8% for binary. Its accuracy is 78.78% for five classes. RF resulted \\naccuracy of 99.7% for binary class and accuracy of 78.3% for five-class. XGBoost resulted \\naccuracy of 98.96% for binary class. The accuracy of XGBoost is 82.56% for five-class \\nwhich is the highest. DT resulted accuracy of 98.5% for binary class and accuracy of \\n77.5% for five-class. Then, the two feature selection methods were applied to the two \\ndatasets. Both SVM and RF with RFECV resulted the highest accuracy for the binary \\nclass. XGBoost has the highest accuracy of 82.56% for five-class. The result is promising \\nand we believe that it can be deployed to support medical experts to identify the disease'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 16, 'page_label': '17', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='class. XGBoost has the highest accuracy of 82.56% for five-class. The result is promising \\nand we believe that it can be deployed to support medical experts to identify the disease \\nfast and accurate. Thus, SVM, RF with RFECV and XGBoost is recommended in our \\nstudy based on its accuracy, f1score, and other performance evaluation for binary class \\nand five-class classifications.\\nConclusion\\nEarly prediction is very crucial for both experts and patients to prevent and slow down \\nthe progress of chronic kidney disease to kidney failure. In this study three machine-\\nlearning models RF, SV, DT, and two feature selection methods RFECV and UFS were \\nused to build proposed models. The evaluation of models were done using tenfold cross-\\nvalidation. First, the four machine learning algorithms were applied to original datasets \\nwith all 19 features. Applying the models on the original dataset, we have got the high -'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 16, 'page_label': '17', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='validation. First, the four machine learning algorithms were applied to original datasets \\nwith all 19 features. Applying the models on the original dataset, we have got the high -\\nest accuracy with RF, SVM, and XGBoost. The accuracy was 99.8% for the binary class \\nand 82.56% for five-class. DT produced lowest performance compared to RF. RF also \\nproduced the highest f1_score values. SVM and RF with RFECV produced the highest \\naccuracy of 99.8%for binary class. XGBoost has 82.56% accuracy for five-class datasets \\nwhich is the highest. Hencewe believe that multi classification work was very important'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 17, 'page_label': '18', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 18 of 19Debal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\nto know the stages of the disease and suggest needed treatments for the patients in order \\nto save their lives.\\nFuture works\\nThis study used a supervised machine-learning algorithm, feature selection methods to \\nselect the best subset features to develop the models. It is better to see the difference \\nin performance results using unsupervised or deep learning algorithms models. The \\nproposed model supports the experts to give the fast decision, it is better to make it a \\nmobile-based system that enables the experts to follow the status of the patients and \\nhelp the patients to use the system to know their status.\\nAcknowledgements\\nNot applicable.\\nAuthor contributions\\nThe development of the basic research questions, identifying the problems and selecting appropriate machine learning \\nalgorithms, data collection, data analysis, interpretation, and critical review of the paper have been done by DA. The'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 17, 'page_label': '18', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='algorithms, data collection, data analysis, interpretation, and critical review of the paper have been done by DA. The \\nedition of the overall progress of the work was supported and guided by TM. Both authors read and approved the final \\nmanuscript.\\nFunding\\nNot applicable.\\nAvailability of data and materials\\nThe data to conduct this research was collected from the patient history data attends their treatment or died during \\na period 2018 to 2019 from St. Paulo’s Hospital. There are no restrictions on the availability of data and the authors are \\nwilling to provide the code as well.\\nDeclarations\\nEthics approval and consent to participate\\nNot applicable.\\nConsent for publication\\nNot applicable.\\nCompeting interests\\nThe authors declare that they have no competing interests.\\nReceived: 31 January 2022   Accepted: 24 October 2022\\nReferences\\n 1. Radhakrishnan J, Mohan S. KI Reports and World Kidney Day. Kidney Int Reports. 2017;2(2):125–6.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 17, 'page_label': '18', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Received: 31 January 2022   Accepted: 24 October 2022\\nReferences\\n 1. Radhakrishnan J, Mohan S. KI Reports and World Kidney Day. Kidney Int Reports. 2017;2(2):125–6.\\n 2. George C, Mogueo A, Okpechi I, Echouffo-Tcheugui JB, Kengne AP . Chronic kidney disease in low-income to \\nmiddle-income countries: The case f increased screening. BMJ Glob Heal. 2017;2(2):1–10.\\n 3. Ethiopia: kidney disease. https:// www. world lifee xpect ancy. com/ ethio pia- kidney- disea se. Accessed 07 Feb 2020.\\n 4. Stanifer JW, et al. The epidemiology of chronic kidney disease in sub-Saharan Africa: A systematic review and meta-\\nanalysis. Lancet Glob Heal. 2014;2(3):e174–81.\\n 5. AbdElhafeez S, Bolignano D, D’Arrigo G, Dounousi E, Tripepi G, Zoccali C. Prevalence and burden of chronic kidney \\ndisease among the general population and high-risk groups in Africa: A systematic review. BMJ Open. 2018;8:1.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 17, 'page_label': '18', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='disease among the general population and high-risk groups in Africa: A systematic review. BMJ Open. 2018;8:1.\\n 6. Molla MD, et al. Assessment of serum electrolytes and kidney function test for screening of chronic kidney disease \\namong Ethiopian Public Health Institute staff members, Addis Ababa, Ethiopia. BMC Nephrol. 2020;21(1):494.\\n 7. Agrawal A, Agrawal H, Mittal S, Sharma M. Disease Prediction Using Machine Learning. SSRN Electron J. \\n2018;5:6937–8.\\n 8. Charleonnan A, Fufaung T, Niyomwong T, Chokchueypattanakit W, Suwannawach S, Ninchawee N. Predictive \\nanalytics for chronic kidney disease using machine learning techniques. Manag Innov Technol Int Conf MITiCON. \\n2016;80–83:2017.\\n 9. Salekin A, Stankovic J. Detection of Chronic Kidney Disease and Selecting Important Predictive Attributes. In: Proc. - \\n2016 IEEE Int. Conf. Healthc. Informatics, ICHI 2016, pp. 262–270, 2016.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 17, 'page_label': '18', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='9. Salekin A, Stankovic J. Detection of Chronic Kidney Disease and Selecting Important Predictive Attributes. In: Proc. - \\n2016 IEEE Int. Conf. Healthc. Informatics, ICHI 2016, pp. 262–270, 2016.\\n 10. Tekale S, Shingavi P , Wandhekar S, Chatorikar A. Prediction of chronic kidney disease using machine learning algo-\\nrithm. Disease. 2018;7(10):92–6.\\n 11. Xiao J, et al. Comparison and development of machine learning tools in the prediction of chronic kidney disease \\nprogression. J Transl Med. 2019;17(1):1–13.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 18, 'page_label': '19', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='Page 19 of 19\\nDebal\\xa0and Sitote  Journal of Big Data           (2022) 9:109 \\n \\n 12. Priyanka K, Science BC. Chronic kidney disease prediction based on naive Bayes technique. 2019. p. 1653–9.\\n 13. Almasoud M, Ward TE. Detection of chronic kidney disease using machine learning algorithms with least number of \\npredictors. Int J Adv Computer. 2019;10(8):89–96.\\n 14. Yashfi SY. Risk Prediction Of Chronic Kidney Disease Using Machine Learning Algorithms. 2020.\\n 15. Rady EA, Anwar AS. Informatics in Medicine Unlocked Prediction of kidney disease stages using data mining algo-\\nrithms. Informatics Med. 2019;15(2018):100178.\\n 16. Alsuhibany SA, et al. Ensemble of deep learning based clinical decision support system for chronic kidney disease \\ndiagnosis in medical internet of things environment. Comput Intell Neurosci. 2021;3:2021.\\n 17. Poonia RC, et al. Intelligent Diagnostic Prediction and Classification Models for Detection of Kidney Disease. Health-\\ncare. 2022;10:2.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 18, 'page_label': '19', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='17. Poonia RC, et al. Intelligent Diagnostic Prediction and Classification Models for Detection of Kidney Disease. Health-\\ncare. 2022;10:2.\\n 18. Kumar V. Evaluation of computationally intelligent techniques for breast cancer diagnosis. Neural Comput Appl. \\n2021;33(8):3195–208.\\n 19. Jasim A, Kaky M. Iintelligent systems approach for classification and management of by. 2017.\\n 20. Saar-tsechansky M, ProvostF. Handling Missing Values when Applying Classi … cation Models. vol. 1, 2007.\\n 21. Data Preparation for Statistical Modeling and Machine Learning. https:// www. featu reran king. com/ tutor ials/ machi \\nne- learn ing- tutor ials/ data- prepa ration- for- machi ne- learn ing/. Accessed 12 Oct 2020.\\n 22. Oliver T. Machine Learning For Absolute Beginners. 2017.\\n 23. ZarPhyu T, Oo NN. Performance comparison of feature selection methods. MATEC Web Conf. 2016;42:2–5.\\n 24. Koshy S. Feature selection for improving multi-label classification using MEKA. Res J. 2017;12(24):14774–82.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 18, 'page_label': '19', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='24. Koshy S. Feature selection for improving multi-label classification using MEKA. Res J. 2017;12(24):14774–82.\\n 25. Vidhya A. Introduction to Feature Selection methods with an example (or how to select the right variables?). https:// \\nwww. analy ticsv idhya. com/ blog/ 2016/ 12/ intro ducti on- to- featu re- selec tion- metho ds- with- an- examp le- or- how- to- \\nselect- the- right- varia bles/. Accessed 24 Mar 2020.\\n 26. Misra P , Yadav AS. Improving the classification accuracy using recursive feature elimination with cross-validation. Int \\nJ Emerg Technol. 2020;11(3):659–65.\\n 27. Aqlan F, Markle R, Shamsan A. Data mining for chronic kidney disease prediction. 67th Annu Conf Expo Inst Ind Eng. \\n2017;2017:1789–94.\\n 28. Subas A, Alickovic E, Kevric J. Diagnosis of chronic kidney disease by using random forest. IFMBE Proc. \\n2017;62(1):589–94.\\n 29. Kapoor S, Verma R, Panda SN. Detecting kidney disease using Naïve bayes and decision tree in machine learning. Int'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 18, 'page_label': '19', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='2017;62(1):589–94.\\n 29. Kapoor S, Verma R, Panda SN. Detecting kidney disease using Naïve bayes and decision tree in machine learning. Int \\nJ Innov Technol Explor Eng. 2019;9(1):498–501.\\n 30. Vijayarani S, Dhayanand S. Data Mining Classification Algorithms for Kidney Disease Prediction. Int J Cybern Infor-\\nmatics. 2015;4(4):13–25.\\n 31. Drall S, Drall GS, Singh S. Chronic kidney disease prediction using machine learning : a new approach bharat \\nBhushan Naib. Learn. 2014;8(278):278–87.\\n 32. KadamVinay R, Soujanya KLS, Singh P . Disease prediction by using deep learning based on patient treatment history. \\nInt J Recent Technol Eng. 2019;7(6):745–54.\\n 33. Ramya S, Radha N. Diagnosis of Chronic Kidney Disease Using. pp. 812–820, 2016.\\n 34. Osisanwo FY, Akinsola JET, Awodele O, Hinmikaiye JO, Olakanmi O, Akinjobi J. Supervised machine learning algo-\\nrithms: classification and comparison. Int J Comput Trends Technol. 2017;48(3):128–38.'),\n",
              " Document(metadata={'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)', 'creator': 'Springer', 'creationdate': '2022-11-12T22:22:16+05:30', 'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)', 'crossmarkdomains[1]': 'springer.com', 'moddate': '2022-11-20T22:44:42+01:00', 'crossmarkmajorversiondate': '2010-04-23', 'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5', 'author': 'Dibaba Adeba Debal', 'title': 'Chronic kidney disease prediction using machine learning techniques', 'crossmarkdomainexclusive': 'true', 'robots': 'noindex', 'doi': '10.1186/s40537-022-00657-5', 'crossmarkdomains[2]': 'springerlink.com', 'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf', 'total_pages': 19, 'page': 18, 'page_label': '19', 'source_file': 's40537-022-00657-5.pdf', 'file_type': 'pdf'}, page_content='rithms: classification and comparison. Int J Comput Trends Technol. 2017;48(3):128–38.\\n 35. Acharya A. Comparative Study of Machine Learning Algorithms for Heart Disease Prediction 2017.\\n 36. Amirgaliyev Y. Analysis of chronic kidney disease dataset by applying machine learning methods. In: 2018 IEEE 12th \\nInternational Conference Application Information Communication Technology, pp. 1–4, 2010.\\nPublisher’s Note\\nSpringer Nature remains neutral with regard to jurisdictional claims in published maps and institutional affiliations.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 0, 'page_label': '1', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='BMJ Open Diab Res Care 2022;10:e002560. doi:10.1136/bmjdrc-2021-002560\\nOpen access \\n1\\nOpen access \\nPrediction of diabetic kidney disease \\nwith machine learning algorithms, upon \\nthe initial diagnosis of type 2 \\ndiabetes mellitus\\nAngier Allen    , Zohora Iqbal    , Abigail Green- Saxena    , Myrna Hurtado    , \\nJana Hoffman    , Qingqing Mao    , Ritankar Das    \\nResearch and Development, \\nDascena, Houston, Texas, USA\\nCorrespondence to\\nDr Myrna Hurtado;  \\n lhurtado@ dascena. com\\nTo cite: Allen\\xa0A, Iqbal\\xa0Z, \\nGreen- Saxena\\xa0A, et\\xa0al. \\nPrediction of diabetic kidney \\ndisease with machine learning \\nalgorithms, upon the initial \\ndiagnosis of type 2 diabetes \\nmellitus. BMJ Open Diab \\nRes Care 2022;10:e002560. \\ndoi:10.1136/\\nbmjdrc-2021-002560\\n ► Additional supplemental \\nmaterial is published online \\nonly. To view, please visit the \\njournal online (http:// dx. doi. \\norg/ 10. 1136/ bmjdrc- 2021- \\n002560).\\nAA and ZI contributed equally.\\nReceived 24 August 2021\\nAccepted 27 December 2021'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 0, 'page_label': '1', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='only. To view, please visit the \\njournal online (http:// dx. doi. \\norg/ 10. 1136/ bmjdrc- 2021- \\n002560).\\nAA and ZI contributed equally.\\nReceived 24 August 2021\\nAccepted 27 December 2021\\nOriginal research\\nEmerging technologies, pharmacology and therapeutics\\n© Author(s) (or their \\nemployer(s)) 2022. Re- use \\npermitted under CC BY- NC. No \\ncommercial re- use. See rights \\nand permissions. Published \\nby BMJ.\\nABSTRACT\\nIntroduction Diabetic kidney disease (DKD) accounts for \\nthe majority of increased risk of mortality for patients with \\ndiabetes, and eventually manifests in approximately half \\nof those patients diagnosed with type 2 diabetes mellitus \\n(T2DM). Although increased screening frequency can avoid \\ndelayed diagnoses, this is not uniformly implemented. The \\npurpose of this study was to develop and retrospectively \\nvalidate a machine learning algorithm (MLA) that predicts \\nstages of DKD within 5 years upon diagnosis of T2DM.\\nResearch design and methods Two MLAs were trained'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 0, 'page_label': '1', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='validate a machine learning algorithm (MLA) that predicts \\nstages of DKD within 5 years upon diagnosis of T2DM.\\nResearch design and methods Two MLAs were trained \\nto predict stages of DKD severity, and compared with the \\nCenters for Disease Control and Prevention (CDC) risk \\nscore to evaluate performance. The models were validated \\non a hold- out test set as well as an external dataset \\nsourced from separate facilities.\\nResults The MLAs outperformed the CDC risk score \\nin both the hold- out test and external datasets. Our \\nalgorithms achieved an area under the receiver operating \\ncharacteristic curve (AUROC) of 0.75 on the hold- out set \\nfor prediction of any- stage DKD and an AUROC of over 0.82 \\nfor more severe endpoints, compared with the CDC risk \\nscore with an AUROC <0.70 on all test sets and endpoints.\\nConclusion This retrospective study shows that an MLA \\ncan provide timely predictions of DKD among patients with \\nrecently diagnosed T2DM.\\nINTRODUCTION'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 0, 'page_label': '1', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='Conclusion This retrospective study shows that an MLA \\ncan provide timely predictions of DKD among patients with \\nrecently diagnosed T2DM.\\nINTRODUCTION\\nChronic kidney disease (CKD) is a general \\nterm for describing any disorders that lead \\nto the gradual loss of kidney function or \\nstructure.1 CKD is defined by impaired renal \\nfunction and/or increased urinary albumin \\nexcretion and strongly associated with excess \\nmorbidity and cardiovascular as well as all- \\ncause mortality,2–5 and is a common complica-\\ntion for patients with type 2 diabetes mellitus \\n(T2DM).3 CKD due to diabetes is also \\nreferred to as diabetic kidney disease (DKD), \\nor diabetic nephropathy,3 6 and accounts for \\nthe majority of increased risk of mortality \\nfor patients with diabetes. 2 T2DM results \\nin long- term hyperglycemia and hyperten-\\nsion, which are the main drivers behind \\npathophysiological and metabolic glomer -\\nular changes, and subsequent renal deterio-\\nration in DKD. 7 Several studies have shown'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 0, 'page_label': '1', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='sion, which are the main drivers behind \\npathophysiological and metabolic glomer -\\nular changes, and subsequent renal deterio-\\nration in DKD. 7 Several studies have shown \\nthat mortality risk increases significantly in \\npatients with glomerular filtration rate (GFR) \\nlevels consistent with CKD stages 3–5. 8 9  In \\n1990–2012, global mortality resulting from \\nDKD increased by over 90%. 10 11  With \\nSignificance of this study\\nWhat is already known about this subject?\\n ► Type 2 diabetes mellitus (T2DM) is a risk factor for im-\\npaired renal function due to long- term hyperglycemia \\nand hypertension affecting kidney function, resulting in \\ndiabetic kidney disease (DKD). DKD cases have steadily \\nincreased over the last three decades and are expected \\nto continue rising worldwide.\\n ► Most individuals with early stages of DKD either exhibit \\nnon- specific symptoms or are asymptomatic, contrib-\\nuting to missed diagnoses. There is a lack of accurate'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 0, 'page_label': '1', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='to continue rising worldwide.\\n ► Most individuals with early stages of DKD either exhibit \\nnon- specific symptoms or are asymptomatic, contrib-\\nuting to missed diagnoses. There is a lack of accurate \\nearly risk prediction of DKD development in patients at \\nthe time of T2DM diagnosis.\\nWhat are the new findings?\\n ► We developed machine learning algorithms (MLAs) to \\npredict risk within a 5- year time frame for DKD develop-\\nment at the time of T2DM diagnosis, using 1 year of prior \\nelectronic health record data.\\n ► The MLAs had improved performance compared with \\nthe Centers for Disease Control and Prevention (CDC) \\nrisk score.\\nHow might these results change the focus of \\nresearch or clinical practice?\\n ► Use of these MLAs in medical practice may help sup-\\nport clinicians in their decision- making. Early DKD risk \\nprediction can facilitate intervention and improve patient \\noutcomes for DKD.\\n ► Data used for MLAs may be automatically extracted'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 0, 'page_label': '1', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='port clinicians in their decision- making. Early DKD risk \\nprediction can facilitate intervention and improve patient \\noutcomes for DKD.\\n ► Data used for MLAs may be automatically extracted \\nfrom electronic health records. This enables broad \\nscreening and may increase identification of patients at \\nrisk of DKD, and removes the burden of manually calcu-\\nlating the risk assessment of DKD with current standard \\nmodels, such as the CDC risk score.\\nBMJ Open Diabetes Research & Care: first published as 10.1136/bmjdrc-2021-002560 on 19 January 2022. Downloaded from https://drc.bmj.com on 27 September 2025 by guest.\\nProtected by copyright, including for uses related to text and data mining, AI training, and similar technologies.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 1, 'page_label': '2', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='2\\nBMJ Open Diab Res Care 2022;10:e002560. doi:10.1136/bmjdrc-2021-002560\\nEmerging technologies, pharmacology and therapeutics\\napproximately half of patients with T2DM developing \\nkidney disease, 3 the global rise in T2DM 12 13  imposes a \\nsignificant cost to patients as well as healthcare systems.\\nAlthough early detection of DKD may prevent its \\nprogression,14 15  routine screening is not universally \\nfeasible; this can lead to missed or delayed diagnoses. \\nDKD diagnosis is based on measurement of renal func-\\ntion and albumin levels in urine along with assessment \\nby a clinician. DKD is defined by: estimated GFR (eGFR) \\n<60 mL/min/1.73 m2 and albuminuria/creatinine ratio \\n>300 mg/g.16 Diabetic retinopathy may also be concur -\\nrent; more than 25% of patients develop retinopathy \\nwithin 2 years of T2DM diagnosis. 17 Despite that these \\nmeasurements are basic clinical and laboratory measure-\\nments, screening for DKD is not uniformly implemented.4'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 1, 'page_label': '2', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='within 2 years of T2DM diagnosis. 17 Despite that these \\nmeasurements are basic clinical and laboratory measure-\\nments, screening for DKD is not uniformly implemented.4 \\nBecause individuals with T2DM have an increased suscep-\\ntibility to development of DKD, it is critical for clinicians \\nto rapidly identify those who are at high risk. Prompt and \\naccurate risk stratification may warrant thorough exam-\\nination and increased screening frequency in high- risk \\npatients for earlier DKD identification.\\nEarly DKD prediction could lead to therapeutic inter -\\nventions and lifestyle changes, prevention of progression \\nto higher stages, and reduction of dialysis dependency \\nas well as costly healthcare spending. 18 Risk scores 19 20  \\nand machine learning (ML) 21–23 approaches have been \\nvalidated for CKD progression, including the Centers for \\nDisease Control and Prevention (CDC) CKD risk score, \\nwhich is based on demographic information and pre-'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 1, 'page_label': '2', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='validated for CKD progression, including the Centers for \\nDisease Control and Prevention (CDC) CKD risk score, \\nwhich is based on demographic information and pre- \\nexisting conditions.18 However, there remains a need for \\nkidney disease prediction for patients newly diagnosed \\nwith T2DM who are at high risk of DKD development. \\nThis is critical as patients who are unaware of their high \\nrisk may be less likely to undergo routine screening, \\nincreasing their odds of missed or delayed diagnosis. We \\nhave developed ML algorithms (MLAs) for patients at \\nthe time of T2DM diagnosis to predict development of \\nDKD within a 5- year time frame.\\nRESEARCH DESIGN AND METHODS\\nData source and data processing\\nRetrospective analysis was performed on patient elec-\\ntronic health records (EHRs) data extracted from a large, \\nproprietary database representing over 700 healthcare \\nsites across the USA between 2007 and 2020. All patient \\ndata were de- identified in compliance with the Health'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 1, 'page_label': '2', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='proprietary database representing over 700 healthcare \\nsites across the USA between 2007 and 2020. All patient \\ndata were de- identified in compliance with the Health \\nInsurance Portability and Accountability Act. The dataset \\nwas split into training, training validation, and hold- out \\ntesting sets (see figure 1).\\nAlgorithm models were tuned with hyperparameter \\noptimization (HPO), fitting each hyperparameter combi-\\nnation on the training set and evaluating its performance \\non the training validation set. The hyperparameter \\ncombination which yielded the highest average precision \\nwas then used to train the final model on both training \\nand training validation sets, as described in the Machine \\nlearning model section below. We report performance \\nof the model on the hold- out test data (not used during \\nthe model development process) and the external vali-\\ndation data. The external validation data come from \\nhealthcare sites and patients separate from those used for'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 1, 'page_label': '2', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='the model development process) and the external vali-\\ndation data. The external validation data come from \\nhealthcare sites and patients separate from those used for \\nmodel selection and training. Each model estimates the \\nrisk of developing DKD in the 5 years following T2DM \\ndiagnosis. Tree- based models use decision trees to build \\nmore complex ensembles, which can allow for a desir -\\nable balance of speed, complexity, and interpretability. \\nTwo variations of this model type were fitted to the data \\nto assess different tree- based techniques, random forests \\n(RF) and gradient boosted trees (XGB). RF fit many deci-\\nsion trees to the data, which combine their predictions \\ndemocratically. XGB sequentially fit trees that improve \\non previous errors to generate their predictions.\\nGold standard\\nAll patients with T2DM were identified using the Inter -\\nnational Classification of Diseases (ICD- 9 and ICD- 10) \\ncodes. Within this population, patients with at least 5'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 1, 'page_label': '2', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='Gold standard\\nAll patients with T2DM were identified using the Inter -\\nnational Classification of Diseases (ICD- 9 and ICD- 10) \\ncodes. Within this population, patients with at least 5 \\nyears of medical data post- T2DM diagnosis, age over 18 \\nyears old, and with at least one of each required measure-\\nments in the year prior to T2DM diagnosis were included \\nin the study (see table 1). We included patients with albu-\\nminuria or reduced eGFR at the start of the study. The \\npositive class, patients who developed DKD within the 5 \\nyears after T2DM diagnosis, were defined by ICD codes \\nFigure 1 Patient inclusion diagram. Hold- out test set and \\nexternal validation set both consist of patients who are not \\nseen during training and validation of the MLAs. The external \\nvalidation set consists only of patients from clinical sites that \\nare not used in training, validation and hold- out test sets. \\nMLAs, machine learning algorithms; T2DM, type 2 diabetes \\nmellitus.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 1, 'page_label': '2', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='validation set consists only of patients from clinical sites that \\nare not used in training, validation and hold- out test sets. \\nMLAs, machine learning algorithms; T2DM, type 2 diabetes \\nmellitus.\\nBMJ Open Diabetes Research & Care: first published as 10.1136/bmjdrc-2021-002560 on 19 January 2022. Downloaded from https://drc.bmj.com on 27 September 2025 by guest.\\nProtected by copyright, including for uses related to text and data mining, AI training, and similar technologies.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 2, 'page_label': '3', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='3\\nBMJ Open Diab Res Care 2022;10:e002560. doi:10.1136/bmjdrc-2021-002560\\nEmerging technologies, pharmacology and therapeutics\\nas reported in online supplemental table 1. Patients with \\nT2DM who did not have an associated ICD code for \\nDKD within the 5- year window were in the negative class. \\nPatients were excluded if they had been diagnosed with \\nCKD or had a renal transplant before T2DM diagnosis \\ntime.\\nIn addition to an any- stage DKD endpoint, we evaluated \\nmodel performance on endpoints defined as reaching \\nDKD stages 3–5 as well as reaching DKD stages 4–5 within \\n5 years following T2DM diagnosis. The endpoint for the \\nCDC CKD risk score is stages 3–5. Patients reaching stages \\n4–5 require close monitoring of kidney function as well \\nas assessment for potential kidney transplants or dialysis.\\nInput selection\\nTo generate the inputs, we first conducted a compre-\\nhensive search through previous literature for CKD \\nrisk factors. This list included age, sex, diabetes, hyper -'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 2, 'page_label': '3', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='Input selection\\nTo generate the inputs, we first conducted a compre-\\nhensive search through previous literature for CKD \\nrisk factors. This list included age, sex, diabetes, hyper -\\ntension, cardiovascular diseases, smoking, obesity, age, \\nalcohol use, cholesterol levels, white cell counts, genetic \\ndisposition, etc.24–27 We then narrowed the list of features \\ndown by what is available in the EHR. For example, \\ngenetic information and socioeconomic status, though \\nthey affect the risk of CKD, are not typically found in the \\nEHR. Finally, we trimmed the model of features that did \\nnot significantly affect the model performance, that is, \\nmalignancy, HIV infection and triglyceride levels.\\nML model\\nTwo MLAs were developed and evaluated: an XGB \\n(XGBoost)28 model and an RF model. The RF model was \\ndeveloped using the Python library Scikit- learn.29 Input \\nfeatures for both models consisted of demographics, clin-\\nical measurements, laboratory values, and patient history'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 2, 'page_label': '3', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='developed using the Python library Scikit- learn.29 Input \\nfeatures for both models consisted of demographics, clin-\\nical measurements, laboratory values, and patient history \\nas reported in table 1. Demographics, clinical measure-\\nments, and laboratory values were averaged over the year \\nprior to T2DM diagnosis as described below. The eGFR was \\nprecalculated in the dataset using the following equation: \\n eGFR = 186× S−1.154\\nCr × age−0.203 ×\\n(\\n1.212 if black\\n)\\n×\\n(\\n0.742 if female\\n)\\n , where S Cr is serum creatinine in mg/dL. 30 The devel-\\noped models were compared with the CDC CKD risk \\nscore based on pre- existing conditions and demographic \\ninformation.31 HPO was performed using the Python \\nlibrary Hyperopt 32 for all models except the CDC CKD \\nrisk score, which does not require training.\\nThe non- external data were split into training, training \\nvalidation, and test sets with a 50:25:25 split. HPO was \\nperformed by fitting the model on the any- stage training'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 2, 'page_label': '3', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='The non- external data were split into training, training \\nvalidation, and test sets with a 50:25:25 split. HPO was \\nperformed by fitting the model on the any- stage training \\ndata, then testing on the any- stage training validation \\ndata. The combination of hyperparameters which yielded \\nthe highest area under the precision- recall curve on the \\nany- stage training validation data was then used to test on \\nthe hold- out testing data and the external validation data. \\nThe other endpoints of stages 3–5 and stages 4–5 kidney \\ndisease were also tested on the hold- out and external vali-\\ndation dataset, but were not used during model training. \\nHyperparameters for each model can be found in the \\nonline supplemental table 2.\\nInput features for the models were averaged over the \\n1- year input time window using combinations of feature \\nmedian, 5th and 95th percentiles, and last available \\nmeasurement when applicable. In the RF model, features'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 2, 'page_label': '3', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='1- year input time window using combinations of feature \\nmedian, 5th and 95th percentiles, and last available \\nmeasurement when applicable. In the RF model, features \\nwere standardized to have mean 0 and variance 1 using \\nTable 1 Measurements used as inputs for machine learning algorithms (MLAs) and for calculating CDC risk score.\\nMeasurements used as inputs to MLA Measurements used as inputs to CDC risk score\\nDemographics Age Age\\nSex Sex\\nClinical measurements BMI None\\nBlood pressure (systolic and diastolic)   \\nLaboratory values Blood urea nitrogen None\\nCreatinine   \\neGFR   \\nCholesterol (HDL and LDL)   \\nWhite cell count   \\nMedical history Presence of past acute kidney injury Cardiovascular disease\\nHistory of chronic heart failure Congestive heart failure\\nReported smoking history Peripheral vascular disease\\nReported alcohol history Proteinuria\\nDemographics (age, sex), clinical measurements (BMI, blood pressure (systolic and diastolic)), laboratory values (blood urea nitrogen,'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 2, 'page_label': '3', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='Reported alcohol history Proteinuria\\nDemographics (age, sex), clinical measurements (BMI, blood pressure (systolic and diastolic)), laboratory values (blood urea nitrogen, \\ncreatinine and eGFR, cholesterol (high- density lipoprotein and low- density lipoprotein), white cell count), and medical history (presence of \\npast acute kidney injury, history of chronic heart failure, reported smoking history, reported alcohol history) served as input features for the \\nMLA models. The clinical and laboratory measurement values were pooled using 5th and 95th percentiles, median, and last available result \\nover 1 year prior to T2DM diagnosis.\\nBMI, body mass index; CDC, Centers for Disease Control and Prevention; eGFR, estimated glomerular filtration rate; HDL, high- density \\nlipoprotein; LDL, low- density lipoprotein; T2DM, type 2 diabetes mellitus.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 2, 'page_label': '3', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='lipoprotein; LDL, low- density lipoprotein; T2DM, type 2 diabetes mellitus.\\nBMJ Open Diabetes Research & Care: first published as 10.1136/bmjdrc-2021-002560 on 19 January 2022. Downloaded from https://drc.bmj.com on 27 September 2025 by guest.\\nProtected by copyright, including for uses related to text and data mining, AI training, and similar technologies.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 3, 'page_label': '4', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='4\\nBMJ Open Diab Res Care 2022;10:e002560. doi:10.1136/bmjdrc-2021-002560\\nEmerging technologies, pharmacology and therapeutics\\nstatistics from the training data, and missing features were \\nimputed with the training data averages. The XGB model \\nassesses missing values as inputs and does not require \\nfeature standardization. The option to standardize and \\nimpute features was thus given as an option to be selected \\nin HPO for XGB, but was not required (however, the final \\nmodel did select standardization and imputation during \\nhyperparameter optimization). The CDC CKD model \\nrequired no imputation, as inputs are based on demo-\\ngraphic and diagnostic information which were available \\nfor all patients.\\nFor each endpoint, model performance was evalu-\\nated on a hold- out testing set not seen during the model \\ntraining process. An additional test set from a unique \\nsource was also used for external validation of the models \\nand endpoints. The models were assessed based on'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 3, 'page_label': '4', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='training process. An additional test set from a unique \\nsource was also used for external validation of the models \\nand endpoints. The models were assessed based on \\narea under the receiver operating characteristic curve \\n(AUROC), sensitivity, specificity, positive and negative \\nlikelihood ratios, and diagnostic odds ratio (DOR).\\nRESULTS\\nA total of 6 918 247 patients with T2DM were available \\nin our dataset. Patients were filtered based on availability \\nof 5 years of data following T2DM diagnosis, resulting in \\n2 248 457 patients. The dataset was further filtered for \\npatients who have age and required input laboratory \\nmeasurements and clinical data (eg, body mass index \\nand creatinine) available within the prior year, resulting \\nin 111 046 patients. From this patient population, 23 073 \\npatients from clinical sites not used in the training and \\ntesting of the MLAs were isolated and used as an external \\nvalidation hold- out test set. A total of 87 973 patients'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 3, 'page_label': '4', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='patients from clinical sites not used in the training and \\ntesting of the MLAs were isolated and used as an external \\nvalidation hold- out test set. A total of 87 973 patients \\nwere randomly split into training (62 994), validation (17 \\n323), and test sets (7 656), where the test set consists of \\npatients not seen by the algorithm during training and \\nvalidation (figure 1).\\nUrinary albumin is typically used for diagnosing DKD. \\nHowever, these measurements are not always available and \\nmay limit screening generalizability. Thus, our model did \\nnot use urinary albumin to make DKD predictions. eGFR \\nwas an included input feature. Before inclusion criteria \\nwere applied to our dataset, 30.96% of the patients were \\nmissing urinary albumin measurements and 11.13% of \\npatients were missing eGFR measurements.\\nDemographics of both the hold- out test set and \\nexternal validation set at the time of T2DM diagnosis are \\npresented in online supplemental tables 3 and 4, respec-'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 3, 'page_label': '4', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='Demographics of both the hold- out test set and \\nexternal validation set at the time of T2DM diagnosis are \\npresented in online supplemental tables 3 and 4, respec-\\ntively. Most patients in the positive class exhibiting DKD \\nare aged 50 years and above. Most common comorbid-\\nities included hypertension, cardiovascular disease, and \\ndyslipidemia in both the positive and negative class.\\nPerformance of the MLA models (RF, XGB) for DKD \\nstages 3–5 was compared with the CDC CKD scoring \\nsystem. The AUROC curves are presented in figure 2, \\nfor (a) the hold- out test dataset and (b) external vali-\\ndation dataset, demonstrating that both MLA models \\noutperformed the CDC CKD comparator in terms of \\nthe model’s ability to discriminate between classes. Both \\nmodels also outperformed the CDC CKD comparator \\nin terms of sensitivity and specificity on both test sets. \\nAUROC curves for the MLA models (RF and XGB) for \\nany- stage or stages 4–5 DKD are compared with the CDC'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 3, 'page_label': '4', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='in terms of sensitivity and specificity on both test sets. \\nAUROC curves for the MLA models (RF and XGB) for \\nany- stage or stages 4–5 DKD are compared with the CDC \\nscoring system and shown in online supplemental figures \\n1 and 2, respectively. For both of the other endpoints, the \\nMLAs also outperformed the CDC risk score in terms of \\nAUROC as well as sensitivity and specificity.\\nFigure 2 Area under the receiver operating characteristic curve (AUROC) plots of machine learning models random forest (RF) \\nand gradient boosted tree (XGB), and Centers for Disease Control and Prevention (CDC) CKD scoring system for (A) hold- out \\ndataset and (B) external validation dataset for prediction of DKD stages 3–5 in the 5 years following T2DM diagnosis. A random \\nclassifier was used as the baseline. CKD, chronic kidney disease; DKD, diabetic kidney disease; T2DM, type 2 diabetes \\nmellitus.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 3, 'page_label': '4', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='classifier was used as the baseline. CKD, chronic kidney disease; DKD, diabetic kidney disease; T2DM, type 2 diabetes \\nmellitus.\\nBMJ Open Diabetes Research & Care: first published as 10.1136/bmjdrc-2021-002560 on 19 January 2022. Downloaded from https://drc.bmj.com on 27 September 2025 by guest.\\nProtected by copyright, including for uses related to text and data mining, AI training, and similar technologies.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 4, 'page_label': '5', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='5\\nBMJ Open Diab Res Care 2022;10:e002560. doi:10.1136/bmjdrc-2021-002560\\nEmerging technologies, pharmacology and therapeutics\\nTables 2 and 3 summarize the performance for RF, \\nXGB and the CDC CKD score for the hold- out test set \\nand external validation set, respectively. XGB and RF \\nachieved similar results in terms of discrimination and \\nclassification performance, with the RF performing more \\nconsistently across the two test sets.\\nDISCUSSION\\nWe have developed and evaluated ML DKD screening tools \\nusing data easily accessible in the EHR, which provide a \\nrobust method of predicting DKD within a 5- year window \\nfor patients at the time of their T2DM diagnosis. Our MLA \\nmodels, which use only demographics, clinical measure-\\nments, laboratory measurements, and patient history \\ndrawn from the EHR outperform the CDC CKD scoring \\nsystem. Urinary albumin is commonly used for kidney \\ndisease diagnosis, however it is not routinely collected'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 4, 'page_label': '5', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='drawn from the EHR outperform the CDC CKD scoring \\nsystem. Urinary albumin is commonly used for kidney \\ndisease diagnosis, however it is not routinely collected \\ndata for all patients. Therefore, to enable screening for \\nDKD on a broad patient population, it was not included \\nas an input. Data for the MLAs can be automatically \\nextracted from the EHR, removing the burden of manu-\\nally calculating CKD risk assessment with the CDC CKD \\nscoring system. These algorithms may provide warning of \\nDKD to physicians for improved patient care by deter -\\nmining who is at high risk and allow for earlier detection \\nand intervention. Routine screening for CKD is essential \\nfor those at high risk, particularly in patients with T2DM, \\nwho have a higher propensity to develop DKD. However, \\nstandard detection of early DKD in patients with T2DM \\nis poor,33 resulting in inadequate management of disease \\nstate and higher healthcare costs. Early warning systems'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 4, 'page_label': '5', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='standard detection of early DKD in patients with T2DM \\nis poor,33 resulting in inadequate management of disease \\nstate and higher healthcare costs. Early warning systems \\naugment clinical expertise to enable clinicians to make \\nimproved treatment and intervention decisions. Predic-\\ntion and early diagnostic methods of DKD offer a lifetime \\nof benefits including prevention of stage progression and \\ndevelopment of associated comorbidities, deterrence \\nof dialysis dependency, and an overall extension of life \\nexpectancy, as well as a reduction in spending on health-\\ncare resources. 18 Additionally, early intervention could \\nsignificantly improve patient quality of life as patients \\nTable 2 Results on hold- out test set\\nXGB RF CDC\\nAny- stage DKD\\n  AUROC 0.750 0.748 0.634\\n  Sensitivity 0.700 0.700 0.633\\n  Specificity 0.670 0.662 0.560\\n  LR+ 2.120 2.071 1.440\\n  LR− 0.447 0.453 0.655\\n  DOR 4.738 4.575 2.197\\nDKD stages 3–5\\n  AUROC 0.825 0.823 0.672\\n  Sensitivity 0.750 0.750 0.637'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 4, 'page_label': '5', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='Specificity 0.670 0.662 0.560\\n  LR+ 2.120 2.071 1.440\\n  LR− 0.447 0.453 0.655\\n  DOR 4.738 4.575 2.197\\nDKD stages 3–5\\n  AUROC 0.825 0.823 0.672\\n  Sensitivity 0.750 0.750 0.637\\n  Specificity 0.742 0.739 0.614\\n  LR+ 2.906 2.870 1.652\\n  LR− 0.336 0.338 0.591\\n  DOR 8.638 8.492 2.794\\nDKD stages 4–5\\n  AUROC 0.830 0.821 0.617\\n  Sensitivity 0.751 0.751 0.581\\n  Specificity 0.739 0.712 0.581\\n  LR+ 2.876 2.606 1.387\\n  LR− 0.337 0.349 0.721\\n  DOR 8.544 7.461 1.923\\nComparison of XGB, RF and the CDC DKD performance includes \\nAUROC, sensitivity, specificity, LR+ and LR‒, and DOR. Prediction \\nof DKD within 5 years following T2DM diagnosis is divided into \\nany- stage DKD, DKD stages 3–5 and DKD stages 4–5.\\nAUROC, area under the receiver operating characteristic; CDC, \\nCenters for Disease Control and Prevention; DKD, diabetic kidney \\ndisease; DOR, diagnostic OR; LR+, positive likelihood ratio; LR−, \\nnegative likelihood ratio; RF , random forest; T2DM, type 2 diabetes'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 4, 'page_label': '5', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='disease; DOR, diagnostic OR; LR+, positive likelihood ratio; LR−, \\nnegative likelihood ratio; RF , random forest; T2DM, type 2 diabetes \\nmellitus; XGB, gradient boosted tree.\\nTable 3 Results on external validation set\\nXGB RF CDC\\nAny- stage DKD\\n  AUROC 0.769 0.769 0.643\\n  Sensitivity 0.761 0.758 0.651\\n  Specificity 0.622 0.619 0.573\\n  LR+ 2.015 1.989 1.522\\n  LR− 0.384 0.391 0.610\\n  DOR 5.251 5.089 2.496\\nDKD stages 3–5\\n  AUROC 0.831 0.832 0.676\\n  Sensitivity 0.807 0.804 0.640\\n  Specificity 0.690 0.692 0.623\\n  LR+ 2.605 2.608 1.697\\n  LR− 0.279 0.283 0.578\\n  DOR 9.322 9.215 2.937\\nDKD stages 4–5\\n  AUROC 0.826 0.827 0.620\\n  Sensitivity 0.819 0.826 0.608\\n  Specificity 0.664 0.643 0.576\\n  LR+ 2.438 2.313 1.436\\n  LR− 0.273 0.270 0.680\\n  DOR 8.933 8.555 2.111\\nComparison of XGB, RF and the CDC DKD performance includes \\nAUROC, sensitivity, specificity, LR+ and LR‒, DOR, and threshold. \\nPrediction of DKD within 5 years following T2DM diagnosis is'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 4, 'page_label': '5', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='Comparison of XGB, RF and the CDC DKD performance includes \\nAUROC, sensitivity, specificity, LR+ and LR‒, DOR, and threshold. \\nPrediction of DKD within 5 years following T2DM diagnosis is \\ndivided into any- stage DKD, DKD stages 3–5 and DKD stages 4–5.\\nAUROC, area under the receiver operating characteristic; CDC, \\nCenters for Disease Control and Prevention; DKD, diabetic kidney \\ndisease; DOR, diagnostic OR; LR−, negative likelihood ratio; LR+, \\npositive likelihood ratio; RF , random forest; T2DM, type 2 diabetes \\nmellitus; XGB, gradient boosted tree.\\nBMJ Open Diabetes Research & Care: first published as 10.1136/bmjdrc-2021-002560 on 19 January 2022. Downloaded from https://drc.bmj.com on 27 September 2025 by guest.\\nProtected by copyright, including for uses related to text and data mining, AI training, and similar technologies.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='6\\nBMJ Open Diab Res Care 2022;10:e002560. doi:10.1136/bmjdrc-2021-002560\\nEmerging technologies, pharmacology and therapeutics\\nwith CKD report disease and management affecting not \\nonly their physical health, but also mental and social \\nhealth.34 As established in previous studies, 20 22 35  we \\nchose to assess DKD risk over a 5- year window to remain \\nwithin a time frame that would allow improvements in \\noutcome through lifestyle or treatment plan changes.\\nPrevious MLA- based approaches to CKD prediction \\ninclude that of Ravizza et al, who forecast CKD within 3 \\nyears of a recent diagnosis of diabetes, using 2 years of \\nprior data.21 Their performance, which was based on a \\npredicted outcome which included all stages of CKD, \\ndropped from an AUROC of 0.79 to an AUROC of \\n0.72 when prediction was restricted to the more severe \\noutcomes defined by Dunkler et al.36 More recently, Chan \\net al developed a model using EHR data along with three'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='0.72 when prediction was restricted to the more severe \\noutcomes defined by Dunkler et al.36 More recently, Chan \\net al developed a model using EHR data along with three \\nplasma biomarkers that achieved an AUROC of 0.77 \\nfor predicting the progression of DKD in patients with \\ndiabetes who have early DKD.22 However, early awareness \\nand prevention is a major obstacle for DKD; thus, devel-\\noping a model only for patients who are already diagnosed \\nwith DKD is a critical limitation and does not address the \\ncurrent clinical challenges. Moreover, the use of plasma \\nbiomarkers also poses a challenge for this method to be \\nwidely implemented, as these are not routinely screened \\nfor or part of typical EHR data. Additional testing for \\nplasma biomarkers would increase the labor burden and \\ncost of care. Further, several new biomarkers have been \\nproposed for DKD diagnosis and prognosis, but enough \\nevidence for their clinical implementation is still lacking;'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='cost of care. Further, several new biomarkers have been \\nproposed for DKD diagnosis and prognosis, but enough \\nevidence for their clinical implementation is still lacking; \\nstudies are typically performed on small cohorts and not \\nexternally validated.37 Our algorithms use 1 year of prior \\npatient data to predict the development of DKD within the \\nnext 5 years at the time of T2DM diagnosis, and achieved \\nAUROC values of 0.77 for any- stage DKD and 0.83 for \\nDKD stages 3–5 on an external validation dataset. Both \\nRF and XGB performed similarly in terms of AUROC \\nand sensitivity/specificity. Results for the RF models were \\nmore consistent between hold- out test set and external \\ndatasets, likely due to a higher resistance to overfitting \\nthan XGB models, because RF models combine many full \\ntrees’ decisions democratically as opposed to building \\na single output from weak- learning smaller trees as in \\nXGB. These results may support the use of RF models for'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='trees’ decisions democratically as opposed to building \\na single output from weak- learning smaller trees as in \\nXGB. These results may support the use of RF models for \\ngreater generalizability across different clinical settings.\\nMLAs are at their best in clinical medicine when used \\nto supplement medical expertise. Tools that inform clini-\\ncians of risk and allow their clinical judgment to be used \\nproactively rather than reactively are highly beneficial for \\npatient outcomes. This data- driven information, when \\npresented to the clinician in an easy- to- use manner, can \\naugment the use of their clinical knowledge and experi-\\nence. We have previously demonstrated the utility of this \\napproach for detecting sepsis in intensive care units. 38 \\nAdditionally, we have also shown that use of ML- based \\ntechniques in healthcare may lead to considerable cost- \\nsavings.39 Development and adoption of MLA models \\nin clinical settings may significantly improve diagnosis'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='techniques in healthcare may lead to considerable cost- \\nsavings.39 Development and adoption of MLA models \\nin clinical settings may significantly improve diagnosis \\nand treatment options for patients. The use of MLA for \\ndisease prediction and diagnosis is especially useful for \\ndiseases which would benefit from early diagnosis and \\nintervention such as DKD.\\nThere were several limitations to this study. First, this is \\na retrospective study and therefore we cannot guarantee \\nthe same performance in a clinical study. The dataset used \\nfor our models had a diverse demographic sample, yet, we \\ncannot guarantee how it will perform in clinical settings \\nwith other patient populations. We generated the patient \\npopulation with diabetes and subsets of populations with \\nCKD based on ICD codes. Although previous studies \\nhave demonstrated that use of ICD codes to determine \\nand classify patient populations with diabetes are highly'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='CKD based on ICD codes. Although previous studies \\nhave demonstrated that use of ICD codes to determine \\nand classify patient populations with diabetes are highly \\nreliable,40–43 we note that there is a possibility of bias that \\ncould arise from human error or under- reporting in ICD \\ncoding. Furthermore, although it has the potential to \\nimprove DKD risk evaluation and patient outcomes, we \\ncannot determine how clinicians would react to the use \\nof MLA models. Future studies should include evaluation \\nof our MLA performance in a prospective clinical prac-\\ntice and assess patient outcome. This research provides \\ninteresting preliminary data and we hope to do more \\nstudies in the future to validate its use.\\nIn this retrospective study, we have developed and eval-\\nuated MLAs for the prediction of DKD risk over the next \\n5 years, for patients recently diagnosed with T2DM. The \\nMLAs use commonly available data extracted from the \\npatient’s prior year EHR data. Our algorithm provides'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='5 years, for patients recently diagnosed with T2DM. The \\nMLAs use commonly available data extracted from the \\npatient’s prior year EHR data. Our algorithm provides \\nincreased accuracy over the CDC score. MLAs may be \\nhelpful in clinical settings to enable early interventions \\nto improve patient outcomes.\\nContributors AA performed the data analysis and created the tables and figures. \\nZI, AG- S, JH and MH contributed to the experimental design and writing. QM and \\nRD obtained the data and developed the project idea. QM is the guarantor for this \\nstudy.\\nFunding The authors have not declared a specific grant for this research from any \\nfunding agency in the public, commercial or not- for- profit sectors.\\nCompeting interests All authors who have affiliations listed with Dascena \\n(Houston, Texas, USA) are employees or contractors of Dascena.\\nPatient consent for publication Not required.\\nEthics approval This study does not involve human participants.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='(Houston, Texas, USA) are employees or contractors of Dascena.\\nPatient consent for publication Not required.\\nEthics approval This study does not involve human participants.\\nProvenance and peer review Not commissioned; externally peer reviewed.\\nData availability statement Data are available upon reasonable request. Data \\nare available from the corresponding author upon reasonable request. Restrictions \\napply to the availability of these data, which were used under license for the \\ncurrent study and so are not publicly available.\\nSupplemental material This content has been supplied by the author(s). It \\nhas not been vetted by BMJ Publishing Group Limited (BMJ) and may not have \\nbeen peer- reviewed. Any opinions or recommendations discussed are solely \\nthose of the author(s) and are not endorsed by BMJ. BMJ disclaims all liability \\nand responsibility arising from any reliance placed on the content. Where the'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='those of the author(s) and are not endorsed by BMJ. BMJ disclaims all liability \\nand responsibility arising from any reliance placed on the content. Where the \\ncontent includes any translated material, BMJ does not warrant the accuracy and \\nreliability of the translations (including but not limited to local regulations, clinical \\nguidelines, terminology, drug names and drug dosages), and is not responsible \\nfor any error and/or omissions arising from translation and adaptation or \\notherwise.\\nOpen access This is an open access article distributed in accordance with the \\nCreative Commons Attribution Non Commercial (CC BY- NC 4.0) license, which \\npermits others to distribute, remix, adapt, build upon this work non- commercially, \\nand license their derivative works on different terms, provided the original work is \\nBMJ Open Diabetes Research & Care: first published as 10.1136/bmjdrc-2021-002560 on 19 January 2022. Downloaded from https://drc.bmj.com on 27 September 2025 by guest.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 5, 'page_label': '6', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='BMJ Open Diabetes Research & Care: first published as 10.1136/bmjdrc-2021-002560 on 19 January 2022. Downloaded from https://drc.bmj.com on 27 September 2025 by guest.\\nProtected by copyright, including for uses related to text and data mining, AI training, and similar technologies.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='7\\nBMJ Open Diab Res Care 2022;10:e002560. doi:10.1136/bmjdrc-2021-002560\\nEmerging technologies, pharmacology and therapeutics\\nproperly cited, appropriate credit is given, any changes made indicated, and the \\nuse is non- commercial. See:\\xa0http://creativecommons.org/licenses/by-nc/4.0/.\\nORCID iDs\\nAngier\\xa0Allen http://orcid.org/0000-0001-7808-6244\\nZohora\\xa0Iqbal http://orcid.org/0000-0001-7065-8367\\nAbigail\\xa0Green- Saxena http://orcid.org/0000-0002-8502-6589\\nMyrna\\xa0Hurtado http://orcid.org/0000-0003-0704-8384\\nJana\\xa0Hoffman http://orcid.org/0000-0002-7745-3900\\nQingqing\\xa0Mao http://orcid.org/0000-0001-6001-6723\\nRitankar\\xa0Das http://orcid.org/0000-0002-1326-844X\\nREFERENCES\\n 1 Levey AS, Coresh J. Chronic kidney disease. Lancet \\n2012;379:165–80.\\n 2 Afkarian M, Sachs MC, Kestenbaum B, et\\xa0al. Kidney disease and \\nincreased mortality risk in type 2 diabetes. JASN 2013;24:302–8.\\n 3 Thomas MC, Brownlee M, Susztak K, et\\xa0al. Diabetic kidney disease. \\nNat Rev Dis Primers 2015;1:1–20.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='increased mortality risk in type 2 diabetes. JASN 2013;24:302–8.\\n 3 Thomas MC, Brownlee M, Susztak K, et\\xa0al. Diabetic kidney disease. \\nNat Rev Dis Primers 2015;1:1–20.\\n 4 Persson F , Rossing P . Diagnosis of diabetic kidney disease: state \\nof the art and future perspective. Kidney International Supplements \\n2018;8:2–7.\\n 5 Webster AC, Nagler EV, Morton RL, et\\xa0al. Chronic kidney disease. \\nLancet 2017;389:1238–52.\\n 6 Anders H- J, Huber TB, Isermann B, et\\xa0al. Ckd in diabetes: diabetic \\nkidney disease versus nondiabetic kidney disease. Nat Rev Nephrol \\n2018;14:361–77.\\n 7 Gnudi L. Cellular and molecular mechanisms of diabetic \\nglomerulopathy. Nephrol Dial Transplant 2012;27:2642–9.\\n 8 Go AS, Chertow GM, Fan D. Chronic kidney disease and the risks \\nof death, cardiovascular events, and hospitalization. N Engl J Med \\n2004;351:1296–305.\\n 9 Patel UD, Young EW, Ojo AO, et\\xa0al. Ckd progression and \\nmortality among older patients with diabetes. Am J Kidney Dis \\n2005;46:406–14.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='2004;351:1296–305.\\n 9 Patel UD, Young EW, Ojo AO, et\\xa0al. Ckd progression and \\nmortality among older patients with diabetes. Am J Kidney Dis \\n2005;46:406–14.\\n 10 Alicic RZ, Rooney MT, Tuttle KR. Diabetic kidney disease: \\nchallenges, progress, and possibilities. Clin J Am Soc Nephrol \\n2017;12:2032–45.\\n 11 Lozano R, Naghavi M, Foreman K, et\\xa0al. Global and regional \\nmortality from 235 causes of death for 20 age groups in 1990 and \\n2010: a systematic analysis for the global burden of disease study \\n2010. The Lancet 2012;380:2095–128.\\n 12 Molitch ME, Adler AI, Flyvbjerg A, et\\xa0al. Diabetic kidney disease: a \\nclinical update from kidney disease: improving global outcomes. \\nKidney Int 2015;87:20–30.\\n 13 Zimmet P , Alberti KGMM, Shaw J. Global and societal implications \\nof the diabetes epidemic. Nature 2001;414:782–7.\\n 14 Ruggenenti P , Fassi A, Ilieva AP , et\\xa0al. Preventing microalbuminuria in \\ntype 2 diabetes. N Engl J Med 2004;351:1941–51.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='of the diabetes epidemic. Nature 2001;414:782–7.\\n 14 Ruggenenti P , Fassi A, Ilieva AP , et\\xa0al. Preventing microalbuminuria in \\ntype 2 diabetes. N Engl J Med 2004;351:1941–51.\\n 15 Haller H, Ito S, Izzo JL, et\\xa0al. Olmesartan for the delay or \\nprevention of microalbuminuria in type 2 diabetes. N Engl J Med \\n2011;364:907–17.\\n 16 Tuttle KR, Bakris GL, Bilous RW, et\\xa0al. Diabetic kidney disease: \\na report from an ADA consensus conference. Diabetes Care \\n2014;37:2864–83.\\n 17 Jawa A, Kcomt J, Fonseca VA. Diabetic nephropathy and \\nretinopathy. Med Clin North Am 2004;88:1001–36.\\n 18 Levin A, Stevens PE. Early detection of CKD: the benefits, limitations \\nand effects on prognosis. Nat Rev Nephrol 2011;7:446–57.\\n 19 Bang H, Vupputuri S, Shoham DA. Screening for occult renal disease \\n(scored): a simple prediction model for chronic kidney disease. Arch \\nIntern Med 2007;167:374–81.\\n 20 Nelson RG, Grams ME, Ballew SH, et\\xa0al. Development of risk'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='(scored): a simple prediction model for chronic kidney disease. Arch \\nIntern Med 2007;167:374–81.\\n 20 Nelson RG, Grams ME, Ballew SH, et\\xa0al. Development of risk \\nprediction equations for incident chronic kidney disease. JAMA \\n2019;322:2104.\\n 21 Ravizza S, Huschto T, Adamov A, et\\xa0al. Predicting the early risk of \\nchronic kidney disease in patients with diabetes using real- world \\ndata. Nat Med 2019;25:57–9.\\n 22 Chan L, Nadkarni GN, Fleming F , et\\xa0al. Derivation and validation \\nof a machine learning risk score using biomarker and electronic \\npatient data to predict progression of diabetic kidney disease. \\nDiabetologia2021;64:1504–15.\\n 23 Makino M, Yoshimoto R, Ono M, et\\xa0al. Artificial intelligence predicts \\nthe progression of diabetic kidney disease using big data machine \\nlearning. Sci Rep 2019;9:11862.\\n 24 Hannan M, Ansari S, Meza N, et\\xa0al. Risk factors for CKD \\nprogression: overview of findings from the CRIC study. Clin J Am \\nSoc Nephrol 2021;16:648–59.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='learning. Sci Rep 2019;9:11862.\\n 24 Hannan M, Ansari S, Meza N, et\\xa0al. Risk factors for CKD \\nprogression: overview of findings from the CRIC study. Clin J Am \\nSoc Nephrol 2021;16:648–59.\\n 25 Kazancioğlu R. Risk factors for chronic kidney disease: an update. \\nKidney International Supplements 2013;3:368–71.\\n 26 Leung RKK, Wang Y , Ma RCW, et\\xa0al. Using a multi- staged \\nstrategy based on machine learning and mathematical modeling \\nto predict genotype- phenotype risk patterns in diabetic kidney \\ndisease: a prospective case–control cohort analysis. BMC Nephrol \\n2013;14:162.\\n 27 CDC. Chronic kidney disease basics | chronic kidney disease \\ninitiative, 2021. Available: https://www.cdc.gov/kidneydisease/ \\nbasics.html\\n 28 Chen T, Guestrin C. XGBoost: a scalable tree boosting system. \\nproceedings of the 22nd ACM SIGKDD International Conference on \\nknowledge discovery and data mining, ACM, 2016:785–94.\\n 29 Pedregosa F . Scikit- learn: machine learning in python. Mach. Learn. \\nPYTHON 6.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='knowledge discovery and data mining, ACM, 2016:785–94.\\n 29 Pedregosa F . Scikit- learn: machine learning in python. Mach. Learn. \\nPYTHON 6.\\n 30 Kellum JA. KDIGO clinical practice guideline for acute kidney injury. \\nKidney Int Suppl 2012;2:1–138.\\n 31 Centers for Disease Control and Prevention. Chronic kidney disease \\n(CKD) surveillance system. Available: https://nccd.cdc.gov/CKD/ \\nCalculators.aspx\\n 32 Bergstra J, Yamins D, Cox DD. Making a science of model search: \\nhyperparameter optimization in hundreds of dimensions for vision \\narchitectures. Proceedings of the 30th International Conference on \\nInternational Conference on Machine Learning - Volume 28,  JMLR. \\norg, 2013:I- 115–0.\\n 33 Szczech LA, Stewart RC, Su H- L, et\\xa0al. Primary care detection of \\nchronic kidney disease in adults with type- 2 diabetes: the ADD- CKD \\nstudy (awareness, detection and drug therapy in type 2 diabetes and \\nchronic kidney disease). PLoS One 2014;9:e110535.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='chronic kidney disease in adults with type- 2 diabetes: the ADD- CKD \\nstudy (awareness, detection and drug therapy in type 2 diabetes and \\nchronic kidney disease). PLoS One 2014;9:e110535.\\n 34 Hussien H, Apetrii M, Covic A. Health- Related quality of life in \\npatients with chronic kidney disease. Expert Rev Pharmacoecon \\nOutcomes Res 2021;21:43–54.\\n 35 Jardine MJ, Hata J, Woodward M, et\\xa0al. Prediction of kidney- \\nrelated outcomes in patients with type 2 diabetes. Am J Kidney Dis \\n2012;60:770–8.\\n 36 Dunkler D, Gao P , Lee SF , et\\xa0al. Risk prediction for early CKD in type \\n2 diabetes. CJASN 2015;10:1371–9.\\n 37 Jim B, Santos J, Spath F , et\\xa0al. Biomarkers of diabetic nephropathy, \\nthe present and the future. Curr. Diabetes Rev 2012;8:317–28.\\n 38 Desautels T, Calvert J, Hoffman J, et\\xa0al. Prediction of sepsis in the \\nintensive care unit with minimal electronic health record data: a \\nmachine learning approach. JMIR Med Inform 2016;4:e28.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='38 Desautels T, Calvert J, Hoffman J, et\\xa0al. Prediction of sepsis in the \\nintensive care unit with minimal electronic health record data: a \\nmachine learning approach. JMIR Med Inform 2016;4:e28.\\n 39 Calvert J, Hoffman J, Barton C, et\\xa0al. Cost and mortality impact \\nof an algorithm- driven sepsis prediction system. J Med Econ \\n2017;20:646–51.\\n 40 Chi GC, Li X, Tartof SY , et\\xa0al. Validity of ICD- 10- CM codes for \\ndetermination of diabetes type for persons with youth- onset type 1 \\nand type 2 diabetes. BMJ Open Diab Res Care 2019;7:e000547.\\n 41 Lenoir KM, Wagenknecht LE, Divers J, et\\xa0al. Determining diagnosis \\ndate of diabetes using structured electronic health record (EHR) \\ndata: the search for diabetes in youth study. BMC Med Res \\nMethodol 2021;21:210.\\n 42 Klompas M, Eggleston E, McVetta J, et\\xa0al. Automated detection and \\nclassification of type 1 versus type 2 diabetes using electronic health \\nrecord data. Diabetes Care 2013;36:914–21.'),\n",
              " Document(metadata={'producer': 'pdf-lib (https://github.com/Hopding/pdf-lib)', 'creator': 'Adobe InDesign 15.1 (Windows)', 'creationdate': '2022-01-19T09:22:58+05:30', 'title': 'Prediction of diabetic kidney disease with machine learning algorithms, upon the initial diagnosis of type 2 diabetes mellitus', 'trapped': '/False', 'jobid': 'cfad47da-4ee7-4473-b44a-7c1d96f1bed6', 'moddate': '2025-09-27T21:45:15+00:00', 'source': '/content/drive/MyDrive/Rag_Data/e002560.full.pdf', 'total_pages': 7, 'page': 6, 'page_label': '7', 'source_file': 'e002560.full.pdf', 'file_type': 'pdf'}, page_content='42 Klompas M, Eggleston E, McVetta J, et\\xa0al. Automated detection and \\nclassification of type 1 versus type 2 diabetes using electronic health \\nrecord data. Diabetes Care 2013;36:914–21.\\n 43 Chen G, Khan N, Walker R, et\\xa0al. Validating ICD coding algorithms \\nfor diabetes mellitus from administrative data. Diabetes Res Clin \\nPract 2010;89:189–95.\\nBMJ Open Diabetes Research & Care: first published as 10.1136/bmjdrc-2021-002560 on 19 January 2022. Downloaded from https://drc.bmj.com on 27 September 2025 by guest.\\nProtected by copyright, including for uses related to text and data mining, AI training, and similar technologies.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 0, 'page_label': '1', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Sabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  1 of 14\\nPrediction of diabetic kidney disease \\nrisk using machine learning models: \\nA population- based cohort study of \\nAsian\\xa0adults\\nCharumathi Sabanayagam1,2*, Feng He1, Simon Nusinovici1, Jialiang Li3, \\nCynthia Lim4, Gavin Tan1,2, Ching Yu Cheng1,2\\n1Singapore Eye Research Institute, Singapore, Singapore; 2Ophthalmology and \\nVisual Sciences Academic Clinical Program, Duke- NUS Medical School, Singapore, \\nSingapore; 3Department of Statistics and Data Science, National University of \\nSingapore, Singapore, Singapore; 4Department of Renal Medicine, Singapore General \\nHospital, Singapore, Singapore\\nAbstract\\nBackground: Machine learning (ML) techniques improve disease prediction by identifying the \\nmost relevant features in multidimensional data. We compared the accuracy of ML algorithms for \\npredicting incident diabetic kidney disease (DKD).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 0, 'page_label': '1', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='most relevant features in multidimensional data. We compared the accuracy of ML algorithms for \\npredicting incident diabetic kidney disease (DKD).\\nMethods: We utilized longitudinal data from 1365 Chinese, Malay, and Indian participants aged \\n40–80\\xa0y with diabetes but free of DKD who participated in the baseline and 6- year follow- up visit \\nof the Singapore Epidemiology of Eye Diseases Study (2004–2017). Incident DKD (11.9%) was \\ndefined as an estimated glomerular filtration rate (eGFR)\\xa0<60\\xa0mL/min/1.73 m 2 with at least 25% \\ndecrease in eGFR at follow- up from baseline. A total of 339 features, including participant char -\\nacteristics, retinal imaging, and genetic and blood metabolites, were used as predictors. Perfor -\\nmances of several ML models were compared to each other and to logistic regression (LR) model \\nbased on established features of DKD (age, sex, ethnicity, duration of diabetes, systolic blood'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 0, 'page_label': '1', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='mances of several ML models were compared to each other and to logistic regression (LR) model \\nbased on established features of DKD (age, sex, ethnicity, duration of diabetes, systolic blood \\npressure, HbA1c, and body mass index) using area under the receiver operating characteristic \\ncurve (AUC).\\nResults: ML model Elastic Net (EN) had the best AUC (95% CI) of 0.851 (0.847–0.856), which was \\n7.0% relatively higher than by LR 0.795 (0.790–0.801). Sensitivity and specificity of EN were 88.2 and \\n65.9% vs. 73.0 and 72.8% by LR. The top 15 predictors included age, ethnicity, antidiabetic medi-\\ncation, hypertension, diabetic retinopathy, systolic blood pressure, HbA1c, eGFR, and metabolites \\nrelated to lipids, lipoproteins, fatty acids, and ketone bodies.\\nConclusions: Our results showed that ML, together with feature selection, improves prediction \\naccuracy of DKD risk in an asymptomatic stable population and identifies novel risk factors, including \\nmetabolites.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 0, 'page_label': '1', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content=\"accuracy of DKD risk in an asymptomatic stable population and identifies novel risk factors, including \\nmetabolites.\\nFunding: This study was supported by the Singapore Ministry of Health’s National Medical Research \\nCouncil, NMRC/OFLCG/MOH- 001327- 03 and NMRC/HCSAINV/MOH- 001019- 00. The funders \\nhad no role in study design, data collection and analysis, decision to publish, or preparation of the \\nmanuscript.\\nEditor's evaluation\\nRESEARCH ARTICLE\\n*For correspondence: \\ncharumathi.sabanayagam@seri. \\ncom.sg\\nCompeting interest: The authors \\ndeclare that no competing \\ninterests exist.\\nFunding: See page 11\\nReceived: 14 July 2022\\nPreprinted: 18 August 2022\\nAccepted: 12 September 2023\\nPublished: 14 September 2023\\nReviewing Editor: Girish N \\nNadkarni, Icahn School of \\nMedicine at Mount Sinai, United \\nStates\\n   Copyright Sabanayagam \\net\\xa0al. This article is distributed \\nunder the terms of the Creative \\nCommons Attribution License, \\nwhich permits unrestricted use\"),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 0, 'page_label': '1', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Medicine at Mount Sinai, United \\nStates\\n   Copyright Sabanayagam \\net\\xa0al. This article is distributed \\nunder the terms of the Creative \\nCommons Attribution License, \\nwhich permits unrestricted use \\nand redistribution provided that \\nthe original author and source \\nare credited.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 1, 'page_label': '2', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  2 of 14\\nThe authors wanted to see which patients with diabetes develop kidney disease and outcomes. They \\nused clinical characteristics, eye pictures, genetic factors and blood levels of metabolites, and they \\nfound a combination of these factors predicted kidney disease in people with diabetes.\\nIntroduction\\nDiabetes affected an estimated 415\\xa0million people worldwide in 2015, and this number is expected \\nto increase to 642\\xa0million by 2040, with the greatest increase expected in Asia, particularly in India \\nand China (Ogurtsova et\\xa0al., 2017). With the rising prevalence of diabetes and an aging popu-\\nlation, the burden of diabetic kidney disease (DKD), a leading cause of end- stage kidney disease \\n(ESKD), cardiovascular disease (CVD), and premature deaths, is also set to rise in parallel. Diabetes'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 1, 'page_label': '2', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='(ESKD), cardiovascular disease (CVD), and premature deaths, is also set to rise in parallel. Diabetes \\naccounts for 30–50% of all chronic kidney disease (CKD) cases, affecting 285\\xa0million people world-\\nwide (Webster et\\xa0al., 2017). As CKD is asymptomatic till more than 50% of kidney function decline, \\nearly detection of individuals with diabetes who are at risk of developing DKD may facilitate preven-\\ntion and appropriate intervention for DKD (Hill et\\xa0 al., 2016; Hirst et\\xa0 al., 2020). However, early \\nidentification of individuals at risk of developing CKD in type 2 diabetes is challenging (Alicic et\\xa0al., \\n2017). Although most ESKD cases are due to diabetes, awareness of diabetes as a risk factor for \\nCKD is markedly lower in several countries, which may constitute a barrier for early detection of CKD \\n(White et\\xa0al., 2008; Hussain et\\xa0al., 2019; Couser et\\xa0al., 2011). Moreover, in people with diabetes,'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 1, 'page_label': '2', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='(White et\\xa0al., 2008; Hussain et\\xa0al., 2019; Couser et\\xa0al., 2011). Moreover, in people with diabetes, \\nadherence to annual screening for DKD with estimated glomerular filtration rate (eGFR) and urine \\nalbumin- creatinine ratio (UACR) remains a challenge (Manski- Nankervis et\\xa0al., 2018). Therefore, \\nthere is an urgent need for the characterization of new biomarkers to identify individuals at risk of \\nprogressive eGFR decline and enable timely intervention for improving outcomes in DKD (Alicic \\net\\xa0al., 2017).\\nSeveral risk prediction models have been developed in the past for predicting progression to \\nESKD, but studies predicting onset of CKD in diabetic populations are limited. These studies were \\nfocused on clinical populations utilizing data from clinical trials (Dunkler et\\xa0al., 2015) or heteroge-\\nneous cohorts of patients with different CKD definitions (Jiang et\\xa0al., 2020). Dunkler et al. showed'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 1, 'page_label': '2', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='neous cohorts of patients with different CKD definitions (Jiang et\\xa0al., 2020). Dunkler et al. showed \\nthat albuminuria and eGFR were the key predictors and addition of demographic, clinical, or labora-\\ntory variables did not improve predictive performance beyond 69% (Dunkler et\\xa0al., 2015). Current \\nCKD risk prediction models developed using traditional regression models (e.g., logistic regression \\n[LR] or linear regression) perform well when there are only small or moderate numbers of variables or \\npredictors but tend to overfit if there is a large number of variables. Machine learning (ML) methods \\nusing ‘Big data,’ or multidimensional data, may improve prediction as they have fewer restrictive \\nstatistical assumptions compared to traditional regression models that assume linear relationships \\nbetween risk factors and the logit of the outcomes and absence of multicollinearity among explana-\\ntory variables (Sundström and Schön, 2020; Doupe et\\xa0al., 2019; Bi et\\xa0al., 2019).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 1, 'page_label': '2', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='between risk factors and the logit of the outcomes and absence of multicollinearity among explana-\\ntory variables (Sundström and Schön, 2020; Doupe et\\xa0al., 2019; Bi et\\xa0al., 2019).\\nDiabetes is a metabolic disorder, and the metabolic changes associated with diabetes can lead to \\nglomerular hypertrophy, glomerulosclerosis, tubulointerstitial inflammation, and fibrosis (Alicic et\\xa0al., \\n2017). Several blood metabolites have been shown to be associated with DKD (Colhoun and Marcov-\\necchio, 2018). Similarly, genetic abnormalities in diabetes have also been linked to an increased \\nrisk of DKD (Cole and Florez, 2020). We, and several others, have previously shown that retinal \\nmicrovascular changes, including retinopathy, vessel narrowing, or dilation, and vessel tortuosity, were \\nassociated with CKD (Yau et\\xa0 al., 2011; Yip et\\xa0 al., 2017). Integrating high- dimensional data from \\nmultiple domains, including patient characteristics, clinical and ‘Omics’ data, has the potential to aid'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 1, 'page_label': '2', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='multiple domains, including patient characteristics, clinical and ‘Omics’ data, has the potential to aid \\nin risk stratification, prediction of future risk, and provide insights into the pathogenesis (Eddy et\\xa0al., \\n2020). These features may contribute to prediction in very complicated ways, and they may not fully \\nsatisfy the requirement for a simple linear logistic model. It is thus more appropriate to consider ML \\napproaches for a comprehensive study.\\nIn the current study, we aimed to evaluate the performance of a set of the most common ML \\nmodels, including traditional LR, for predicting the 6- year risk of DKD and identifying important \\npredictors of DKD in a large population- based cohort study in Singapore with multidimensional data, \\nincluding imaging, metabolites, and genetic biomarkers.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 2, 'page_label': '3', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  3 of 14\\nMethods\\nStudy population\\nData for this study was derived from the Singapore Epidemiology of Eye Diseases (SEED) study, a \\npopulation- based prospective study of eye diseases in 10,033 Asian adults aged 40–80\\xa0y in Singa-\\npore. The follow- up study was conducted after a median duration of 6.08\\xa0y (interquartile range: [5.56, \\n6.79]), with 6762 participants. The detailed methodology of SEED has been published elsewhere \\n(Majithia et\\xa0al., 2021). Briefly, the name list of adults residing in the southwestern part of Singa-\\npore was provided by the Ministry of Home Affairs, and then an age- stratified random sampling \\nprocedure was conducted. A total of 3280 Malays (2004–2007) (Foong et\\xa0al., 2007), 3400 Indians \\n(2007–2009), and 3353 Chinese (2009–2011) (Lavanya et\\xa0al., 2009) participated in the baseline study'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 2, 'page_label': '3', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='procedure was conducted. A total of 3280 Malays (2004–2007) (Foong et\\xa0al., 2007), 3400 Indians \\n(2007–2009), and 3353 Chinese (2009–2011) (Lavanya et\\xa0al., 2009) participated in the baseline study \\nwith response rates of 78.7, 75.6 and 72.8%, respectively. As all three studies followed the same \\nmethodology and were conducted in the same study clinic, we combined the three populations for \\nthe present study. For the current analysis, we included only those with diabetes, defined as random \\nglucose\\xa0≥11.1\\xa0mmol/L, HbA1c ≥ 6.5% (48 mmol/mol), self- reported antidiabetic medication use, or \\nhaving been diagnosed with diabetes by a physician. Of the 6762 participants who attended both \\nbaseline and follow- up visit, after excluding those without diabetes (n = 5307), prevalent CKD (n \\n= 315), missing information on eGFR (n = 90), the final sample size for analysis was 1365 (47.5% \\nIndians, 27.8% Malays, and 24.7% Chinese). The sample size available for each dataset after removing'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 2, 'page_label': '3', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='= 315), missing information on eGFR (n = 90), the final sample size for analysis was 1365 (47.5% \\nIndians, 27.8% Malays, and 24.7% Chinese). The sample size available for each dataset after removing \\nparticipants missing\\xa0>10%\\xa0data was between 976 and 1364 (Supplementary file 1- Table 1a). SEED \\nwas conducted in accordance with the Declaration of Helsinki and was approved by the SingHealth \\nCentralised Institutional Review Board (2018/2717, 2018/2921, 2012/487/A, 2015/2279, 2018/2006, \\n2018/2594, 2018/2570). Informed consent was obtained from all participants.\\nAssessment of DKD\\nIncident DKD was defined as an eGFR\\xa0<60\\xa0mL/min/1.73 m 2 with at least a 25% decrease in eGFR \\nat follow- up in participants who had eGFR\\xa0≥ 60\\xa0mL/min/1.73 m 2 at baseline. Combining change in \\neGFR category together with a minimal percent change ensures that small changes in eGFR, for \\nexample, from 61 to 59\\xa0mL/min/1.73 m 2, are not misinterpreted as incident CKD as the eGFR is\\xa0<'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 2, 'page_label': '3', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='eGFR category together with a minimal percent change ensures that small changes in eGFR, for \\nexample, from 61 to 59\\xa0mL/min/1.73 m 2, are not misinterpreted as incident CKD as the eGFR is\\xa0< \\n60\\xa0mL/min/1.73 m2 (Yip et\\xa0al., 2017; Stevens et\\xa0al., 2013). The reduction in eGFR at follow- up was \\ncalculated as a percentage of the baseline eGFR as (eGFR at baseline – eGFR at follow- up)/eGFR at \\nbaseline * 100%. GFR was estimated from plasma creatinine using the Chronic Kidney Disease Epide-\\nmiology Collaboration (CKD- EPI) equation (Levey et\\xa0al., 2009). Blood creatinine was measured by the \\nJaffe method on the Beckman DXC800 analyzer calibrated to the Isotope Dilution Mass Spectrometry \\n(IDMS) method using the National Institute of Standards and Technology (NIST) Reference material.\\nVariables for prediction\\nWe evaluated 339 features such as demographic, lifestyle, socioeconomic, physical, laboratory,'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 2, 'page_label': '3', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Variables for prediction\\nWe evaluated 339 features such as demographic, lifestyle, socioeconomic, physical, laboratory, \\nretinal imaging, genetic and blood metabolomics profile. The entire list of variables is presented in \\nSupplementary file 1- Table 1b. We organized the variables into five different domains: traditional risk \\nfactors, extended risk factors, imaging parameters, genetic parameters, and blood metabolites. For \\nML analysis, based on different combinations of the five domains, we tested six models (A to F): A = \\ntraditional\\xa0risk factors; B = A + extended risk factors; C = B + imaging parameters; D = B + genetic \\nparameters; E = B + blood metabolites; F = B + imaging parameters + blood metabolites + genetic \\nparameters.\\nTraditional risk factors (n = 7)\\nAge, sex, ethnicity (Chinese, Malays, and Indians), body mass index (BMI, kg/m2), systolic blood pres-\\nsure (BP , mm Hg), duration of diabetes (years), and HbA1c% were included as traditional risk factors.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 2, 'page_label': '3', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='sure (BP , mm Hg), duration of diabetes (years), and HbA1c% were included as traditional risk factors.\\nExtended risk factors (n = 22)\\nParticipant information was collected using an interviewer- administered questionnaire (demo-\\ngraphic, socioeconomic, lifestyle factors, and personal history of diseases and medication use), phys-\\nical examination (height, weight, BP), and laboratory examination (blood glucose, creatinine, lipid \\nprofile) (Majithia et\\xa0 al., 2021). Marital status, educational level (primary/below [≤6\\xa0 y], secondary/'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 3, 'page_label': '4', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  4 of 14\\nabove [7\\xa0y and above of education]), monthly income, smoking status (current smokers vs. former \\nand nonsmokers), alcohol consumption (ever vs. nondrinkers), history of CVD (self- reported history \\nof myocardial infarction, stroke or angina), hypertension status, diastolic BP , pulse pressure, blood \\nglucose, total, high- density lipoprotein (HDL) and low- density lipoprotein (LDL) cholesterol levels, \\nantidiabetic including oral hypoglycemia drugs and insulin, antihypertensive, and anticholesterol \\nmedication use were included as part of extended risk factors.\\nBlood metabolites (n = 223)\\nWe quantified 228 metabolic measures from stored serum/plasma samples at baseline using a \\nhigh- throughput NMR metabolomics platform (Nightingale Health, Helsinki, Finland). The metab-'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 3, 'page_label': '4', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='We quantified 228 metabolic measures from stored serum/plasma samples at baseline using a \\nhigh- throughput NMR metabolomics platform (Nightingale Health, Helsinki, Finland). The metab-\\nolites included routine lipids, lipoprotein subclasses with lipid concentrations within 14 subclasses, \\nfatty acids, amino acids, ketone bodies, and glycolysis- related metabolites. The 14 lipoprotein \\nsubclasses include six subclasses of VLDL (extremely large, very large, large, medium, small, very \\nsmall), IDL, three subclasses of LDL (large, medium, small), and four subclasses of HDL (very large, \\nlarge, medium, small). Lipid concentration within each lipoprotein particle included triacylglyc-\\nerol, total cholesterol, non- esterified cholesterol and cholesteryl ester levels, and phospholipid \\nconcentrations (Quek et\\xa0al., 2021). Of the 228 metabolites, pyruvate, glycerol, and glycine were \\nnot available in Malays. In addition, creatinine and glucose were measured as part of the blood'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 3, 'page_label': '4', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='concentrations (Quek et\\xa0al., 2021). Of the 228 metabolites, pyruvate, glycerol, and glycine were \\nnot available in Malays. In addition, creatinine and glucose were measured as part of the blood \\nbiochemistry. After excluding these five metabolites, 223 were included under the metabolites \\ndataset.\\nGenetic parameters (n = 76)\\nWe included 76 type 2 diabetes- associated single- nucleotide polymorphisms (SNPs) identified in the \\nlargest meta- analysis of type 2 diabetes genome- wide association studies by the DIAbetes Genetics \\nReplication and Meta- analysis consortium (Chong et\\xa0al., 2017).\\nFigure 1. Comparison of nine machine learning models for diabetic kidney disease (DKD) incidence prediction using different sets of features (Panel A- \\nF). Abbreviations: CART, classification and regression tree; EN, elastic net; GBDT, gradient boosting decision tree; LASSO, least absolute shrinkage and'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 3, 'page_label': '4', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='F). Abbreviations: CART, classification and regression tree; EN, elastic net; GBDT, gradient boosting decision tree; LASSO, least absolute shrinkage and \\nselection operator; LR, logistic regression; NB, naïve Bayes; RF , random forest; SVM, support vector machine; XGB, extreme gradient boosting.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 4, 'page_label': '5', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  5 of 14\\nImaging parameters (n = 11)\\nUsing a semi- automated computer program (Singapore I Vessel Assessment, SIVA), we quantified \\nretinal imaging parameters from digital retinal photographs. The parameters included retinal arte-\\nriolar and venular diameters, vessel tortuosity, branching angle, fractal dimension, etc. (Yip et\\xa0al., \\n2017). Diabetic retinopathy (DR) was assessed by trained graders using a standard protocol (Saban-\\nayagam et\\xa0al., 2019).\\nMachine learning algorithms\\nWe tested nine different ML algorithms, including LR, least absolute shrinkage and selection oper -\\nator (LASSO), elastic net (EN), classification and regression tree (CART), random forest (RF), gradient \\nboosting decision tree (GBDT), extreme gradient boosting (XGB), support vector machine (SVM), and \\nnaïve Bayes (NB) (Hastie et\\xa0al., 2009).\\nModel development'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 4, 'page_label': '5', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='boosting decision tree (GBDT), extreme gradient boosting (XGB), support vector machine (SVM), and \\nnaïve Bayes (NB) (Hastie et\\xa0al., 2009).\\nModel development\\nWe split the study samples randomly into training (80%) and test sets (20%) of equal CKD case rate by \\nstratified sampling, with 40 random repeats of 5- fold cross- validation to evaluate the model perfor -\\nmance. Predictive accuracy was assessed using metrics such as area under the receiver operating char-\\nacteristic curve (AUC) with 95% confidence interval (CI) , sensitivity and specificity calculated at the \\noptimal cut- point (determined by Youden’s index). In preliminary analyses, testing different combina-\\ntions of features (Figure\\xa01A–F), performance of all ML models was below 0.80 in dataset D including \\ngenetic features (best AUC = 0.785 by RF) and dataset F including all 339 features (best AUC = 0.788 \\nby XGB). Hence, we dropped these two datasets (D and F) from further analyses. The performance of'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 4, 'page_label': '5', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='by XGB). Hence, we dropped these two datasets (D and F) from further analyses. The performance of \\nall ML models based on AUC (IQR) in datasets 1A–F is shown in Supplementary file 1- Table 1c and \\nbased on sensitivity and specificity is shown in Supplementary file 1- Table 1d.\\nOf the ML models, performances of CART, SVM, and NB were lower compared to other models, \\nhence these models were also dropped. Consequently, ML models EN, GBDT, LASSO, XGB, and RF \\nwere considered for subsequent analyses using datasets A, B, C, and E including 252 features.\\nFeature selection\\nAll algorithms included in the current study can perform feature selection but using different selection \\ncriteria. In LR, stepwise selection according to the Akaike information criterion (AIC) is widely used \\nbut it lacks stability. LASSO is an extension of LR with L1 regularization to drop the less important vari-\\nables. EN is like LASSO but with a milder regularization, resulting in a larger number of retained vari-'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 4, 'page_label': '5', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='ables. EN is like LASSO but with a milder regularization, resulting in a larger number of retained vari-\\nables. In order to select only the most predictive features, we recursively apply EN until the retained \\nvariable subset is optimized, that is, recursive feature selection (RFE). In RF , GBDT, and XGB, the most \\npredictive variables were identified based on their relative importance to model performance. Feature \\nselection was also performed according to their selection frequency during repeated cross- validation. \\nWe identified the top 15 predictors by each of the best- performing ML models, then compared the \\nperformance of the ML models based on the top variables with that of LR based on seven traditional \\nrisk factors (age, sex, ethnicity, BMI, HbA1c, duration of diabetes, and systolic BP) in another 40 \\nrandom repeats of 5- fold cross- validation. Subgroup analyses were conducted for the three ethnic \\ngroups separately.\\nStatistical analyses'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 4, 'page_label': '5', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='random repeats of 5- fold cross- validation. Subgroup analyses were conducted for the three ethnic \\ngroups separately.\\nStatistical analyses\\nWe compared the baseline characteristics of participants with diabetes by incident DKD status using \\nχ2 test or Mann–Whitney U test as appropriate for the variable and compared the socioeconomic \\nstatus by ethnicity using χ 2 test and Kruskal–Wallis test as appropriate for the variable. Statistical \\nsignificance was defined as a p- value<0.05. Subgroup numbers such as DR status may not add up \\ndue to the presence of missing data. For modeling, we used mean values/modes for missing value \\nimputation as appropriate for each variable because the missing proportions were all below 10%. \\nImprovement in prediction accuracy by ML over the traditional risk factor model was calculated as \\n(ML AUC – traditional model AUC)/traditional model AUC * 100%. All analyses were conducted using'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 4, 'page_label': '5', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Improvement in prediction accuracy by ML over the traditional risk factor model was calculated as \\n(ML AUC – traditional model AUC)/traditional model AUC * 100%. All analyses were conducted using \\nR software version 4.0.2. To assess whether the features selected by ML models were meaningful, we \\nvisualized the association of top 15 variables with incident DKD in forest plots or a variable importance \\nplot as appropriate for the algorithm.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 5, 'page_label': '6', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  6 of 14\\nResults\\nThe 6- year incidence of DKD was 11.9% in the study population. The incidence of DKD was the \\nhighest in Malays (18.4%), followed by Chinese (12.8%). Although Indians represent nearly half of the \\ntotal diabetic population (648 of the 1365 diabetic participants, 47.5%), DKD incidence was the lowest \\nin Indians (7.6%).\\nAs shown in Table\\xa01, compared to those without incident DKD, those with were significantly older, \\nmore likely to be Malays, primary/below educated, had higher prevalence of hypertension, DR, CVD, \\nantidiabetic medication use; had longer duration of diabetes, and higher levels of systolic BP and \\nHbA1c%.\\nTable 1. Baseline characteristics of SEED diabetic participants by incident DKD status.\\nCharacteristics\\nNo DKD\\n(n = 1203)\\nDKD\\n(n = 162) p- value\\nOverall\\n(n = 1365)\\nAge (years) 57.95 (8.78) 64.63 (7.98) <0.001 58.74 (8.95)'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 5, 'page_label': '6', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Characteristics\\nNo DKD\\n(n = 1203)\\nDKD\\n(n = 162) p- value\\nOverall\\n(n = 1365)\\nAge (years) 57.95 (8.78) 64.63 (7.98) <0.001 58.74 (8.95)\\nSex, female 580 (48.2) 87 (53.7) 0.219 667 (48.9)\\nEthnicity <0.001\\n  Indians (ref) 599 (49.8) 49 (30.2) 648 (47.5)\\n  Malays 310 (25.8) 70 (43.2) 380 (27.8)\\n  Chinese 294 (24.4) 43 (26.5) 337 (24.7)\\nPrimary/below education (%) 706 (58.7) 121 (74.7) <0.001 827 (60.6)\\nCurrent smoker (%) 173 (14.4) 16 (9.9) 0.15 189 (13.9)\\nAlcohol consumption (%) 111 (9.2) 11 (6.8) 0.389 122 (9.0)\\nHypertension (%) 845 (70.4) 155 (95.7) <0.001 1000 (73.4)\\nDiabetic retinopathy (%) 228 (19.2) 56 (35.4) <0.001 284 (21.1)\\nCardiovascular disease (%) 153 (12.7) 32 (19.8) 0.02 185 (13.6)\\nDuration of diabetes (years) 2.68 [0.00, 8.56] 6.08 [1.44, 11.63] <0.001 3.20 [0.00, 9.37]\\nAntidiabetic medication (%) 681 (56.6) 122 (75.3) <0.001 803 (58.8)\\nInsulin use (%) 39 (3.3) 11 (7.1) 0.036 50 (3.8)\\nBody mass index (kg/m2) 26.96 (4.62) 27.05 (4.36) 0.764 26.97 (4.59)'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 5, 'page_label': '6', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Antidiabetic medication (%) 681 (56.6) 122 (75.3) <0.001 803 (58.8)\\nInsulin use (%) 39 (3.3) 11 (7.1) 0.036 50 (3.8)\\nBody mass index (kg/m2) 26.96 (4.62) 27.05 (4.36) 0.764 26.97 (4.59)\\nSystolic blood pressure (mm Hg) 139.42 (18.95) 155.24 (20.01) <0.001 141.29 (19.74)\\nDiastolic blood pressure (mm Hg) 78.25 (9.74) 79.14 (10.70) 0.278 78.35 (9.85)\\nRandom blood glucose (mmol/L) 9.53 (4.26) 10.44 (5.01) 0.052 9.64 (4.36)\\nHbA1c (%) 7.61 (1.58) 8.04 (1.83) 0.003 7.66 (1.62)\\nBlood total cholesterol (mmol/L) 5.14 (1.14) 4.98 (1.15) 0.124 5.12 (1.15)\\nBlood HDL cholesterol (mmol/L) 1.12 (0.31) 1.16 (0.35) 0.178 1.12 (0.32)\\neGFR (mL/min/1.73\\xa0m2) 89.98 (14.34) 79.40 (11.69) <0.001 88.72 (14.46)\\nValues for categorical variables are presented as number (percentages); values for continuous variables are given \\nas mean (SD) or median [IQR]. p- values are given by χ2 test or Mann–Whitney U test as appropriate for the \\nvariable.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 5, 'page_label': '6', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='as mean (SD) or median [IQR]. p- values are given by χ2 test or Mann–Whitney U test as appropriate for the \\nvariable.\\nDKD, diabetic kidney disease; HDL, high- density lipoprotein cholesterol; IQR, interquartile range; SD, standard \\ndeviation; SEED, Singapore Epidemiology of Eye Diseases.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 6, 'page_label': '7', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  7 of 14\\nPerformance of LR using traditional risk factors (reference) and other \\ndomain features\\nThe LR using the seven traditional risk factors (age, sex, ethnicity, BMI, HbA1c, duration of diabetes, \\nand systolic BP) had an AUC of 0.796. The performance of LR improved to 0.821 using the tradi-\\ntional\\xa0+ extended\\xa0risk factors. With additional features, the performance of LR dropped significantly \\n(AUC of 0.622 in E and 0.811 in C).\\nPerformance of ML models using multidimensional data\\nUsing datasets, A, B, C, and E, the performances of the five\\xa0ML models (Figure\\xa01A–C and E) were as \\nfollows:\\n1. EN ranked first in performance in all four datasets with AUCs ranging from 0.797 in A to 0.843 \\nin E\\n2. LASSO ranged from 0.781 in A to 0.814 in E\\n3. GBDT ranged from 0.789 in A to 0.809 in E\\n4. RF ranged from 0.772 in E to 0.817 in C'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 6, 'page_label': '7', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='in E\\n2. LASSO ranged from 0.781 in A to 0.814 in E\\n3. GBDT ranged from 0.789 in A to 0.809 in E\\n4. RF ranged from 0.772 in E to 0.817 in C\\n5. XGB ranged from 0.764 in A to 0.804 in C\\nFigure\\xa02 shows the AUCs of the top 3 performing models. Using the top 15 predictors generated \\nby feature selection, the performance of EN improved further with an AUC (95%\\xa0CI) of 0.851 (0.847–\\n0.856), sensitivity and specificity of 88.2 and 65.9% compared to LR using seven established features \\nwith AUC of 0.795 (0.790–0.801), and sensitivity and specificity of 73.0 and 72.8%.\\nFigure 2. Performance of the top 3 machine learning (ML) models based on selected variables in dataset E (risk \\nfactors + blood metabolites) compared to LR using seven established features. Abbreviations: EN, Elastic net; \\nGBDT, gradient boosting decision tree; LASSO; least absolute shrinkage and selection operator; LR, logistic \\nregression.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 7, 'page_label': '8', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  8 of 14\\nThe corresponding estimates for LASSO were 0.820 (0.816–0.825), 84.4% and 67.0%; 0.819 \\n(0.814–0.824), 80.6% and 70.1% for GBDT. The AUCs of EN, LASSO, and GBDT were 7.0, 3.1, and \\n3.0% relatively higher than that of LR.\\nTop 15 predictors\\nFigure\\xa03 shows the top 15 predictors visualized using forest plots for EN and LASSO and a variable \\nimportance plot for GBDT. We found low collinearity in EN- selected features (Spearman’s correlation \\ncoefficients: –0.49 to 0.43), while LASSO and GBDT selected some variables of higher correlation \\n(e.g., systolic BP and pulse pressure).\\nAmong the traditional and extended risk factors, all three models chose age, SBP , DR, and lower \\nlevels of eGFR as the top 15 predictors. In addition, antidiabetic medication use, HbA1c, hyperten-'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 7, 'page_label': '8', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Among the traditional and extended risk factors, all three models chose age, SBP , DR, and lower \\nlevels of eGFR as the top 15 predictors. In addition, antidiabetic medication use, HbA1c, hyperten-\\nsion, and ethnicity (Malays and Chinese compared to Indians) were chosen as risk factors by EN and \\nLASSO; antihypertensive medication and low housing type by LASSO; and duration of diabetes, BMI, \\nand HDL cholesterol by GBDT. Among the metabolites, phospholipids to total lipids ratio in MHDL \\nand DHA were selected by all three models. Free cholesterol to total lipids ratio in small HDL/XSVLDL, \\nand cholesterol esters to total lipids ratio in IDL/LLDL/XLHDL were also found to be of high frequency. \\nAdditionally, higher levels of acetate were shown to be protective by LR based on EN- selected vari-\\nables, while tyrosine and lactate were identified as important factors by GBDT. Source data for the'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 7, 'page_label': '8', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='ables, while tyrosine and lactate were identified as important factors by GBDT. Source data for the \\nforest plots are shown in Supplementary file 1- Table 1e. Using the same approach, we also identi-\\nfied the top variables by each ethnicity in Supplementary file 1- Table 1f, and found baseline eGFR \\nlinked to incident DKD in all three subgroups, whereas acetate, SBP , antidiabetic medication use, and \\nhousing type were important in two ethnic subgroups.\\nDiscussion\\nThe results of the current study suggest that prediction using ML models with selected features \\nprovided improved prediction compared to LR model based on seven established features in this \\nFigure 3. Association of the top 15 machine learning (ML)- selected predictors with incident diabetic kidney disease (DKD). Abbreviations: LASSO, \\nleast absolute shrinkage and selection operator; GBDT, Gradient boosting decision tree. Variables: anti- HTN Meds, anti hypertensive medications;'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 7, 'page_label': '8', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='least absolute shrinkage and selection operator; GBDT, Gradient boosting decision tree. Variables: anti- HTN Meds, anti hypertensive medications; \\nDM, diabetes mellitus; GFR- EPI, glomerular filtration rate estimated using CKD- EPI equation; HDL, high- density lipoprotein. Metabolites: DHA: 22:6, \\ndocosahexaenoic acid; DHAFA, Ratio of 22:6 docosahexaenoic acid to total fatty acids; IDL- CE%, Cholesterol esters to total lipids ratio in IDL; M- HDL- \\nPL%, Phospholipids to total lipids ratio in medium HDL; M- VLDL- PL%, Phospholipids to total lipids ratio in medium VLDL; S- HDL- FC% Free cholesterol \\nto total lipids ratio in small HDL; XL- HDL- CE%, Cholesterol esters to total lipids ratio in very large HDL; L- LDL- CE%, Cholesterol esters to total lipids \\nratio in large LDL; M- HDL- PL%, Phospholipids to total lipids ratio in medium HDL; IDL- C% Total cholesterol to total lipids ratio in IDL; M- HDL- PL% \\nPhospholipids to total lipids ratio in medium HDL.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 8, 'page_label': '9', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  9 of 14\\nextensively phenotyped large- scale epidemiological study. The best performance was obtained by \\nthe EN model based on dataset E including risk factors and metabolites with an AUC of 0.851, which \\nwas 7.0% higher than that of LR using seven established risk factors. Sensitivity was also higher by EN \\n(88.2 and 65.9%) compared to LR (73.0 and 72.8%). The top 15 predictors by EN using RFE identi-\\nfied several metabolites related to lipid concentration, lipoprotein subclasses, fatty acids, and ketone \\nbodies as novel predictors besides confirming traditional predictors, including age, ethnicity, antidi-\\nabetic medication use, presence of hypertension, DR, higher levels of systolic BP , HbA1c, and lower \\nlevels of eGFR. Contrary to conventional risk factors, sex, BMI, and duration of diabetes did not come \\nin the top 15- predictors.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 8, 'page_label': '9', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='levels of eGFR. Contrary to conventional risk factors, sex, BMI, and duration of diabetes did not come \\nin the top 15- predictors.\\nOur results showed that ML models combined with feature selection improved the accuracy for \\npredicting incident DKD in high- dimensional datasets. The AUC of MLs based on dataset E including \\nmetabolites (+risk factors) scored the highest, while the one based on dataset D including genetic \\nfeatures scored the lowest compared to other domain features. This finding suggests that modifiable \\nrisk factors and metabolites predict DKD risk better than genetic features. The predictive performance \\nwas the best by EN, followed by LASSO and GBDT. The top 15 predictors selected by LASSO and \\nGBDT were largely consistent to that by EN.\\nFew previous studies have evaluated the performance of ML models for predicting the risk of \\nincident DKD (Table\\xa02). Ravizza et al. identified seven key features (age, BMI, eGFR, concentration of'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 8, 'page_label': '9', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Few previous studies have evaluated the performance of ML models for predicting the risk of \\nincident DKD (Table\\xa02). Ravizza et al. identified seven key features (age, BMI, eGFR, concentration of \\ncreatinine, glucose, albumin, and HbA1c%) by a data- driven feature selection strategy for predicting \\nDKD using electronic health records (EHR) data from 417,912 people with diabetes retrieved from \\nthe IBM Explorys Database and developed a random forest model in 82,912 people with diabetes \\nretrieved from the Indiana Network for Patient Care (INPC). The RF algorithm using seven prioritized \\nkey features achieved an AUC of 0.833 compared to 0.827 by LR (Ravizza et\\xa0al., 2019).\\nSong et al. predicted 1- year risk of DKD based on electronic health records (EHR) data using \\ngradient boosting machine (GBM) algorithm with an AUC of 83% (Song et\\xa0al., 2020). As the median \\nduration of development of DKD was\\xa0~10\\xa0y since the onset of diabetes, predicting 1- year risk may'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 8, 'page_label': '9', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='gradient boosting machine (GBM) algorithm with an AUC of 83% (Song et\\xa0al., 2020). As the median \\nduration of development of DKD was\\xa0~10\\xa0y since the onset of diabetes, predicting 1- year risk may \\nnot be sufficient. Huang et al. predicted DKD risk in 1838 adults with diabetes and prediabetes who \\nparticipated in the KORA study in Germany. The authors used ML models SVM, RF , and Ada Boost \\nbased on 14 clinical factors and 125 metabolites. The best achieved AUC was 0.857, which is similar \\nto that of our model using EN (AUC = 0.851) (Huang et\\xa0al., 2020a).\\nTable 2. Machine learning model for predicting incident CKD in literature.\\nAuthor, journal\\nStudy cohort,\\ncountry\\nStudy population\\nFollow- up\\nCKD definition \\nand incidence\\nNumber of \\npredictors ML performance\\nRavizza et\\xa0al., 2019, \\nNature Medicine\\nEHR data from the \\nIBM Explorys and \\nINPC datasets, the \\nUnited States\\nDevelopment cohort (IBM): \\n>500,000 adults with diabetes. \\nValidation (INPC) = 82,912'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 8, 'page_label': '9', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Ravizza et\\xa0al., 2019, \\nNature Medicine\\nEHR data from the \\nIBM Explorys and \\nINPC datasets, the \\nUnited States\\nDevelopment cohort (IBM): \\n>500,000 adults with diabetes. \\nValidation (INPC) = 82,912 \\nadults with T2DM; FU = 3\\xa0y. ICD 9/10 codes 300 features\\nBased on seven prioritized \\nfeatures, AUC by RF = \\n0.833\\xa0and the Roche/IBM \\nsupervised algorithm by LR = \\n0.827\\nSong et\\xa0al., 2020., \\nJMIR\\nEHR data, the United \\nStates (2007–2017)\\n14,039 adults with T2DM.\\nFU = 1\\xa0y.\\neGFR\\xa0< 60\\xa0or \\nUACR\\xa0≥30\\xa0mg/g;\\n34.1% >3000\\nGBM\\nAUC = 0.83\\nHuang et\\xa0al., 2020a., \\nDiabetes\\nKORA cohort, \\nGermany\\n1838 adults with prediabetes \\nand T2DM.\\nFU = 6.5 y.\\neGFR\\xa0< 60\\xa0or \\nUACR\\xa0≥30\\xa0mg/g \\nat FU;\\n10.9%\\n125 mets + 14 \\nclinical factors\\nSVM, RF , Ada Boost\\nBest set: Mets- SM and PC\\xa0+ \\nage, TC, FPG, eGFR, UACR, \\nAUC = 0.857\\nTraditional LR using 14 \\nvariables, AUC = 0.809\\nSabanayagam et al., \\n2023, current study SEED population data, \\nSingapore\\n1365 adults with diabetes.\\nFU = 6\\xa0y.\\neGFR\\xa0< 60 + 25% \\ndecline in eGFR'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 8, 'page_label': '9', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Traditional LR using 14 \\nvariables, AUC = 0.809\\nSabanayagam et al., \\n2023, current study SEED population data, \\nSingapore\\n1365 adults with diabetes.\\nFU = 6\\xa0y.\\neGFR\\xa0< 60 + 25% \\ndecline in eGFR \\nfrom baseline 339 features\\nEN + RFE selected 15 features, \\nAUC = 0.851 vs. 0.795 using \\nseven features by traditional LR\\nAUC, area under the receiver operating characteristic curve; CKD, chronic kidney disease; eGFR, estimated glomerular filtration rate; EHR, electronic \\nhealth records; EN, Elastic Net; FPG, Fasting plasma glucose; FU, follow- up; GBM, Gradient Boosting Machine; ICD, International Classification of \\nDiseases; INPC, Indiana Network for Patient Care; LR, logistic regression; ML, machine learning; RF , random forest; RFE, recursive feature selection; \\nSEED, Singapore Epidemiology of Eye Diseases; SVM, support vector machine; T2DM, type 2 diabetes mellitus; TC, total cholesterol; UACR, urine \\nalbumin- creatinine ratio.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 9, 'page_label': '10', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  10 of 14\\nIn the current study, we observed that when the features were limited to the traditional risk factors, \\nthe performance of LR was similar to that of the best ML model EN. However, when the number \\nof features was large, LR’s performance dropped significantly compared to the top- performing ML \\nmodels, including EN, LASSO, and GBDT. While ML models are capable of addressing complex vari-\\nable effects and nonlinearity issues, herein we found regularized regression models (EN and LASSO) \\noutperforming other more sophisticated models. This suggests that the pathophysiological progres-\\nsion from diabetes to CKD may not be as nonlinear as previously assumed. At CKD stage 4, heteroge-\\nneity in terms of disease trajectory is primarily low, and nonadditive effects are likely to be negligible,'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 9, 'page_label': '10', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='neity in terms of disease trajectory is primarily low, and nonadditive effects are likely to be negligible, \\nwhich could explain the superior performance of EN and LASSO in our study. In a previous study \\nbased on the same dataset as the current study, Nusinovici et al. tested the performance of several \\nML models utilizing 20 risk factors alone and found that the performance of LR (AUC = 0.905) was \\nsimilar to that of the best ML model, GBDT (AUC = 0.903), for predicting incident CKD in those with \\nand without diabetes (Nusinovici et\\xa0al., 2020). When a large number of features are present, more \\nadvanced ML methods may capture the complex functional dependency of the incident CKD outcome \\nmuch better than the linear approach used in LR.\\nThe top- performing ML models (EN, LASSO, GBDT) identified established risk factors for DKD, such \\nas age, ethnicity, antidiabetic medication use, presence of hypertension, DR, higher levels of systolic'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 9, 'page_label': '10', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='as age, ethnicity, antidiabetic medication use, presence of hypertension, DR, higher levels of systolic \\nBP , HbA1c, and lower levels of eGFR. Additionally, antihypertensive medication use and low housing \\ntype were identified by LASSO while BMI and duration of diabetes by GBDT. Increasing age, longer \\nduration of diabetes, higher levels of HbA1c, systolic BP/hypertension are well- known risk factors of \\nDKD. A meta- analysis conducted by Nelson et al., including 15 multinational cohorts with diabetes as \\npart of the CKD Prognosis Consortium (Nelson et\\xa0al., 2019), also identified older age, hypertension, \\nlower eGFR, higher levels of BMI, HbA1c, and antidiabetic medication use as significant risk factors \\nfor incident CKD in those with diabetes. While black ethnicity was a risk factor for CKD in the meta- \\nanalysis, in our study, we found Chinese and Malay ethnicity to be at higher risk of developing incident'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 9, 'page_label': '10', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='analysis, in our study, we found Chinese and Malay ethnicity to be at higher risk of developing incident \\nDKD compared to Indian ethnicity. Compared to the Indian population, the Chinese population was \\ngenerally older, while Malay participants had lower education levels and higher prevalence of smoking \\n(Supplementary file 1 -Table 1g). Both Chinese and Malays had a higher prevalence of hypertension \\nand lower levels of antidiabetic medication use, which may contribute to the ethnic difference in DKD \\nincidence. Another reason for the lower risk of developing DKD in the Indian ethnicity could be that \\nas a high- risk group for diabetes they may be more aware of the risk and comply with screening, medi-\\ncation, and other measures that can reduce their risk of developing DKD. Malay ethnicity has been \\nidentified to be a high- risk group for CKD by several studies conducted in Singapore (Sabanayagam'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 9, 'page_label': '10', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='cation, and other measures that can reduce their risk of developing DKD. Malay ethnicity has been \\nidentified to be a high- risk group for CKD by several studies conducted in Singapore (Sabanayagam \\net\\xa0al., 2010; Lim et\\xa0al., 2021; National Registry of Diseases Office, 2020). Surprisingly, sex was not \\nidentified to be a risk factor by any of the three\\xa0ML models, which is consistent with the findings of the \\nRavizza et al. algorithm based on data- driven feature selection that did not pick up sex as one of the \\npriority features (Ravizza et\\xa0al., 2019).\\nIn the current study, several new predictors from the metabolites domain were identified. Specifi-\\ncally, lipid metabolites, including phospholipids in HDL and VLDL subclasses, cholesterol esters, and \\nfree cholesterol in HDL subclasses, were found to be associated with an increased risk of DKD, while \\ncholesterol esters in IDL were found to be protective against DKD. Furthermore, higher levels of DHA,'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 9, 'page_label': '10', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='cholesterol esters in IDL were found to be protective against DKD. Furthermore, higher levels of DHA, \\nacetate, and tyrosine also showed a protective association (odds ratios not shown). These findings \\nare consistent with previous studies. For example, in the ADVANCE trial, higher tyrosine levels were \\nassociated with an increased risk of microvascular complications in diabetic participants. DHA, a n- 3 \\npolyunsaturated fatty acid (PUFA), has been shown to reduce renal inflammation and fibrosis and slow \\ndown the progression of CKD in animal models with type 2 diabetes (Mathi Thumilan et\\xa0al., 2016). \\nPUFA supplementation has also been shown to reduce hyperglycemia- induced pathogenic mech-\\nanisms by its anti- inflammatory and antioxidant properties, and improve renal function in diabetic \\nnephropathy patients . Furthermore, higher levels of the short- chain fatty acid acetate have been'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 9, 'page_label': '10', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='nephropathy patients . Furthermore, higher levels of the short- chain fatty acid acetate have been \\nshown to be inversely associated with diabetic nephropathy in type 2 diabetic patients (Huang et\\xa0al., \\n2020b), and have beneficial effects in mice models with type 2 diabetes by reducing oxidative stress \\nand inflammation.\\nThe strengths of our study include a multiethnic Asian population with a long follow- up and the \\navailability of a wealth of information. The use of RFE for dimension reduction and feature selection \\nreduced overfitting of data. ML models identified the relative importance of one domain over the'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 10, 'page_label': '11', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  11 of 14\\nother domains (like metabolite features in our study compared to genetic features) and the best \\npredictors within one domain. However, our study results should be interpreted in light of a few \\nlimitations. First, our definition of DKD was based on the measurement of a single blood creatinine \\nboth at baseline and follow- up. This could have resulted in some misclassification, but the bias would \\nbe nondifferential and similar across both outcomes. Second, albuminuria, an important predictor of \\nDKD, was not included as it was missing in a substantial number of participants. Third, external vali-\\ndation was not performed. Fourth, ML models are computationally intensive compared to traditional \\nregression models.\\nIn conclusion, in a population- based sample of multiethnic Asian adults, we found that EN with'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 10, 'page_label': '11', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='regression models.\\nIn conclusion, in a population- based sample of multiethnic Asian adults, we found that EN with \\nspecific metabolites outperformed the current DKD risk prediction models using demographic and \\nclinical variables. Our results provide evidence that combining metabolites and ML models could \\nimprove the prediction accuracy for DKD and that increasing the use of ML techniques may discover \\nnew risk factors for DKD. Further testing in external populations would support the validity of the \\nmodel.\\nAdditional information\\nFunding\\nFunder Grant reference number Author\\nSingapore Ministry of \\nHealth’s National Medical \\nResearch Council\\nNMRC/OFLCG/MOH-\\n001327-03\\nGavin Tan\\nSingapore Ministry of \\nHealth’s National Medical \\nResearch Council\\nNMRC/HCSAINV/MOH-\\n001019-00\\nCharumathi Sabanayagam\\nThe funders had no role in study design, data collection and interpretation, or the \\ndecision to submit the work for publication.\\nAuthor contributions'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 10, 'page_label': '11', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='001019-00\\nCharumathi Sabanayagam\\nThe funders had no role in study design, data collection and interpretation, or the \\ndecision to submit the work for publication.\\nAuthor contributions\\nCharumathi Sabanayagam, Conceptualization, Data curation, Supervision, Funding acquisition, \\nMethodology, Writing – original draft, Writing – review and editing; Feng He, Data curation, Soft-\\nware, Formal analysis, Methodology; Simon Nusinovici, Cynthia Lim, Gavin Tan, Writing – review and \\nediting; Jialiang Li, Supervision, Writing – review and editing; Ching Yu Cheng, Funding acquisition, \\nWriting – review and editing\\nAuthor ORCIDs\\nCharumathi Sabanayagam  \\n   https://orcid.org/0000-0002-4042-4719\\nFeng He  \\n   http://orcid.org/0000-0002-4717-2243\\nEthics\\nSEED was conducted in accordance with the Declaration of Helsinki and was approved by the Sing-\\nHealth Centralised Institutional Review Board [2018/2717, 2018/ 2921, 2012/487/A, 2015/2279,'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 10, 'page_label': '11', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Ethics\\nSEED was conducted in accordance with the Declaration of Helsinki and was approved by the Sing-\\nHealth Centralised Institutional Review Board [2018/2717, 2018/ 2921, 2012/487/A, 2015/2279, \\n2018/2006, 2018/2594, 2018/2570]. Informed consent was obtained from all participants.\\nDecision letter and Author response\\nDecision letter https://doi.org/10.7554/eLife.81878.sa1\\nAuthor response https://doi.org/10.7554/eLife.81878.sa2\\nAdditional files\\nSupplementary files\\n•  Supplementary file 1. Description of variables in each dataset and results from supplementary \\nanalyses (Table 1a- 1g). (Table 1a) Characteristics of the preprocessed datasets A–F . (Table 1b) List \\nof variables used for DKD prediction. (Table 1c) Median AUC [IQR] performance of the ML models \\nusing datasets A–F . (Table 1d) Median SN%/SP% performance of the ML models using datasets A–F .'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 11, 'page_label': '12', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  12 of 14\\n(Table 1e) Source data linked to Figure\\xa03. (Table 1f) Top ML- selected predictors for incident DKD in \\neach of the three ethnic groups by EN and RFE. (Table 1g) Baseline characteristics of SEED diabetic \\nparticipants by ethnicity (n = 1365).\\n•  MDAR checklist \\nData availability\\nThe authors declare that the data supporting the findings of this study are available within the article \\nand its supplementary materials. The individual participant- level data cannot be shared publicly due \\nto privacy and confidentiality concerns. However, de- identified data may be available upon request \\nwith a project proposal to qualified individuals, subject to approval by the Singapore Eye Research \\nInstitute and access will require a data sharing agreement. Interested researchers can send data access'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 11, 'page_label': '12', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Institute and access will require a data sharing agreement. Interested researchers can send data access \\nrequests to Prof. Ching- Yu Cheng at the Singapore Eye Research Institute using the following email \\naddress:  cheng. ching. yu@ seri. com. sg. Processed version of the datasets are provided in Supplemen-\\ntary file 1- Table 1a- 1g.\\nReferences\\nAlicic RZ, Rooney MT, Tuttle KR. 2017. Diabetic Kidney Disease: challenges, progress, and possibilities. Clinical \\nJournal of the American Society of Nephrology 12:2032–2045. DOI: https://doi.org/10.2215/CJN.11491116, \\nPMID: 28522654\\nBi Q, Goodman KE, Kaminsky J, Lessler J. 2019. What is machine learning? A primer for the epidemiologist. \\nAmerican Journal of Epidemiology 188:2222–2239. DOI: https://doi.org/10.1093/aje/kwz189, PMID: 31509183\\nChong YH, Fan Q, Tham YC, Gan A, Tan SP , Tan G, Wang JJ, Mitchell P , Wong TY , Cheng CY . 2017. Type 2'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 11, 'page_label': '12', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Chong YH, Fan Q, Tham YC, Gan A, Tan SP , Tan G, Wang JJ, Mitchell P , Wong TY , Cheng CY . 2017. Type 2 \\ndiabetes genetic variants and risk of diabetic retinopathy. Ophthalmology 124:336–342. DOI: https://doi.org/ \\n10.1016/j.ophtha.2016.11.016, PMID: 28038984\\nCole JB, Florez JC. 2020. Genetics of diabetes mellitus and diabetes complications. Nature Reviews. Nephrology \\n16:377–390. DOI: https://doi.org/10.1038/s41581-020-0278-5, PMID: 32398868\\nColhoun HM, Marcovecchio ML. 2018. Biomarkers of diabetic kidney disease. Diabetologia 61:996–1011. DOI: \\nhttps://doi.org/10.1007/s00125-018-4567-5, PMID: 29520581\\nCouser WG, Remuzzi G, Mendis S, Tonelli M. 2011. The contribution of chronic kidney disease to the global \\nburden of major noncommunicable diseases. Kidney International 80:1258–1270. DOI: https://doi.org/10.1038/ \\nki.2011.368, PMID: 21993585\\nDoupe P, Faghmous J, Basu S. 2019. Machine learning for health services researchers. Value in Health 22:808–'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 11, 'page_label': '12', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='ki.2011.368, PMID: 21993585\\nDoupe P, Faghmous J, Basu S. 2019. Machine learning for health services researchers. Value in Health 22:808–\\n815. DOI: https://doi.org/10.1016/j.jval.2019.02.012, PMID: 31277828\\nDunkler D, Gao P , Lee SF , Heinze G, Clase CM, Tobe S, Teo KK, Gerstein H, Mann JFE, Oberbauer R, \\nONTARGET and ORIGIN Investigators. 2015. Risk Prediction for Early CKD in Type 2 Diabetes. Clinical Journal \\nof the American Society of Nephrology 10:1371–1379. DOI: https://doi.org/10.2215/CJN.10321014, PMID: \\n26175542\\nEddy S, Mariani LH, Kretzler M. 2020. Integrated multi- omics approaches to improve classification of chronic \\nkidney disease. Nature Reviews. Nephrology 16:657–668. DOI: https://doi.org/10.1038/s41581-020-0286-5, \\nPMID: 32424281\\nFoong AWP, Saw SM, Loo JL, Shen S, Loon SC, Rosman M, Aung T, Tan DTH, Tai ES, Wong TY . 2007. Rationale \\nand methodology for a population- based study of eye diseases in Malay people: The Singapore Malay eye'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 11, 'page_label': '12', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='and methodology for a population- based study of eye diseases in Malay people: The Singapore Malay eye \\nstudy (SiMES). Ophthalmic Epidemiology 14:25–35. DOI: https://doi.org/10.1080/09286580600878844, PMID: \\n17365815\\nHastie T, Tibshirani R, Friedman J. 2009. The elements of statistical learning. The Elements of Statistical \\nLearning: Data Mining, Inference, and Prediction New York, NY: Springer. DOI: https://doi.org/10.1007/978-0- \\n387-84858-7\\nHill NR, Fatoba ST, Oke JL, Hirst JA, O’Callaghan CA, Lasserson DS, Hobbs FDR. 2016. Global Prevalence of \\nChronic Kidney Disease - a systematic review and meta- analysis. PLOS ONE 11:e0158765. DOI: https://doi. \\norg/10.1371/journal.pone.0158765, PMID: 27383068\\nHirst JA, Hill N, O’Callaghan CA, Lasserson D, McManus RJ, Ogburn E, Mena JMO, Shine B, Taylor CJ, \\nVazquez- Montes MD, Yang Y , Hobbs FR. 2020. Prevalence of chronic kidney disease in the community using'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 11, 'page_label': '12', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Vazquez- Montes MD, Yang Y , Hobbs FR. 2020. Prevalence of chronic kidney disease in the community using \\ndata from OxRen: a UK population- based cohort study. The British Journal of General Practice 70:e285–e293. \\nDOI: https://doi.org/10.3399/bjgp20X708245, PMID: 32041766\\nHuang J, Huth C, Covic M, Troll M, Adam J, Zukunft S, Prehn C, Wang L, Nano J, Scheerer MF , Neschen S, \\nKastenmüller G, Suhre K, Laxy M, Schliess F , Gieger C, Adamski J, Hrabe de Angelis M, Peters A, \\nWang- Sattler R. 2020a. Machine learning approaches reveal metabolic signatures of incident chronic kidney \\ndisease in individuals with prediabetes and type 2 diabetes. Diabetes 69:2756–2765. DOI: https://doi.org/10. \\n2337/db20-0586, PMID: 33024004\\nHuang W, Man Y , Gao C, Zhou L, Gu J, Xu H, Wan Q, Long Y , Chai L, Xu Y , Xu Y . 2020b. Short- Chain Fatty Acids \\nAmeliorate Diabetic Nephropathy via GPR43- Mediated inhibition of Oxidative Stress and NF-κB Signaling.'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 11, 'page_label': '12', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Ameliorate Diabetic Nephropathy via GPR43- Mediated inhibition of Oxidative Stress and NF-κB Signaling. \\nOxidative Medicine and Cellular Longevity 2020:4074832. DOI: https://doi.org/10.1155/2020/4074832, PMID: \\n32831998'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 12, 'page_label': '13', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  13 of 14\\nHussain S, Habib A, Najmi AK. 2019. Limited knowledge of Chronic Kidney Disease among Type 2 Diabetes \\nMellitus Patients in India. International Journal of Environmental Research and Public Health 16:1443. DOI: \\nhttps://doi.org/10.3390/ijerph16081443, PMID: 31018581\\nJiang W, Wang J, Shen X, Lu W, Wang Y , Li W, Gao Z, Xu J, Li X, Liu R, Zheng M, Chang B, Li J, Yang J, Chang B. \\n2020. Establishment and validation of a risk prediction model for early diabetic kidney disease based on a \\nsystematic review and meta- analysis of 20 cohorts. Diabetes Care 43:925–933. DOI: https://doi.org/10.2337/ \\ndc19-1897, PMID: 32198286\\nLavanya R, Jeganathan VSE, Zheng Y , Raju P , Cheung N, Tai ES, Wang JJ, Lamoureux E, Mitchell P , Young TL, \\nCajucom- Uy H, Foster PJ, Aung T, Saw SM, Wong TY . 2009. Methodology of the Singapore Indian Chinese'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 12, 'page_label': '13', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Cajucom- Uy H, Foster PJ, Aung T, Saw SM, Wong TY . 2009. Methodology of the Singapore Indian Chinese \\nCohort (SICC) eye study: quantifying ethnic variations in the epidemiology of eye diseases in Asians. \\nOphthalmic Epidemiology 16:325–336. DOI: https://doi.org/10.3109/09286580903144738, PMID: 19995197\\nLevey AS, Stevens LA, Schmid CH, Zhang YL, Castro AF , Feldman HI, Kusek JW, Eggers P , Van Lente F , Greene T, \\nCoresh J, CKD- EPI (Chronic Kidney Disease Epidemiology Collaboration). 2009. A new equation to estimate \\nglomerular filtration rate. Annals of Internal Medicine 150:604–612. DOI: https://doi.org/10.7326/0003-4819- \\n150-9-200905050-00006, PMID: 19414839\\nLim CC, He F , Li J, Tham YC, Tan CS, Cheng CY , Wong TY , Sabanayagam C. 2021. Application of machine \\nlearning techniques to understand ethnic differences and risk factors for incident chronic kidney disease in \\nAsians. BMJ Open Diabetes Research & Care 9:e002364. DOI: https://doi.org/10.1136/bmjdrc-2021-002364,'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 12, 'page_label': '13', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Asians. BMJ Open Diabetes Research & Care 9:e002364. DOI: https://doi.org/10.1136/bmjdrc-2021-002364, \\nPMID: 34952839\\nMajithia S, Tham YC, Chee ML, Nusinovici S, Teo CL, Chee ML, Thakur S, Soh ZD, Kumari N, Lamoureux E, \\nSabanayagam C, Wong TY , Cheng CY . 2021. Corrigendum to: Cohort profile: the Singapore Epidemiology of \\nEye Diseases Study (SEED). International Journal of Epidemiology 50:1401. DOI: https://doi.org/10.1093/ije/ \\ndyab114, PMID: 34179972\\nManski- Nankervis JAE, Thuraisingam S, Lau P , Blackberry I, Sluggett JK, Ilomaki J, Bell JS, Furler J. 2018. \\nScreening and diagnosis of chronic kidney disease in people with type 2 diabetes attending Australian general \\npractice. Australian Journal of Primary Health 24:280–286. DOI: https://doi.org/10.1071/PY17156, PMID: \\n29807557\\nMathi Thumilan B, Sajeevan RS, Biradar J, Madhuri T, N Nataraja K, Sreeman SM. 2016. Development and'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 12, 'page_label': '13', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='29807557\\nMathi Thumilan B, Sajeevan RS, Biradar J, Madhuri T, N Nataraja K, Sreeman SM. 2016. Development and \\ncharacterization of genic SSR markers from Indian Mulberry transcriptome and their transferability to related \\nSpecies of Moraceae. PLOS ONE 11:e0162909. DOI: https://doi.org/10.1371/journal.pone.0162909, PMID: \\n27669004\\nNational Registry of Diseases Office. 2020. Singapore Renal Registry Annual Report 2020. https://www.nrdo. \\ngov.sg/docs/librariesprovider3/default-document-library/srr-annual-report-2020.pdf?sfvrsn=658dd717_0 \\n[Accessed December 30, 2022].\\nNelson RG, Grams ME, Ballew SH, Sang Y , Azizi F , Chadban SJ, Chaker L, Dunning SC, Fox C, Hirakawa Y , \\nIseki K, Ix J, Jafar TH, Köttgen A, Naimark DMJ, Ohkubo T, Prescott GJ, Rebholz CM, Sabanayagam C, \\nSairenchi T, et\\xa0al. 2019. Development of risk prediction equations for incident chronic kidney disease. JAMA \\n322:2104–2114. DOI: https://doi.org/10.1001/jama.2019.17379, PMID: 31703124'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 12, 'page_label': '13', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Sairenchi T, et\\xa0al. 2019. Development of risk prediction equations for incident chronic kidney disease. JAMA \\n322:2104–2114. DOI: https://doi.org/10.1001/jama.2019.17379, PMID: 31703124\\nNusinovici S, Tham YC, Chak Yan MY , Wei Ting DS, Li J, Sabanayagam C, Wong TY , Cheng C- Y . 2020. Logistic \\nregression was as good as machine learning for predicting major chronic diseases. Journal of Clinical \\nEpidemiology 122:56–69. DOI: https://doi.org/10.1016/j.jclinepi.2020.03.002, PMID: 32169597\\nOgurtsova K, da Rocha Fernandes JD, Huang Y , Linnenkamp U, Guariguata L, Cho NH, Cavan D, Shaw JE, \\nMakaroff LE. 2017. IDF Diabetes Atlas: Global estimates for the prevalence of diabetes for 2015 and 2040. \\nDiabetes Research and Clinical Practice 128:40–50. DOI: https://doi.org/10.1016/j.diabres.2017.03.024, PMID: \\n28437734\\nQuek DQY, He F , Sultana R, Banu R, Chee ML, Nusinovici S, Thakur S, Qian C, Cheng CY , Wong TY ,'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 12, 'page_label': '13', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='28437734\\nQuek DQY, He F , Sultana R, Banu R, Chee ML, Nusinovici S, Thakur S, Qian C, Cheng CY , Wong TY , \\nSabanayagam C. 2021. Novel serum and urinary metabolites associated with diabetic retinopathy in three asian \\ncohorts. Metabolites 11:614. DOI: https://doi.org/10.3390/metabo11090614, PMID: 34564429\\nRavizza S, Huschto T, Adamov A, Böhm L, Büsser A, Flöther FF , Hinzmann R, König H, McAhren SM, \\nRobertson DH, Schleyer T, Schneidinger B, Petrich W. 2019. Predicting the early risk of chronic kidney disease \\nin patients with diabetes using real- world data. Nature Medicine 25:57–59. DOI: https://doi.org/10.1038/ \\ns41591-018-0239-8, PMID: 30617317\\nSabanayagam C, Lim SC, Wong TY , Lee J, Shankar A, Tai ES. 2010. Ethnic disparities in prevalence and impact of \\nrisk factors of chronic kidney disease. Nephrology, Dialysis, Transplantation 25:2564–2570. DOI: https://doi. \\norg/10.1093/ndt/gfq084, PMID: 20185856'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 12, 'page_label': '13', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='risk factors of chronic kidney disease. Nephrology, Dialysis, Transplantation 25:2564–2570. DOI: https://doi. \\norg/10.1093/ndt/gfq084, PMID: 20185856\\nSabanayagam C, Chee ML, Banu R, Cheng CY , Lim SC, Tai ES, Coffman T, Wong TY . 2019. Association of \\ndiabetic retinopathy and diabetic kidney disease with all- cause and cardiovascular mortality in a Multiethnic \\nAsian Population. JAMA Network Open 2:e191540. DOI: https://doi.org/10.1001/jamanetworkopen.2019. \\n1540, PMID: 30924904\\nSong X, Waitman LR, Yu AS, Robbins DC, Hu Y , Liu M. 2020. Longitudinal risk prediction of chronic kidney \\ndisease in diabetic patients using a temporal- enhanced gradient boosting machine: retrospective cohort study. \\nJMIR Medical Informatics 8:e15510. DOI: https://doi.org/10.2196/15510, PMID: 32012067\\nStevens PE, Levin A, Kidney Disease: Improving Global Outcomes Chronic Kidney Disease Guideline \\nDevelopment Work Group Members. 2013. Evaluation and management of chronic kidney disease: synopsis of'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 12, 'page_label': '13', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Development Work Group Members. 2013. Evaluation and management of chronic kidney disease: synopsis of \\nthe kidney disease: improving global outcomes 2012 clinical practice guideline. Annals of Internal Medicine \\n158:825–830. DOI: https://doi.org/10.7326/0003-4819-158-11-201306040-00007, PMID: 23732715'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 13, 'page_label': '14', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='Research article      \\nEpidemiology and Global Health\\nSabanayagam et\\xa0al. eLife 2023;12:e81878. DOI: https://doi.org/10.7554/eLife.81878  14 of 14\\nSundström J, Schön TB. 2020. Machine learning in risk prediction. Hypertension 75:1165–1166. DOI: https://doi. \\norg/10.1161/HYPERTENSIONAHA.120.13516, PMID: 32172625\\nWebster AC, Nagler EV, Morton RL, Masson P . 2017. Chronic Kidney Disease. Lancet 389:1238–1252. DOI: \\nhttps://doi.org/10.1016/S0140-6736(16)32064-5, PMID: 27887750\\nWhite SL, Polkinghorne KR, Cass A, Shaw J, Atkins RC, Chadban SJ. 2008. Limited knowledge of kidney disease \\nin a survey of AusDiab study participants. The Medical Journal of Australia 188:204–208. DOI: https://doi.org/ \\n10.5694/j.1326-5377.2008.tb01585.x, PMID: 18279125\\nYau JWY, Xie J, Kawasaki R, Kramer H, Shlipak M, Klein R, Klein B, Cotch MF , Wong TY . 2011. Retinal arteriolar \\nnarrowing and subsequent development of CKD Stage 3: the Multi- Ethnic Study of Atherosclerosis (MESA).'),\n",
              " Document(metadata={'producer': 'Adobe PDF Library 15.0', 'creator': 'Adobe InDesign CC 14.0 (Windows)', 'creationdate': '2024-01-04T09:46:05+00:00', 'moddate': '2024-01-04T09:46:06+00:00', 'trapped': '/False', 'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c', 'source': '/content/drive/MyDrive/Rag_Data/81878.pdf', 'total_pages': 14, 'page': 13, 'page_label': '14', 'source_file': '81878.pdf', 'file_type': 'pdf'}, page_content='narrowing and subsequent development of CKD Stage 3: the Multi- Ethnic Study of Atherosclerosis (MESA). \\nAmerican Journal of Kidney Diseases 58:39–46. DOI: https://doi.org/10.1053/j.ajkd.2011.02.382, PMID: \\n21549464\\nYip W, Ong PG, Teo BW, Cheung CYL, Tai ES, Cheng CY , Lamoureux E, Wong TY , Sabanayagam C. 2017. Retinal \\nvascular imaging markers and incident chronic kidney disease: a prospective cohort study. Scientific Reports \\n7:9374. DOI: https://doi.org/10.1038/s41598-017-09204-2, PMID: 28839244')]"
            ]
          },
          "metadata": {},
          "execution_count": 23
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "### Convert the text to embeddings\n",
        "texts=[doc.page_content for doc in chunks]\n",
        "## Generate the Embeddings\n",
        "embeddings=embedding_manager.generate_embeddings(texts)\n",
        "##store int he vector dtaabase\n",
        "vectorstore.add_documents(chunks,embeddings)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 136,
          "referenced_widgets": [
            "d03d5b9dd8b24318b69b543ad37d1fd4",
            "830cf1fd79e5453781dc6c0ca9fcbaf5",
            "6f4fce1b24b443aaaed222ca0ca11b64",
            "63c578dd205f493b8c2967684303d7f0",
            "a70ea477ceb8437a9142a4d4a16a0d10",
            "35ec037126ab4d318500b75e23eb9c60",
            "eb433b46efb745beb06bcf8a2222879b",
            "6b6c695cd465489aa1e08493ee5fb1d1",
            "8be2370b12f648f5ac582e61ae806cd8",
            "7ff9edd14df345bf9489fdc8f2dd03fe",
            "cc9a08733610408b81be5a882112fe5a"
          ]
        },
        "id": "imSfOmCoZtMw",
        "outputId": "fbecb569-9428-41dc-9c5a-0966353d8f12"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Generating embeddings for 413 texts....\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Batches:   0%|          | 0/13 [00:00<?, ?it/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "d03d5b9dd8b24318b69b543ad37d1fd4"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Embeddings generated successfully. Shape: (413, 384)\n",
            "Adding 413 documents to vector store...\n",
            "Successfully added 413 documents to vector store\n",
            "Total documents in collection: 413\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "class RAGRetriever:\n",
        "    \"\"\"Handles query-based retrieval from the vector store\"\"\"\n",
        "\n",
        "    def __init__(self, vector_store: VectorStore, embedding_manager: EmbeddingManager):\n",
        "        \"\"\"\n",
        "        Initialize the retriever\n",
        "\n",
        "        Args:\n",
        "            vector_store: Vector store containing document embeddings\n",
        "            embedding_manager: Manager for generating query embeddings\n",
        "        \"\"\"\n",
        "        self.vector_store = vector_store\n",
        "        self.embedding_manager = embedding_manager\n",
        "\n",
        "    def retrieve(self, query: str, top_k: int = 5, score_threshold: float = 0.0) -> List[Dict[str, Any]]:\n",
        "        \"\"\"\n",
        "        Retrieve relevant documents for a query\n",
        "\n",
        "        Args:\n",
        "            query: The search query\n",
        "            top_k: Number of top results to return\n",
        "            score_threshold: Minimum similarity score threshold\n",
        "\n",
        "        Returns:\n",
        "            List of dictionaries containing retrieved documents and metadata\n",
        "            \"\"\"\n",
        "        print(f\"Retrieving documents for query: '{query}'\")\n",
        "        print(f\"Top K: {top_k}, Score threshold: {score_threshold}\")\n",
        "\n",
        "        # Generate query embedding\n",
        "        query_embedding = self.embedding_manager.generate_embeddings([query])[0]\n",
        "\n",
        "        # Search in vector store\n",
        "        try:\n",
        "            results = self.vector_store.collection.query(\n",
        "                query_embeddings=[query_embedding.tolist()],\n",
        "                n_results=top_k\n",
        "            )\n",
        "\n",
        "            # Process results\n",
        "            retrieved_docs = []\n",
        "\n",
        "            if results['documents'] and results['documents'][0]:\n",
        "                documents = results['documents'][0]\n",
        "                metadatas = results['metadatas'][0]\n",
        "                distances = results['distances'][0]\n",
        "                ids = results['ids'][0]\n",
        "\n",
        "                for i, (doc_id, document, metadata, distance) in enumerate(zip(ids, documents, metadatas, distances)):\n",
        "                    # Convert distance to similarity score (ChromaDB uses cosine distance)\n",
        "                    similarity_score = 1 - distance\n",
        "\n",
        "                    if similarity_score >= score_threshold:\n",
        "                      retrieved_docs.append({\n",
        "                            'id': doc_id,\n",
        "                            'content': document,\n",
        "                            'metadata': metadata,\n",
        "                            'similarity_score': similarity_score,\n",
        "                            'distance': distance,\n",
        "                            'rank': i + 1\n",
        "                        })\n",
        "\n",
        "                print(f\"Retrieved {len(retrieved_docs)} documents (after filtering)\")\n",
        "            else:\n",
        "                print(\"No documents found\")\n",
        "\n",
        "            return retrieved_docs\n",
        "\n",
        "        except Exception as e:\n",
        "            print(f\"Error during retrieval: {e}\")\n",
        "            return []\n",
        "\n",
        "rag_retriever=RAGRetriever(vectorstore,embedding_manager)"
      ],
      "metadata": {
        "id": "NBe5ITXwZ_E2"
      },
      "execution_count": 25,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "rag_retriever"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YQo6T8oubLBD",
        "outputId": "17815469-df06-4347-c82d-7c0826927da4"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<__main__.RAGRetriever at 0x7c94e81aba40>"
            ]
          },
          "metadata": {},
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "rag_retriever.retrieve(\"What is a Chronic Kidney Disease?\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000,
          "referenced_widgets": [
            "d16cb3fb9499455f88cb89b3efc0c1c5",
            "1670f5066d02400891f787719ec1b86f",
            "444b866babf045869bc004b41bfb604a",
            "51cc30baefac4a94b00c30432c1f012f",
            "70fa290b0f2142c58bbd3e38bb1337eb",
            "eca118b4e9a44faeb243d0ef6369ea04",
            "f743bddb5e6d4832b76d6ada022758d2",
            "18e899bf051f457390875f54986ca345",
            "534bb57c4abc440b95ef3f59d4b4ecca",
            "f4a8e387a07d411f83ce23138a5516e2",
            "9d7b50d40d0d4f1ba3523aff1c2ec1b6"
          ]
        },
        "id": "E9CuwX_cbZ8-",
        "outputId": "3f56b764-84de-4d22-863b-69fa1ac66146"
      },
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Retrieving documents for query: 'What is a Chronic Kidney Disease?'\n",
            "Top K: 5, Score threshold: 0.0\n",
            "Generating embeddings for 1 texts....\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Batches:   0%|          | 0/1 [00:00<?, ?it/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "d16cb3fb9499455f88cb89b3efc0c1c5"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Embeddings generated successfully. Shape: (1, 384)\n",
            "Retrieved 5 documents (after filtering)\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[{'id': 'doc_fa088aef_129',\n",
              "  'content': '1103Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\nIntroduction\\nChronic Kidney Disease (CKD) is a state of progressive loss \\nof kidney function ultimately resulting in the need for renal \\nreplacement therapy (dialysis or transplantation) [1 ]. It is \\ndefined as the presence of kidney damage or an estimated \\nglomerular filtration rate less than 60\\xa0ml/min per 1.73  m2, \\npersisting for 3\\xa0months or more [2]. CKD prevalence is grow-\\ning worldwide, along with demographic and epidemiologi-\\ncal transitions [3]. The implications of this disease are enor-\\nmous for our society in terms of quality of life and the overall \\nsustainability of national health systems. Worldwide, CKD \\naccounted for 2,968,600 (1%) disability-adjusted life-years \\nand 2,546,700 (1% to 3%) life-years lost in 2012 [4]. There-\\nfore, it is of the utmost importance to assess how to promptly \\nand adequately diagnose and treat patients with CKD.\\nThe causes of CKD vary globally. The most common pri-',\n",
              "  'metadata': {'crossmarkmajorversiondate': '2010-04-23',\n",
              "   'page_label': '1103',\n",
              "   'creator': 'Springer',\n",
              "   'content_length': 969,\n",
              "   'total_pages': 17,\n",
              "   'producer': 'Acrobat Distiller 10.1.8 (Windows)',\n",
              "   'source_file': 's40620-023-01573-4.pdf',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf',\n",
              "   'doc_index': 129,\n",
              "   'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4',\n",
              "   'creationdate': '2023-02-22T14:59:25+05:30',\n",
              "   'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review',\n",
              "   'robots': 'noindex',\n",
              "   'file_type': 'pdf',\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'page': 2,\n",
              "   'crossmarkdomains[1]': 'springer.com',\n",
              "   'crossmarkdomains[2]': 'springerlink.com',\n",
              "   'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review',\n",
              "   'doi': '10.1007/s40620-023-01573-4',\n",
              "   'moddate': '2023-05-23T18:34:02+05:30',\n",
              "   'author': 'Francesco Sanmarchi'},\n",
              "  'similarity_score': 0.3406142592430115,\n",
              "  'distance': 0.6593857407569885,\n",
              "  'rank': 1},\n",
              " {'id': 'doc_fcdf9ab8_410',\n",
              "  'content': 'Development Work Group Members. 2013. Evaluation and management of chronic kidney disease: synopsis of \\nthe kidney disease: improving global outcomes 2012 clinical practice guideline. Annals of Internal Medicine \\n158:825–830. DOI: https://doi.org/10.7326/0003-4819-158-11-201306040-00007, PMID: 23732715',\n",
              "  'metadata': {'producer': 'Adobe PDF Library 15.0',\n",
              "   'total_pages': 14,\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/81878.pdf',\n",
              "   'page': 12,\n",
              "   'creator': 'Adobe InDesign CC 14.0 (Windows)',\n",
              "   'doc_index': 410,\n",
              "   'jobid': '58d5703c-43e3-40ee-a282-404cf350e13c',\n",
              "   'trapped': '/False',\n",
              "   'file_type': 'pdf',\n",
              "   'creationdate': '2024-01-04T09:46:05+00:00',\n",
              "   'content_length': 303,\n",
              "   'page_label': '13',\n",
              "   'source_file': '81878.pdf',\n",
              "   'moddate': '2024-01-04T09:46:06+00:00'},\n",
              "  'similarity_score': 0.2970777750015259,\n",
              "  'distance': 0.7029222249984741,\n",
              "  'rank': 2},\n",
              " {'id': 'doc_79ec0225_184',\n",
              "  'content': '1114 Journal of Nephrology (2023) 36:1101–1117\\n1 3\\n 2. Chen TK, Knicely DH, Grams ME (2019) Chronic kidney dis-\\nease diagnosis and management. JAMA 322(13):1294–1304. \\nhttps:// doi. org/ 10. 1001/ jama. 2019. 14745\\n 3. Global, regional, and national burden of chronic kidney disease, \\n1990–2017: a systematic analysis for the Global Burden of Dis-\\nease Study 2017. Lancet Lond Engl. 2020;395(10225):709–733. \\ndoi:https:// doi. org/ 10. 1016/ S0140- 6736(20) 30045-3\\n 4. Vaidya SR, Aeddula NR. Chronic Renal Failure. In: StatPearls. \\nStatPearls Publishing; 2022. Accessed July 28, 2022. http:// www. \\nncbi. nlm. nih. gov/ books/ NBK53 5404/\\n 5. Romagnani P, Remuzzi G, Glassock R et\\xa0al (2017) Chronic \\nkidney disease. Nat Rev Dis Primer 3:17088. https:// doi. org/ 10. \\n1038/ nrdp. 2017. 88\\n 6. Thomas R, Kanso A, Sedor JR (2008) Chronic kidney disease \\nand its complications. Prim Care 35(2):329–vii. https:// doi. org/ \\n10. 1016/j. pop. 2008. 01. 008',\n",
              "  'metadata': {'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4',\n",
              "   'doi': '10.1007/s40620-023-01573-4',\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review',\n",
              "   'creator': 'Springer',\n",
              "   'robots': 'noindex',\n",
              "   'crossmarkmajorversiondate': '2010-04-23',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf',\n",
              "   'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review',\n",
              "   'author': 'Francesco Sanmarchi',\n",
              "   'creationdate': '2023-02-22T14:59:25+05:30',\n",
              "   'file_type': 'pdf',\n",
              "   'moddate': '2023-05-23T18:34:02+05:30',\n",
              "   'crossmarkdomains[1]': 'springer.com',\n",
              "   'source_file': 's40620-023-01573-4.pdf',\n",
              "   'producer': 'Acrobat Distiller 10.1.8 (Windows)',\n",
              "   'content_length': 951,\n",
              "   'total_pages': 17,\n",
              "   'doc_index': 184,\n",
              "   'page': 13,\n",
              "   'page_label': '1114',\n",
              "   'crossmarkdomains[2]': 'springerlink.com'},\n",
              "  'similarity_score': 0.25014567375183105,\n",
              "  'distance': 0.749854326248169,\n",
              "  'rank': 3},\n",
              " {'id': 'doc_71fd64cc_216',\n",
              "  'content': 'year due to kidney related disease. Currently, it is the 6th fastest-growing cause of death \\nworldwideCKD is becoming a challenging public health problem with increasing prev -\\nalence worldwide. Its burden is even higher in low-income countries where detection, \\nAbstract \\nGoal three of the UN’s Sustainable Development Goal is good health and well-being \\nwhere it clearly emphasized that non-communicable diseases is emerging challenge. \\nOne of the objectives is to reduce premature mortality from non-communicable dis-\\nease by third in 2030. Chronic kidney disease (CKD) is among the significant contribu-\\ntor to morbidity and mortality from non-communicable diseases that can affected \\n10–15% of the global population. Early and accurate detection of the stages of CKD \\nis believed to be vital to minimize impacts of patient’s health complications such \\nas hypertension, anemia (low blood count), mineral bone disorder, poor nutritional',\n",
              "  'metadata': {'source_file': 's40537-022-00657-5.pdf',\n",
              "   'moddate': '2022-11-20T22:44:42+01:00',\n",
              "   'doi': '10.1186/s40537-022-00657-5',\n",
              "   'total_pages': 19,\n",
              "   'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)',\n",
              "   'author': 'Dibaba Adeba Debal',\n",
              "   'file_type': 'pdf',\n",
              "   'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)',\n",
              "   'title': 'Chronic kidney disease prediction using machine learning techniques',\n",
              "   'page_label': '1',\n",
              "   'crossmarkdomains[2]': 'springerlink.com',\n",
              "   'page': 0,\n",
              "   'creator': 'Springer',\n",
              "   'content_length': 939,\n",
              "   'crossmarkmajorversiondate': '2010-04-23',\n",
              "   'doc_index': 216,\n",
              "   'creationdate': '2022-11-12T22:22:16+05:30',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf',\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5',\n",
              "   'robots': 'noindex',\n",
              "   'crossmarkdomains[1]': 'springer.com'},\n",
              "  'similarity_score': 0.23177069425582886,\n",
              "  'distance': 0.7682293057441711,\n",
              "  'rank': 4},\n",
              " {'id': 'doc_a1e8e17c_2',\n",
              "  'content': 'validation and improving the models with additional predictor variables.\\nChronic kidney disease (CKD) is a significant healthcare burden that affects billions of individuals  worldwide1,2 \\nand makes a profound impact on global morbidity and  mortality3–5. In the United States, approximately 11% of \\nthe population or 37 million people suffer from CKD that results in an annual Medicare cost of $84  billion6. The \\nprevalence of this disease is estimated at 10.8% in China, affecting about 119.5 million  people7.\\nGradual loss of the kidney function can lead to end stage kidney disease (ESKD) in CKD patients, precipitat-\\ning the need for kidney replacement therapy (KRT). Timely intervention in those CKD patients who have a high \\nrisk of ESKD may not only improve these patients’ quality of life by delaying the disease progression, but also \\nreduce the morbidity, mortality and healthcare costs resulting from  KRT8,9. Because the disease progression is',\n",
              "  'metadata': {'crossmarkmajorversiondate': '2010-04-23',\n",
              "   'creationdate': '2022-05-19T09:40:50+05:30',\n",
              "   'author': 'Qiong Bai',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s41598-022-12316-z.pdf',\n",
              "   'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)',\n",
              "   'doi': '10.1038/s41598-022-12316-z',\n",
              "   'content_length': 957,\n",
              "   'moddate': '2022-05-19T08:13:56+02:00',\n",
              "   'page_label': '1',\n",
              "   'file_type': 'pdf',\n",
              "   'total_pages': 8,\n",
              "   'doc_index': 2,\n",
              "   'crossmarkdomains[1]': 'springer.com',\n",
              "   'page': 0,\n",
              "   'creator': 'Springer',\n",
              "   'source_file': 's41598-022-12316-z.pdf',\n",
              "   'robots': 'noindex',\n",
              "   'title': 'Machine learning to predict end stage kidney disease in chronic kidney disease',\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'crossmarkdomains[2]': 'springerlink.com',\n",
              "   'subject': 'Scientific Reports, https://doi.org/10.1038/s41598-022-12316-z'},\n",
              "  'similarity_score': 0.22165870666503906,\n",
              "  'distance': 0.7783412933349609,\n",
              "  'rank': 5}]"
            ]
          },
          "metadata": {},
          "execution_count": 32
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "rag_retriever.retrieve(\"Best ML model for classifying Kidney Disease?\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000,
          "referenced_widgets": [
            "6698907a3cde41d58cd638d315a8964e",
            "8ea9ba784d5b47a6b217f6e7defd0b1d",
            "2c5423b6cc0e48b99209d800edaffb76",
            "38487f3ff0cb44608847013d8ddd9b51",
            "c6eed94d3e9745b7ac9f48f4ffb47c6d",
            "8ad56b7bed1c4cafad3e1daccf4e9c97",
            "34d68f3c63fb4834b8b6dd0a2ab2913c",
            "d9d1dad68ce2458a9321fb6acabd406c",
            "40b554f8d4244cceb030a04d8343eee4",
            "5ac5560d9c074be08f8d9b51e8e4375c",
            "4d459b47925b42a48938bdf82f2c5386"
          ]
        },
        "id": "I5bfqZKCdJHV",
        "outputId": "774189b8-a542-4c21-d181-0f9475abfa13"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Retrieving documents for query: 'Best ML model for classifying Kidney Disease?'\n",
            "Top K: 5, Score threshold: 0.0\n",
            "Generating embeddings for 1 texts....\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Batches:   0%|          | 0/1 [00:00<?, ?it/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "6698907a3cde41d58cd638d315a8964e"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Embeddings generated successfully. Shape: (1, 384)\n",
            "Retrieved 5 documents (after filtering)\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[{'id': 'doc_99461e9a_194',\n",
              "  'content': '1115Journal of Nephrology (2023) 36:1101–1117 \\n1 3\\n 36. Chen Z, Zhang X, Zhang Z (2016) Clinical risk assessment \\nof patients with chronic kidney disease by using clinical data \\nand multivariate models. Int Urol Nephrol 48(12):2069–2075. \\nhttps:// doi. org/ 10. 1007/ s11255- 016- 1346-4\\n 37. Rashed-Al-Mahfuz M, Haque A, Azad A et\\xa0al (2021) Clinically \\napplicable machine learning approaches to identify attributes \\nof chronic kidney disease (CKD) for use in low-cost diagnostic \\nscreening. IEEE J Transl Eng Health Med 9:4900511. https://  \\ndoi. org/ 10. 1109/ JTEHM. 2021. 30736 29\\n 38. Roth JA, Radevski G, Marzolini C et\\xa0al (2021) Cohort-derived \\nmachine learning models for individual prediction of chronic \\nkidney disease in people living with human immunodeficiency \\nvirus: a prospective multicenter cohort study. J Infect Dis \\n224(7):1198–1208. https:// doi. org/ 10. 1093/ infdis/ jiaa2 36\\n 39. Huang ML, Chou YC (2019) Combining a gravitational search',\n",
              "  'metadata': {'author': 'Francesco Sanmarchi',\n",
              "   'file_type': 'pdf',\n",
              "   'crossmarkdomains[1]': 'springer.com',\n",
              "   'crossmarkmajorversiondate': '2010-04-23',\n",
              "   'title': 'Predict, diagnose, and treat chronic kidney disease with machine learning: a systematic literature review',\n",
              "   'doc_index': 194,\n",
              "   'creator': 'Springer',\n",
              "   'robots': 'noindex',\n",
              "   'page_label': '1115',\n",
              "   'doi': '10.1007/s40620-023-01573-4',\n",
              "   'source_file': 's40620-023-01573-4.pdf',\n",
              "   'creationdate': '2023-02-22T14:59:25+05:30',\n",
              "   'content_length': 962,\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'moddate': '2023-05-23T18:34:02+05:30',\n",
              "   'subject': 'Journal of Nephrology, https://doi.org/10.1007/s40620-023-01573-4',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s40620-023-01573-4.pdf',\n",
              "   'keywords': 'Chronic kidney disease; Machine learning; Artificial intelligence; Systematic review',\n",
              "   'page': 14,\n",
              "   'crossmarkdomains[2]': 'springerlink.com',\n",
              "   'producer': 'Acrobat Distiller 10.1.8 (Windows)',\n",
              "   'total_pages': 17},\n",
              "  'similarity_score': 0.4355783462524414,\n",
              "  'distance': 0.5644216537475586,\n",
              "  'rank': 1},\n",
              " {'id': 'doc_73542535_224',\n",
              "  'content': 'stated as a better model with an accuracy of 96.75%.\\nXiao et\\xa0 al. [11] proposed prediction of\\xa0 chronic kidney disease progression using \\nlogistic regression, Elastic Net, lasso regression, ridge regression, support vector \\nmachine, random forest, XGBoost, neural network and k-nearest neighbor and com -\\npared the models based on their performance. They have used 551 patients’ history \\ndata with proteinuria with 18 features and classified the outcome as mild, moderate,',\n",
              "  'metadata': {'doc_index': 224,\n",
              "   'source_file': 's40537-022-00657-5.pdf',\n",
              "   'content_length': 471,\n",
              "   'page_label': '2',\n",
              "   'creationdate': '2022-11-12T22:22:16+05:30',\n",
              "   'moddate': '2022-11-20T22:44:42+01:00',\n",
              "   'subject': 'Journal of Big Data, https://doi.org/10.1186/s40537-022-00657-5',\n",
              "   'creator': 'Springer',\n",
              "   'robots': 'noindex',\n",
              "   'crossmarkdomains[2]': 'springerlink.com',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s40537-022-00657-5.pdf',\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'page': 1,\n",
              "   'title': 'Chronic kidney disease prediction using machine learning techniques',\n",
              "   'crossmarkmajorversiondate': '2010-04-23',\n",
              "   'keywords': 'Chronic Kidney Disease (CKD); Machine Learning; Random Forest (RF); Support Vector Machine (SVM)',\n",
              "   'file_type': 'pdf',\n",
              "   'producer': 'Acrobat Distiller 10.1.8 (Windows); modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)',\n",
              "   'crossmarkdomains[1]': 'springer.com',\n",
              "   'doi': '10.1186/s40537-022-00657-5',\n",
              "   'total_pages': 19,\n",
              "   'author': 'Dibaba Adeba Debal'},\n",
              "  'similarity_score': 0.4307621717453003,\n",
              "  'distance': 0.5692378282546997,\n",
              "  'rank': 2},\n",
              " {'id': 'doc_3295a3fc_74',\n",
              "  'content': 'Page 5 of 16\\nLei\\xa0et\\xa0al. BMC Medical Informatics and Decision Making          (2022) 22:205 \\n \\nwere used to construct progression models with 6 types \\nof predictors—demographics, comorbidities, laboratory \\ndata from blood and urine samples, renal biopsy pathol -\\nogy, and therapeutic regimen. These algorithms’ accuracy \\nvaried, as did the evaluation indexes used to evaluate the \\naccuracy (see Table\\xa01).\\nKidney disease types\\nThe various kidney diseases in the included articles could \\nbe classified into 3 categories: CKD, Immunoglobulin A \\nNephropathy (IgAN) and diabetic nephropathy. Studies \\non CKD (43.75%) and IgAN (37.5%) accounted for the \\nlargest proportions.\\nThe CKD sample size was 17,862, with a mean age of \\n61.93\\xa0years old and stage 3–4 in 7 articles. Utilization of \\nthe Logistics Regression (LR), Support Vector Machine \\n(SVM) and Random Forest (RF) algorithm accounted for \\nproportions of 15.3%, 10.7% and 9.2%, respectively. The',\n",
              "  'metadata': {'creator': 'Adobe InDesign 15.0 (Windows)',\n",
              "   'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy',\n",
              "   'page': 4,\n",
              "   'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1',\n",
              "   'crossmarkdomains[1]': 'springer.com',\n",
              "   'trapped': '/False',\n",
              "   'creationdate': '2022-08-01T07:19:39+00:00',\n",
              "   'moddate': '2022-08-01T10:44:42+02:00',\n",
              "   'author': 'Nuo Lei',\n",
              "   'robots': 'noindex',\n",
              "   'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis',\n",
              "   'source_file': 's12911-022-01951-1.pdf',\n",
              "   'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)',\n",
              "   'crossmarkmajorversiondate': '2010-04-23',\n",
              "   'content_length': 945,\n",
              "   'total_pages': 16,\n",
              "   'crossmarkdomains[2]': 'springerlink.com',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf',\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'doc_index': 74,\n",
              "   'doi': '10.1186/s12911-022-01951-1',\n",
              "   'page_label': '5',\n",
              "   'file_type': 'pdf'},\n",
              "  'similarity_score': 0.41704612970352173,\n",
              "  'distance': 0.5829538702964783,\n",
              "  'rank': 3},\n",
              " {'id': 'doc_6c7ad74b_64',\n",
              "  'content': 'Reviews) with the CRD (Centre of Reviews and Dissemi -\\nnation), report number CRD42020156213.\\nEligibility criteria\\nThe inclusion criteria were:\\n1. Clinical studies of diagnostic tests of accuracy.\\n2. Participants with kidney disease, aged 18\\xa0 years or \\nolder.\\n3. Studies that used ML algorithms.\\n4. Study outcome reflected kidney disease deteriora -\\ntion, including the doubling of serum creatinine, \\nsudden estimated Glomerular Filtration Rate (eGFR) \\ndeterioration, urinary protein level aggravation, \\nESRD occurrence, RRT initiation, cardiovascular \\nevents and all-cause mortality.\\nConclusion: Taking advantage of big data, ML algorithm-based prediction models have high accuracy in predicting \\nkidney disease progression, we recommend ML algorithms as an auxiliary tool for clinicians to determine proper treat-\\nment and disease management strategies.\\nKeywords: Artificial intelligence, Machine learning algorithm, Prediction models, Chronic kidney disease, CKD',\n",
              "  'metadata': {'author': 'Nuo Lei',\n",
              "   'crossmarkdomains[2]': 'springerlink.com',\n",
              "   'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis',\n",
              "   'creationdate': '2022-08-01T07:19:39+00:00',\n",
              "   'page': 1,\n",
              "   'source_file': 's12911-022-01951-1.pdf',\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'doc_index': 64,\n",
              "   'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy',\n",
              "   'doi': '10.1186/s12911-022-01951-1',\n",
              "   'content_length': 965,\n",
              "   'file_type': 'pdf',\n",
              "   'robots': 'noindex',\n",
              "   'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)',\n",
              "   'page_label': '2',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf',\n",
              "   'moddate': '2022-08-01T10:44:42+02:00',\n",
              "   'creator': 'Adobe InDesign 15.0 (Windows)',\n",
              "   'total_pages': 16,\n",
              "   'trapped': '/False',\n",
              "   'crossmarkdomains[1]': 'springer.com',\n",
              "   'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1',\n",
              "   'crossmarkmajorversiondate': '2010-04-23'},\n",
              "  'similarity_score': 0.4116290211677551,\n",
              "  'distance': 0.5883709788322449,\n",
              "  'rank': 4},\n",
              " {'id': 'doc_949e703c_54',\n",
              "  'content': 'Lei\\xa0et\\xa0al. \\nBMC Medical Informatics and Decision Making          (2022) 22:205  \\nhttps://doi.org/10.1186/s12911-022-01951-1\\nRESEARCH\\nMachine learning algorithms’ accuracy \\nin\\xa0predicting kidney disease progression: \\na\\xa0systematic review and\\xa0meta-analysis\\nNuo Lei1†, Xianlong Zhang2†, Mengting Wei1, Beini Lao1, Xueyi Xu1, Min Zhang1, Huifen Chen1, Yanmin Xu1, \\nBingqing Xia1, Dingjun Zhang1, Chendi Dong1, Lizhe Fu3, Fang Tang3 and Yifan Wu2* \\nAbstract \\nBackground: Kidney disease progression rates vary among patients. Rapid and accurate prediction of kidney disease \\noutcomes is crucial for disease management. In recent years, various prediction models using Machine Learning \\n(ML) algorithms have been established in nephrology. However, their accuracy have been inconsistent. Therefore, we \\nconducted a systematic review and meta-analysis to investigate the diagnostic accuracy of ML algorithms for kidney \\ndisease progression.',\n",
              "  'metadata': {'crossmarkdomains[1]': 'springer.com',\n",
              "   'crossmarkdomainexclusive': 'true',\n",
              "   'content_length': 930,\n",
              "   'creationdate': '2022-08-01T07:19:39+00:00',\n",
              "   'source': '/content/drive/MyDrive/Rag_Data/s12911-022-01951-1.pdf',\n",
              "   'trapped': '/False',\n",
              "   'moddate': '2022-08-01T10:44:42+02:00',\n",
              "   'title': 'Machine learning algorithms’ accuracy in predicting kidney disease progression: a systematic review and meta-analysis',\n",
              "   'doc_index': 54,\n",
              "   'doi': '10.1186/s12911-022-01951-1',\n",
              "   'subject': 'BMC Medical Informatics and Decision Making, https://doi.org/10.1186/s12911-022-01951-1',\n",
              "   'page_label': '1',\n",
              "   'total_pages': 16,\n",
              "   'author': 'Nuo Lei',\n",
              "   'producer': 'Adobe PDF Library 15.0; modified using iText® 5.3.5 ©2000-2012 1T3XT BVBA (SPRINGER SBM; licensed version)',\n",
              "   'robots': 'noindex',\n",
              "   'creator': 'Adobe InDesign 15.0 (Windows)',\n",
              "   'file_type': 'pdf',\n",
              "   'keywords': 'Artificial intelligence;Machine learning algorithm;Prediction models;Chronic kidney disease;CKD progression;Immunoglobulin A nephropathy',\n",
              "   'crossmarkmajorversiondate': '2010-04-23',\n",
              "   'crossmarkdomains[2]': 'springerlink.com',\n",
              "   'source_file': 's12911-022-01951-1.pdf',\n",
              "   'page': 0},\n",
              "  'similarity_score': 0.3964105248451233,\n",
              "  'distance': 0.6035894751548767,\n",
              "  'rank': 5}]"
            ]
          },
          "metadata": {},
          "execution_count": 29
        }
      ]
    }
  ]
}